// @pulumi/azure-native/insights/v20170501preview/diagnosticSetting.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiagnosticSetting = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The diagnostic setting resource.
 */
class DiagnosticSetting extends pulumi.CustomResource {
    /**
     * Create a DiagnosticSetting resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceUri === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceUri'");
            }
            inputs["eventHubAuthorizationRuleId"] = args ? args.eventHubAuthorizationRuleId : undefined;
            inputs["eventHubName"] = args ? args.eventHubName : undefined;
            inputs["logAnalyticsDestinationType"] = args ? args.logAnalyticsDestinationType : undefined;
            inputs["logs"] = args ? args.logs : undefined;
            inputs["metrics"] = args ? args.metrics : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceUri"] = args ? args.resourceUri : undefined;
            inputs["serviceBusRuleId"] = args ? args.serviceBusRuleId : undefined;
            inputs["storageAccountId"] = args ? args.storageAccountId : undefined;
            inputs["workspaceId"] = args ? args.workspaceId : undefined;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["eventHubAuthorizationRuleId"] = undefined /*out*/;
            inputs["eventHubName"] = undefined /*out*/;
            inputs["logAnalyticsDestinationType"] = undefined /*out*/;
            inputs["logs"] = undefined /*out*/;
            inputs["metrics"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["serviceBusRuleId"] = undefined /*out*/;
            inputs["storageAccountId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["workspaceId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:insights/v20170501preview:DiagnosticSetting" }, { type: "azure-native:insights:DiagnosticSetting" }, { type: "azure-nextgen:insights:DiagnosticSetting" }, { type: "azure-native:insights/v20210501preview:DiagnosticSetting" }, { type: "azure-nextgen:insights/v20210501preview:DiagnosticSetting" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DiagnosticSetting.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DiagnosticSetting resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DiagnosticSetting(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DiagnosticSetting.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DiagnosticSetting.__pulumiType;
    }
}
exports.DiagnosticSetting = DiagnosticSetting;
/** @internal */
DiagnosticSetting.__pulumiType = 'azure-native:insights/v20170501preview:DiagnosticSetting';
