// @pulumi/azure-native/marketplace/v20210601/privateStoreCollection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateStoreCollection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The Collection data structure.
 */
class PrivateStoreCollection extends pulumi.CustomResource {
    /**
     * Create a PrivateStoreCollection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.privateStoreId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'privateStoreId'");
            }
            inputs["allSubscriptions"] = args ? args.allSubscriptions : undefined;
            inputs["claim"] = args ? args.claim : undefined;
            inputs["collectionId"] = args ? args.collectionId : undefined;
            inputs["collectionName"] = args ? args.collectionName : undefined;
            inputs["enabled"] = args ? args.enabled : undefined;
            inputs["privateStoreId"] = args ? args.privateStoreId : undefined;
            inputs["subscriptionsList"] = args ? args.subscriptionsList : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["numberOfOffers"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["allSubscriptions"] = undefined /*out*/;
            inputs["claim"] = undefined /*out*/;
            inputs["collectionId"] = undefined /*out*/;
            inputs["collectionName"] = undefined /*out*/;
            inputs["enabled"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["numberOfOffers"] = undefined /*out*/;
            inputs["subscriptionsList"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:marketplace/v20210601:PrivateStoreCollection" }, { type: "azure-native:marketplace:PrivateStoreCollection" }, { type: "azure-nextgen:marketplace:PrivateStoreCollection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateStoreCollection.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing PrivateStoreCollection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateStoreCollection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivateStoreCollection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateStoreCollection.__pulumiType;
    }
}
exports.PrivateStoreCollection = PrivateStoreCollection;
/** @internal */
PrivateStoreCollection.__pulumiType = 'azure-native:marketplace/v20210601:PrivateStoreCollection';
