// @pulumi/azure-native/machinelearningservices/machineLearningDataset.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MachineLearningDataset = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Machine Learning dataset object wrapped into ARM resource envelope.
 * API Version: 2020-05-01-preview.
 */
class MachineLearningDataset extends pulumi.CustomResource {
    /**
     * Create a MachineLearningDataset resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.datasetType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'datasetType'");
            }
            if ((!args || args.parameters === undefined) && !opts.urn) {
                throw new Error("Missing required property 'parameters'");
            }
            if ((!args || args.registration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'registration'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["datasetName"] = args ? args.datasetName : undefined;
            inputs["datasetType"] = args ? args.datasetType : undefined;
            inputs["parameters"] = args ? args.parameters : undefined;
            inputs["registration"] = args ? args.registration : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["skipValidation"] = (_a = (args ? args.skipValidation : undefined)) !== null && _a !== void 0 ? _a : false;
            inputs["timeSeries"] = args ? args.timeSeries : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["identity"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["identity"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:machinelearningservices:MachineLearningDataset" }, { type: "azure-native:machinelearningservices/v20200501preview:MachineLearningDataset" }, { type: "azure-nextgen:machinelearningservices/v20200501preview:MachineLearningDataset" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MachineLearningDataset.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing MachineLearningDataset resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MachineLearningDataset(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MachineLearningDataset.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MachineLearningDataset.__pulumiType;
    }
}
exports.MachineLearningDataset = MachineLearningDataset;
/** @internal */
MachineLearningDataset.__pulumiType = 'azure-native:machinelearningservices:MachineLearningDataset';
