// @pulumi/azure-native/types/enums/devtestlab/v20180915/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WindowsOsState = exports.UsagePermissionType = exports.TransportProtocol = exports.StorageType = exports.SourceControlType = exports.PremiumDataDisk = exports.PolicyStatus = exports.PolicyFactName = exports.PolicyEvaluatorType = exports.NotificationChannelEventType = exports.ManagedIdentityType = exports.LinuxOsState = exports.HostCachingOptions = exports.EnvironmentPermission = exports.EnableStatus = exports.CustomImageOsType = void 0;
exports.CustomImageOsType = {
    Windows: "Windows",
    Linux: "Linux",
    None: "None",
};
exports.EnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.EnvironmentPermission = {
    Reader: "Reader",
    Contributor: "Contributor",
};
exports.HostCachingOptions = {
    None: "None",
    ReadOnly: "ReadOnly",
    ReadWrite: "ReadWrite",
};
exports.LinuxOsState = {
    NonDeprovisioned: "NonDeprovisioned",
    DeprovisionRequested: "DeprovisionRequested",
    DeprovisionApplied: "DeprovisionApplied",
};
exports.ManagedIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.NotificationChannelEventType = {
    AutoShutdown: "AutoShutdown",
    Cost: "Cost",
};
exports.PolicyEvaluatorType = {
    AllowedValuesPolicy: "AllowedValuesPolicy",
    MaxValuePolicy: "MaxValuePolicy",
};
exports.PolicyFactName = {
    UserOwnedLabVmCount: "UserOwnedLabVmCount",
    UserOwnedLabPremiumVmCount: "UserOwnedLabPremiumVmCount",
    LabVmCount: "LabVmCount",
    LabPremiumVmCount: "LabPremiumVmCount",
    LabVmSize: "LabVmSize",
    GalleryImage: "GalleryImage",
    UserOwnedLabVmCountInSubnet: "UserOwnedLabVmCountInSubnet",
    LabTargetCost: "LabTargetCost",
    EnvironmentTemplate: "EnvironmentTemplate",
    ScheduleEditPermission: "ScheduleEditPermission",
};
exports.PolicyStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PremiumDataDisk = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.SourceControlType = {
    VsoGit: "VsoGit",
    GitHub: "GitHub",
    StorageAccount: "StorageAccount",
};
exports.StorageType = {
    Standard: "Standard",
    Premium: "Premium",
    StandardSSD: "StandardSSD",
};
exports.TransportProtocol = {
    Tcp: "Tcp",
    Udp: "Udp",
};
exports.UsagePermissionType = {
    Default: "Default",
    Deny: "Deny",
    Allow: "Allow",
};
exports.WindowsOsState = {
    NonSysprepped: "NonSysprepped",
    SysprepRequested: "SysprepRequested",
    SysprepApplied: "SysprepApplied",
};
