// @pulumi/azure-native/dbforpostgresql/serverSecurityAlertPolicy.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerSecurityAlertPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A server security alert policy.
 * API Version: 2017-12-01.
 */
class ServerSecurityAlertPolicy extends pulumi.CustomResource {
    /**
     * Create a ServerSecurityAlertPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serverName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverName'");
            }
            if ((!args || args.state === undefined) && !opts.urn) {
                throw new Error("Missing required property 'state'");
            }
            inputs["disabledAlerts"] = args ? args.disabledAlerts : undefined;
            inputs["emailAccountAdmins"] = args ? args.emailAccountAdmins : undefined;
            inputs["emailAddresses"] = args ? args.emailAddresses : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["retentionDays"] = args ? args.retentionDays : undefined;
            inputs["securityAlertPolicyName"] = args ? args.securityAlertPolicyName : undefined;
            inputs["serverName"] = args ? args.serverName : undefined;
            inputs["state"] = args ? args.state : undefined;
            inputs["storageAccountAccessKey"] = args ? args.storageAccountAccessKey : undefined;
            inputs["storageEndpoint"] = args ? args.storageEndpoint : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["disabledAlerts"] = undefined /*out*/;
            inputs["emailAccountAdmins"] = undefined /*out*/;
            inputs["emailAddresses"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["retentionDays"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
            inputs["storageAccountAccessKey"] = undefined /*out*/;
            inputs["storageEndpoint"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:dbforpostgresql:ServerSecurityAlertPolicy" }, { type: "azure-native:dbforpostgresql/v20171201:ServerSecurityAlertPolicy" }, { type: "azure-nextgen:dbforpostgresql/v20171201:ServerSecurityAlertPolicy" }, { type: "azure-native:dbforpostgresql/v20171201preview:ServerSecurityAlertPolicy" }, { type: "azure-nextgen:dbforpostgresql/v20171201preview:ServerSecurityAlertPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ServerSecurityAlertPolicy.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ServerSecurityAlertPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ServerSecurityAlertPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ServerSecurityAlertPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServerSecurityAlertPolicy.__pulumiType;
    }
}
exports.ServerSecurityAlertPolicy = ServerSecurityAlertPolicy;
/** @internal */
ServerSecurityAlertPolicy.__pulumiType = 'azure-native:dbforpostgresql:ServerSecurityAlertPolicy';
