// @pulumi/azure-native/types/enums/guestconfiguration/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Kind = exports.ConfigurationMode = exports.AssignmentType = exports.ActionAfterReboot = exports.v20210125 = exports.v20200625 = exports.v20181120 = exports.v20180630preview = void 0;
// Export sub-modules:
const v20180630preview = require("./v20180630preview");
exports.v20180630preview = v20180630preview;
const v20181120 = require("./v20181120");
exports.v20181120 = v20181120;
const v20200625 = require("./v20200625");
exports.v20200625 = v20200625;
const v20210125 = require("./v20210125");
exports.v20210125 = v20210125;
exports.ActionAfterReboot = {
    ContinueConfiguration: "ContinueConfiguration",
    StopConfiguration: "StopConfiguration",
};
exports.AssignmentType = {
    Audit: "Audit",
    DeployAndAutoCorrect: "DeployAndAutoCorrect",
    ApplyAndAutoCorrect: "ApplyAndAutoCorrect",
    ApplyAndMonitor: "ApplyAndMonitor",
};
exports.ConfigurationMode = {
    ApplyOnly: "ApplyOnly",
    ApplyAndMonitor: "ApplyAndMonitor",
    ApplyAndAutoCorrect: "ApplyAndAutoCorrect",
};
exports.Kind = {
    DSC: "DSC",
};
