// @pulumi/azure-native/testbase/v20201216preview/customerEvent.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerEvent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The Customer Notification Event resource.
 */
class CustomerEvent extends pulumi.CustomResource {
    /**
     * Create a CustomerEvent resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.eventName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'eventName'");
            }
            if ((!args || args.receivers === undefined) && !opts.urn) {
                throw new Error("Missing required property 'receivers'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.testBaseAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'testBaseAccountName'");
            }
            inputs["customerEventName"] = args ? args.customerEventName : undefined;
            inputs["eventName"] = args ? args.eventName : undefined;
            inputs["receivers"] = args ? args.receivers : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["testBaseAccountName"] = args ? args.testBaseAccountName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["eventName"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["receivers"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:testbase/v20201216preview:CustomerEvent" }, { type: "azure-native:testbase:CustomerEvent" }, { type: "azure-nextgen:testbase:CustomerEvent" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CustomerEvent.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing CustomerEvent resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CustomerEvent(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CustomerEvent.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomerEvent.__pulumiType;
    }
}
exports.CustomerEvent = CustomerEvent;
/** @internal */
CustomerEvent.__pulumiType = 'azure-native:testbase/v20201216preview:CustomerEvent';
