// @pulumi/azure-native/testbase/v20201216preview/package.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Package = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The Test Base Package resource.
 */
class Package extends pulumi.CustomResource {
    /**
     * Create a Package resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.applicationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'applicationName'");
            }
            if ((!args || args.blobPath === undefined) && !opts.urn) {
                throw new Error("Missing required property 'blobPath'");
            }
            if ((!args || args.flightingRing === undefined) && !opts.urn) {
                throw new Error("Missing required property 'flightingRing'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.targetOSList === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetOSList'");
            }
            if ((!args || args.testBaseAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'testBaseAccountName'");
            }
            if ((!args || args.tests === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tests'");
            }
            if ((!args || args.version === undefined) && !opts.urn) {
                throw new Error("Missing required property 'version'");
            }
            inputs["applicationName"] = args ? args.applicationName : undefined;
            inputs["blobPath"] = args ? args.blobPath : undefined;
            inputs["flightingRing"] = args ? args.flightingRing : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["packageName"] = args ? args.packageName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["targetOSList"] = args ? args.targetOSList : undefined;
            inputs["testBaseAccountName"] = args ? args.testBaseAccountName : undefined;
            inputs["tests"] = args ? args.tests : undefined;
            inputs["version"] = args ? args.version : undefined;
            inputs["etag"] = undefined /*out*/;
            inputs["isEnabled"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["packageStatus"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["testTypes"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["validationResults"] = undefined /*out*/;
        }
        else {
            inputs["applicationName"] = undefined /*out*/;
            inputs["blobPath"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["flightingRing"] = undefined /*out*/;
            inputs["isEnabled"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["packageStatus"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["targetOSList"] = undefined /*out*/;
            inputs["testTypes"] = undefined /*out*/;
            inputs["tests"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["validationResults"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:testbase/v20201216preview:Package" }, { type: "azure-native:testbase:Package" }, { type: "azure-nextgen:testbase:Package" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Package.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Package resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Package(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Package.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Package.__pulumiType;
    }
}
exports.Package = Package;
/** @internal */
Package.__pulumiType = 'azure-native:testbase/v20201216preview:Package';
