// @pulumi/azure-native/storage/v20180301preview/storageAccountManagementPolicies.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageAccountManagementPolicies = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The Get Storage Account ManagementPolicies operation response.
 */
class StorageAccountManagementPolicies extends pulumi.CustomResource {
    /**
     * Create a StorageAccountManagementPolicies resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["managementPolicyName"] = args ? args.managementPolicyName : undefined;
            inputs["policy"] = args ? args.policy : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["policy"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:storage/v20180301preview:StorageAccountManagementPolicies" }, { type: "azure-native:storage:StorageAccountManagementPolicies" }, { type: "azure-nextgen:storage:StorageAccountManagementPolicies" }, { type: "azure-native:storage/v20181101:StorageAccountManagementPolicies" }, { type: "azure-nextgen:storage/v20181101:StorageAccountManagementPolicies" }, { type: "azure-native:storage/v20190401:StorageAccountManagementPolicies" }, { type: "azure-nextgen:storage/v20190401:StorageAccountManagementPolicies" }, { type: "azure-native:storage/v20190601:StorageAccountManagementPolicies" }, { type: "azure-nextgen:storage/v20190601:StorageAccountManagementPolicies" }, { type: "azure-native:storage/v20200801preview:StorageAccountManagementPolicies" }, { type: "azure-nextgen:storage/v20200801preview:StorageAccountManagementPolicies" }, { type: "azure-native:storage/v20210101:StorageAccountManagementPolicies" }, { type: "azure-nextgen:storage/v20210101:StorageAccountManagementPolicies" }, { type: "azure-native:storage/v20210201:StorageAccountManagementPolicies" }, { type: "azure-nextgen:storage/v20210201:StorageAccountManagementPolicies" }, { type: "azure-native:storage/v20210401:StorageAccountManagementPolicies" }, { type: "azure-nextgen:storage/v20210401:StorageAccountManagementPolicies" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(StorageAccountManagementPolicies.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing StorageAccountManagementPolicies resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StorageAccountManagementPolicies(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StorageAccountManagementPolicies.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StorageAccountManagementPolicies.__pulumiType;
    }
}
exports.StorageAccountManagementPolicies = StorageAccountManagementPolicies;
/** @internal */
StorageAccountManagementPolicies.__pulumiType = 'azure-native:storage/v20180301preview:StorageAccountManagementPolicies';
