// @pulumi/azure-native/desktopvirtualization/sessionHostConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionHostConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents a SessionHostConfiguration definition.
 * API Version: 2021-05-13-preview.
 */
class SessionHostConfiguration extends pulumi.CustomResource {
    /**
     * Create a SessionHostConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.hostPoolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hostPoolName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["diskType"] = args ? args.diskType : undefined;
            inputs["domainInfo"] = args ? args.domainInfo : undefined;
            inputs["hostPoolName"] = args ? args.hostPoolName : undefined;
            inputs["imageInfo"] = args ? args.imageInfo : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["vMSizeId"] = args ? args.vMSizeId : undefined;
            inputs["vmCustomConfigurationUri"] = args ? args.vmCustomConfigurationUri : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        else {
            inputs["diskType"] = undefined /*out*/;
            inputs["domainInfo"] = undefined /*out*/;
            inputs["imageInfo"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["vMSizeId"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
            inputs["vmCustomConfigurationUri"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:desktopvirtualization:SessionHostConfiguration" }, { type: "azure-native:desktopvirtualization/v20210513preview:SessionHostConfiguration" }, { type: "azure-nextgen:desktopvirtualization/v20210513preview:SessionHostConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SessionHostConfiguration.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SessionHostConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SessionHostConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SessionHostConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SessionHostConfiguration.__pulumiType;
    }
}
exports.SessionHostConfiguration = SessionHostConfiguration;
/** @internal */
SessionHostConfiguration.__pulumiType = 'azure-native:desktopvirtualization:SessionHostConfiguration';
