// @pulumi/azure-native/solutions/v20190701/applicationDefinition.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Information about managed application definition.
 */
class ApplicationDefinition extends pulumi.CustomResource {
    /**
     * Create a ApplicationDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.lockLevel === undefined) && !opts.urn) {
                throw new Error("Missing required property 'lockLevel'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["applicationDefinitionName"] = args ? args.applicationDefinitionName : undefined;
            inputs["artifacts"] = args ? args.artifacts : undefined;
            inputs["authorizations"] = args ? args.authorizations : undefined;
            inputs["createUiDefinition"] = args ? args.createUiDefinition : undefined;
            inputs["deploymentPolicy"] = args ? args.deploymentPolicy : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["isEnabled"] = args ? args.isEnabled : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["lockLevel"] = args ? args.lockLevel : undefined;
            inputs["lockingPolicy"] = args ? args.lockingPolicy : undefined;
            inputs["mainTemplate"] = args ? args.mainTemplate : undefined;
            inputs["managedBy"] = args ? args.managedBy : undefined;
            inputs["managementPolicy"] = args ? args.managementPolicy : undefined;
            inputs["notificationPolicy"] = args ? args.notificationPolicy : undefined;
            inputs["packageFileUri"] = args ? args.packageFileUri : undefined;
            inputs["policies"] = args ? args.policies : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["artifacts"] = undefined /*out*/;
            inputs["authorizations"] = undefined /*out*/;
            inputs["createUiDefinition"] = undefined /*out*/;
            inputs["deploymentPolicy"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["isEnabled"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["lockLevel"] = undefined /*out*/;
            inputs["lockingPolicy"] = undefined /*out*/;
            inputs["mainTemplate"] = undefined /*out*/;
            inputs["managedBy"] = undefined /*out*/;
            inputs["managementPolicy"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["notificationPolicy"] = undefined /*out*/;
            inputs["packageFileUri"] = undefined /*out*/;
            inputs["policies"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:solutions/v20190701:ApplicationDefinition" }, { type: "azure-native:solutions:ApplicationDefinition" }, { type: "azure-nextgen:solutions:ApplicationDefinition" }, { type: "azure-native:solutions/v20160901preview:ApplicationDefinition" }, { type: "azure-nextgen:solutions/v20160901preview:ApplicationDefinition" }, { type: "azure-native:solutions/v20170901:ApplicationDefinition" }, { type: "azure-nextgen:solutions/v20170901:ApplicationDefinition" }, { type: "azure-native:solutions/v20180601:ApplicationDefinition" }, { type: "azure-nextgen:solutions/v20180601:ApplicationDefinition" }, { type: "azure-native:solutions/v20200821preview:ApplicationDefinition" }, { type: "azure-nextgen:solutions/v20200821preview:ApplicationDefinition" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ApplicationDefinition.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ApplicationDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ApplicationDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ApplicationDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ApplicationDefinition.__pulumiType;
    }
}
exports.ApplicationDefinition = ApplicationDefinition;
/** @internal */
ApplicationDefinition.__pulumiType = 'azure-native:solutions/v20190701:ApplicationDefinition';
