// @pulumi/azure-native/types/enums/peering/v20210101/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationState = exports.Tier = exports.Size = exports.SessionAddressProvider = exports.Role = exports.Kind = exports.Family = exports.DirectPeeringType = void 0;
exports.DirectPeeringType = {
    Edge: "Edge",
    Transit: "Transit",
    Cdn: "Cdn",
    Internal: "Internal",
    Ix: "Ix",
    IxRs: "IxRs",
    Voice: "Voice",
};
exports.Family = {
    Direct: "Direct",
    Exchange: "Exchange",
};
exports.Kind = {
    Direct: "Direct",
    Exchange: "Exchange",
};
exports.Role = {
    Noc: "Noc",
    Policy: "Policy",
    Technical: "Technical",
    Service: "Service",
    Escalation: "Escalation",
    Other: "Other",
};
exports.SessionAddressProvider = {
    Microsoft: "Microsoft",
    Peer: "Peer",
};
exports.Size = {
    Free: "Free",
    Metered: "Metered",
    Unlimited: "Unlimited",
};
exports.Tier = {
    Basic: "Basic",
    Premium: "Premium",
};
exports.ValidationState = {
    None: "None",
    Pending: "Pending",
    Approved: "Approved",
    Failed: "Failed",
};
