// @pulumi/azure-native/azurestackhci/arcSetting.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ArcSetting = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ArcSetting details.
 * API Version: 2021-01-01-preview.
 */
class ArcSetting extends pulumi.CustomResource {
    /**
     * Create a ArcSetting resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["arcSettingName"] = args ? args.arcSettingName : undefined;
            inputs["clusterName"] = args ? args.clusterName : undefined;
            inputs["createdAt"] = args ? args.createdAt : undefined;
            inputs["createdBy"] = args ? args.createdBy : undefined;
            inputs["createdByType"] = args ? args.createdByType : undefined;
            inputs["lastModifiedAt"] = args ? args.lastModifiedAt : undefined;
            inputs["lastModifiedBy"] = args ? args.lastModifiedBy : undefined;
            inputs["lastModifiedByType"] = args ? args.lastModifiedByType : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["aggregateState"] = undefined /*out*/;
            inputs["arcInstanceResourceGroup"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["perNodeDetails"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["aggregateState"] = undefined /*out*/;
            inputs["arcInstanceResourceGroup"] = undefined /*out*/;
            inputs["createdAt"] = undefined /*out*/;
            inputs["createdBy"] = undefined /*out*/;
            inputs["createdByType"] = undefined /*out*/;
            inputs["lastModifiedAt"] = undefined /*out*/;
            inputs["lastModifiedBy"] = undefined /*out*/;
            inputs["lastModifiedByType"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["perNodeDetails"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:azurestackhci:ArcSetting" }, { type: "azure-native:azurestackhci/v20210101preview:ArcSetting" }, { type: "azure-nextgen:azurestackhci/v20210101preview:ArcSetting" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ArcSetting.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ArcSetting resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ArcSetting(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ArcSetting.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ArcSetting.__pulumiType;
    }
}
exports.ArcSetting = ArcSetting;
/** @internal */
ArcSetting.__pulumiType = 'azure-native:azurestackhci:ArcSetting';
