// @pulumi/azure-native/azurestackhci/cluster.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Cluster details.
 * API Version: 2020-10-01.
 */
class Cluster extends pulumi.CustomResource {
    /**
     * Create a Cluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.aadClientId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'aadClientId'");
            }
            if ((!args || args.aadTenantId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'aadTenantId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["aadClientId"] = args ? args.aadClientId : undefined;
            inputs["aadTenantId"] = args ? args.aadTenantId : undefined;
            inputs["clusterName"] = args ? args.clusterName : undefined;
            inputs["createdAt"] = args ? args.createdAt : undefined;
            inputs["createdBy"] = args ? args.createdBy : undefined;
            inputs["createdByType"] = args ? args.createdByType : undefined;
            inputs["lastModifiedAt"] = args ? args.lastModifiedAt : undefined;
            inputs["lastModifiedBy"] = args ? args.lastModifiedBy : undefined;
            inputs["lastModifiedByType"] = args ? args.lastModifiedByType : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["billingModel"] = undefined /*out*/;
            inputs["cloudId"] = undefined /*out*/;
            inputs["lastBillingTimestamp"] = undefined /*out*/;
            inputs["lastSyncTimestamp"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["registrationTimestamp"] = undefined /*out*/;
            inputs["reportedProperties"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["trialDaysRemaining"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["aadClientId"] = undefined /*out*/;
            inputs["aadTenantId"] = undefined /*out*/;
            inputs["billingModel"] = undefined /*out*/;
            inputs["cloudId"] = undefined /*out*/;
            inputs["createdAt"] = undefined /*out*/;
            inputs["createdBy"] = undefined /*out*/;
            inputs["createdByType"] = undefined /*out*/;
            inputs["lastBillingTimestamp"] = undefined /*out*/;
            inputs["lastModifiedAt"] = undefined /*out*/;
            inputs["lastModifiedBy"] = undefined /*out*/;
            inputs["lastModifiedByType"] = undefined /*out*/;
            inputs["lastSyncTimestamp"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["registrationTimestamp"] = undefined /*out*/;
            inputs["reportedProperties"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["trialDaysRemaining"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:azurestackhci:Cluster" }, { type: "azure-native:azurestackhci/v20200301preview:Cluster" }, { type: "azure-nextgen:azurestackhci/v20200301preview:Cluster" }, { type: "azure-native:azurestackhci/v20201001:Cluster" }, { type: "azure-nextgen:azurestackhci/v20201001:Cluster" }, { type: "azure-native:azurestackhci/v20210101preview:Cluster" }, { type: "azure-nextgen:azurestackhci/v20210101preview:Cluster" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Cluster.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Cluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Cluster(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Cluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Cluster.__pulumiType;
    }
}
exports.Cluster = Cluster;
/** @internal */
Cluster.__pulumiType = 'azure-native:azurestackhci:Cluster';
