// @pulumi/azure-native/types/enums/dbforpostgresql/v20201005privatepreview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerRole = exports.ServerEdition = exports.PostgreSQLVersion = exports.CreateMode = exports.CitusVersion = void 0;
exports.CitusVersion = {
    CitusVersion_8_3: "8.3",
    CitusVersion_9_0: "9.0",
    CitusVersion_9_1: "9.1",
    CitusVersion_9_2: "9.2",
    CitusVersion_9_3: "9.3",
    CitusVersion_9_4: "9.4",
    CitusVersion_9_5: "9.5",
};
exports.CreateMode = {
    Default: "Default",
    PointInTimeRestore: "PointInTimeRestore",
    ReadReplica: "ReadReplica",
};
exports.PostgreSQLVersion = {
    PostgreSQLVersion_11: "11",
    PostgreSQLVersion_12: "12",
};
exports.ServerEdition = {
    GeneralPurpose: "GeneralPurpose",
    MemoryOptimized: "MemoryOptimized",
};
exports.ServerRole = {
    Coordinator: "Coordinator",
    Worker: "Worker",
};
