// @pulumi/azure-native/types/enums/resources/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScriptType = exports.ResourceIdentityType = exports.OnErrorDeploymentType = exports.ManagedServiceIdentityType = exports.ExpressionEvaluationOptionsScopeType = exports.DeploymentMode = exports.CleanupOptions = exports.v20210401 = exports.v20210101 = exports.v20201001 = exports.v20200801 = exports.v20200601 = exports.v20191001preview = exports.v20191001 = exports.v20190801 = exports.v20190701 = exports.v20190601preview = exports.v20190510 = exports.v20190501 = exports.v20190301 = exports.v20180501 = exports.v20180201 = exports.v20170510 = exports.v20160901 = exports.v20160701 = exports.v20160201 = exports.v20151101 = void 0;
// Export sub-modules:
const v20151101 = require("./v20151101");
exports.v20151101 = v20151101;
const v20160201 = require("./v20160201");
exports.v20160201 = v20160201;
const v20160701 = require("./v20160701");
exports.v20160701 = v20160701;
const v20160901 = require("./v20160901");
exports.v20160901 = v20160901;
const v20170510 = require("./v20170510");
exports.v20170510 = v20170510;
const v20180201 = require("./v20180201");
exports.v20180201 = v20180201;
const v20180501 = require("./v20180501");
exports.v20180501 = v20180501;
const v20190301 = require("./v20190301");
exports.v20190301 = v20190301;
const v20190501 = require("./v20190501");
exports.v20190501 = v20190501;
const v20190510 = require("./v20190510");
exports.v20190510 = v20190510;
const v20190601preview = require("./v20190601preview");
exports.v20190601preview = v20190601preview;
const v20190701 = require("./v20190701");
exports.v20190701 = v20190701;
const v20190801 = require("./v20190801");
exports.v20190801 = v20190801;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20191001preview = require("./v20191001preview");
exports.v20191001preview = v20191001preview;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20200801 = require("./v20200801");
exports.v20200801 = v20200801;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20210401 = require("./v20210401");
exports.v20210401 = v20210401;
exports.CleanupOptions = {
    Always: "Always",
    OnSuccess: "OnSuccess",
    OnExpiration: "OnExpiration",
};
exports.DeploymentMode = {
    Incremental: "Incremental",
    Complete: "Complete",
};
exports.ExpressionEvaluationOptionsScopeType = {
    NotSpecified: "NotSpecified",
    Outer: "Outer",
    Inner: "Inner",
};
exports.ManagedServiceIdentityType = {
    UserAssigned: "UserAssigned",
};
exports.OnErrorDeploymentType = {
    LastSuccessful: "LastSuccessful",
    SpecificDeployment: "SpecificDeployment",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.ScriptType = {
    AzurePowerShell: "AzurePowerShell",
    AzureCLI: "AzureCLI",
};
