// @pulumi/azure-native/types/enums/containerservice/v20210301/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeekDay = exports.UpgradeChannel = exports.ScaleSetPriority = exports.ScaleSetEvictionPolicy = exports.ResourceIdentityType = exports.OutboundType = exports.OSType = exports.OSSKU = exports.OSDiskType = exports.NetworkPolicy = exports.NetworkPlugin = exports.NetworkMode = exports.ManagedClusterSKUTier = exports.ManagedClusterSKUName = exports.LoadBalancerSku = exports.LicenseType = exports.KubeletDiskType = exports.GPUInstanceProfile = exports.ExtendedLocationTypes = exports.Expander = exports.ConnectionStatus = exports.AgentPoolType = exports.AgentPoolMode = void 0;
exports.AgentPoolMode = {
    System: "System",
    User: "User",
};
exports.AgentPoolType = {
    VirtualMachineScaleSets: "VirtualMachineScaleSets",
    AvailabilitySet: "AvailabilitySet",
};
exports.ConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.Expander = {
    Least_waste: "least-waste",
    Most_pods: "most-pods",
    Priority: "priority",
    Random: "random",
};
exports.ExtendedLocationTypes = {
    EdgeZone: "EdgeZone",
};
exports.GPUInstanceProfile = {
    MIG1g: "MIG1g",
    MIG2g: "MIG2g",
    MIG3g: "MIG3g",
    MIG4g: "MIG4g",
    MIG7g: "MIG7g",
};
exports.KubeletDiskType = {
    OS: "OS",
    Temporary: "Temporary",
};
exports.LicenseType = {
    None: "None",
    Windows_Server: "Windows_Server",
};
exports.LoadBalancerSku = {
    Standard: "standard",
    Basic: "basic",
};
exports.ManagedClusterSKUName = {
    Basic: "Basic",
};
exports.ManagedClusterSKUTier = {
    Paid: "Paid",
    Free: "Free",
};
exports.NetworkMode = {
    Transparent: "transparent",
    Bridge: "bridge",
};
exports.NetworkPlugin = {
    Azure: "azure",
    Kubenet: "kubenet",
};
exports.NetworkPolicy = {
    Calico: "calico",
    Azure: "azure",
};
exports.OSDiskType = {
    Managed: "Managed",
    Ephemeral: "Ephemeral",
};
exports.OSSKU = {
    Ubuntu: "Ubuntu",
    CBLMariner: "CBLMariner",
};
exports.OSType = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.OutboundType = {
    LoadBalancer: "loadBalancer",
    UserDefinedRouting: "userDefinedRouting",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    None: "None",
};
exports.ScaleSetEvictionPolicy = {
    Delete: "Delete",
    Deallocate: "Deallocate",
};
exports.ScaleSetPriority = {
    Spot: "Spot",
    Regular: "Regular",
};
exports.UpgradeChannel = {
    Rapid: "rapid",
    Stable: "stable",
    Patch: "patch",
    Node_image: "node-image",
    None: "none",
};
exports.WeekDay = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
