// @pulumi/azure-native/datashare/kustoClusterDataSet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.KustoClusterDataSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A kusto cluster data set.
 * API Version: 2020-09-01.
 */
class KustoClusterDataSet extends pulumi.CustomResource {
    /**
     * Create a KustoClusterDataSet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.kustoClusterResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kustoClusterResourceId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.shareName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shareName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["dataSetName"] = args ? args.dataSetName : undefined;
            inputs["kind"] = "KustoCluster";
            inputs["kustoClusterResourceId"] = args ? args.kustoClusterResourceId : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["shareName"] = args ? args.shareName : undefined;
            inputs["dataSetId"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["dataSetId"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["kustoClusterResourceId"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:datashare:KustoClusterDataSet" }, { type: "azure-native:datashare/v20181101preview:KustoClusterDataSet" }, { type: "azure-nextgen:datashare/v20181101preview:KustoClusterDataSet" }, { type: "azure-native:datashare/v20191101:KustoClusterDataSet" }, { type: "azure-nextgen:datashare/v20191101:KustoClusterDataSet" }, { type: "azure-native:datashare/v20200901:KustoClusterDataSet" }, { type: "azure-nextgen:datashare/v20200901:KustoClusterDataSet" }, { type: "azure-native:datashare/v20201001preview:KustoClusterDataSet" }, { type: "azure-nextgen:datashare/v20201001preview:KustoClusterDataSet" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(KustoClusterDataSet.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing KustoClusterDataSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new KustoClusterDataSet(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of KustoClusterDataSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KustoClusterDataSet.__pulumiType;
    }
}
exports.KustoClusterDataSet = KustoClusterDataSet;
/** @internal */
KustoClusterDataSet.__pulumiType = 'azure-native:datashare:KustoClusterDataSet';
