// @pulumi/azure-native/migrate/v20210101/moveResource.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MoveResource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Defines the move resource.
 */
class MoveResource extends pulumi.CustomResource {
    /**
     * Create a MoveResource resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.moveCollectionName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'moveCollectionName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["moveCollectionName"] = args ? args.moveCollectionName : undefined;
            inputs["moveResourceName"] = args ? args.moveResourceName : undefined;
            inputs["properties"] = args ? args.properties : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:migrate/v20210101:MoveResource" }, { type: "azure-native:migrate:MoveResource" }, { type: "azure-nextgen:migrate:MoveResource" }, { type: "azure-native:migrate/v20191001preview:MoveResource" }, { type: "azure-nextgen:migrate/v20191001preview:MoveResource" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MoveResource.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing MoveResource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MoveResource(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MoveResource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MoveResource.__pulumiType;
    }
}
exports.MoveResource = MoveResource;
/** @internal */
MoveResource.__pulumiType = 'azure-native:migrate/v20210101:MoveResource';
