// @pulumi/azure-native/types/enums/network/v20190501/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionAffinityEnabledState = exports.RoutingRuleEnabledState = exports.HealthProbeEnabled = exports.FrontDoorRedirectType = exports.FrontDoorRedirectProtocol = exports.FrontDoorQuery = exports.FrontDoorProtocol = exports.FrontDoorHealthProbeMethod = exports.FrontDoorForwardingProtocol = exports.FrontDoorEnabledState = exports.EnforceCertificateNameCheckEnabledState = exports.DynamicCompressionEnabled = exports.BackendEnabledState = void 0;
exports.BackendEnabledState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.DynamicCompressionEnabled = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.EnforceCertificateNameCheckEnabledState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.FrontDoorEnabledState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.FrontDoorForwardingProtocol = {
    HttpOnly: "HttpOnly",
    HttpsOnly: "HttpsOnly",
    MatchRequest: "MatchRequest",
};
exports.FrontDoorHealthProbeMethod = {
    GET: "GET",
    HEAD: "HEAD",
};
exports.FrontDoorProtocol = {
    Http: "Http",
    Https: "Https",
};
exports.FrontDoorQuery = {
    StripNone: "StripNone",
    StripAll: "StripAll",
};
exports.FrontDoorRedirectProtocol = {
    HttpOnly: "HttpOnly",
    HttpsOnly: "HttpsOnly",
    MatchRequest: "MatchRequest",
};
exports.FrontDoorRedirectType = {
    Moved: "Moved",
    Found: "Found",
    TemporaryRedirect: "TemporaryRedirect",
    PermanentRedirect: "PermanentRedirect",
};
exports.HealthProbeEnabled = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RoutingRuleEnabledState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SessionAffinityEnabledState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
