// @pulumi/azure-native/insights/v20210401/dataCollectionEndpoint.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataCollectionEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Definition of ARM tracked top level resource.
 */
class DataCollectionEndpoint extends pulumi.CustomResource {
    /**
     * Create a DataCollectionEndpoint resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["dataCollectionEndpointName"] = args ? args.dataCollectionEndpointName : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["immutableId"] = args ? args.immutableId : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["networkAcls"] = args ? args.networkAcls : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["configurationAccess"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["logsIngestion"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["configurationAccess"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["immutableId"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["logsIngestion"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["networkAcls"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:insights/v20210401:DataCollectionEndpoint" }, { type: "azure-native:insights:DataCollectionEndpoint" }, { type: "azure-nextgen:insights:DataCollectionEndpoint" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DataCollectionEndpoint.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DataCollectionEndpoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DataCollectionEndpoint(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DataCollectionEndpoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataCollectionEndpoint.__pulumiType;
    }
}
exports.DataCollectionEndpoint = DataCollectionEndpoint;
/** @internal */
DataCollectionEndpoint.__pulumiType = 'azure-native:insights/v20210401:DataCollectionEndpoint';
