// @pulumi/azure-native/types/enums/aad/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TlsV1 = exports.SyncOnPremPasswords = exports.SyncNtlmPasswords = exports.SyncKerberosPasswords = exports.NtlmV1 = exports.NotifyGlobalAdmins = exports.NotifyDcAdmins = exports.Ldaps = exports.KerberosRc4Encryption = exports.KerberosArmoring = exports.FilteredSync = exports.ExternalAccess = exports.v20210501 = exports.v20210301 = exports.v20200101 = exports.v20170601 = exports.v20170101 = void 0;
// Export sub-modules:
const v20170101 = require("./v20170101");
exports.v20170101 = v20170101;
const v20170601 = require("./v20170601");
exports.v20170601 = v20170601;
const v20200101 = require("./v20200101");
exports.v20200101 = v20200101;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210501 = require("./v20210501");
exports.v20210501 = v20210501;
exports.ExternalAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.FilteredSync = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.KerberosArmoring = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.KerberosRc4Encryption = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.Ldaps = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.NotifyDcAdmins = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.NotifyGlobalAdmins = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.NtlmV1 = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SyncKerberosPasswords = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SyncNtlmPasswords = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SyncOnPremPasswords = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.TlsV1 = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
