// @pulumi/azure-native/network/v20210201preview/connectivityConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectivityConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The network manager connectivity configuration resource
 */
class ConnectivityConfiguration extends pulumi.CustomResource {
    /**
     * Create a ConnectivityConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.connectivityTopology === undefined) && !opts.urn) {
                throw new Error("Missing required property 'connectivityTopology'");
            }
            if ((!args || args.networkManagerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkManagerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["appliesToGroups"] = args ? args.appliesToGroups : undefined;
            inputs["configurationName"] = args ? args.configurationName : undefined;
            inputs["connectivityTopology"] = args ? args.connectivityTopology : undefined;
            inputs["deleteExistingPeering"] = args ? args.deleteExistingPeering : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["hubId"] = args ? args.hubId : undefined;
            inputs["isGlobal"] = args ? args.isGlobal : undefined;
            inputs["networkManagerName"] = args ? args.networkManagerName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["appliesToGroups"] = undefined /*out*/;
            inputs["connectivityTopology"] = undefined /*out*/;
            inputs["deleteExistingPeering"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["hubId"] = undefined /*out*/;
            inputs["isGlobal"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20210201preview:ConnectivityConfiguration" }, { type: "azure-native:network:ConnectivityConfiguration" }, { type: "azure-nextgen:network:ConnectivityConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ConnectivityConfiguration.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ConnectivityConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ConnectivityConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ConnectivityConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConnectivityConfiguration.__pulumiType;
    }
}
exports.ConnectivityConfiguration = ConnectivityConfiguration;
/** @internal */
ConnectivityConfiguration.__pulumiType = 'azure-native:network/v20210201preview:ConnectivityConfiguration';
