// @pulumi/azure-native/network/v20210201preview/defaultAdminRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DefaultAdminRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Network default admin rule.
 */
class DefaultAdminRule extends pulumi.CustomResource {
    /**
     * Create a DefaultAdminRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.configurationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'configurationName'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.networkManagerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkManagerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.ruleCollectionName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ruleCollectionName'");
            }
            inputs["configurationName"] = args ? args.configurationName : undefined;
            inputs["flag"] = args ? args.flag : undefined;
            inputs["kind"] = "Default";
            inputs["networkManagerName"] = args ? args.networkManagerName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["ruleCollectionName"] = args ? args.ruleCollectionName : undefined;
            inputs["ruleName"] = args ? args.ruleName : undefined;
            inputs["access"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["destinationPortRanges"] = undefined /*out*/;
            inputs["destinations"] = undefined /*out*/;
            inputs["direction"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["priority"] = undefined /*out*/;
            inputs["protocol"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["sourcePortRanges"] = undefined /*out*/;
            inputs["sources"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["access"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["destinationPortRanges"] = undefined /*out*/;
            inputs["destinations"] = undefined /*out*/;
            inputs["direction"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["flag"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["priority"] = undefined /*out*/;
            inputs["protocol"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["sourcePortRanges"] = undefined /*out*/;
            inputs["sources"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20210201preview:DefaultAdminRule" }, { type: "azure-native:network:DefaultAdminRule" }, { type: "azure-nextgen:network:DefaultAdminRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DefaultAdminRule.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DefaultAdminRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DefaultAdminRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DefaultAdminRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DefaultAdminRule.__pulumiType;
    }
}
exports.DefaultAdminRule = DefaultAdminRule;
/** @internal */
DefaultAdminRule.__pulumiType = 'azure-native:network/v20210201preview:DefaultAdminRule';
