// @pulumi/azure-native/types/enums/logic/v20180701preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.X12TimeFormat = exports.X12DateFormat = exports.X12CharacterSet = exports.WorkflowState = exports.UsageIndicator = exports.TrailingSeparatorPolicy = exports.SkuName = exports.SigningAlgorithm = exports.SegmentTerminatorSuffix = exports.SchemaType = exports.RecurrenceFrequency = exports.PartnerType = exports.ParameterType = exports.MessageFilterType = exports.MapType = exports.KeyType = exports.IntegrationAccountSkuName = exports.HashingAlgorithm = exports.EncryptionAlgorithm = exports.EdifactDecimalIndicator = exports.EdifactCharacterSet = exports.DaysOfWeek = exports.DayOfWeek = exports.AgreementType = void 0;
exports.AgreementType = {
    NotSpecified: "NotSpecified",
    AS2: "AS2",
    X12: "X12",
    Edifact: "Edifact",
};
exports.DayOfWeek = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
exports.DaysOfWeek = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
exports.EdifactCharacterSet = {
    NotSpecified: "NotSpecified",
    UNOB: "UNOB",
    UNOA: "UNOA",
    UNOC: "UNOC",
    UNOD: "UNOD",
    UNOE: "UNOE",
    UNOF: "UNOF",
    UNOG: "UNOG",
    UNOH: "UNOH",
    UNOI: "UNOI",
    UNOJ: "UNOJ",
    UNOK: "UNOK",
    UNOX: "UNOX",
    UNOY: "UNOY",
    KECA: "KECA",
};
exports.EdifactDecimalIndicator = {
    NotSpecified: "NotSpecified",
    Comma: "Comma",
    Decimal: "Decimal",
};
exports.EncryptionAlgorithm = {
    NotSpecified: "NotSpecified",
    None: "None",
    DES3: "DES3",
    RC2: "RC2",
    AES128: "AES128",
    AES192: "AES192",
    AES256: "AES256",
};
exports.HashingAlgorithm = {
    NotSpecified: "NotSpecified",
    None: "None",
    MD5: "MD5",
    SHA1: "SHA1",
    SHA2256: "SHA2256",
    SHA2384: "SHA2384",
    SHA2512: "SHA2512",
};
exports.IntegrationAccountSkuName = {
    NotSpecified: "NotSpecified",
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
};
exports.KeyType = {
    NotSpecified: "NotSpecified",
    Primary: "Primary",
    Secondary: "Secondary",
};
exports.MapType = {
    NotSpecified: "NotSpecified",
    Xslt: "Xslt",
    Xslt20: "Xslt20",
    Xslt30: "Xslt30",
    Liquid: "Liquid",
};
exports.MessageFilterType = {
    NotSpecified: "NotSpecified",
    Include: "Include",
    Exclude: "Exclude",
};
exports.ParameterType = {
    NotSpecified: "NotSpecified",
    String: "String",
    SecureString: "SecureString",
    Int: "Int",
    Float: "Float",
    Bool: "Bool",
    Array: "Array",
    Object: "Object",
    SecureObject: "SecureObject",
};
exports.PartnerType = {
    NotSpecified: "NotSpecified",
    B2B: "B2B",
};
exports.RecurrenceFrequency = {
    NotSpecified: "NotSpecified",
    Second: "Second",
    Minute: "Minute",
    Hour: "Hour",
    Day: "Day",
    Week: "Week",
    Month: "Month",
    Year: "Year",
};
exports.SchemaType = {
    NotSpecified: "NotSpecified",
    Xml: "Xml",
};
exports.SegmentTerminatorSuffix = {
    NotSpecified: "NotSpecified",
    None: "None",
    CR: "CR",
    LF: "LF",
    CRLF: "CRLF",
};
exports.SigningAlgorithm = {
    NotSpecified: "NotSpecified",
    Default: "Default",
    SHA1: "SHA1",
    SHA2256: "SHA2256",
    SHA2384: "SHA2384",
    SHA2512: "SHA2512",
};
exports.SkuName = {
    NotSpecified: "NotSpecified",
    Free: "Free",
    Shared: "Shared",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.TrailingSeparatorPolicy = {
    NotSpecified: "NotSpecified",
    NotAllowed: "NotAllowed",
    Optional: "Optional",
    Mandatory: "Mandatory",
};
exports.UsageIndicator = {
    NotSpecified: "NotSpecified",
    Test: "Test",
    Information: "Information",
    Production: "Production",
};
exports.WorkflowState = {
    NotSpecified: "NotSpecified",
    Completed: "Completed",
    Enabled: "Enabled",
    Disabled: "Disabled",
    Deleted: "Deleted",
    Suspended: "Suspended",
};
exports.X12CharacterSet = {
    NotSpecified: "NotSpecified",
    Basic: "Basic",
    Extended: "Extended",
    UTF8: "UTF8",
};
exports.X12DateFormat = {
    NotSpecified: "NotSpecified",
    CCYYMMDD: "CCYYMMDD",
    YYMMDD: "YYMMDD",
};
exports.X12TimeFormat = {
    NotSpecified: "NotSpecified",
    HHMM: "HHMM",
    HHMMSS: "HHMMSS",
    HHMMSSdd: "HHMMSSdd",
    HHMMSSd: "HHMMSSd",
};
