// @pulumi/azure-native/netapp/v20191101/volume.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Volume = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Volume resource
 */
class Volume extends pulumi.CustomResource {
    /**
     * Create a Volume resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.creationToken === undefined) && !opts.urn) {
                throw new Error("Missing required property 'creationToken'");
            }
            if ((!args || args.poolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'poolName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.subnetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subnetId'");
            }
            if ((!args || args.usageThreshold === undefined) && !opts.urn) {
                throw new Error("Missing required property 'usageThreshold'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["creationToken"] = args ? args.creationToken : undefined;
            inputs["dataProtection"] = args ? args.dataProtection : undefined;
            inputs["exportPolicy"] = args ? args.exportPolicy : undefined;
            inputs["isRestoring"] = args ? args.isRestoring : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["poolName"] = args ? args.poolName : undefined;
            inputs["protocolTypes"] = args ? args.protocolTypes : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serviceLevel"] = (_a = (args ? args.serviceLevel : undefined)) !== null && _a !== void 0 ? _a : "Premium";
            inputs["snapshotId"] = args ? args.snapshotId : undefined;
            inputs["subnetId"] = args ? args.subnetId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["usageThreshold"] = (_b = (args ? args.usageThreshold : undefined)) !== null && _b !== void 0 ? _b : 107374182400;
            inputs["volumeName"] = args ? args.volumeName : undefined;
            inputs["volumeType"] = args ? args.volumeType : undefined;
            inputs["baremetalTenantId"] = undefined /*out*/;
            inputs["fileSystemId"] = undefined /*out*/;
            inputs["mountTargets"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["usedBytes"] = undefined /*out*/;
        }
        else {
            inputs["baremetalTenantId"] = undefined /*out*/;
            inputs["creationToken"] = undefined /*out*/;
            inputs["dataProtection"] = undefined /*out*/;
            inputs["exportPolicy"] = undefined /*out*/;
            inputs["fileSystemId"] = undefined /*out*/;
            inputs["isRestoring"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["mountTargets"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["protocolTypes"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["serviceLevel"] = undefined /*out*/;
            inputs["snapshotId"] = undefined /*out*/;
            inputs["subnetId"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["usageThreshold"] = undefined /*out*/;
            inputs["usedBytes"] = undefined /*out*/;
            inputs["volumeType"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:netapp/v20191101:Volume" }, { type: "azure-native:netapp:Volume" }, { type: "azure-nextgen:netapp:Volume" }, { type: "azure-native:netapp/v20170815:Volume" }, { type: "azure-nextgen:netapp/v20170815:Volume" }, { type: "azure-native:netapp/v20190501:Volume" }, { type: "azure-nextgen:netapp/v20190501:Volume" }, { type: "azure-native:netapp/v20190601:Volume" }, { type: "azure-nextgen:netapp/v20190601:Volume" }, { type: "azure-native:netapp/v20190701:Volume" }, { type: "azure-nextgen:netapp/v20190701:Volume" }, { type: "azure-native:netapp/v20190801:Volume" }, { type: "azure-nextgen:netapp/v20190801:Volume" }, { type: "azure-native:netapp/v20191001:Volume" }, { type: "azure-nextgen:netapp/v20191001:Volume" }, { type: "azure-native:netapp/v20200201:Volume" }, { type: "azure-nextgen:netapp/v20200201:Volume" }, { type: "azure-native:netapp/v20200301:Volume" }, { type: "azure-nextgen:netapp/v20200301:Volume" }, { type: "azure-native:netapp/v20200501:Volume" }, { type: "azure-nextgen:netapp/v20200501:Volume" }, { type: "azure-native:netapp/v20200601:Volume" }, { type: "azure-nextgen:netapp/v20200601:Volume" }, { type: "azure-native:netapp/v20200701:Volume" }, { type: "azure-nextgen:netapp/v20200701:Volume" }, { type: "azure-native:netapp/v20200801:Volume" }, { type: "azure-nextgen:netapp/v20200801:Volume" }, { type: "azure-native:netapp/v20200901:Volume" }, { type: "azure-nextgen:netapp/v20200901:Volume" }, { type: "azure-native:netapp/v20201101:Volume" }, { type: "azure-nextgen:netapp/v20201101:Volume" }, { type: "azure-native:netapp/v20201201:Volume" }, { type: "azure-nextgen:netapp/v20201201:Volume" }, { type: "azure-native:netapp/v20210201:Volume" }, { type: "azure-nextgen:netapp/v20210201:Volume" }, { type: "azure-native:netapp/v20210401:Volume" }, { type: "azure-nextgen:netapp/v20210401:Volume" }, { type: "azure-native:netapp/v20210401preview:Volume" }, { type: "azure-nextgen:netapp/v20210401preview:Volume" }, { type: "azure-native:netapp/v20210601:Volume" }, { type: "azure-nextgen:netapp/v20210601:Volume" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Volume.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Volume resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Volume(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Volume.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Volume.__pulumiType;
    }
}
exports.Volume = Volume;
/** @internal */
Volume.__pulumiType = 'azure-native:netapp/v20191101:Volume';
