// @pulumi/azure-native/signalrservice/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210601preview = exports.v20210401preview = exports.v20200701preview = exports.v20200501 = exports.v20181001 = exports.v20180301preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./getSignalR"), exports);
__exportStar(require("./getSignalRPrivateEndpointConnection"), exports);
__exportStar(require("./getSignalRSharedPrivateLinkResource"), exports);
__exportStar(require("./listSignalRKeys"), exports);
__exportStar(require("./signalR"), exports);
__exportStar(require("./signalRPrivateEndpointConnection"), exports);
__exportStar(require("./signalRSharedPrivateLinkResource"), exports);
// Export enums:
__exportStar(require("../types/enums/signalrservice"), exports);
// Export sub-modules:
const v20180301preview = require("./v20180301preview");
exports.v20180301preview = v20180301preview;
const v20181001 = require("./v20181001");
exports.v20181001 = v20181001;
const v20200501 = require("./v20200501");
exports.v20200501 = v20200501;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
// Import resources to register:
const signalR_1 = require("./signalR");
const signalRPrivateEndpointConnection_1 = require("./signalRPrivateEndpointConnection");
const signalRSharedPrivateLinkResource_1 = require("./signalRSharedPrivateLinkResource");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:signalrservice:SignalR":
                return new signalR_1.SignalR(name, undefined, { urn });
            case "azure-native:signalrservice:SignalRPrivateEndpointConnection":
                return new signalRPrivateEndpointConnection_1.SignalRPrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:signalrservice:SignalRSharedPrivateLinkResource":
                return new signalRSharedPrivateLinkResource_1.SignalRSharedPrivateLinkResource(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "signalrservice", _module);
