// @pulumi/azure-native/types/enums/compute/v20180930/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SnapshotStorageAccountTypes = exports.OperatingSystemTypes = exports.HyperVGeneration = exports.DiskStorageAccountTypes = exports.DiskCreateOption = void 0;
exports.DiskCreateOption = {
    Empty: "Empty",
    Attach: "Attach",
    FromImage: "FromImage",
    Import: "Import",
    Copy: "Copy",
    Restore: "Restore",
    Upload: "Upload",
};
exports.DiskStorageAccountTypes = {
    Standard_LRS: "Standard_LRS",
    Premium_LRS: "Premium_LRS",
    StandardSSD_LRS: "StandardSSD_LRS",
    UltraSSD_LRS: "UltraSSD_LRS",
};
exports.HyperVGeneration = {
    V1: "V1",
    V2: "V2",
};
exports.OperatingSystemTypes = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.SnapshotStorageAccountTypes = {
    Standard_LRS: "Standard_LRS",
    Premium_LRS: "Premium_LRS",
    Standard_ZRS: "Standard_ZRS",
};
