// @pulumi/azure-native/automation/v20190601/softwareUpdateConfigurationByName.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SoftwareUpdateConfigurationByName = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Software update configuration properties.
 */
class SoftwareUpdateConfigurationByName extends pulumi.CustomResource {
    /**
     * Create a SoftwareUpdateConfigurationByName resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.automationAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'automationAccountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.scheduleInfo === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scheduleInfo'");
            }
            if ((!args || args.updateConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'updateConfiguration'");
            }
            inputs["automationAccountName"] = args ? args.automationAccountName : undefined;
            inputs["error"] = args ? args.error : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["scheduleInfo"] = args ? args.scheduleInfo : undefined;
            inputs["softwareUpdateConfigurationName"] = args ? args.softwareUpdateConfigurationName : undefined;
            inputs["tasks"] = args ? args.tasks : undefined;
            inputs["updateConfiguration"] = args ? args.updateConfiguration : undefined;
            inputs["createdBy"] = undefined /*out*/;
            inputs["creationTime"] = undefined /*out*/;
            inputs["lastModifiedBy"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["createdBy"] = undefined /*out*/;
            inputs["creationTime"] = undefined /*out*/;
            inputs["error"] = undefined /*out*/;
            inputs["lastModifiedBy"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["scheduleInfo"] = undefined /*out*/;
            inputs["tasks"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["updateConfiguration"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:automation/v20190601:SoftwareUpdateConfigurationByName" }, { type: "azure-native:automation:SoftwareUpdateConfigurationByName" }, { type: "azure-nextgen:automation:SoftwareUpdateConfigurationByName" }, { type: "azure-native:automation/v20170515preview:SoftwareUpdateConfigurationByName" }, { type: "azure-nextgen:automation/v20170515preview:SoftwareUpdateConfigurationByName" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SoftwareUpdateConfigurationByName.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SoftwareUpdateConfigurationByName resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SoftwareUpdateConfigurationByName(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SoftwareUpdateConfigurationByName.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SoftwareUpdateConfigurationByName.__pulumiType;
    }
}
exports.SoftwareUpdateConfigurationByName = SoftwareUpdateConfigurationByName;
/** @internal */
SoftwareUpdateConfigurationByName.__pulumiType = 'azure-native:automation/v20190601:SoftwareUpdateConfigurationByName';
