// @pulumi/azure-native/automation/v20190601/sourceControl.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceControl = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Definition of the source control.
 */
class SourceControl extends pulumi.CustomResource {
    /**
     * Create a SourceControl resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.automationAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'automationAccountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["autoSync"] = args ? args.autoSync : undefined;
            inputs["automationAccountName"] = args ? args.automationAccountName : undefined;
            inputs["branch"] = args ? args.branch : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["folderPath"] = args ? args.folderPath : undefined;
            inputs["publishRunbook"] = args ? args.publishRunbook : undefined;
            inputs["repoUrl"] = args ? args.repoUrl : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["securityToken"] = args ? args.securityToken : undefined;
            inputs["sourceControlName"] = args ? args.sourceControlName : undefined;
            inputs["sourceType"] = args ? args.sourceType : undefined;
            inputs["creationTime"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["autoSync"] = undefined /*out*/;
            inputs["branch"] = undefined /*out*/;
            inputs["creationTime"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["folderPath"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["publishRunbook"] = undefined /*out*/;
            inputs["repoUrl"] = undefined /*out*/;
            inputs["sourceType"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:automation/v20190601:SourceControl" }, { type: "azure-native:automation:SourceControl" }, { type: "azure-nextgen:automation:SourceControl" }, { type: "azure-native:automation/v20170515preview:SourceControl" }, { type: "azure-nextgen:automation/v20170515preview:SourceControl" }, { type: "azure-native:automation/v20200113preview:SourceControl" }, { type: "azure-nextgen:automation/v20200113preview:SourceControl" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SourceControl.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SourceControl resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SourceControl(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SourceControl.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SourceControl.__pulumiType;
    }
}
exports.SourceControl = SourceControl;
/** @internal */
SourceControl.__pulumiType = 'azure-native:automation/v20190601:SourceControl';
