// @pulumi/azure-native/types/enums/costmanagement/v20191101/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeframeType = exports.StatusType = exports.ReportType = exports.ReportTimeframeType = exports.ReportGranularityType = exports.ReportConfigColumnType = exports.RecurrenceType = exports.QueryColumnType = exports.PivotTypeType = exports.OperatorType = exports.MetricType = exports.KpiTypeType = exports.GranularityType = exports.FunctionType = exports.FormatType = exports.ExportType = exports.ChartType = exports.AccumulatedType = void 0;
exports.AccumulatedType = {
    True: "true",
    False: "false",
};
exports.ChartType = {
    Area: "Area",
    Line: "Line",
    StackedColumn: "StackedColumn",
    GroupedColumn: "GroupedColumn",
    Table: "Table",
};
exports.ExportType = {
    Usage: "Usage",
    ActualCost: "ActualCost",
    AmortizedCost: "AmortizedCost",
};
exports.FormatType = {
    Csv: "Csv",
};
exports.FunctionType = {
    Avg: "Avg",
    Max: "Max",
    Min: "Min",
    Sum: "Sum",
};
exports.GranularityType = {
    Daily: "Daily",
};
exports.KpiTypeType = {
    Forecast: "Forecast",
    Budget: "Budget",
};
exports.MetricType = {
    ActualCost: "ActualCost",
    AmortizedCost: "AmortizedCost",
    AHUB: "AHUB",
};
exports.OperatorType = {
    In: "In",
    Contains: "Contains",
};
exports.PivotTypeType = {
    Dimension: "Dimension",
    TagKey: "TagKey",
};
exports.QueryColumnType = {
    Tag: "Tag",
    Dimension: "Dimension",
};
exports.RecurrenceType = {
    Daily: "Daily",
    Weekly: "Weekly",
    Monthly: "Monthly",
    Annually: "Annually",
};
exports.ReportConfigColumnType = {
    Tag: "Tag",
    Dimension: "Dimension",
};
exports.ReportGranularityType = {
    Daily: "Daily",
    Monthly: "Monthly",
};
exports.ReportTimeframeType = {
    WeekToDate: "WeekToDate",
    MonthToDate: "MonthToDate",
    YearToDate: "YearToDate",
    Custom: "Custom",
};
exports.ReportType = {
    Usage: "Usage",
};
exports.StatusType = {
    Active: "Active",
    Inactive: "Inactive",
};
exports.TimeframeType = {
    MonthToDate: "MonthToDate",
    BillingMonthToDate: "BillingMonthToDate",
    TheLastMonth: "TheLastMonth",
    TheLastBillingMonth: "TheLastBillingMonth",
    WeekToDate: "WeekToDate",
    Custom: "Custom",
};
