// @pulumi/azure-native/synapse/v20201201/sqlPoolWorkloadClassifier.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlPoolWorkloadClassifier = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Workload classifier operations for a data warehouse
 */
class SqlPoolWorkloadClassifier extends pulumi.CustomResource {
    /**
     * Create a SqlPoolWorkloadClassifier resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.memberName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'memberName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sqlPoolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sqlPoolName'");
            }
            if ((!args || args.workloadGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workloadGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["context"] = args ? args.context : undefined;
            inputs["endTime"] = args ? args.endTime : undefined;
            inputs["importance"] = args ? args.importance : undefined;
            inputs["label"] = args ? args.label : undefined;
            inputs["memberName"] = args ? args.memberName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sqlPoolName"] = args ? args.sqlPoolName : undefined;
            inputs["startTime"] = args ? args.startTime : undefined;
            inputs["workloadClassifierName"] = args ? args.workloadClassifierName : undefined;
            inputs["workloadGroupName"] = args ? args.workloadGroupName : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["context"] = undefined /*out*/;
            inputs["endTime"] = undefined /*out*/;
            inputs["importance"] = undefined /*out*/;
            inputs["label"] = undefined /*out*/;
            inputs["memberName"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["startTime"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:synapse/v20201201:SqlPoolWorkloadClassifier" }, { type: "azure-native:synapse:SqlPoolWorkloadClassifier" }, { type: "azure-nextgen:synapse:SqlPoolWorkloadClassifier" }, { type: "azure-native:synapse/v20190601preview:SqlPoolWorkloadClassifier" }, { type: "azure-nextgen:synapse/v20190601preview:SqlPoolWorkloadClassifier" }, { type: "azure-native:synapse/v20210301:SqlPoolWorkloadClassifier" }, { type: "azure-nextgen:synapse/v20210301:SqlPoolWorkloadClassifier" }, { type: "azure-native:synapse/v20210401preview:SqlPoolWorkloadClassifier" }, { type: "azure-nextgen:synapse/v20210401preview:SqlPoolWorkloadClassifier" }, { type: "azure-native:synapse/v20210501:SqlPoolWorkloadClassifier" }, { type: "azure-nextgen:synapse/v20210501:SqlPoolWorkloadClassifier" }, { type: "azure-native:synapse/v20210601preview:SqlPoolWorkloadClassifier" }, { type: "azure-nextgen:synapse/v20210601preview:SqlPoolWorkloadClassifier" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlPoolWorkloadClassifier.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SqlPoolWorkloadClassifier resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SqlPoolWorkloadClassifier(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlPoolWorkloadClassifier.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlPoolWorkloadClassifier.__pulumiType;
    }
}
exports.SqlPoolWorkloadClassifier = SqlPoolWorkloadClassifier;
/** @internal */
SqlPoolWorkloadClassifier.__pulumiType = 'azure-native:synapse/v20201201:SqlPoolWorkloadClassifier';
