// @pulumi/azure-native/windowsesu/multipleActivationKey.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultipleActivationKey = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * MAK key details.
 * API Version: 2019-09-16-preview.
 */
class MultipleActivationKey extends pulumi.CustomResource {
    /**
     * Create a MultipleActivationKey resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["agreementNumber"] = args ? args.agreementNumber : undefined;
            inputs["installedServerNumber"] = args ? args.installedServerNumber : undefined;
            inputs["isEligible"] = args ? args.isEligible : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["multipleActivationKeyName"] = args ? args.multipleActivationKeyName : undefined;
            inputs["osType"] = args ? args.osType : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["supportType"] = (_a = (args ? args.supportType : undefined)) !== null && _a !== void 0 ? _a : "SupplementalServicing";
            inputs["tags"] = args ? args.tags : undefined;
            inputs["expirationDate"] = undefined /*out*/;
            inputs["multipleActivationKey"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["agreementNumber"] = undefined /*out*/;
            inputs["expirationDate"] = undefined /*out*/;
            inputs["installedServerNumber"] = undefined /*out*/;
            inputs["isEligible"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["multipleActivationKey"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["osType"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["supportType"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:windowsesu:MultipleActivationKey" }, { type: "azure-native:windowsesu/v20190916preview:MultipleActivationKey" }, { type: "azure-nextgen:windowsesu/v20190916preview:MultipleActivationKey" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MultipleActivationKey.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing MultipleActivationKey resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MultipleActivationKey(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MultipleActivationKey.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MultipleActivationKey.__pulumiType;
    }
}
exports.MultipleActivationKey = MultipleActivationKey;
/** @internal */
MultipleActivationKey.__pulumiType = 'azure-native:windowsesu:MultipleActivationKey';
