// @pulumi/azure-native/streamanalytics/v20170401preview/output.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Output = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An output object, containing all information associated with the named output. All outputs are contained under a streaming job.
 */
class Output extends pulumi.CustomResource {
    /**
     * Create a Output resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.jobName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'jobName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["datasource"] = args ? args.datasource : undefined;
            inputs["jobName"] = args ? args.jobName : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["outputName"] = args ? args.outputName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serialization"] = args ? args.serialization : undefined;
            inputs["sizeWindow"] = args ? args.sizeWindow : undefined;
            inputs["timeWindow"] = args ? args.timeWindow : undefined;
            inputs["diagnostics"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["datasource"] = undefined /*out*/;
            inputs["diagnostics"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["serialization"] = undefined /*out*/;
            inputs["sizeWindow"] = undefined /*out*/;
            inputs["timeWindow"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:streamanalytics/v20170401preview:Output" }, { type: "azure-native:streamanalytics:Output" }, { type: "azure-nextgen:streamanalytics:Output" }, { type: "azure-native:streamanalytics/v20160301:Output" }, { type: "azure-nextgen:streamanalytics/v20160301:Output" }, { type: "azure-native:streamanalytics/v20200301:Output" }, { type: "azure-nextgen:streamanalytics/v20200301:Output" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Output.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Output resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Output(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Output.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Output.__pulumiType;
    }
}
exports.Output = Output;
/** @internal */
Output.__pulumiType = 'azure-native:streamanalytics/v20170401preview:Output';
