// @pulumi/azure-native/web/v20210115/webAppSiteExtensionSlot.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppSiteExtensionSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Site Extension Information.
 */
class WebAppSiteExtensionSlot extends pulumi.CustomResource {
    /**
     * Create a WebAppSiteExtensionSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["siteExtensionId"] = args ? args.siteExtensionId : undefined;
            inputs["slot"] = args ? args.slot : undefined;
            inputs["authors"] = undefined /*out*/;
            inputs["comment"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["downloadCount"] = undefined /*out*/;
            inputs["extensionId"] = undefined /*out*/;
            inputs["extensionType"] = undefined /*out*/;
            inputs["extensionUrl"] = undefined /*out*/;
            inputs["feedUrl"] = undefined /*out*/;
            inputs["iconUrl"] = undefined /*out*/;
            inputs["installedDateTime"] = undefined /*out*/;
            inputs["installerCommandLineParams"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["licenseUrl"] = undefined /*out*/;
            inputs["localIsLatestVersion"] = undefined /*out*/;
            inputs["localPath"] = undefined /*out*/;
            inputs["projectUrl"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["publishedDateTime"] = undefined /*out*/;
            inputs["summary"] = undefined /*out*/;
            inputs["title"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        else {
            inputs["authors"] = undefined /*out*/;
            inputs["comment"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["downloadCount"] = undefined /*out*/;
            inputs["extensionId"] = undefined /*out*/;
            inputs["extensionType"] = undefined /*out*/;
            inputs["extensionUrl"] = undefined /*out*/;
            inputs["feedUrl"] = undefined /*out*/;
            inputs["iconUrl"] = undefined /*out*/;
            inputs["installedDateTime"] = undefined /*out*/;
            inputs["installerCommandLineParams"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["licenseUrl"] = undefined /*out*/;
            inputs["localIsLatestVersion"] = undefined /*out*/;
            inputs["localPath"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["projectUrl"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["publishedDateTime"] = undefined /*out*/;
            inputs["summary"] = undefined /*out*/;
            inputs["title"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20210115:WebAppSiteExtensionSlot" }, { type: "azure-native:web:WebAppSiteExtensionSlot" }, { type: "azure-nextgen:web:WebAppSiteExtensionSlot" }, { type: "azure-native:web/v20160801:WebAppSiteExtensionSlot" }, { type: "azure-nextgen:web/v20160801:WebAppSiteExtensionSlot" }, { type: "azure-native:web/v20180201:WebAppSiteExtensionSlot" }, { type: "azure-nextgen:web/v20180201:WebAppSiteExtensionSlot" }, { type: "azure-native:web/v20181101:WebAppSiteExtensionSlot" }, { type: "azure-nextgen:web/v20181101:WebAppSiteExtensionSlot" }, { type: "azure-native:web/v20190801:WebAppSiteExtensionSlot" }, { type: "azure-nextgen:web/v20190801:WebAppSiteExtensionSlot" }, { type: "azure-native:web/v20200601:WebAppSiteExtensionSlot" }, { type: "azure-nextgen:web/v20200601:WebAppSiteExtensionSlot" }, { type: "azure-native:web/v20200901:WebAppSiteExtensionSlot" }, { type: "azure-nextgen:web/v20200901:WebAppSiteExtensionSlot" }, { type: "azure-native:web/v20201001:WebAppSiteExtensionSlot" }, { type: "azure-nextgen:web/v20201001:WebAppSiteExtensionSlot" }, { type: "azure-native:web/v20201201:WebAppSiteExtensionSlot" }, { type: "azure-nextgen:web/v20201201:WebAppSiteExtensionSlot" }, { type: "azure-native:web/v20210101:WebAppSiteExtensionSlot" }, { type: "azure-nextgen:web/v20210101:WebAppSiteExtensionSlot" }, { type: "azure-native:web/v20210201:WebAppSiteExtensionSlot" }, { type: "azure-nextgen:web/v20210201:WebAppSiteExtensionSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppSiteExtensionSlot.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WebAppSiteExtensionSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppSiteExtensionSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppSiteExtensionSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppSiteExtensionSlot.__pulumiType;
    }
}
exports.WebAppSiteExtensionSlot = WebAppSiteExtensionSlot;
/** @internal */
WebAppSiteExtensionSlot.__pulumiType = 'azure-native:web/v20210115:WebAppSiteExtensionSlot';
