// @pulumi/azure-native/web/v20210115/webAppHostNameBinding.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppHostNameBinding = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A hostname binding object.
 */
class WebAppHostNameBinding extends pulumi.CustomResource {
    /**
     * Create a WebAppHostNameBinding resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["azureResourceName"] = args ? args.azureResourceName : undefined;
            inputs["azureResourceType"] = args ? args.azureResourceType : undefined;
            inputs["customHostNameDnsRecordType"] = args ? args.customHostNameDnsRecordType : undefined;
            inputs["domainId"] = args ? args.domainId : undefined;
            inputs["hostName"] = args ? args.hostName : undefined;
            inputs["hostNameType"] = args ? args.hostNameType : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["siteName"] = args ? args.siteName : undefined;
            inputs["sslState"] = args ? args.sslState : undefined;
            inputs["thumbprint"] = args ? args.thumbprint : undefined;
            inputs["type"] = undefined /*out*/;
            inputs["virtualIP"] = undefined /*out*/;
        }
        else {
            inputs["azureResourceName"] = undefined /*out*/;
            inputs["azureResourceType"] = undefined /*out*/;
            inputs["customHostNameDnsRecordType"] = undefined /*out*/;
            inputs["domainId"] = undefined /*out*/;
            inputs["hostNameType"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["siteName"] = undefined /*out*/;
            inputs["sslState"] = undefined /*out*/;
            inputs["thumbprint"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["virtualIP"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20210115:WebAppHostNameBinding" }, { type: "azure-native:web:WebAppHostNameBinding" }, { type: "azure-nextgen:web:WebAppHostNameBinding" }, { type: "azure-native:web/v20150801:WebAppHostNameBinding" }, { type: "azure-nextgen:web/v20150801:WebAppHostNameBinding" }, { type: "azure-native:web/v20160801:WebAppHostNameBinding" }, { type: "azure-nextgen:web/v20160801:WebAppHostNameBinding" }, { type: "azure-native:web/v20180201:WebAppHostNameBinding" }, { type: "azure-nextgen:web/v20180201:WebAppHostNameBinding" }, { type: "azure-native:web/v20181101:WebAppHostNameBinding" }, { type: "azure-nextgen:web/v20181101:WebAppHostNameBinding" }, { type: "azure-native:web/v20190801:WebAppHostNameBinding" }, { type: "azure-nextgen:web/v20190801:WebAppHostNameBinding" }, { type: "azure-native:web/v20200601:WebAppHostNameBinding" }, { type: "azure-nextgen:web/v20200601:WebAppHostNameBinding" }, { type: "azure-native:web/v20200901:WebAppHostNameBinding" }, { type: "azure-nextgen:web/v20200901:WebAppHostNameBinding" }, { type: "azure-native:web/v20201001:WebAppHostNameBinding" }, { type: "azure-nextgen:web/v20201001:WebAppHostNameBinding" }, { type: "azure-native:web/v20201201:WebAppHostNameBinding" }, { type: "azure-nextgen:web/v20201201:WebAppHostNameBinding" }, { type: "azure-native:web/v20210101:WebAppHostNameBinding" }, { type: "azure-nextgen:web/v20210101:WebAppHostNameBinding" }, { type: "azure-native:web/v20210201:WebAppHostNameBinding" }, { type: "azure-nextgen:web/v20210201:WebAppHostNameBinding" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppHostNameBinding.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WebAppHostNameBinding resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppHostNameBinding(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppHostNameBinding.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppHostNameBinding.__pulumiType;
    }
}
exports.WebAppHostNameBinding = WebAppHostNameBinding;
/** @internal */
WebAppHostNameBinding.__pulumiType = 'azure-native:web/v20210115:WebAppHostNameBinding';
