// @pulumi/azure-native/network/v20200401/ipAllocation.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IpAllocation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * IpAllocation resource.
 */
class IpAllocation extends pulumi.CustomResource {
    /**
     * Create a IpAllocation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["allocationTags"] = args ? args.allocationTags : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["ipAllocationName"] = args ? args.ipAllocationName : undefined;
            inputs["ipamAllocationId"] = args ? args.ipamAllocationId : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["prefix"] = args ? args.prefix : undefined;
            inputs["prefixLength"] = (_a = (args ? args.prefixLength : undefined)) !== null && _a !== void 0 ? _a : 0;
            inputs["prefixType"] = args ? args.prefixType : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["type"] = args ? args.type : undefined;
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["subnet"] = undefined /*out*/;
            inputs["virtualNetwork"] = undefined /*out*/;
        }
        else {
            inputs["allocationTags"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["ipamAllocationId"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["prefix"] = undefined /*out*/;
            inputs["prefixLength"] = undefined /*out*/;
            inputs["prefixType"] = undefined /*out*/;
            inputs["subnet"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["virtualNetwork"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20200401:IpAllocation" }, { type: "azure-native:network:IpAllocation" }, { type: "azure-nextgen:network:IpAllocation" }, { type: "azure-native:network/v20200301:IpAllocation" }, { type: "azure-nextgen:network/v20200301:IpAllocation" }, { type: "azure-native:network/v20200501:IpAllocation" }, { type: "azure-nextgen:network/v20200501:IpAllocation" }, { type: "azure-native:network/v20200601:IpAllocation" }, { type: "azure-nextgen:network/v20200601:IpAllocation" }, { type: "azure-native:network/v20200701:IpAllocation" }, { type: "azure-nextgen:network/v20200701:IpAllocation" }, { type: "azure-native:network/v20200801:IpAllocation" }, { type: "azure-nextgen:network/v20200801:IpAllocation" }, { type: "azure-native:network/v20201101:IpAllocation" }, { type: "azure-nextgen:network/v20201101:IpAllocation" }, { type: "azure-native:network/v20210201:IpAllocation" }, { type: "azure-nextgen:network/v20210201:IpAllocation" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IpAllocation.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing IpAllocation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IpAllocation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IpAllocation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IpAllocation.__pulumiType;
    }
}
exports.IpAllocation = IpAllocation;
/** @internal */
IpAllocation.__pulumiType = 'azure-native:network/v20200401:IpAllocation';
