// @pulumi/azure-native/types/enums/deploymentmanager/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StepType = exports.RestRequestMethod = exports.RestMatchQuantifier = exports.RestAuthType = exports.RestAuthLocation = exports.DeploymentMode = exports.v20191101preview = exports.v20180901preview = void 0;
// Export sub-modules:
const v20180901preview = require("./v20180901preview");
exports.v20180901preview = v20180901preview;
const v20191101preview = require("./v20191101preview");
exports.v20191101preview = v20191101preview;
exports.DeploymentMode = {
    Incremental: "Incremental",
    Complete: "Complete",
};
exports.RestAuthLocation = {
    Query: "Query",
    Header: "Header",
};
exports.RestAuthType = {
    ApiKey: "ApiKey",
    RolloutIdentity: "RolloutIdentity",
};
exports.RestMatchQuantifier = {
    All: "All",
    Any: "Any",
};
exports.RestRequestMethod = {
    GET: "GET",
    POST: "POST",
};
exports.StepType = {
    Wait: "Wait",
    HealthCheck: "HealthCheck",
};
