// @pulumi/azure-native/kusto/script.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Script = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Class representing a database script.
 * API Version: 2021-01-01.
 */
class Script extends pulumi.CustomResource {
    /**
     * Create a Script resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.scriptUrl === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scriptUrl'");
            }
            if ((!args || args.scriptUrlSasToken === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scriptUrlSasToken'");
            }
            inputs["clusterName"] = args ? args.clusterName : undefined;
            inputs["continueOnErrors"] = (_a = (args ? args.continueOnErrors : undefined)) !== null && _a !== void 0 ? _a : false;
            inputs["databaseName"] = args ? args.databaseName : undefined;
            inputs["forceUpdateTag"] = args ? args.forceUpdateTag : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["scriptName"] = args ? args.scriptName : undefined;
            inputs["scriptUrl"] = args ? args.scriptUrl : undefined;
            inputs["scriptUrlSasToken"] = args ? args.scriptUrlSasToken : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["continueOnErrors"] = undefined /*out*/;
            inputs["forceUpdateTag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["scriptUrl"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:kusto:Script" }, { type: "azure-native:kusto/v20210101:Script" }, { type: "azure-nextgen:kusto/v20210101:Script" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Script.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Script resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Script(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Script.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Script.__pulumiType;
    }
}
exports.Script = Script;
/** @internal */
Script.__pulumiType = 'azure-native:kusto:Script';
