// @pulumi/azure-native/logic/workflowAccessKey.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowAccessKey = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * API Version: 2015-02-01-preview.
 */
class WorkflowAccessKey extends pulumi.CustomResource {
    /**
     * Create a WorkflowAccessKey resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workflowName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workflowName'");
            }
            inputs["accessKeyName"] = args ? args.accessKeyName : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["notAfter"] = args ? args.notAfter : undefined;
            inputs["notBefore"] = args ? args.notBefore : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["workflowName"] = args ? args.workflowName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["name"] = undefined /*out*/;
            inputs["notAfter"] = undefined /*out*/;
            inputs["notBefore"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:logic:WorkflowAccessKey" }, { type: "azure-native:logic/v20150201preview:WorkflowAccessKey" }, { type: "azure-nextgen:logic/v20150201preview:WorkflowAccessKey" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WorkflowAccessKey.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WorkflowAccessKey resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WorkflowAccessKey(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WorkflowAccessKey.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkflowAccessKey.__pulumiType;
    }
}
exports.WorkflowAccessKey = WorkflowAccessKey;
/** @internal */
WorkflowAccessKey.__pulumiType = 'azure-native:logic:WorkflowAccessKey';
