// @pulumi/azure-native/types/enums/dbforpostgresql/v20210615privatepreview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.ServerVersion = exports.ResourceIdentityType = exports.HighAvailabilityMode = exports.GeoRedundantBackupEnum = exports.CreateMode = void 0;
exports.CreateMode = {
    Default: "Default",
    Create: "Create",
    Update: "Update",
    PointInTimeRestore: "PointInTimeRestore",
};
exports.GeoRedundantBackupEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.HighAvailabilityMode = {
    Disabled: "Disabled",
    ZoneRedundant: "ZoneRedundant",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.ServerVersion = {
    ServerVersion_13: "13",
    ServerVersion_12: "12",
    ServerVersion_11: "11",
};
exports.SkuTier = {
    Burstable: "Burstable",
    GeneralPurpose: "GeneralPurpose",
    MemoryOptimized: "MemoryOptimized",
};
