// @pulumi/azure-native/managednetwork/v20190601preview/managedNetworkGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedNetworkGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The Managed Network Group resource
 */
class ManagedNetworkGroup extends pulumi.CustomResource {
    /**
     * Create a ManagedNetworkGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.managedNetworkName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managedNetworkName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["kind"] = args ? args.kind : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["managedNetworkGroupName"] = args ? args.managedNetworkGroupName : undefined;
            inputs["managedNetworkName"] = args ? args.managedNetworkName : undefined;
            inputs["managementGroups"] = args ? args.managementGroups : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["subnets"] = args ? args.subnets : undefined;
            inputs["subscriptions"] = args ? args.subscriptions : undefined;
            inputs["virtualNetworks"] = args ? args.virtualNetworks : undefined;
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["etag"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["managementGroups"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["subnets"] = undefined /*out*/;
            inputs["subscriptions"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["virtualNetworks"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:managednetwork/v20190601preview:ManagedNetworkGroup" }, { type: "azure-native:managednetwork:ManagedNetworkGroup" }, { type: "azure-nextgen:managednetwork:ManagedNetworkGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagedNetworkGroup.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ManagedNetworkGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ManagedNetworkGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagedNetworkGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagedNetworkGroup.__pulumiType;
    }
}
exports.ManagedNetworkGroup = ManagedNetworkGroup;
/** @internal */
ManagedNetworkGroup.__pulumiType = 'azure-native:managednetwork/v20190601preview:ManagedNetworkGroup';
