// @pulumi/azure-native/insights/v20150501/analyticsItem.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnalyticsItem = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Properties that define an Analytics item that is associated to an Application Insights component.
 */
class AnalyticsItem extends pulumi.CustomResource {
    /**
     * Create a AnalyticsItem resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            if ((!args || args.scopePath === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scopePath'");
            }
            inputs["content"] = args ? args.content : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["overrideItem"] = args ? args.overrideItem : undefined;
            inputs["properties"] = args ? args.properties : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["resourceName"] = args ? args.resourceName : undefined;
            inputs["scope"] = args ? args.scope : undefined;
            inputs["scopePath"] = args ? args.scopePath : undefined;
            inputs["type"] = args ? args.type : undefined;
            inputs["timeCreated"] = undefined /*out*/;
            inputs["timeModified"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        else {
            inputs["content"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["scope"] = undefined /*out*/;
            inputs["timeCreated"] = undefined /*out*/;
            inputs["timeModified"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:insights/v20150501:AnalyticsItem" }, { type: "azure-native:insights:AnalyticsItem" }, { type: "azure-nextgen:insights:AnalyticsItem" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AnalyticsItem.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing AnalyticsItem resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AnalyticsItem(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AnalyticsItem.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AnalyticsItem.__pulumiType;
    }
}
exports.AnalyticsItem = AnalyticsItem;
/** @internal */
AnalyticsItem.__pulumiType = 'azure-native:insights/v20150501:AnalyticsItem';
