// @pulumi/azure-native/apimanagement/v20210101preview/subscription.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Subscription details.
 */
class Subscription extends pulumi.CustomResource {
    /**
     * Create a Subscription resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.scope === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scope'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            inputs["allowTracing"] = args ? args.allowTracing : undefined;
            inputs["appType"] = args ? args.appType : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["notify"] = args ? args.notify : undefined;
            inputs["ownerId"] = args ? args.ownerId : undefined;
            inputs["primaryKey"] = args ? args.primaryKey : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["scope"] = args ? args.scope : undefined;
            inputs["secondaryKey"] = args ? args.secondaryKey : undefined;
            inputs["serviceName"] = args ? args.serviceName : undefined;
            inputs["sid"] = args ? args.sid : undefined;
            inputs["state"] = args ? args.state : undefined;
            inputs["createdDate"] = undefined /*out*/;
            inputs["endDate"] = undefined /*out*/;
            inputs["expirationDate"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["notificationDate"] = undefined /*out*/;
            inputs["startDate"] = undefined /*out*/;
            inputs["stateComment"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["allowTracing"] = undefined /*out*/;
            inputs["createdDate"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["endDate"] = undefined /*out*/;
            inputs["expirationDate"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["notificationDate"] = undefined /*out*/;
            inputs["ownerId"] = undefined /*out*/;
            inputs["primaryKey"] = undefined /*out*/;
            inputs["scope"] = undefined /*out*/;
            inputs["secondaryKey"] = undefined /*out*/;
            inputs["startDate"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
            inputs["stateComment"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:apimanagement/v20210101preview:Subscription" }, { type: "azure-native:apimanagement:Subscription" }, { type: "azure-nextgen:apimanagement:Subscription" }, { type: "azure-native:apimanagement/v20160707:Subscription" }, { type: "azure-nextgen:apimanagement/v20160707:Subscription" }, { type: "azure-native:apimanagement/v20161010:Subscription" }, { type: "azure-nextgen:apimanagement/v20161010:Subscription" }, { type: "azure-native:apimanagement/v20170301:Subscription" }, { type: "azure-nextgen:apimanagement/v20170301:Subscription" }, { type: "azure-native:apimanagement/v20180101:Subscription" }, { type: "azure-nextgen:apimanagement/v20180101:Subscription" }, { type: "azure-native:apimanagement/v20180601preview:Subscription" }, { type: "azure-nextgen:apimanagement/v20180601preview:Subscription" }, { type: "azure-native:apimanagement/v20190101:Subscription" }, { type: "azure-nextgen:apimanagement/v20190101:Subscription" }, { type: "azure-native:apimanagement/v20191201:Subscription" }, { type: "azure-nextgen:apimanagement/v20191201:Subscription" }, { type: "azure-native:apimanagement/v20191201preview:Subscription" }, { type: "azure-nextgen:apimanagement/v20191201preview:Subscription" }, { type: "azure-native:apimanagement/v20200601preview:Subscription" }, { type: "azure-nextgen:apimanagement/v20200601preview:Subscription" }, { type: "azure-native:apimanagement/v20201201:Subscription" }, { type: "azure-nextgen:apimanagement/v20201201:Subscription" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Subscription.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Subscription resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Subscription(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Subscription.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Subscription.__pulumiType;
    }
}
exports.Subscription = Subscription;
/** @internal */
Subscription.__pulumiType = 'azure-native:apimanagement/v20210101preview:Subscription';
