// @pulumi/azure-native/types/enums/videoanalyzer/v20210501preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountEncryptionKeyType = exports.AccessPolicyRsaAlgo = exports.AccessPolicyRole = exports.AccessPolicyEccAlgo = void 0;
exports.AccessPolicyEccAlgo = {
    /**
     * ES265
     */
    ES256: "ES256",
    /**
     * ES384
     */
    ES384: "ES384",
    /**
     * ES512
     */
    ES512: "ES512",
};
exports.AccessPolicyRole = {
    /**
     * Reader role allows for read-only operations to be performed through the client APIs.
     */
    Reader: "Reader",
};
exports.AccessPolicyRsaAlgo = {
    /**
     * RS256
     */
    RS256: "RS256",
    /**
     * RS384
     */
    RS384: "RS384",
    /**
     * RS512
     */
    RS512: "RS512",
};
exports.AccountEncryptionKeyType = {
    /**
     * The Account Key is encrypted with a System Key.
     */
    SystemKey: "SystemKey",
    /**
     * The Account Key is encrypted with a Customer Key.
     */
    CustomerKey: "CustomerKey",
};
