// @pulumi/azure-native/machinelearningservices/v20200515preview/endpointVariant.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointVariant = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Machine Learning service object wrapped into ARM resource envelope.
 */
class EndpointVariant extends pulumi.CustomResource {
    /**
     * Create a EndpointVariant resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.computeType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'computeType'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["computeType"] = "Custom";
            inputs["description"] = args ? args.description : undefined;
            inputs["environmentImageRequest"] = args ? args.environmentImageRequest : undefined;
            inputs["isDefault"] = args ? args.isDefault : undefined;
            inputs["keys"] = args ? args.keys : undefined;
            inputs["kvTags"] = args ? args.kvTags : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["properties"] = args ? args.properties : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serviceName"] = args ? args.serviceName : undefined;
            inputs["trafficPercentile"] = args ? args.trafficPercentile : undefined;
            inputs["type"] = args ? args.type : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["identity"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
        }
        else {
            inputs["identity"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:machinelearningservices/v20200515preview:EndpointVariant" }, { type: "azure-native:machinelearningservices:EndpointVariant" }, { type: "azure-nextgen:machinelearningservices:EndpointVariant" }, { type: "azure-native:machinelearningservices/v20200501preview:EndpointVariant" }, { type: "azure-nextgen:machinelearningservices/v20200501preview:EndpointVariant" }, { type: "azure-native:machinelearningservices/v20200901preview:EndpointVariant" }, { type: "azure-nextgen:machinelearningservices/v20200901preview:EndpointVariant" }, { type: "azure-native:machinelearningservices/v20210101:EndpointVariant" }, { type: "azure-nextgen:machinelearningservices/v20210101:EndpointVariant" }, { type: "azure-native:machinelearningservices/v20210401:EndpointVariant" }, { type: "azure-nextgen:machinelearningservices/v20210401:EndpointVariant" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(EndpointVariant.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing EndpointVariant resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EndpointVariant(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EndpointVariant.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EndpointVariant.__pulumiType;
    }
}
exports.EndpointVariant = EndpointVariant;
/** @internal */
EndpointVariant.__pulumiType = 'azure-native:machinelearningservices/v20200515preview:EndpointVariant';
