// @pulumi/azure-native/synapse/v20210601preview/attachedDatabaseConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttachedDatabaseConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Class representing an attached database configuration.
 */
class AttachedDatabaseConfiguration extends pulumi.CustomResource {
    /**
     * Create a AttachedDatabaseConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterResourceId'");
            }
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.defaultPrincipalsModificationKind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'defaultPrincipalsModificationKind'");
            }
            if ((!args || args.kustoPoolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kustoPoolName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["attachedDatabaseConfigurationName"] = args ? args.attachedDatabaseConfigurationName : undefined;
            inputs["clusterResourceId"] = args ? args.clusterResourceId : undefined;
            inputs["databaseName"] = args ? args.databaseName : undefined;
            inputs["defaultPrincipalsModificationKind"] = args ? args.defaultPrincipalsModificationKind : undefined;
            inputs["kustoPoolName"] = args ? args.kustoPoolName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tableLevelSharingProperties"] = args ? args.tableLevelSharingProperties : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["attachedDatabaseNames"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["attachedDatabaseNames"] = undefined /*out*/;
            inputs["clusterResourceId"] = undefined /*out*/;
            inputs["databaseName"] = undefined /*out*/;
            inputs["defaultPrincipalsModificationKind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tableLevelSharingProperties"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:synapse/v20210601preview:AttachedDatabaseConfiguration" }, { type: "azure-native:synapse:AttachedDatabaseConfiguration" }, { type: "azure-nextgen:synapse:AttachedDatabaseConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AttachedDatabaseConfiguration.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing AttachedDatabaseConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AttachedDatabaseConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AttachedDatabaseConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AttachedDatabaseConfiguration.__pulumiType;
    }
}
exports.AttachedDatabaseConfiguration = AttachedDatabaseConfiguration;
/** @internal */
AttachedDatabaseConfiguration.__pulumiType = 'azure-native:synapse/v20210601preview:AttachedDatabaseConfiguration';
