// @pulumi/azure-native/network/v20170901/zone.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Zone = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Describes a DNS zone.
 */
class Zone extends pulumi.CustomResource {
    /**
     * Create a Zone resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["etag"] = args ? args.etag : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["zoneName"] = args ? args.zoneName : undefined;
            inputs["zoneType"] = (_a = (args ? args.zoneType : undefined)) !== null && _a !== void 0 ? _a : "Public";
            inputs["maxNumberOfRecordSets"] = undefined /*out*/;
            inputs["maxNumberOfRecordsPerRecordSet"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["nameServers"] = undefined /*out*/;
            inputs["numberOfRecordSets"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["etag"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["maxNumberOfRecordSets"] = undefined /*out*/;
            inputs["maxNumberOfRecordsPerRecordSet"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["nameServers"] = undefined /*out*/;
            inputs["numberOfRecordSets"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["zoneType"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20170901:Zone" }, { type: "azure-native:network:Zone" }, { type: "azure-nextgen:network:Zone" }, { type: "azure-native:network/v20150504preview:Zone" }, { type: "azure-nextgen:network/v20150504preview:Zone" }, { type: "azure-native:network/v20160401:Zone" }, { type: "azure-nextgen:network/v20160401:Zone" }, { type: "azure-native:network/v20171001:Zone" }, { type: "azure-nextgen:network/v20171001:Zone" }, { type: "azure-native:network/v20180301preview:Zone" }, { type: "azure-nextgen:network/v20180301preview:Zone" }, { type: "azure-native:network/v20180501:Zone" }, { type: "azure-nextgen:network/v20180501:Zone" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Zone.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Zone resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Zone(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Zone.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Zone.__pulumiType;
    }
}
exports.Zone = Zone;
/** @internal */
Zone.__pulumiType = 'azure-native:network/v20170901:Zone';
