// @pulumi/azure-native/automation/v20151031/module.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Module = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Definition of the module type.
 */
class Module extends pulumi.CustomResource {
    /**
     * Create a Module resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.automationAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'automationAccountName'");
            }
            if ((!args || args.contentLink === undefined) && !opts.urn) {
                throw new Error("Missing required property 'contentLink'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["automationAccountName"] = args ? args.automationAccountName : undefined;
            inputs["contentLink"] = args ? args.contentLink : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["moduleName"] = args ? args.moduleName : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["activityCount"] = undefined /*out*/;
            inputs["creationTime"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["error"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["isComposite"] = undefined /*out*/;
            inputs["isGlobal"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["sizeInBytes"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        else {
            inputs["activityCount"] = undefined /*out*/;
            inputs["contentLink"] = undefined /*out*/;
            inputs["creationTime"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["error"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["isComposite"] = undefined /*out*/;
            inputs["isGlobal"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["sizeInBytes"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:automation/v20151031:Module" }, { type: "azure-native:automation:Module" }, { type: "azure-nextgen:automation:Module" }, { type: "azure-native:automation/v20190601:Module" }, { type: "azure-nextgen:automation/v20190601:Module" }, { type: "azure-native:automation/v20200113preview:Module" }, { type: "azure-nextgen:automation/v20200113preview:Module" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Module.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Module resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Module(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Module.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Module.__pulumiType;
    }
}
exports.Module = Module;
/** @internal */
Module.__pulumiType = 'azure-native:automation/v20151031:Module';
