// @pulumi/azure-native/types/enums/containerregistry/v20191201preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookStatus = exports.WebhookAction = exports.TrustPolicyType = exports.TriggerStatus = exports.SkuName = exports.ResourceIdentityType = exports.PublicNetworkAccess = exports.PolicyStatus = exports.PipelineSourceType = exports.PipelineRunTargetType = exports.PipelineRunSourceType = exports.PipelineOptions = exports.NetworkRuleBypassOptions = exports.EncryptionStatus = exports.DefaultAction = exports.ConnectionStatus = exports.ActionsRequired = exports.Action = void 0;
exports.Action = {
    Allow: "Allow",
};
exports.ActionsRequired = {
    None: "None",
    Recreate: "Recreate",
};
exports.ConnectionStatus = {
    Approved: "Approved",
    Pending: "Pending",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.DefaultAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.EncryptionStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.NetworkRuleBypassOptions = {
    AzureServices: "AzureServices",
    None: "None",
};
exports.PipelineOptions = {
    OverwriteTags: "OverwriteTags",
    OverwriteBlobs: "OverwriteBlobs",
    DeleteSourceBlobOnSuccess: "DeleteSourceBlobOnSuccess",
    ContinueOnErrors: "ContinueOnErrors",
};
exports.PipelineRunSourceType = {
    AzureStorageBlob: "AzureStorageBlob",
};
exports.PipelineRunTargetType = {
    AzureStorageBlob: "AzureStorageBlob",
};
exports.PipelineSourceType = {
    AzureStorageBlobContainer: "AzureStorageBlobContainer",
};
exports.PolicyStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.SkuName = {
    Classic: "Classic",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.TriggerStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.TrustPolicyType = {
    Notary: "Notary",
};
exports.WebhookAction = {
    Push: "push",
    Delete: "delete",
    Quarantine: "quarantine",
    Chart_push: "chart_push",
    Chart_delete: "chart_delete",
};
exports.WebhookStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
