// @pulumi/azure-native/types/enums/powerbi/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceProvisioningState = exports.PersistedConnectionStatus = exports.AzureSkuTier = exports.AzureSkuName = exports.v20200601 = exports.v20160129 = void 0;
// Export sub-modules:
const v20160129 = require("./v20160129");
exports.v20160129 = v20160129;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
exports.AzureSkuName = {
    S1: "S1",
};
exports.AzureSkuTier = {
    Standard: "Standard",
};
exports.PersistedConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.ResourceProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
