// @pulumi/azure-native/management/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210401 = exports.v20201001 = exports.v20200501 = exports.v20200201 = exports.v20191101 = exports.v20180301preview = exports.v20180101preview = exports.v20171101preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./getEntity"), exports);
__exportStar(require("./getHierarchySetting"), exports);
__exportStar(require("./getManagementGroup"), exports);
__exportStar(require("./getManagementGroupSubscription"), exports);
__exportStar(require("./hierarchySetting"), exports);
__exportStar(require("./managementGroup"), exports);
__exportStar(require("./managementGroupSubscription"), exports);
// Export sub-modules:
const v20171101preview = require("./v20171101preview");
exports.v20171101preview = v20171101preview;
const v20180101preview = require("./v20180101preview");
exports.v20180101preview = v20180101preview;
const v20180301preview = require("./v20180301preview");
exports.v20180301preview = v20180301preview;
const v20191101 = require("./v20191101");
exports.v20191101 = v20191101;
const v20200201 = require("./v20200201");
exports.v20200201 = v20200201;
const v20200501 = require("./v20200501");
exports.v20200501 = v20200501;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20210401 = require("./v20210401");
exports.v20210401 = v20210401;
// Import resources to register:
const hierarchySetting_1 = require("./hierarchySetting");
const managementGroup_1 = require("./managementGroup");
const managementGroupSubscription_1 = require("./managementGroupSubscription");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:management:HierarchySetting":
                return new hierarchySetting_1.HierarchySetting(name, undefined, { urn });
            case "azure-native:management:ManagementGroup":
                return new managementGroup_1.ManagementGroup(name, undefined, { urn });
            case "azure-native:management:ManagementGroupSubscription":
                return new managementGroupSubscription_1.ManagementGroupSubscription(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "management", _module);
