// @pulumi/azure-native/search/v20200801preview/sharedPrivateLinkResource.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedPrivateLinkResource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Describes a Shared Private Link Resource managed by the Azure Cognitive Search service.
 */
class SharedPrivateLinkResource extends pulumi.CustomResource {
    /**
     * Create a SharedPrivateLinkResource resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.searchServiceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'searchServiceName'");
            }
            inputs["properties"] = args ? args.properties : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["searchServiceName"] = args ? args.searchServiceName : undefined;
            inputs["sharedPrivateLinkResourceName"] = args ? args.sharedPrivateLinkResourceName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:search/v20200801preview:SharedPrivateLinkResource" }, { type: "azure-native:search:SharedPrivateLinkResource" }, { type: "azure-nextgen:search:SharedPrivateLinkResource" }, { type: "azure-native:search/v20200801:SharedPrivateLinkResource" }, { type: "azure-nextgen:search/v20200801:SharedPrivateLinkResource" }, { type: "azure-native:search/v20210401preview:SharedPrivateLinkResource" }, { type: "azure-nextgen:search/v20210401preview:SharedPrivateLinkResource" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SharedPrivateLinkResource.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SharedPrivateLinkResource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SharedPrivateLinkResource(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SharedPrivateLinkResource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SharedPrivateLinkResource.__pulumiType;
    }
}
exports.SharedPrivateLinkResource = SharedPrivateLinkResource;
/** @internal */
SharedPrivateLinkResource.__pulumiType = 'azure-native:search/v20200801preview:SharedPrivateLinkResource';
