// @pulumi/azure-native/redhatopenshift/openShiftCluster.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenShiftCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * OpenShiftCluster represents an Azure Red Hat OpenShift cluster.
 * API Version: 2020-04-30.
 */
class OpenShiftCluster extends pulumi.CustomResource {
    /**
     * Create a OpenShiftCluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["apiserverProfile"] = args ? args.apiserverProfile : undefined;
            inputs["clusterProfile"] = args ? args.clusterProfile : undefined;
            inputs["consoleProfile"] = args ? args.consoleProfile : undefined;
            inputs["ingressProfiles"] = args ? args.ingressProfiles : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["masterProfile"] = args ? args.masterProfile : undefined;
            inputs["networkProfile"] = args ? args.networkProfile : undefined;
            inputs["provisioningState"] = args ? args.provisioningState : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["resourceName"] = args ? args.resourceName : undefined;
            inputs["servicePrincipalProfile"] = args ? args.servicePrincipalProfile : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["workerProfiles"] = args ? args.workerProfiles : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["apiserverProfile"] = undefined /*out*/;
            inputs["clusterProfile"] = undefined /*out*/;
            inputs["consoleProfile"] = undefined /*out*/;
            inputs["ingressProfiles"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["masterProfile"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["networkProfile"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["servicePrincipalProfile"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["workerProfiles"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:redhatopenshift:OpenShiftCluster" }, { type: "azure-native:redhatopenshift/v20200430:OpenShiftCluster" }, { type: "azure-nextgen:redhatopenshift/v20200430:OpenShiftCluster" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(OpenShiftCluster.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing OpenShiftCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new OpenShiftCluster(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of OpenShiftCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OpenShiftCluster.__pulumiType;
    }
}
exports.OpenShiftCluster = OpenShiftCluster;
/** @internal */
OpenShiftCluster.__pulumiType = 'azure-native:redhatopenshift:OpenShiftCluster';
