// @pulumi/azure-native/apimanagement/gatewayHostnameConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GatewayHostnameConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Gateway hostname configuration details.
 * API Version: 2020-12-01.
 */
class GatewayHostnameConfiguration extends pulumi.CustomResource {
    /**
     * Create a GatewayHostnameConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.gatewayId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'gatewayId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            inputs["certificateId"] = args ? args.certificateId : undefined;
            inputs["gatewayId"] = args ? args.gatewayId : undefined;
            inputs["hcId"] = args ? args.hcId : undefined;
            inputs["hostname"] = args ? args.hostname : undefined;
            inputs["http2Enabled"] = args ? args.http2Enabled : undefined;
            inputs["negotiateClientCertificate"] = args ? args.negotiateClientCertificate : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serviceName"] = args ? args.serviceName : undefined;
            inputs["tls10Enabled"] = args ? args.tls10Enabled : undefined;
            inputs["tls11Enabled"] = args ? args.tls11Enabled : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["certificateId"] = undefined /*out*/;
            inputs["hostname"] = undefined /*out*/;
            inputs["http2Enabled"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["negotiateClientCertificate"] = undefined /*out*/;
            inputs["tls10Enabled"] = undefined /*out*/;
            inputs["tls11Enabled"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:apimanagement:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20191201:GatewayHostnameConfiguration" }, { type: "azure-nextgen:apimanagement/v20191201:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20191201preview:GatewayHostnameConfiguration" }, { type: "azure-nextgen:apimanagement/v20191201preview:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20200601preview:GatewayHostnameConfiguration" }, { type: "azure-nextgen:apimanagement/v20200601preview:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20201201:GatewayHostnameConfiguration" }, { type: "azure-nextgen:apimanagement/v20201201:GatewayHostnameConfiguration" }, { type: "azure-native:apimanagement/v20210101preview:GatewayHostnameConfiguration" }, { type: "azure-nextgen:apimanagement/v20210101preview:GatewayHostnameConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GatewayHostnameConfiguration.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing GatewayHostnameConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GatewayHostnameConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GatewayHostnameConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GatewayHostnameConfiguration.__pulumiType;
    }
}
exports.GatewayHostnameConfiguration = GatewayHostnameConfiguration;
/** @internal */
GatewayHostnameConfiguration.__pulumiType = 'azure-native:apimanagement:GatewayHostnameConfiguration';
