// @pulumi/azure-native/apimanagement/gatewayApiEntityTag.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GatewayApiEntityTag = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Api details.
 * API Version: 2020-12-01.
 */
class GatewayApiEntityTag extends pulumi.CustomResource {
    /**
     * Create a GatewayApiEntityTag resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.gatewayId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'gatewayId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            inputs["apiId"] = args ? args.apiId : undefined;
            inputs["gatewayId"] = args ? args.gatewayId : undefined;
            inputs["provisioningState"] = args ? args.provisioningState : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serviceName"] = args ? args.serviceName : undefined;
            inputs["apiRevision"] = undefined /*out*/;
            inputs["apiRevisionDescription"] = undefined /*out*/;
            inputs["apiType"] = undefined /*out*/;
            inputs["apiVersion"] = undefined /*out*/;
            inputs["apiVersionDescription"] = undefined /*out*/;
            inputs["apiVersionSet"] = undefined /*out*/;
            inputs["apiVersionSetId"] = undefined /*out*/;
            inputs["authenticationSettings"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["isCurrent"] = undefined /*out*/;
            inputs["isOnline"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["path"] = undefined /*out*/;
            inputs["protocols"] = undefined /*out*/;
            inputs["serviceUrl"] = undefined /*out*/;
            inputs["sourceApiId"] = undefined /*out*/;
            inputs["subscriptionKeyParameterNames"] = undefined /*out*/;
            inputs["subscriptionRequired"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["apiRevision"] = undefined /*out*/;
            inputs["apiRevisionDescription"] = undefined /*out*/;
            inputs["apiType"] = undefined /*out*/;
            inputs["apiVersion"] = undefined /*out*/;
            inputs["apiVersionDescription"] = undefined /*out*/;
            inputs["apiVersionSet"] = undefined /*out*/;
            inputs["apiVersionSetId"] = undefined /*out*/;
            inputs["authenticationSettings"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["isCurrent"] = undefined /*out*/;
            inputs["isOnline"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["path"] = undefined /*out*/;
            inputs["protocols"] = undefined /*out*/;
            inputs["serviceUrl"] = undefined /*out*/;
            inputs["sourceApiId"] = undefined /*out*/;
            inputs["subscriptionKeyParameterNames"] = undefined /*out*/;
            inputs["subscriptionRequired"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:apimanagement:GatewayApiEntityTag" }, { type: "azure-native:apimanagement/v20191201:GatewayApiEntityTag" }, { type: "azure-nextgen:apimanagement/v20191201:GatewayApiEntityTag" }, { type: "azure-native:apimanagement/v20191201preview:GatewayApiEntityTag" }, { type: "azure-nextgen:apimanagement/v20191201preview:GatewayApiEntityTag" }, { type: "azure-native:apimanagement/v20200601preview:GatewayApiEntityTag" }, { type: "azure-nextgen:apimanagement/v20200601preview:GatewayApiEntityTag" }, { type: "azure-native:apimanagement/v20201201:GatewayApiEntityTag" }, { type: "azure-nextgen:apimanagement/v20201201:GatewayApiEntityTag" }, { type: "azure-native:apimanagement/v20210101preview:GatewayApiEntityTag" }, { type: "azure-nextgen:apimanagement/v20210101preview:GatewayApiEntityTag" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GatewayApiEntityTag.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing GatewayApiEntityTag resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GatewayApiEntityTag(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GatewayApiEntityTag.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GatewayApiEntityTag.__pulumiType;
    }
}
exports.GatewayApiEntityTag = GatewayApiEntityTag;
/** @internal */
GatewayApiEntityTag.__pulumiType = 'azure-native:apimanagement:GatewayApiEntityTag';
