// @pulumi/azure-native/eventgrid/partnerNamespace.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartnerNamespace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * EventGrid Partner Namespace.
 * API Version: 2021-06-01-preview.
 */
class PartnerNamespace extends pulumi.CustomResource {
    /**
     * Create a PartnerNamespace resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["disableLocalAuth"] = (_a = (args ? args.disableLocalAuth : undefined)) !== null && _a !== void 0 ? _a : false;
            inputs["inboundIpRules"] = args ? args.inboundIpRules : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["partnerNamespaceName"] = args ? args.partnerNamespaceName : undefined;
            inputs["partnerRegistrationFullyQualifiedId"] = args ? args.partnerRegistrationFullyQualifiedId : undefined;
            inputs["publicNetworkAccess"] = (_b = (args ? args.publicNetworkAccess : undefined)) !== null && _b !== void 0 ? _b : "Enabled";
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["endpoint"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["privateEndpointConnections"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["disableLocalAuth"] = undefined /*out*/;
            inputs["endpoint"] = undefined /*out*/;
            inputs["inboundIpRules"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["partnerRegistrationFullyQualifiedId"] = undefined /*out*/;
            inputs["privateEndpointConnections"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["publicNetworkAccess"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:eventgrid:PartnerNamespace" }, { type: "azure-native:eventgrid/v20200401preview:PartnerNamespace" }, { type: "azure-nextgen:eventgrid/v20200401preview:PartnerNamespace" }, { type: "azure-native:eventgrid/v20201015preview:PartnerNamespace" }, { type: "azure-nextgen:eventgrid/v20201015preview:PartnerNamespace" }, { type: "azure-native:eventgrid/v20210601preview:PartnerNamespace" }, { type: "azure-nextgen:eventgrid/v20210601preview:PartnerNamespace" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PartnerNamespace.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing PartnerNamespace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PartnerNamespace(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PartnerNamespace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PartnerNamespace.__pulumiType;
    }
}
exports.PartnerNamespace = PartnerNamespace;
/** @internal */
PartnerNamespace.__pulumiType = 'azure-native:eventgrid:PartnerNamespace';
