// @pulumi/azure-native/desktopvirtualization/v20201019preview/msixpackage.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MSIXPackage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Schema for MSIX Package properties.
 */
class MSIXPackage extends pulumi.CustomResource {
    /**
     * Create a MSIXPackage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.hostPoolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hostPoolName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["hostPoolName"] = args ? args.hostPoolName : undefined;
            inputs["imagePath"] = args ? args.imagePath : undefined;
            inputs["isActive"] = args ? args.isActive : undefined;
            inputs["isRegularRegistration"] = args ? args.isRegularRegistration : undefined;
            inputs["lastUpdated"] = args ? args.lastUpdated : undefined;
            inputs["msixPackageFullName"] = args ? args.msixPackageFullName : undefined;
            inputs["packageApplications"] = args ? args.packageApplications : undefined;
            inputs["packageDependencies"] = args ? args.packageDependencies : undefined;
            inputs["packageFamilyName"] = args ? args.packageFamilyName : undefined;
            inputs["packageName"] = args ? args.packageName : undefined;
            inputs["packageRelativePath"] = args ? args.packageRelativePath : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["version"] = args ? args.version : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["displayName"] = undefined /*out*/;
            inputs["imagePath"] = undefined /*out*/;
            inputs["isActive"] = undefined /*out*/;
            inputs["isRegularRegistration"] = undefined /*out*/;
            inputs["lastUpdated"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["packageApplications"] = undefined /*out*/;
            inputs["packageDependencies"] = undefined /*out*/;
            inputs["packageFamilyName"] = undefined /*out*/;
            inputs["packageName"] = undefined /*out*/;
            inputs["packageRelativePath"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:desktopvirtualization/v20201019preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization:MSIXPackage" }, { type: "azure-nextgen:desktopvirtualization:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20200921preview:MSIXPackage" }, { type: "azure-nextgen:desktopvirtualization/v20200921preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20201102preview:MSIXPackage" }, { type: "azure-nextgen:desktopvirtualization/v20201102preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20201110preview:MSIXPackage" }, { type: "azure-nextgen:desktopvirtualization/v20201110preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20210114preview:MSIXPackage" }, { type: "azure-nextgen:desktopvirtualization/v20210114preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20210201preview:MSIXPackage" }, { type: "azure-nextgen:desktopvirtualization/v20210201preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20210309preview:MSIXPackage" }, { type: "azure-nextgen:desktopvirtualization/v20210309preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20210401preview:MSIXPackage" }, { type: "azure-nextgen:desktopvirtualization/v20210401preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20210513preview:MSIXPackage" }, { type: "azure-nextgen:desktopvirtualization/v20210513preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20210712:MSIXPackage" }, { type: "azure-nextgen:desktopvirtualization/v20210712:MSIXPackage" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MSIXPackage.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing MSIXPackage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MSIXPackage(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MSIXPackage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MSIXPackage.__pulumiType;
    }
}
exports.MSIXPackage = MSIXPackage;
/** @internal */
MSIXPackage.__pulumiType = 'azure-native:desktopvirtualization/v20201019preview:MSIXPackage';
