// @pulumi/azure-native/types/enums/scheduler/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuDefinition = exports.ServiceBusTransportType = exports.ServiceBusAuthenticationType = exports.RetryType = exports.RecurrenceFrequency = exports.JobState = exports.JobScheduleDay = exports.JobCollectionState = exports.JobActionType = exports.HttpAuthenticationType = exports.DayOfWeek = exports.v20160301 = exports.v20160101 = exports.v20140801preview = void 0;
// Export sub-modules:
const v20140801preview = require("./v20140801preview");
exports.v20140801preview = v20140801preview;
const v20160101 = require("./v20160101");
exports.v20160101 = v20160101;
const v20160301 = require("./v20160301");
exports.v20160301 = v20160301;
exports.DayOfWeek = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
exports.HttpAuthenticationType = {
    NotSpecified: "NotSpecified",
    ClientCertificate: "ClientCertificate",
    ActiveDirectoryOAuth: "ActiveDirectoryOAuth",
    Basic: "Basic",
};
exports.JobActionType = {
    Http: "Http",
    Https: "Https",
    StorageQueue: "StorageQueue",
    ServiceBusQueue: "ServiceBusQueue",
    ServiceBusTopic: "ServiceBusTopic",
};
exports.JobCollectionState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
    Suspended: "Suspended",
    Deleted: "Deleted",
};
exports.JobScheduleDay = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.JobState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
    Faulted: "Faulted",
    Completed: "Completed",
};
exports.RecurrenceFrequency = {
    Minute: "Minute",
    Hour: "Hour",
    Day: "Day",
    Week: "Week",
    Month: "Month",
};
exports.RetryType = {
    None: "None",
    Fixed: "Fixed",
};
exports.ServiceBusAuthenticationType = {
    NotSpecified: "NotSpecified",
    SharedAccessKey: "SharedAccessKey",
};
exports.ServiceBusTransportType = {
    NotSpecified: "NotSpecified",
    NetMessaging: "NetMessaging",
    AMQP: "AMQP",
};
exports.SkuDefinition = {
    Standard: "Standard",
    Free: "Free",
    P10Premium: "P10Premium",
    P20Premium: "P20Premium",
};
