// @pulumi/azure-native/types/enums/relay/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.SkuName = exports.Relaytype = exports.PrivateEndpointServiceConnectionStatus = exports.AccessRights = exports.v20180101preview = exports.v20170401 = exports.v20160701 = void 0;
// Export sub-modules:
const v20160701 = require("./v20160701");
exports.v20160701 = v20160701;
const v20170401 = require("./v20170401");
exports.v20170401 = v20170401;
const v20180101preview = require("./v20180101preview");
exports.v20180101preview = v20180101preview;
exports.AccessRights = {
    Manage: "Manage",
    Send: "Send",
    Listen: "Listen",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.Relaytype = {
    NetTcp: "NetTcp",
    Http: "Http",
};
exports.SkuName = {
    Standard: "Standard",
};
exports.SkuTier = {
    Standard: "Standard",
};
