// @pulumi/azure-native/databox/v20210501/job.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Job = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Job Resource.
 */
class Job extends pulumi.CustomResource {
    /**
     * Create a Job resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            if ((!args || args.transferType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'transferType'");
            }
            inputs["deliveryInfo"] = args ? args.deliveryInfo : undefined;
            inputs["deliveryType"] = (_a = (args ? args.deliveryType : undefined)) !== null && _a !== void 0 ? _a : "NonScheduled";
            inputs["details"] = args ? args.details : undefined;
            inputs["identity"] = args ? args.identity : undefined;
            inputs["jobName"] = args ? args.jobName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["transferType"] = args ? args.transferType : undefined;
            inputs["cancellationReason"] = undefined /*out*/;
            inputs["error"] = undefined /*out*/;
            inputs["isCancellable"] = undefined /*out*/;
            inputs["isCancellableWithoutFee"] = undefined /*out*/;
            inputs["isDeletable"] = undefined /*out*/;
            inputs["isPrepareToShipEnabled"] = undefined /*out*/;
            inputs["isShippingAddressEditable"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["startTime"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["cancellationReason"] = undefined /*out*/;
            inputs["deliveryInfo"] = undefined /*out*/;
            inputs["deliveryType"] = undefined /*out*/;
            inputs["details"] = undefined /*out*/;
            inputs["error"] = undefined /*out*/;
            inputs["identity"] = undefined /*out*/;
            inputs["isCancellable"] = undefined /*out*/;
            inputs["isCancellableWithoutFee"] = undefined /*out*/;
            inputs["isDeletable"] = undefined /*out*/;
            inputs["isPrepareToShipEnabled"] = undefined /*out*/;
            inputs["isShippingAddressEditable"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["startTime"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["transferType"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:databox/v20210501:Job" }, { type: "azure-native:databox:Job" }, { type: "azure-nextgen:databox:Job" }, { type: "azure-native:databox/v20180101:Job" }, { type: "azure-nextgen:databox/v20180101:Job" }, { type: "azure-native:databox/v20190901:Job" }, { type: "azure-nextgen:databox/v20190901:Job" }, { type: "azure-native:databox/v20200401:Job" }, { type: "azure-nextgen:databox/v20200401:Job" }, { type: "azure-native:databox/v20201101:Job" }, { type: "azure-nextgen:databox/v20201101:Job" }, { type: "azure-native:databox/v20210301:Job" }, { type: "azure-nextgen:databox/v20210301:Job" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Job.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Job resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Job(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Job.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Job.__pulumiType;
    }
}
exports.Job = Job;
/** @internal */
Job.__pulumiType = 'azure-native:databox/v20210501:Job';
