// @pulumi/azure-native/solutions/v20160901preview/applianceDefinition.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplianceDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Information about appliance definition.
 */
class ApplianceDefinition extends pulumi.CustomResource {
    /**
     * Create a ApplianceDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.authorizations === undefined) && !opts.urn) {
                throw new Error("Missing required property 'authorizations'");
            }
            if ((!args || args.lockLevel === undefined) && !opts.urn) {
                throw new Error("Missing required property 'lockLevel'");
            }
            if ((!args || args.packageFileUri === undefined) && !opts.urn) {
                throw new Error("Missing required property 'packageFileUri'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["applianceDefinitionName"] = args ? args.applianceDefinitionName : undefined;
            inputs["artifacts"] = args ? args.artifacts : undefined;
            inputs["authorizations"] = args ? args.authorizations : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["identity"] = args ? args.identity : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["lockLevel"] = args ? args.lockLevel : undefined;
            inputs["managedBy"] = args ? args.managedBy : undefined;
            inputs["packageFileUri"] = args ? args.packageFileUri : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["artifacts"] = undefined /*out*/;
            inputs["authorizations"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["identity"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["lockLevel"] = undefined /*out*/;
            inputs["managedBy"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["packageFileUri"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:solutions/v20160901preview:ApplianceDefinition" }, { type: "azure-native:solutions:ApplianceDefinition" }, { type: "azure-nextgen:solutions:ApplianceDefinition" }, { type: "azure-native:solutions/v20170901:ApplianceDefinition" }, { type: "azure-nextgen:solutions/v20170901:ApplianceDefinition" }, { type: "azure-native:solutions/v20180601:ApplianceDefinition" }, { type: "azure-nextgen:solutions/v20180601:ApplianceDefinition" }, { type: "azure-native:solutions/v20190701:ApplianceDefinition" }, { type: "azure-nextgen:solutions/v20190701:ApplianceDefinition" }, { type: "azure-native:solutions/v20200821preview:ApplianceDefinition" }, { type: "azure-nextgen:solutions/v20200821preview:ApplianceDefinition" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ApplianceDefinition.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ApplianceDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ApplianceDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ApplianceDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ApplianceDefinition.__pulumiType;
    }
}
exports.ApplianceDefinition = ApplianceDefinition;
/** @internal */
ApplianceDefinition.__pulumiType = 'azure-native:solutions/v20160901preview:ApplianceDefinition';
