// @pulumi/azure-native/sql/v20170301preview/backupLongTermRetentionPolicy.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackupLongTermRetentionPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A long term retention policy.
 */
class BackupLongTermRetentionPolicy extends pulumi.CustomResource {
    /**
     * Create a BackupLongTermRetentionPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serverName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverName'");
            }
            inputs["databaseName"] = args ? args.databaseName : undefined;
            inputs["monthlyRetention"] = args ? args.monthlyRetention : undefined;
            inputs["policyName"] = args ? args.policyName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serverName"] = args ? args.serverName : undefined;
            inputs["weekOfYear"] = args ? args.weekOfYear : undefined;
            inputs["weeklyRetention"] = args ? args.weeklyRetention : undefined;
            inputs["yearlyRetention"] = args ? args.yearlyRetention : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["monthlyRetention"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["weekOfYear"] = undefined /*out*/;
            inputs["weeklyRetention"] = undefined /*out*/;
            inputs["yearlyRetention"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:sql/v20170301preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql:BackupLongTermRetentionPolicy" }, { type: "azure-nextgen:sql:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20200202preview:BackupLongTermRetentionPolicy" }, { type: "azure-nextgen:sql/v20200202preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20200801preview:BackupLongTermRetentionPolicy" }, { type: "azure-nextgen:sql/v20200801preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20201101preview:BackupLongTermRetentionPolicy" }, { type: "azure-nextgen:sql/v20201101preview:BackupLongTermRetentionPolicy" }, { type: "azure-native:sql/v20210201preview:BackupLongTermRetentionPolicy" }, { type: "azure-nextgen:sql/v20210201preview:BackupLongTermRetentionPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BackupLongTermRetentionPolicy.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing BackupLongTermRetentionPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BackupLongTermRetentionPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BackupLongTermRetentionPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BackupLongTermRetentionPolicy.__pulumiType;
    }
}
exports.BackupLongTermRetentionPolicy = BackupLongTermRetentionPolicy;
/** @internal */
BackupLongTermRetentionPolicy.__pulumiType = 'azure-native:sql/v20170301preview:BackupLongTermRetentionPolicy';
