// @pulumi/azure-native/types/enums/costmanagement/v20190901/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeframeType = exports.StatusType = exports.SortDirection = exports.RecurrenceType = exports.QueryColumnType = exports.OperatorType = exports.GranularityType = exports.FunctionType = exports.FormatType = exports.ExportType = void 0;
exports.ExportType = {
    Usage: "Usage",
};
exports.FormatType = {
    Csv: "Csv",
};
exports.FunctionType = {
    Sum: "Sum",
};
exports.GranularityType = {
    Daily: "Daily",
    Hourly: "Hourly",
};
exports.OperatorType = {
    In: "In",
};
exports.QueryColumnType = {
    Tag: "Tag",
    Dimension: "Dimension",
};
exports.RecurrenceType = {
    Daily: "Daily",
    Weekly: "Weekly",
    Monthly: "Monthly",
    Annually: "Annually",
};
exports.SortDirection = {
    Ascending: "Ascending",
    Descending: "Descending",
};
exports.StatusType = {
    Active: "Active",
    Inactive: "Inactive",
};
exports.TimeframeType = {
    WeekToDate: "WeekToDate",
    MonthToDate: "MonthToDate",
    YearToDate: "YearToDate",
    TheLastWeek: "TheLastWeek",
    TheLastMonth: "TheLastMonth",
    TheLastYear: "TheLastYear",
    Custom: "Custom",
};
