// @pulumi/azure-native/documentdb/v20190801/cassandraResourceCassandraKeyspace.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CassandraResourceCassandraKeyspace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An Azure Cosmos DB Cassandra keyspace.
 */
class CassandraResourceCassandraKeyspace extends pulumi.CustomResource {
    /**
     * Create a CassandraResourceCassandraKeyspace resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.options === undefined) && !opts.urn) {
                throw new Error("Missing required property 'options'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["keyspaceName"] = args ? args.keyspaceName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["options"] = args ? args.options : undefined;
            inputs["resource"] = args ? args.resource : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["resource"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:documentdb/v20190801:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20150401:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20150401:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20150408:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20150408:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20151106:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20151106:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20160319:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20160319:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20160331:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20160331:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20191212:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20191212:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20200301:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20200301:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20200401:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20200401:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20200601preview:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20200601preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20200901:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20200901:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210115:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20210115:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210301preview:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20210301preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210315:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20210315:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210401preview:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20210401preview:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210415:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20210415:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210515:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20210515:CassandraResourceCassandraKeyspace" }, { type: "azure-native:documentdb/v20210615:CassandraResourceCassandraKeyspace" }, { type: "azure-nextgen:documentdb/v20210615:CassandraResourceCassandraKeyspace" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CassandraResourceCassandraKeyspace.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing CassandraResourceCassandraKeyspace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CassandraResourceCassandraKeyspace(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CassandraResourceCassandraKeyspace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CassandraResourceCassandraKeyspace.__pulumiType;
    }
}
exports.CassandraResourceCassandraKeyspace = CassandraResourceCassandraKeyspace;
/** @internal */
CassandraResourceCassandraKeyspace.__pulumiType = 'azure-native:documentdb/v20190801:CassandraResourceCassandraKeyspace';
