// @pulumi/azure-native/documentdb/notebookWorkspace.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotebookWorkspace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A notebook workspace resource
 * API Version: 2021-03-15.
 */
class NotebookWorkspace extends pulumi.CustomResource {
    /**
     * Create a NotebookWorkspace resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["notebookWorkspaceName"] = args ? args.notebookWorkspaceName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["notebookServerEndpoint"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["name"] = undefined /*out*/;
            inputs["notebookServerEndpoint"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:documentdb:NotebookWorkspace" }, { type: "azure-native:documentdb/v20190801:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20190801:NotebookWorkspace" }, { type: "azure-native:documentdb/v20191212:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20191212:NotebookWorkspace" }, { type: "azure-native:documentdb/v20200301:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20200301:NotebookWorkspace" }, { type: "azure-native:documentdb/v20200401:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20200401:NotebookWorkspace" }, { type: "azure-native:documentdb/v20200601preview:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20200601preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20200901:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20200901:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210115:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20210115:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210301preview:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20210301preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210315:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20210315:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210401preview:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20210401preview:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210415:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20210415:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210515:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20210515:NotebookWorkspace" }, { type: "azure-native:documentdb/v20210615:NotebookWorkspace" }, { type: "azure-nextgen:documentdb/v20210615:NotebookWorkspace" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NotebookWorkspace.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing NotebookWorkspace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NotebookWorkspace(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NotebookWorkspace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NotebookWorkspace.__pulumiType;
    }
}
exports.NotebookWorkspace = NotebookWorkspace;
/** @internal */
NotebookWorkspace.__pulumiType = 'azure-native:documentdb:NotebookWorkspace';
