// @pulumi/azure-native/powerplatform/v20201030preview/enterprisePolicy.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnterprisePolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Definition of the EnterprisePolicy.
 */
class EnterprisePolicy extends pulumi.CustomResource {
    /**
     * Create a EnterprisePolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["encryption"] = args ? args.encryption : undefined;
            inputs["enterprisePolicyName"] = args ? args.enterprisePolicyName : undefined;
            inputs["identity"] = args ? args.identity : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["lockbox"] = args ? args.lockbox : undefined;
            inputs["networkInjection"] = args ? args.networkInjection : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["encryption"] = undefined /*out*/;
            inputs["identity"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["lockbox"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["networkInjection"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:powerplatform/v20201030preview:EnterprisePolicy" }, { type: "azure-native:powerplatform:EnterprisePolicy" }, { type: "azure-nextgen:powerplatform:EnterprisePolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(EnterprisePolicy.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing EnterprisePolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EnterprisePolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EnterprisePolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EnterprisePolicy.__pulumiType;
    }
}
exports.EnterprisePolicy = EnterprisePolicy;
/** @internal */
EnterprisePolicy.__pulumiType = 'azure-native:powerplatform/v20201030preview:EnterprisePolicy';
