// @pulumi/azure-native/documentdb/v20200401/mongoDBResourceMongoDBCollection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MongoDBResourceMongoDBCollection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An Azure Cosmos DB MongoDB collection.
 */
class MongoDBResourceMongoDBCollection extends pulumi.CustomResource {
    /**
     * Create a MongoDBResourceMongoDBCollection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.options === undefined) && !opts.urn) {
                throw new Error("Missing required property 'options'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["collectionName"] = args ? args.collectionName : undefined;
            inputs["databaseName"] = args ? args.databaseName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["options"] = args ? args.options : undefined;
            inputs["resource"] = args ? args.resource : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["options"] = undefined /*out*/;
            inputs["resource"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:documentdb/v20200401:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20150401:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20150401:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20150408:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20150408:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20151106:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20151106:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20160319:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20160319:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20160331:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20160331:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20190801:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20190801:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20191212:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20191212:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20200301:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20200301:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20200601preview:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20200601preview:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20200901:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20200901:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20210115:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20210115:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20210301preview:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20210301preview:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20210315:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20210315:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20210401preview:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20210401preview:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20210415:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20210415:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20210515:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20210515:MongoDBResourceMongoDBCollection" }, { type: "azure-native:documentdb/v20210615:MongoDBResourceMongoDBCollection" }, { type: "azure-nextgen:documentdb/v20210615:MongoDBResourceMongoDBCollection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MongoDBResourceMongoDBCollection.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing MongoDBResourceMongoDBCollection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MongoDBResourceMongoDBCollection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MongoDBResourceMongoDBCollection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MongoDBResourceMongoDBCollection.__pulumiType;
    }
}
exports.MongoDBResourceMongoDBCollection = MongoDBResourceMongoDBCollection;
/** @internal */
MongoDBResourceMongoDBCollection.__pulumiType = 'azure-native:documentdb/v20200401:MongoDBResourceMongoDBCollection';
