// @pulumi/azure-native/insights/exportConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExportConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Properties that define a Continuous Export configuration.
 * API Version: 2015-05-01.
 */
class ExportConfiguration extends pulumi.CustomResource {
    /**
     * Create a ExportConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            inputs["destinationAccountId"] = args ? args.destinationAccountId : undefined;
            inputs["destinationAddress"] = args ? args.destinationAddress : undefined;
            inputs["destinationStorageLocationId"] = args ? args.destinationStorageLocationId : undefined;
            inputs["destinationStorageSubscriptionId"] = args ? args.destinationStorageSubscriptionId : undefined;
            inputs["destinationType"] = args ? args.destinationType : undefined;
            inputs["exportId"] = args ? args.exportId : undefined;
            inputs["isEnabled"] = args ? args.isEnabled : undefined;
            inputs["notificationQueueEnabled"] = args ? args.notificationQueueEnabled : undefined;
            inputs["notificationQueueUri"] = args ? args.notificationQueueUri : undefined;
            inputs["recordTypes"] = args ? args.recordTypes : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["resourceName"] = args ? args.resourceName : undefined;
            inputs["applicationName"] = undefined /*out*/;
            inputs["containerName"] = undefined /*out*/;
            inputs["exportStatus"] = undefined /*out*/;
            inputs["instrumentationKey"] = undefined /*out*/;
            inputs["isUserEnabled"] = undefined /*out*/;
            inputs["lastGapTime"] = undefined /*out*/;
            inputs["lastSuccessTime"] = undefined /*out*/;
            inputs["lastUserUpdate"] = undefined /*out*/;
            inputs["permanentErrorReason"] = undefined /*out*/;
            inputs["resourceGroup"] = undefined /*out*/;
            inputs["storageName"] = undefined /*out*/;
            inputs["subscriptionId"] = undefined /*out*/;
        }
        else {
            inputs["applicationName"] = undefined /*out*/;
            inputs["containerName"] = undefined /*out*/;
            inputs["destinationAccountId"] = undefined /*out*/;
            inputs["destinationStorageLocationId"] = undefined /*out*/;
            inputs["destinationStorageSubscriptionId"] = undefined /*out*/;
            inputs["destinationType"] = undefined /*out*/;
            inputs["exportId"] = undefined /*out*/;
            inputs["exportStatus"] = undefined /*out*/;
            inputs["instrumentationKey"] = undefined /*out*/;
            inputs["isUserEnabled"] = undefined /*out*/;
            inputs["lastGapTime"] = undefined /*out*/;
            inputs["lastSuccessTime"] = undefined /*out*/;
            inputs["lastUserUpdate"] = undefined /*out*/;
            inputs["notificationQueueEnabled"] = undefined /*out*/;
            inputs["permanentErrorReason"] = undefined /*out*/;
            inputs["recordTypes"] = undefined /*out*/;
            inputs["resourceGroup"] = undefined /*out*/;
            inputs["storageName"] = undefined /*out*/;
            inputs["subscriptionId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:insights:ExportConfiguration" }, { type: "azure-native:insights/v20150501:ExportConfiguration" }, { type: "azure-nextgen:insights/v20150501:ExportConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ExportConfiguration.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ExportConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ExportConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ExportConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExportConfiguration.__pulumiType;
    }
}
exports.ExportConfiguration = ExportConfiguration;
/** @internal */
ExportConfiguration.__pulumiType = 'azure-native:insights:ExportConfiguration';
