// @pulumi/azure-native/types/enums/databox/v20210301/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransportShipmentTypes = exports.TransferType = exports.TransferConfigurationType = exports.SkuName = exports.NotificationStageName = exports.LogCollectionLevel = exports.KekType = exports.JobDeliveryType = exports.FilterFileType = exports.DoubleEncryption = exports.DataAccountType = exports.ClassDiscriminator = exports.AddressType = void 0;
exports.AddressType = {
    /**
     * Address type not known.
     */
    None: "None",
    /**
     * Residential Address.
     */
    Residential: "Residential",
    /**
     * Commercial Address.
     */
    Commercial: "Commercial",
};
exports.ClassDiscriminator = {
    /**
     * Data Box orders.
     */
    DataBox: "DataBox",
    /**
     * Data Box Disk orders.
     */
    DataBoxDisk: "DataBoxDisk",
    /**
     * Data Box Heavy orders.
     */
    DataBoxHeavy: "DataBoxHeavy",
};
exports.DataAccountType = {
    /**
     * Storage Accounts .
     */
    StorageAccount: "StorageAccount",
    /**
     * Azure Managed disk storage.
     */
    ManagedDisk: "ManagedDisk",
};
exports.DoubleEncryption = {
    /**
     * Software-based encryption is enabled.
     */
    Enabled: "Enabled",
    /**
     * Software-based encryption is disabled.
     */
    Disabled: "Disabled",
};
exports.FilterFileType = {
    /**
     * Filter file is of the type AzureBlob.
     */
    AzureBlob: "AzureBlob",
    /**
     * Filter file is of the type AzureFiles.
     */
    AzureFile: "AzureFile",
};
exports.JobDeliveryType = {
    /**
     * Non Scheduled job.
     */
    NonScheduled: "NonScheduled",
    /**
     * Scheduled job.
     */
    Scheduled: "Scheduled",
};
exports.KekType = {
    /**
     * Key encryption key is managed by Microsoft.
     */
    MicrosoftManaged: "MicrosoftManaged",
    /**
     * Key encryption key is managed by the Customer.
     */
    CustomerManaged: "CustomerManaged",
};
exports.LogCollectionLevel = {
    /**
     * Only Errors will be collected in the logs.
     */
    Error: "Error",
    /**
     * Verbose logging (includes Errors, CRC, size information and others).
     */
    Verbose: "Verbose",
};
exports.NotificationStageName = {
    /**
     * Notification at device prepared stage.
     */
    DevicePrepared: "DevicePrepared",
    /**
     * Notification at device dispatched stage.
     */
    Dispatched: "Dispatched",
    /**
     * Notification at device delivered stage.
     */
    Delivered: "Delivered",
    /**
     * Notification at device picked up from user stage.
     */
    PickedUp: "PickedUp",
    /**
     * Notification at device received at Azure datacenter stage.
     */
    AtAzureDC: "AtAzureDC",
    /**
     * Notification at data copy started stage.
     */
    DataCopy: "DataCopy",
};
exports.SkuName = {
    /**
     * Data Box.
     */
    DataBox: "DataBox",
    /**
     * Data Box Disk.
     */
    DataBoxDisk: "DataBoxDisk",
    /**
     * Data Box Heavy.
     */
    DataBoxHeavy: "DataBoxHeavy",
};
exports.TransferConfigurationType = {
    /**
     * Transfer all the data.
     */
    TransferAll: "TransferAll",
    /**
     * Transfer using filter.
     */
    TransferUsingFilter: "TransferUsingFilter",
};
exports.TransferType = {
    /**
     * Import data to azure.
     */
    ImportToAzure: "ImportToAzure",
    /**
     * Export data from azure.
     */
    ExportFromAzure: "ExportFromAzure",
};
exports.TransportShipmentTypes = {
    /**
     * Shipment Logistics is handled by the customer.
     */
    CustomerManaged: "CustomerManaged",
    /**
     * Shipment Logistics is handled by Microsoft.
     */
    MicrosoftManaged: "MicrosoftManaged",
};
