// @pulumi/azure-native/advisor/v20160712preview/suppression.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Suppression = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The details of the snoozed or dismissed rule; for example, the duration, name, and GUID associated with the rule.
 */
class Suppression extends pulumi.CustomResource {
    /**
     * Create a Suppression resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.recommendationId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'recommendationId'");
            }
            if ((!args || args.resourceUri === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceUri'");
            }
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["recommendationId"] = args ? args.recommendationId : undefined;
            inputs["resourceUri"] = args ? args.resourceUri : undefined;
            inputs["suppressionId"] = args ? args.suppressionId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["ttl"] = args ? args.ttl : undefined;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["suppressionId"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["ttl"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:advisor/v20160712preview:Suppression" }, { type: "azure-native:advisor:Suppression" }, { type: "azure-nextgen:advisor:Suppression" }, { type: "azure-native:advisor/v20170331:Suppression" }, { type: "azure-nextgen:advisor/v20170331:Suppression" }, { type: "azure-native:advisor/v20170419:Suppression" }, { type: "azure-nextgen:advisor/v20170419:Suppression" }, { type: "azure-native:advisor/v20200101:Suppression" }, { type: "azure-nextgen:advisor/v20200101:Suppression" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Suppression.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Suppression resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Suppression(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Suppression.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Suppression.__pulumiType;
    }
}
exports.Suppression = Suppression;
/** @internal */
Suppression.__pulumiType = 'azure-native:advisor/v20160712preview:Suppression';
