// @pulumi/azure-native/sql/v20200801preview/serverAdvisor.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerAdvisor = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Database, Server or Elastic Pool Advisor.
 */
class ServerAdvisor extends pulumi.CustomResource {
    /**
     * Create a ServerAdvisor resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.autoExecuteStatus === undefined) && !opts.urn) {
                throw new Error("Missing required property 'autoExecuteStatus'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serverName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverName'");
            }
            inputs["advisorName"] = args ? args.advisorName : undefined;
            inputs["autoExecuteStatus"] = args ? args.autoExecuteStatus : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serverName"] = args ? args.serverName : undefined;
            inputs["advisorStatus"] = undefined /*out*/;
            inputs["autoExecuteStatusInheritedFrom"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["lastChecked"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["recommendationsStatus"] = undefined /*out*/;
            inputs["recommendedActions"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["advisorStatus"] = undefined /*out*/;
            inputs["autoExecuteStatus"] = undefined /*out*/;
            inputs["autoExecuteStatusInheritedFrom"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["lastChecked"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["recommendationsStatus"] = undefined /*out*/;
            inputs["recommendedActions"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:sql/v20200801preview:ServerAdvisor" }, { type: "azure-native:sql:ServerAdvisor" }, { type: "azure-nextgen:sql:ServerAdvisor" }, { type: "azure-native:sql/v20140401:ServerAdvisor" }, { type: "azure-nextgen:sql/v20140401:ServerAdvisor" }, { type: "azure-native:sql/v20150501preview:ServerAdvisor" }, { type: "azure-nextgen:sql/v20150501preview:ServerAdvisor" }, { type: "azure-native:sql/v20200202preview:ServerAdvisor" }, { type: "azure-nextgen:sql/v20200202preview:ServerAdvisor" }, { type: "azure-native:sql/v20201101preview:ServerAdvisor" }, { type: "azure-nextgen:sql/v20201101preview:ServerAdvisor" }, { type: "azure-native:sql/v20210201preview:ServerAdvisor" }, { type: "azure-nextgen:sql/v20210201preview:ServerAdvisor" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ServerAdvisor.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ServerAdvisor resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ServerAdvisor(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ServerAdvisor.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServerAdvisor.__pulumiType;
    }
}
exports.ServerAdvisor = ServerAdvisor;
/** @internal */
ServerAdvisor.__pulumiType = 'azure-native:sql/v20200801preview:ServerAdvisor';
