// @pulumi/azure-native/sql/v20200801preview/syncGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyncGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An Azure SQL Database sync group.
 */
class SyncGroup extends pulumi.CustomResource {
    /**
     * Create a SyncGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serverName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverName'");
            }
            inputs["conflictLoggingRetentionInDays"] = args ? args.conflictLoggingRetentionInDays : undefined;
            inputs["conflictResolutionPolicy"] = args ? args.conflictResolutionPolicy : undefined;
            inputs["databaseName"] = args ? args.databaseName : undefined;
            inputs["enableConflictLogging"] = args ? args.enableConflictLogging : undefined;
            inputs["hubDatabasePassword"] = args ? args.hubDatabasePassword : undefined;
            inputs["hubDatabaseUserName"] = args ? args.hubDatabaseUserName : undefined;
            inputs["interval"] = args ? args.interval : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["schema"] = args ? args.schema : undefined;
            inputs["serverName"] = args ? args.serverName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["syncDatabaseId"] = args ? args.syncDatabaseId : undefined;
            inputs["syncGroupName"] = args ? args.syncGroupName : undefined;
            inputs["usePrivateLinkConnection"] = args ? args.usePrivateLinkConnection : undefined;
            inputs["lastSyncTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["privateEndpointName"] = undefined /*out*/;
            inputs["syncState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["conflictLoggingRetentionInDays"] = undefined /*out*/;
            inputs["conflictResolutionPolicy"] = undefined /*out*/;
            inputs["enableConflictLogging"] = undefined /*out*/;
            inputs["hubDatabaseUserName"] = undefined /*out*/;
            inputs["interval"] = undefined /*out*/;
            inputs["lastSyncTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["privateEndpointName"] = undefined /*out*/;
            inputs["schema"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["syncDatabaseId"] = undefined /*out*/;
            inputs["syncState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["usePrivateLinkConnection"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:sql/v20200801preview:SyncGroup" }, { type: "azure-native:sql:SyncGroup" }, { type: "azure-nextgen:sql:SyncGroup" }, { type: "azure-native:sql/v20150501preview:SyncGroup" }, { type: "azure-nextgen:sql/v20150501preview:SyncGroup" }, { type: "azure-native:sql/v20190601preview:SyncGroup" }, { type: "azure-nextgen:sql/v20190601preview:SyncGroup" }, { type: "azure-native:sql/v20200202preview:SyncGroup" }, { type: "azure-nextgen:sql/v20200202preview:SyncGroup" }, { type: "azure-native:sql/v20201101preview:SyncGroup" }, { type: "azure-nextgen:sql/v20201101preview:SyncGroup" }, { type: "azure-native:sql/v20210201preview:SyncGroup" }, { type: "azure-nextgen:sql/v20210201preview:SyncGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SyncGroup.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SyncGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SyncGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SyncGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SyncGroup.__pulumiType;
    }
}
exports.SyncGroup = SyncGroup;
/** @internal */
SyncGroup.__pulumiType = 'azure-native:sql/v20200801preview:SyncGroup';
