// @pulumi/azure-native/iotsecurity/sensor.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Sensor = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * IoT sensor model
 * API Version: 2021-02-01-preview.
 */
class Sensor extends pulumi.CustomResource {
    /**
     * Create a Sensor resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.scope === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scope'");
            }
            inputs["scope"] = args ? args.scope : undefined;
            inputs["sensorName"] = args ? args.sensorName : undefined;
            inputs["sensorType"] = args ? args.sensorType : undefined;
            inputs["tiAutomaticUpdates"] = args ? args.tiAutomaticUpdates : undefined;
            inputs["zone"] = args ? args.zone : undefined;
            inputs["connectivityTime"] = undefined /*out*/;
            inputs["dynamicLearning"] = undefined /*out*/;
            inputs["learningMode"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["sensorStatus"] = undefined /*out*/;
            inputs["sensorVersion"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tiStatus"] = undefined /*out*/;
            inputs["tiVersion"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["connectivityTime"] = undefined /*out*/;
            inputs["dynamicLearning"] = undefined /*out*/;
            inputs["learningMode"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["sensorStatus"] = undefined /*out*/;
            inputs["sensorType"] = undefined /*out*/;
            inputs["sensorVersion"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tiAutomaticUpdates"] = undefined /*out*/;
            inputs["tiStatus"] = undefined /*out*/;
            inputs["tiVersion"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["zone"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:iotsecurity:Sensor" }, { type: "azure-native:iotsecurity/v20210201preview:Sensor" }, { type: "azure-nextgen:iotsecurity/v20210201preview:Sensor" }, { type: "azure-native:iotsecurity/v20210901preview:Sensor" }, { type: "azure-nextgen:iotsecurity/v20210901preview:Sensor" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Sensor.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Sensor resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Sensor(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Sensor.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Sensor.__pulumiType;
    }
}
exports.Sensor = Sensor;
/** @internal */
Sensor.__pulumiType = 'azure-native:iotsecurity:Sensor';
