// @pulumi/azure-native/types/enums/netapp/v20201101/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevel = exports.SecurityStyle = exports.ReplicationSchedule = exports.QosType = exports.KeySource = exports.EndpointType = void 0;
exports.EndpointType = {
    Src: "src",
    Dst: "dst",
};
exports.KeySource = {
    /**
     * The service manages the keys.
     */
    Microsoft_NetApp: "Microsoft.NetApp",
};
exports.QosType = {
    /**
     * qos type Auto
     */
    Auto: "Auto",
    /**
     * qos type Manual
     */
    Manual: "Manual",
};
exports.ReplicationSchedule = {
    ReplicationSchedule_10minutely: "_10minutely",
    Hourly: "hourly",
    Daily: "daily",
};
exports.SecurityStyle = {
    Ntfs: "ntfs",
    Unix: "unix",
};
exports.ServiceLevel = {
    /**
     * Standard service level
     */
    Standard: "Standard",
    /**
     * Premium service level
     */
    Premium: "Premium",
    /**
     * Ultra service level
     */
    Ultra: "Ultra",
};
