// @pulumi/azure-native/securityinsights/v20210301preview/tiTaxiiDataConnector.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TiTaxiiDataConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Data connector to pull Threat intelligence data from TAXII 2.0/2.1 server
 */
class TiTaxiiDataConnector extends pulumi.CustomResource {
    /**
     * Create a TiTaxiiDataConnector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dataTypes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataTypes'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.operationalInsightsResourceProvider === undefined) && !opts.urn) {
                throw new Error("Missing required property 'operationalInsightsResourceProvider'");
            }
            if ((!args || args.pollingFrequency === undefined) && !opts.urn) {
                throw new Error("Missing required property 'pollingFrequency'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.tenantId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'tenantId'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["collectionId"] = args ? args.collectionId : undefined;
            inputs["dataConnectorId"] = args ? args.dataConnectorId : undefined;
            inputs["dataTypes"] = args ? args.dataTypes : undefined;
            inputs["etag"] = args ? args.etag : undefined;
            inputs["friendlyName"] = args ? args.friendlyName : undefined;
            inputs["kind"] = "ThreatIntelligenceTaxii";
            inputs["operationalInsightsResourceProvider"] = args ? args.operationalInsightsResourceProvider : undefined;
            inputs["password"] = args ? args.password : undefined;
            inputs["pollingFrequency"] = args ? args.pollingFrequency : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["taxiiLookbackPeriod"] = args ? args.taxiiLookbackPeriod : undefined;
            inputs["taxiiServer"] = args ? args.taxiiServer : undefined;
            inputs["tenantId"] = args ? args.tenantId : undefined;
            inputs["userName"] = args ? args.userName : undefined;
            inputs["workspaceId"] = args ? args.workspaceId : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["collectionId"] = undefined /*out*/;
            inputs["dataTypes"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["friendlyName"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["password"] = undefined /*out*/;
            inputs["pollingFrequency"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["taxiiLookbackPeriod"] = undefined /*out*/;
            inputs["taxiiServer"] = undefined /*out*/;
            inputs["tenantId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["userName"] = undefined /*out*/;
            inputs["workspaceId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:securityinsights/v20210301preview:TiTaxiiDataConnector" }, { type: "azure-native:securityinsights:TiTaxiiDataConnector" }, { type: "azure-nextgen:securityinsights:TiTaxiiDataConnector" }, { type: "azure-native:securityinsights/v20190101preview:TiTaxiiDataConnector" }, { type: "azure-nextgen:securityinsights/v20190101preview:TiTaxiiDataConnector" }, { type: "azure-native:securityinsights/v20200101:TiTaxiiDataConnector" }, { type: "azure-nextgen:securityinsights/v20200101:TiTaxiiDataConnector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(TiTaxiiDataConnector.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing TiTaxiiDataConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TiTaxiiDataConnector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TiTaxiiDataConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TiTaxiiDataConnector.__pulumiType;
    }
}
exports.TiTaxiiDataConnector = TiTaxiiDataConnector;
/** @internal */
TiTaxiiDataConnector.__pulumiType = 'azure-native:securityinsights/v20210301preview:TiTaxiiDataConnector';
