// @pulumi/azure-native/automanage/configurationProfileAssignment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigurationProfileAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Configuration profile assignment is an association between a VM and automanage profile configuration.
 * API Version: 2020-06-30-preview.
 */
class ConfigurationProfileAssignment extends pulumi.CustomResource {
    /**
     * Create a ConfigurationProfileAssignment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.vmName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vmName'");
            }
            inputs["configurationProfileAssignmentName"] = args ? args.configurationProfileAssignmentName : undefined;
            inputs["properties"] = args ? args.properties : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["vmName"] = args ? args.vmName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:automanage:ConfigurationProfileAssignment" }, { type: "azure-native:automanage/v20200630preview:ConfigurationProfileAssignment" }, { type: "azure-nextgen:automanage/v20200630preview:ConfigurationProfileAssignment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ConfigurationProfileAssignment.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ConfigurationProfileAssignment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ConfigurationProfileAssignment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ConfigurationProfileAssignment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConfigurationProfileAssignment.__pulumiType;
    }
}
exports.ConfigurationProfileAssignment = ConfigurationProfileAssignment;
/** @internal */
ConfigurationProfileAssignment.__pulumiType = 'azure-native:automanage:ConfigurationProfileAssignment';
