// @pulumi/azure-native/peering/v20200401/peeringService.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PeeringService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Peering Service
 */
class PeeringService extends pulumi.CustomResource {
    /**
     * Create a PeeringService resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["location"] = args ? args.location : undefined;
            inputs["peeringServiceLocation"] = args ? args.peeringServiceLocation : undefined;
            inputs["peeringServiceName"] = args ? args.peeringServiceName : undefined;
            inputs["peeringServiceProvider"] = args ? args.peeringServiceProvider : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["peeringServiceLocation"] = undefined /*out*/;
            inputs["peeringServiceProvider"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:peering/v20200401:PeeringService" }, { type: "azure-native:peering:PeeringService" }, { type: "azure-nextgen:peering:PeeringService" }, { type: "azure-native:peering/v20190801preview:PeeringService" }, { type: "azure-nextgen:peering/v20190801preview:PeeringService" }, { type: "azure-native:peering/v20190901preview:PeeringService" }, { type: "azure-nextgen:peering/v20190901preview:PeeringService" }, { type: "azure-native:peering/v20200101preview:PeeringService" }, { type: "azure-nextgen:peering/v20200101preview:PeeringService" }, { type: "azure-native:peering/v20201001:PeeringService" }, { type: "azure-nextgen:peering/v20201001:PeeringService" }, { type: "azure-native:peering/v20210101:PeeringService" }, { type: "azure-nextgen:peering/v20210101:PeeringService" }, { type: "azure-native:peering/v20210601:PeeringService" }, { type: "azure-nextgen:peering/v20210601:PeeringService" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PeeringService.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing PeeringService resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PeeringService(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PeeringService.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PeeringService.__pulumiType;
    }
}
exports.PeeringService = PeeringService;
/** @internal */
PeeringService.__pulumiType = 'azure-native:peering/v20200401:PeeringService';
