// @pulumi/azure-native/storage/v20210101/fileShare.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileShare = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Properties of the file share, including Id, resource name, resource type, Etag.
 */
class FileShare extends pulumi.CustomResource {
    /**
     * Create a FileShare resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["accessTier"] = args ? args.accessTier : undefined;
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["enabledProtocols"] = args ? args.enabledProtocols : undefined;
            inputs["expand"] = args ? args.expand : undefined;
            inputs["metadata"] = args ? args.metadata : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["rootSquash"] = args ? args.rootSquash : undefined;
            inputs["shareName"] = args ? args.shareName : undefined;
            inputs["shareQuota"] = args ? args.shareQuota : undefined;
            inputs["accessTierChangeTime"] = undefined /*out*/;
            inputs["accessTierStatus"] = undefined /*out*/;
            inputs["deleted"] = undefined /*out*/;
            inputs["deletedTime"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["remainingRetentionDays"] = undefined /*out*/;
            inputs["shareUsageBytes"] = undefined /*out*/;
            inputs["snapshotTime"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        else {
            inputs["accessTier"] = undefined /*out*/;
            inputs["accessTierChangeTime"] = undefined /*out*/;
            inputs["accessTierStatus"] = undefined /*out*/;
            inputs["deleted"] = undefined /*out*/;
            inputs["deletedTime"] = undefined /*out*/;
            inputs["enabledProtocols"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["metadata"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["remainingRetentionDays"] = undefined /*out*/;
            inputs["rootSquash"] = undefined /*out*/;
            inputs["shareQuota"] = undefined /*out*/;
            inputs["shareUsageBytes"] = undefined /*out*/;
            inputs["snapshotTime"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:storage/v20210101:FileShare" }, { type: "azure-native:storage:FileShare" }, { type: "azure-nextgen:storage:FileShare" }, { type: "azure-native:storage/v20190401:FileShare" }, { type: "azure-nextgen:storage/v20190401:FileShare" }, { type: "azure-native:storage/v20190601:FileShare" }, { type: "azure-nextgen:storage/v20190601:FileShare" }, { type: "azure-native:storage/v20200801preview:FileShare" }, { type: "azure-nextgen:storage/v20200801preview:FileShare" }, { type: "azure-native:storage/v20210201:FileShare" }, { type: "azure-nextgen:storage/v20210201:FileShare" }, { type: "azure-native:storage/v20210401:FileShare" }, { type: "azure-nextgen:storage/v20210401:FileShare" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FileShare.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing FileShare resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FileShare(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FileShare.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FileShare.__pulumiType;
    }
}
exports.FileShare = FileShare;
/** @internal */
FileShare.__pulumiType = 'azure-native:storage/v20210101:FileShare';
