// @pulumi/azure-native/web/v20210101/staticSiteUserProvidedFunctionAppForStaticSite.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StaticSiteUserProvidedFunctionAppForStaticSite = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Static Site User Provided Function App ARM resource.
 */
class StaticSiteUserProvidedFunctionAppForStaticSite extends pulumi.CustomResource {
    /**
     * Create a StaticSiteUserProvidedFunctionAppForStaticSite resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["functionAppName"] = args ? args.functionAppName : undefined;
            inputs["functionAppRegion"] = args ? args.functionAppRegion : undefined;
            inputs["functionAppResourceId"] = args ? args.functionAppResourceId : undefined;
            inputs["isForced"] = args ? args.isForced : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["createdOn"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["createdOn"] = undefined /*out*/;
            inputs["functionAppRegion"] = undefined /*out*/;
            inputs["functionAppResourceId"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20210101:StaticSiteUserProvidedFunctionAppForStaticSite" }, { type: "azure-native:web:StaticSiteUserProvidedFunctionAppForStaticSite" }, { type: "azure-nextgen:web:StaticSiteUserProvidedFunctionAppForStaticSite" }, { type: "azure-native:web/v20201201:StaticSiteUserProvidedFunctionAppForStaticSite" }, { type: "azure-nextgen:web/v20201201:StaticSiteUserProvidedFunctionAppForStaticSite" }, { type: "azure-native:web/v20210115:StaticSiteUserProvidedFunctionAppForStaticSite" }, { type: "azure-nextgen:web/v20210115:StaticSiteUserProvidedFunctionAppForStaticSite" }, { type: "azure-native:web/v20210201:StaticSiteUserProvidedFunctionAppForStaticSite" }, { type: "azure-nextgen:web/v20210201:StaticSiteUserProvidedFunctionAppForStaticSite" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(StaticSiteUserProvidedFunctionAppForStaticSite.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing StaticSiteUserProvidedFunctionAppForStaticSite resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StaticSiteUserProvidedFunctionAppForStaticSite(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StaticSiteUserProvidedFunctionAppForStaticSite.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StaticSiteUserProvidedFunctionAppForStaticSite.__pulumiType;
    }
}
exports.StaticSiteUserProvidedFunctionAppForStaticSite = StaticSiteUserProvidedFunctionAppForStaticSite;
/** @internal */
StaticSiteUserProvidedFunctionAppForStaticSite.__pulumiType = 'azure-native:web/v20210101:StaticSiteUserProvidedFunctionAppForStaticSite';
