// @pulumi/azure-native/web/v20210101/staticSite.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StaticSite = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Static Site ARM resource.
 */
class StaticSite extends pulumi.CustomResource {
    /**
     * Create a StaticSite resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["allowConfigFileUpdates"] = args ? args.allowConfigFileUpdates : undefined;
            inputs["branch"] = args ? args.branch : undefined;
            inputs["buildProperties"] = args ? args.buildProperties : undefined;
            inputs["identity"] = args ? args.identity : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["repositoryToken"] = args ? args.repositoryToken : undefined;
            inputs["repositoryUrl"] = args ? args.repositoryUrl : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["stagingEnvironmentPolicy"] = args ? args.stagingEnvironmentPolicy : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["templateProperties"] = args ? args.templateProperties : undefined;
            inputs["contentDistributionEndpoint"] = undefined /*out*/;
            inputs["customDomains"] = undefined /*out*/;
            inputs["defaultHostname"] = undefined /*out*/;
            inputs["keyVaultReferenceIdentity"] = undefined /*out*/;
            inputs["privateEndpointConnections"] = undefined /*out*/;
            inputs["provider"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["userProvidedFunctionApps"] = undefined /*out*/;
        }
        else {
            inputs["allowConfigFileUpdates"] = undefined /*out*/;
            inputs["branch"] = undefined /*out*/;
            inputs["buildProperties"] = undefined /*out*/;
            inputs["contentDistributionEndpoint"] = undefined /*out*/;
            inputs["customDomains"] = undefined /*out*/;
            inputs["defaultHostname"] = undefined /*out*/;
            inputs["identity"] = undefined /*out*/;
            inputs["keyVaultReferenceIdentity"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["privateEndpointConnections"] = undefined /*out*/;
            inputs["provider"] = undefined /*out*/;
            inputs["repositoryToken"] = undefined /*out*/;
            inputs["repositoryUrl"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["stagingEnvironmentPolicy"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["templateProperties"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["userProvidedFunctionApps"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20210101:StaticSite" }, { type: "azure-native:web:StaticSite" }, { type: "azure-nextgen:web:StaticSite" }, { type: "azure-native:web/v20190801:StaticSite" }, { type: "azure-nextgen:web/v20190801:StaticSite" }, { type: "azure-native:web/v20200601:StaticSite" }, { type: "azure-nextgen:web/v20200601:StaticSite" }, { type: "azure-native:web/v20200901:StaticSite" }, { type: "azure-nextgen:web/v20200901:StaticSite" }, { type: "azure-native:web/v20201001:StaticSite" }, { type: "azure-nextgen:web/v20201001:StaticSite" }, { type: "azure-native:web/v20201201:StaticSite" }, { type: "azure-nextgen:web/v20201201:StaticSite" }, { type: "azure-native:web/v20210115:StaticSite" }, { type: "azure-nextgen:web/v20210115:StaticSite" }, { type: "azure-native:web/v20210201:StaticSite" }, { type: "azure-nextgen:web/v20210201:StaticSite" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(StaticSite.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing StaticSite resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StaticSite(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StaticSite.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StaticSite.__pulumiType;
    }
}
exports.StaticSite = StaticSite;
/** @internal */
StaticSite.__pulumiType = 'azure-native:web/v20210101:StaticSite';
