// @pulumi/azure-native/search/v20210401preview/service.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Service = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Describes an Azure Cognitive Search service and its current state.
 */
class Service extends pulumi.CustomResource {
    /**
     * Create a Service resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["authOptions"] = args ? args.authOptions : undefined;
            inputs["disableLocalAuth"] = args ? args.disableLocalAuth : undefined;
            inputs["disabledDataExfiltrationOptions"] = args ? args.disabledDataExfiltrationOptions : undefined;
            inputs["encryptionWithCmk"] = args ? args.encryptionWithCmk : undefined;
            inputs["hostingMode"] = (_a = (args ? args.hostingMode : undefined)) !== null && _a !== void 0 ? _a : "default";
            inputs["identity"] = args ? args.identity : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["networkRuleSet"] = args ? args.networkRuleSet : undefined;
            inputs["partitionCount"] = (_b = (args ? args.partitionCount : undefined)) !== null && _b !== void 0 ? _b : 1;
            inputs["publicNetworkAccess"] = (_c = (args ? args.publicNetworkAccess : undefined)) !== null && _c !== void 0 ? _c : "enabled";
            inputs["replicaCount"] = (_d = (args ? args.replicaCount : undefined)) !== null && _d !== void 0 ? _d : 1;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["searchServiceName"] = args ? args.searchServiceName : undefined;
            inputs["semanticSearch"] = args ? args.semanticSearch : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["eTag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["privateEndpointConnections"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["sharedPrivateLinkResources"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["statusDetails"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["authOptions"] = undefined /*out*/;
            inputs["disableLocalAuth"] = undefined /*out*/;
            inputs["disabledDataExfiltrationOptions"] = undefined /*out*/;
            inputs["eTag"] = undefined /*out*/;
            inputs["encryptionWithCmk"] = undefined /*out*/;
            inputs["hostingMode"] = undefined /*out*/;
            inputs["identity"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["networkRuleSet"] = undefined /*out*/;
            inputs["partitionCount"] = undefined /*out*/;
            inputs["privateEndpointConnections"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["publicNetworkAccess"] = undefined /*out*/;
            inputs["replicaCount"] = undefined /*out*/;
            inputs["semanticSearch"] = undefined /*out*/;
            inputs["sharedPrivateLinkResources"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["statusDetails"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:search/v20210401preview:Service" }, { type: "azure-native:search:Service" }, { type: "azure-nextgen:search:Service" }, { type: "azure-native:search/v20150819:Service" }, { type: "azure-nextgen:search/v20150819:Service" }, { type: "azure-native:search/v20191001preview:Service" }, { type: "azure-nextgen:search/v20191001preview:Service" }, { type: "azure-native:search/v20200313:Service" }, { type: "azure-nextgen:search/v20200313:Service" }, { type: "azure-native:search/v20200801:Service" }, { type: "azure-nextgen:search/v20200801:Service" }, { type: "azure-native:search/v20200801preview:Service" }, { type: "azure-nextgen:search/v20200801preview:Service" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Service.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Service resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Service(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Service.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Service.__pulumiType;
    }
}
exports.Service = Service;
/** @internal */
Service.__pulumiType = 'azure-native:search/v20210401preview:Service';
