// @pulumi/azure-native/datashare/v20181101preview/adlsgen2FileDataSet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ADLSGen2FileDataSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An ADLS Gen 2 file data set.
 */
class ADLSGen2FileDataSet extends pulumi.CustomResource {
    /**
     * Create a ADLSGen2FileDataSet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.filePath === undefined) && !opts.urn) {
                throw new Error("Missing required property 'filePath'");
            }
            if ((!args || args.fileSystem === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fileSystem'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroup === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroup'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.shareName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shareName'");
            }
            if ((!args || args.storageAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageAccountName'");
            }
            if ((!args || args.subscriptionId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subscriptionId'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["dataSetName"] = args ? args.dataSetName : undefined;
            inputs["filePath"] = args ? args.filePath : undefined;
            inputs["fileSystem"] = args ? args.fileSystem : undefined;
            inputs["kind"] = "AdlsGen2File";
            inputs["resourceGroup"] = args ? args.resourceGroup : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["shareName"] = args ? args.shareName : undefined;
            inputs["storageAccountName"] = args ? args.storageAccountName : undefined;
            inputs["subscriptionId"] = args ? args.subscriptionId : undefined;
            inputs["dataSetId"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["dataSetId"] = undefined /*out*/;
            inputs["filePath"] = undefined /*out*/;
            inputs["fileSystem"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["resourceGroup"] = undefined /*out*/;
            inputs["storageAccountName"] = undefined /*out*/;
            inputs["subscriptionId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:datashare/v20181101preview:ADLSGen2FileDataSet" }, { type: "azure-native:datashare:ADLSGen2FileDataSet" }, { type: "azure-nextgen:datashare:ADLSGen2FileDataSet" }, { type: "azure-native:datashare/v20191101:ADLSGen2FileDataSet" }, { type: "azure-nextgen:datashare/v20191101:ADLSGen2FileDataSet" }, { type: "azure-native:datashare/v20200901:ADLSGen2FileDataSet" }, { type: "azure-nextgen:datashare/v20200901:ADLSGen2FileDataSet" }, { type: "azure-native:datashare/v20201001preview:ADLSGen2FileDataSet" }, { type: "azure-nextgen:datashare/v20201001preview:ADLSGen2FileDataSet" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ADLSGen2FileDataSet.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ADLSGen2FileDataSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ADLSGen2FileDataSet(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ADLSGen2FileDataSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ADLSGen2FileDataSet.__pulumiType;
    }
}
exports.ADLSGen2FileDataSet = ADLSGen2FileDataSet;
/** @internal */
ADLSGen2FileDataSet.__pulumiType = 'azure-native:datashare/v20181101preview:ADLSGen2FileDataSet';
