// @pulumi/azure-native/datashare/v20181101preview/kustoClusterDataSetMapping.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.KustoClusterDataSetMapping = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A Kusto cluster data set mapping
 */
class KustoClusterDataSetMapping extends pulumi.CustomResource {
    /**
     * Create a KustoClusterDataSetMapping resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.dataSetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataSetId'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.kustoClusterResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kustoClusterResourceId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.shareSubscriptionName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shareSubscriptionName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["dataSetId"] = args ? args.dataSetId : undefined;
            inputs["dataSetMappingName"] = args ? args.dataSetMappingName : undefined;
            inputs["kind"] = "KustoCluster";
            inputs["kustoClusterResourceId"] = args ? args.kustoClusterResourceId : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["shareSubscriptionName"] = args ? args.shareSubscriptionName : undefined;
            inputs["dataSetMappingStatus"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["dataSetId"] = undefined /*out*/;
            inputs["dataSetMappingStatus"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["kustoClusterResourceId"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:datashare/v20181101preview:KustoClusterDataSetMapping" }, { type: "azure-native:datashare:KustoClusterDataSetMapping" }, { type: "azure-nextgen:datashare:KustoClusterDataSetMapping" }, { type: "azure-native:datashare/v20191101:KustoClusterDataSetMapping" }, { type: "azure-nextgen:datashare/v20191101:KustoClusterDataSetMapping" }, { type: "azure-native:datashare/v20200901:KustoClusterDataSetMapping" }, { type: "azure-nextgen:datashare/v20200901:KustoClusterDataSetMapping" }, { type: "azure-native:datashare/v20201001preview:KustoClusterDataSetMapping" }, { type: "azure-nextgen:datashare/v20201001preview:KustoClusterDataSetMapping" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(KustoClusterDataSetMapping.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing KustoClusterDataSetMapping resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new KustoClusterDataSetMapping(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of KustoClusterDataSetMapping.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KustoClusterDataSetMapping.__pulumiType;
    }
}
exports.KustoClusterDataSetMapping = KustoClusterDataSetMapping;
/** @internal */
KustoClusterDataSetMapping.__pulumiType = 'azure-native:datashare/v20181101preview:KustoClusterDataSetMapping';
