// @pulumi/azure-native/datashare/v20181101preview/synchronizationSetting.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SynchronizationSetting = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A Synchronization Setting data transfer object.
 *
 * @deprecated Please use one of the variants: ScheduledSynchronizationSetting.
 */
class SynchronizationSetting extends pulumi.CustomResource {
    /**
     * Create a SynchronizationSetting resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Please use one of the variants: ScheduledSynchronizationSetting. */
    constructor(name, args, opts) {
        pulumi.log.warn("SynchronizationSetting is deprecated: Please use one of the variants: ScheduledSynchronizationSetting.");
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.shareName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shareName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["shareName"] = args ? args.shareName : undefined;
            inputs["synchronizationSettingName"] = args ? args.synchronizationSettingName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:datashare/v20181101preview:SynchronizationSetting" }, { type: "azure-native:datashare:SynchronizationSetting" }, { type: "azure-nextgen:datashare:SynchronizationSetting" }, { type: "azure-native:datashare/v20191101:SynchronizationSetting" }, { type: "azure-nextgen:datashare/v20191101:SynchronizationSetting" }, { type: "azure-native:datashare/v20200901:SynchronizationSetting" }, { type: "azure-nextgen:datashare/v20200901:SynchronizationSetting" }, { type: "azure-native:datashare/v20201001preview:SynchronizationSetting" }, { type: "azure-nextgen:datashare/v20201001preview:SynchronizationSetting" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SynchronizationSetting.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SynchronizationSetting resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SynchronizationSetting is deprecated: Please use one of the variants: ScheduledSynchronizationSetting.");
        return new SynchronizationSetting(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SynchronizationSetting.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SynchronizationSetting.__pulumiType;
    }
}
exports.SynchronizationSetting = SynchronizationSetting;
/** @internal */
SynchronizationSetting.__pulumiType = 'azure-native:datashare/v20181101preview:SynchronizationSetting';
