// @pulumi/azure-native/network/v20200101/privateZone.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateZone = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Describes a Private DNS zone.
 */
class PrivateZone extends pulumi.CustomResource {
    /**
     * Create a PrivateZone resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["etag"] = args ? args.etag : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["privateZoneName"] = args ? args.privateZoneName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["internalId"] = undefined /*out*/;
            inputs["maxNumberOfRecordSets"] = undefined /*out*/;
            inputs["maxNumberOfVirtualNetworkLinks"] = undefined /*out*/;
            inputs["maxNumberOfVirtualNetworkLinksWithRegistration"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["numberOfRecordSets"] = undefined /*out*/;
            inputs["numberOfVirtualNetworkLinks"] = undefined /*out*/;
            inputs["numberOfVirtualNetworkLinksWithRegistration"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["etag"] = undefined /*out*/;
            inputs["internalId"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["maxNumberOfRecordSets"] = undefined /*out*/;
            inputs["maxNumberOfVirtualNetworkLinks"] = undefined /*out*/;
            inputs["maxNumberOfVirtualNetworkLinksWithRegistration"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["numberOfRecordSets"] = undefined /*out*/;
            inputs["numberOfVirtualNetworkLinks"] = undefined /*out*/;
            inputs["numberOfVirtualNetworkLinksWithRegistration"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20200101:PrivateZone" }, { type: "azure-native:network:PrivateZone" }, { type: "azure-nextgen:network:PrivateZone" }, { type: "azure-native:network/v20180901:PrivateZone" }, { type: "azure-nextgen:network/v20180901:PrivateZone" }, { type: "azure-native:network/v20200601:PrivateZone" }, { type: "azure-nextgen:network/v20200601:PrivateZone" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateZone.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing PrivateZone resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateZone(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivateZone.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateZone.__pulumiType;
    }
}
exports.PrivateZone = PrivateZone;
/** @internal */
PrivateZone.__pulumiType = 'azure-native:network/v20200101:PrivateZone';
