// @pulumi/azure-native/network/v20200101/virtualNetworkLink.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetworkLink = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Describes a link to virtual network for a Private DNS zone.
 */
class VirtualNetworkLink extends pulumi.CustomResource {
    /**
     * Create a VirtualNetworkLink resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.privateZoneName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'privateZoneName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["etag"] = args ? args.etag : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["privateZoneName"] = args ? args.privateZoneName : undefined;
            inputs["registrationEnabled"] = args ? args.registrationEnabled : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["virtualNetwork"] = args ? args.virtualNetwork : undefined;
            inputs["virtualNetworkLinkName"] = args ? args.virtualNetworkLinkName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["virtualNetworkLinkState"] = undefined /*out*/;
        }
        else {
            inputs["etag"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["registrationEnabled"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["virtualNetwork"] = undefined /*out*/;
            inputs["virtualNetworkLinkState"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20200101:VirtualNetworkLink" }, { type: "azure-native:network:VirtualNetworkLink" }, { type: "azure-nextgen:network:VirtualNetworkLink" }, { type: "azure-native:network/v20180901:VirtualNetworkLink" }, { type: "azure-nextgen:network/v20180901:VirtualNetworkLink" }, { type: "azure-native:network/v20200601:VirtualNetworkLink" }, { type: "azure-nextgen:network/v20200601:VirtualNetworkLink" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VirtualNetworkLink.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing VirtualNetworkLink resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VirtualNetworkLink(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualNetworkLink.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualNetworkLink.__pulumiType;
    }
}
exports.VirtualNetworkLink = VirtualNetworkLink;
/** @internal */
VirtualNetworkLink.__pulumiType = 'azure-native:network/v20200101:VirtualNetworkLink';
