// @pulumi/azure-native/datashare/v20201001preview/adlsgen1FileDataSet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ADLSGen1FileDataSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An ADLS Gen 1 file data set.
 */
class ADLSGen1FileDataSet extends pulumi.CustomResource {
    /**
     * Create a ADLSGen1FileDataSet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.fileName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fileName'");
            }
            if ((!args || args.folderPath === undefined) && !opts.urn) {
                throw new Error("Missing required property 'folderPath'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroup === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroup'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.shareName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shareName'");
            }
            if ((!args || args.subscriptionId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subscriptionId'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["dataSetName"] = args ? args.dataSetName : undefined;
            inputs["fileName"] = args ? args.fileName : undefined;
            inputs["folderPath"] = args ? args.folderPath : undefined;
            inputs["kind"] = "AdlsGen1File";
            inputs["resourceGroup"] = args ? args.resourceGroup : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["shareName"] = args ? args.shareName : undefined;
            inputs["subscriptionId"] = args ? args.subscriptionId : undefined;
            inputs["dataSetId"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["accountName"] = undefined /*out*/;
            inputs["dataSetId"] = undefined /*out*/;
            inputs["fileName"] = undefined /*out*/;
            inputs["folderPath"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["resourceGroup"] = undefined /*out*/;
            inputs["subscriptionId"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:datashare/v20201001preview:ADLSGen1FileDataSet" }, { type: "azure-native:datashare:ADLSGen1FileDataSet" }, { type: "azure-nextgen:datashare:ADLSGen1FileDataSet" }, { type: "azure-native:datashare/v20181101preview:ADLSGen1FileDataSet" }, { type: "azure-nextgen:datashare/v20181101preview:ADLSGen1FileDataSet" }, { type: "azure-native:datashare/v20191101:ADLSGen1FileDataSet" }, { type: "azure-nextgen:datashare/v20191101:ADLSGen1FileDataSet" }, { type: "azure-native:datashare/v20200901:ADLSGen1FileDataSet" }, { type: "azure-nextgen:datashare/v20200901:ADLSGen1FileDataSet" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ADLSGen1FileDataSet.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ADLSGen1FileDataSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ADLSGen1FileDataSet(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ADLSGen1FileDataSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ADLSGen1FileDataSet.__pulumiType;
    }
}
exports.ADLSGen1FileDataSet = ADLSGen1FileDataSet;
/** @internal */
ADLSGen1FileDataSet.__pulumiType = 'azure-native:datashare/v20201001preview:ADLSGen1FileDataSet';
