// @pulumi/azure-native/datashare/v20201001preview/blobStorageAccountDataSetMapping.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlobStorageAccountDataSetMapping = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A blob storage account data set mapping.
 */
class BlobStorageAccountDataSetMapping extends pulumi.CustomResource {
    /**
     * Create a BlobStorageAccountDataSetMapping resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.containerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'containerName'");
            }
            if ((!args || args.dataSetId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataSetId'");
            }
            if ((!args || args.folder === undefined) && !opts.urn) {
                throw new Error("Missing required property 'folder'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.shareSubscriptionName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shareSubscriptionName'");
            }
            if ((!args || args.storageAccountResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageAccountResourceId'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["containerName"] = args ? args.containerName : undefined;
            inputs["dataSetId"] = args ? args.dataSetId : undefined;
            inputs["dataSetMappingName"] = args ? args.dataSetMappingName : undefined;
            inputs["folder"] = args ? args.folder : undefined;
            inputs["kind"] = "BlobStorageAccount";
            inputs["mountPath"] = args ? args.mountPath : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["shareSubscriptionName"] = args ? args.shareSubscriptionName : undefined;
            inputs["storageAccountResourceId"] = args ? args.storageAccountResourceId : undefined;
            inputs["dataSetMappingStatus"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["containerName"] = undefined /*out*/;
            inputs["dataSetId"] = undefined /*out*/;
            inputs["dataSetMappingStatus"] = undefined /*out*/;
            inputs["folder"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["mountPath"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["storageAccountResourceId"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:datashare/v20201001preview:BlobStorageAccountDataSetMapping" }, { type: "azure-native:datashare:BlobStorageAccountDataSetMapping" }, { type: "azure-nextgen:datashare:BlobStorageAccountDataSetMapping" }, { type: "azure-native:datashare/v20181101preview:BlobStorageAccountDataSetMapping" }, { type: "azure-nextgen:datashare/v20181101preview:BlobStorageAccountDataSetMapping" }, { type: "azure-native:datashare/v20191101:BlobStorageAccountDataSetMapping" }, { type: "azure-nextgen:datashare/v20191101:BlobStorageAccountDataSetMapping" }, { type: "azure-native:datashare/v20200901:BlobStorageAccountDataSetMapping" }, { type: "azure-nextgen:datashare/v20200901:BlobStorageAccountDataSetMapping" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BlobStorageAccountDataSetMapping.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing BlobStorageAccountDataSetMapping resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BlobStorageAccountDataSetMapping(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BlobStorageAccountDataSetMapping.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BlobStorageAccountDataSetMapping.__pulumiType;
    }
}
exports.BlobStorageAccountDataSetMapping = BlobStorageAccountDataSetMapping;
/** @internal */
BlobStorageAccountDataSetMapping.__pulumiType = 'azure-native:datashare/v20201001preview:BlobStorageAccountDataSetMapping';
