// @pulumi/azure-native/datashare/v20201001preview/blobFolderDataSet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlobFolderDataSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An Azure storage blob folder data set.
 */
class BlobFolderDataSet extends pulumi.CustomResource {
    /**
     * Create a BlobFolderDataSet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.containerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'containerName'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.prefix === undefined) && !opts.urn) {
                throw new Error("Missing required property 'prefix'");
            }
            if ((!args || args.resourceGroup === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroup'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.shareName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shareName'");
            }
            if ((!args || args.storageAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageAccountName'");
            }
            if ((!args || args.subscriptionId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subscriptionId'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["containerName"] = args ? args.containerName : undefined;
            inputs["dataSetName"] = args ? args.dataSetName : undefined;
            inputs["kind"] = "BlobFolder";
            inputs["prefix"] = args ? args.prefix : undefined;
            inputs["resourceGroup"] = args ? args.resourceGroup : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["shareName"] = args ? args.shareName : undefined;
            inputs["storageAccountName"] = args ? args.storageAccountName : undefined;
            inputs["subscriptionId"] = args ? args.subscriptionId : undefined;
            inputs["dataSetId"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["containerName"] = undefined /*out*/;
            inputs["dataSetId"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["prefix"] = undefined /*out*/;
            inputs["resourceGroup"] = undefined /*out*/;
            inputs["storageAccountName"] = undefined /*out*/;
            inputs["subscriptionId"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:datashare/v20201001preview:BlobFolderDataSet" }, { type: "azure-native:datashare:BlobFolderDataSet" }, { type: "azure-nextgen:datashare:BlobFolderDataSet" }, { type: "azure-native:datashare/v20181101preview:BlobFolderDataSet" }, { type: "azure-nextgen:datashare/v20181101preview:BlobFolderDataSet" }, { type: "azure-native:datashare/v20191101:BlobFolderDataSet" }, { type: "azure-nextgen:datashare/v20191101:BlobFolderDataSet" }, { type: "azure-native:datashare/v20200901:BlobFolderDataSet" }, { type: "azure-nextgen:datashare/v20200901:BlobFolderDataSet" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BlobFolderDataSet.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing BlobFolderDataSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BlobFolderDataSet(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BlobFolderDataSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BlobFolderDataSet.__pulumiType;
    }
}
exports.BlobFolderDataSet = BlobFolderDataSet;
/** @internal */
BlobFolderDataSet.__pulumiType = 'azure-native:datashare/v20201001preview:BlobFolderDataSet';
