// @pulumi/azure-native/datashare/v20201001preview/kustoDatabaseDataSet.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.KustoDatabaseDataSet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A kusto database data set.
 */
class KustoDatabaseDataSet extends pulumi.CustomResource {
    /**
     * Create a KustoDatabaseDataSet resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.kustoDatabaseResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kustoDatabaseResourceId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.shareName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shareName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["dataSetName"] = args ? args.dataSetName : undefined;
            inputs["kind"] = "KustoDatabase";
            inputs["kustoDatabaseResourceId"] = args ? args.kustoDatabaseResourceId : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["shareName"] = args ? args.shareName : undefined;
            inputs["dataSetId"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["dataSetId"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["kustoDatabaseResourceId"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:datashare/v20201001preview:KustoDatabaseDataSet" }, { type: "azure-native:datashare:KustoDatabaseDataSet" }, { type: "azure-nextgen:datashare:KustoDatabaseDataSet" }, { type: "azure-native:datashare/v20181101preview:KustoDatabaseDataSet" }, { type: "azure-nextgen:datashare/v20181101preview:KustoDatabaseDataSet" }, { type: "azure-native:datashare/v20191101:KustoDatabaseDataSet" }, { type: "azure-nextgen:datashare/v20191101:KustoDatabaseDataSet" }, { type: "azure-native:datashare/v20200901:KustoDatabaseDataSet" }, { type: "azure-nextgen:datashare/v20200901:KustoDatabaseDataSet" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(KustoDatabaseDataSet.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing KustoDatabaseDataSet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new KustoDatabaseDataSet(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of KustoDatabaseDataSet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KustoDatabaseDataSet.__pulumiType;
    }
}
exports.KustoDatabaseDataSet = KustoDatabaseDataSet;
/** @internal */
KustoDatabaseDataSet.__pulumiType = 'azure-native:datashare/v20201001preview:KustoDatabaseDataSet';
