// @pulumi/azure-native/securityinsights/v20200101/mdatpdataConnector.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MDATPDataConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents MDATP (Microsoft Defender Advanced Threat Protection) data connector.
 */
class MDATPDataConnector extends pulumi.CustomResource {
    /**
     * Create a MDATPDataConnector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["dataConnectorId"] = args ? args.dataConnectorId : undefined;
            inputs["dataTypes"] = args ? args.dataTypes : undefined;
            inputs["etag"] = args ? args.etag : undefined;
            inputs["kind"] = "MicrosoftDefenderAdvancedThreatProtection";
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tenantId"] = args ? args.tenantId : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["dataTypes"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["tenantId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:securityinsights/v20200101:MDATPDataConnector" }, { type: "azure-native:securityinsights:MDATPDataConnector" }, { type: "azure-nextgen:securityinsights:MDATPDataConnector" }, { type: "azure-native:securityinsights/v20190101preview:MDATPDataConnector" }, { type: "azure-nextgen:securityinsights/v20190101preview:MDATPDataConnector" }, { type: "azure-native:securityinsights/v20210301preview:MDATPDataConnector" }, { type: "azure-nextgen:securityinsights/v20210301preview:MDATPDataConnector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MDATPDataConnector.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing MDATPDataConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MDATPDataConnector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MDATPDataConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MDATPDataConnector.__pulumiType;
    }
}
exports.MDATPDataConnector = MDATPDataConnector;
/** @internal */
MDATPDataConnector.__pulumiType = 'azure-native:securityinsights/v20200101:MDATPDataConnector';
