// @pulumi/azure-native/security/alertsSuppressionRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlertsSuppressionRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Describes the suppression rule
 * API Version: 2019-01-01-preview.
 */
class AlertsSuppressionRule extends pulumi.CustomResource {
    /**
     * Create a AlertsSuppressionRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.alertType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'alertType'");
            }
            if ((!args || args.reason === undefined) && !opts.urn) {
                throw new Error("Missing required property 'reason'");
            }
            if ((!args || args.state === undefined) && !opts.urn) {
                throw new Error("Missing required property 'state'");
            }
            inputs["alertType"] = args ? args.alertType : undefined;
            inputs["alertsSuppressionRuleName"] = args ? args.alertsSuppressionRuleName : undefined;
            inputs["comment"] = args ? args.comment : undefined;
            inputs["expirationDateUtc"] = args ? args.expirationDateUtc : undefined;
            inputs["reason"] = args ? args.reason : undefined;
            inputs["state"] = args ? args.state : undefined;
            inputs["suppressionAlertsScope"] = args ? args.suppressionAlertsScope : undefined;
            inputs["lastModifiedUtc"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["alertType"] = undefined /*out*/;
            inputs["comment"] = undefined /*out*/;
            inputs["expirationDateUtc"] = undefined /*out*/;
            inputs["lastModifiedUtc"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["reason"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
            inputs["suppressionAlertsScope"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:security:AlertsSuppressionRule" }, { type: "azure-native:security/v20190101preview:AlertsSuppressionRule" }, { type: "azure-nextgen:security/v20190101preview:AlertsSuppressionRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AlertsSuppressionRule.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing AlertsSuppressionRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AlertsSuppressionRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AlertsSuppressionRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AlertsSuppressionRule.__pulumiType;
    }
}
exports.AlertsSuppressionRule = AlertsSuppressionRule;
/** @internal */
AlertsSuppressionRule.__pulumiType = 'azure-native:security:AlertsSuppressionRule';
