// @pulumi/azure-native/security/iotSecuritySolution.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IotSecuritySolution = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * IoT Security solution configuration and resource information.
 * API Version: 2019-08-01.
 */
class IotSecuritySolution extends pulumi.CustomResource {
    /**
     * Create a IotSecuritySolution resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.iotHubs === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iotHubs'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["additionalWorkspaces"] = args ? args.additionalWorkspaces : undefined;
            inputs["disabledDataSources"] = args ? args.disabledDataSources : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["export"] = args ? args.export : undefined;
            inputs["iotHubs"] = args ? args.iotHubs : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["recommendationsConfiguration"] = args ? args.recommendationsConfiguration : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["solutionName"] = args ? args.solutionName : undefined;
            inputs["status"] = (_a = (args ? args.status : undefined)) !== null && _a !== void 0 ? _a : "Enabled";
            inputs["tags"] = args ? args.tags : undefined;
            inputs["unmaskedIpLoggingStatus"] = (_b = (args ? args.unmaskedIpLoggingStatus : undefined)) !== null && _b !== void 0 ? _b : "Disabled";
            inputs["userDefinedResources"] = args ? args.userDefinedResources : undefined;
            inputs["workspace"] = args ? args.workspace : undefined;
            inputs["autoDiscoveredResources"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["additionalWorkspaces"] = undefined /*out*/;
            inputs["autoDiscoveredResources"] = undefined /*out*/;
            inputs["disabledDataSources"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["export"] = undefined /*out*/;
            inputs["iotHubs"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["recommendationsConfiguration"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["unmaskedIpLoggingStatus"] = undefined /*out*/;
            inputs["userDefinedResources"] = undefined /*out*/;
            inputs["workspace"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:security:IotSecuritySolution" }, { type: "azure-native:security/v20170801preview:IotSecuritySolution" }, { type: "azure-nextgen:security/v20170801preview:IotSecuritySolution" }, { type: "azure-native:security/v20190801:IotSecuritySolution" }, { type: "azure-nextgen:security/v20190801:IotSecuritySolution" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IotSecuritySolution.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing IotSecuritySolution resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IotSecuritySolution(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IotSecuritySolution.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IotSecuritySolution.__pulumiType;
    }
}
exports.IotSecuritySolution = IotSecuritySolution;
/** @internal */
IotSecuritySolution.__pulumiType = 'azure-native:security:IotSecuritySolution';
