// @pulumi/azure-native/databoxedge/device.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Device = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The Data Box Edge/Gateway device.
 * API Version: 2020-12-01.
 */
class Device extends pulumi.CustomResource {
    /**
     * Create a Device resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["dataBoxEdgeDeviceStatus"] = args ? args.dataBoxEdgeDeviceStatus : undefined;
            inputs["deviceName"] = args ? args.deviceName : undefined;
            inputs["etag"] = args ? args.etag : undefined;
            inputs["identity"] = args ? args.identity : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["configuredRoleTypes"] = undefined /*out*/;
            inputs["culture"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["deviceHcsVersion"] = undefined /*out*/;
            inputs["deviceLocalCapacity"] = undefined /*out*/;
            inputs["deviceModel"] = undefined /*out*/;
            inputs["deviceSoftwareVersion"] = undefined /*out*/;
            inputs["deviceType"] = undefined /*out*/;
            inputs["edgeProfile"] = undefined /*out*/;
            inputs["friendlyName"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["modelDescription"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["nodeCount"] = undefined /*out*/;
            inputs["resourceMoveDetails"] = undefined /*out*/;
            inputs["serialNumber"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["timeZone"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["configuredRoleTypes"] = undefined /*out*/;
            inputs["culture"] = undefined /*out*/;
            inputs["dataBoxEdgeDeviceStatus"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["deviceHcsVersion"] = undefined /*out*/;
            inputs["deviceLocalCapacity"] = undefined /*out*/;
            inputs["deviceModel"] = undefined /*out*/;
            inputs["deviceSoftwareVersion"] = undefined /*out*/;
            inputs["deviceType"] = undefined /*out*/;
            inputs["edgeProfile"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["friendlyName"] = undefined /*out*/;
            inputs["identity"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["modelDescription"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["nodeCount"] = undefined /*out*/;
            inputs["resourceMoveDetails"] = undefined /*out*/;
            inputs["serialNumber"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["timeZone"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:databoxedge:Device" }, { type: "azure-native:databoxedge/v20190301:Device" }, { type: "azure-nextgen:databoxedge/v20190301:Device" }, { type: "azure-native:databoxedge/v20190701:Device" }, { type: "azure-nextgen:databoxedge/v20190701:Device" }, { type: "azure-native:databoxedge/v20190801:Device" }, { type: "azure-nextgen:databoxedge/v20190801:Device" }, { type: "azure-native:databoxedge/v20200501preview:Device" }, { type: "azure-nextgen:databoxedge/v20200501preview:Device" }, { type: "azure-native:databoxedge/v20200901:Device" }, { type: "azure-nextgen:databoxedge/v20200901:Device" }, { type: "azure-native:databoxedge/v20200901preview:Device" }, { type: "azure-nextgen:databoxedge/v20200901preview:Device" }, { type: "azure-native:databoxedge/v20201201:Device" }, { type: "azure-nextgen:databoxedge/v20201201:Device" }, { type: "azure-native:databoxedge/v20210201:Device" }, { type: "azure-nextgen:databoxedge/v20210201:Device" }, { type: "azure-native:databoxedge/v20210201preview:Device" }, { type: "azure-nextgen:databoxedge/v20210201preview:Device" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Device.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Device resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Device(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Device.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Device.__pulumiType;
    }
}
exports.Device = Device;
/** @internal */
Device.__pulumiType = 'azure-native:databoxedge:Device';
