// @pulumi/azure-native/databoxedge/ioTRole.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IoTRole = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Compute role.
 * API Version: 2020-12-01.
 */
class IoTRole extends pulumi.CustomResource {
    /**
     * Create a IoTRole resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.deviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deviceName'");
            }
            if ((!args || args.hostPlatform === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hostPlatform'");
            }
            if ((!args || args.ioTDeviceDetails === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ioTDeviceDetails'");
            }
            if ((!args || args.ioTEdgeDeviceDetails === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ioTEdgeDeviceDetails'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.roleStatus === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleStatus'");
            }
            inputs["computeResource"] = args ? args.computeResource : undefined;
            inputs["deviceName"] = args ? args.deviceName : undefined;
            inputs["hostPlatform"] = args ? args.hostPlatform : undefined;
            inputs["ioTDeviceDetails"] = args ? args.ioTDeviceDetails : undefined;
            inputs["ioTEdgeAgentInfo"] = args ? args.ioTEdgeAgentInfo : undefined;
            inputs["ioTEdgeDeviceDetails"] = args ? args.ioTEdgeDeviceDetails : undefined;
            inputs["kind"] = "IOT";
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["roleStatus"] = args ? args.roleStatus : undefined;
            inputs["shareMappings"] = args ? args.shareMappings : undefined;
            inputs["hostPlatformType"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["computeResource"] = undefined /*out*/;
            inputs["hostPlatform"] = undefined /*out*/;
            inputs["hostPlatformType"] = undefined /*out*/;
            inputs["ioTDeviceDetails"] = undefined /*out*/;
            inputs["ioTEdgeAgentInfo"] = undefined /*out*/;
            inputs["ioTEdgeDeviceDetails"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["roleStatus"] = undefined /*out*/;
            inputs["shareMappings"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:databoxedge:IoTRole" }, { type: "azure-native:databoxedge/v20190301:IoTRole" }, { type: "azure-nextgen:databoxedge/v20190301:IoTRole" }, { type: "azure-native:databoxedge/v20190701:IoTRole" }, { type: "azure-nextgen:databoxedge/v20190701:IoTRole" }, { type: "azure-native:databoxedge/v20190801:IoTRole" }, { type: "azure-nextgen:databoxedge/v20190801:IoTRole" }, { type: "azure-native:databoxedge/v20200501preview:IoTRole" }, { type: "azure-nextgen:databoxedge/v20200501preview:IoTRole" }, { type: "azure-native:databoxedge/v20200901:IoTRole" }, { type: "azure-nextgen:databoxedge/v20200901:IoTRole" }, { type: "azure-native:databoxedge/v20200901preview:IoTRole" }, { type: "azure-nextgen:databoxedge/v20200901preview:IoTRole" }, { type: "azure-native:databoxedge/v20201201:IoTRole" }, { type: "azure-nextgen:databoxedge/v20201201:IoTRole" }, { type: "azure-native:databoxedge/v20210201:IoTRole" }, { type: "azure-nextgen:databoxedge/v20210201:IoTRole" }, { type: "azure-native:databoxedge/v20210201preview:IoTRole" }, { type: "azure-nextgen:databoxedge/v20210201preview:IoTRole" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IoTRole.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing IoTRole resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IoTRole(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IoTRole.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IoTRole.__pulumiType;
    }
}
exports.IoTRole = IoTRole;
/** @internal */
IoTRole.__pulumiType = 'azure-native:databoxedge:IoTRole';
