// @pulumi/azure-native/cdn/afdcustomDomain.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AFDCustomDomain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Friendly domain name mapping to the endpoint hostname that the customer provides for branding purposes, e.g. www.contoso.com.
 * API Version: 2020-09-01.
 */
class AFDCustomDomain extends pulumi.CustomResource {
    /**
     * Create a AFDCustomDomain resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.hostName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hostName'");
            }
            if ((!args || args.profileName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'profileName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["azureDnsZone"] = args ? args.azureDnsZone : undefined;
            inputs["customDomainName"] = args ? args.customDomainName : undefined;
            inputs["hostName"] = args ? args.hostName : undefined;
            inputs["profileName"] = args ? args.profileName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tlsSettings"] = args ? args.tlsSettings : undefined;
            inputs["deploymentStatus"] = undefined /*out*/;
            inputs["domainValidationState"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["validationProperties"] = undefined /*out*/;
        }
        else {
            inputs["azureDnsZone"] = undefined /*out*/;
            inputs["deploymentStatus"] = undefined /*out*/;
            inputs["domainValidationState"] = undefined /*out*/;
            inputs["hostName"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tlsSettings"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["validationProperties"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:cdn:AFDCustomDomain" }, { type: "azure-native:cdn/v20200901:AFDCustomDomain" }, { type: "azure-nextgen:cdn/v20200901:AFDCustomDomain" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AFDCustomDomain.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing AFDCustomDomain resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AFDCustomDomain(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AFDCustomDomain.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AFDCustomDomain.__pulumiType;
    }
}
exports.AFDCustomDomain = AFDCustomDomain;
/** @internal */
AFDCustomDomain.__pulumiType = 'azure-native:cdn:AFDCustomDomain';
