// @pulumi/azure-native/eventgrid/v20210601preview/eventChannel.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventChannel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Event Channel.
 */
class EventChannel extends pulumi.CustomResource {
    /**
     * Create a EventChannel resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.partnerNamespaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'partnerNamespaceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["destination"] = args ? args.destination : undefined;
            inputs["eventChannelName"] = args ? args.eventChannelName : undefined;
            inputs["expirationTimeIfNotActivatedUtc"] = args ? args.expirationTimeIfNotActivatedUtc : undefined;
            inputs["filter"] = args ? args.filter : undefined;
            inputs["partnerNamespaceName"] = args ? args.partnerNamespaceName : undefined;
            inputs["partnerTopicFriendlyDescription"] = args ? args.partnerTopicFriendlyDescription : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["source"] = args ? args.source : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["partnerTopicReadinessState"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["destination"] = undefined /*out*/;
            inputs["expirationTimeIfNotActivatedUtc"] = undefined /*out*/;
            inputs["filter"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["partnerTopicFriendlyDescription"] = undefined /*out*/;
            inputs["partnerTopicReadinessState"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["source"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:eventgrid/v20210601preview:EventChannel" }, { type: "azure-native:eventgrid:EventChannel" }, { type: "azure-nextgen:eventgrid:EventChannel" }, { type: "azure-native:eventgrid/v20200401preview:EventChannel" }, { type: "azure-nextgen:eventgrid/v20200401preview:EventChannel" }, { type: "azure-native:eventgrid/v20201015preview:EventChannel" }, { type: "azure-nextgen:eventgrid/v20201015preview:EventChannel" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(EventChannel.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing EventChannel resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new EventChannel(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EventChannel.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EventChannel.__pulumiType;
    }
}
exports.EventChannel = EventChannel;
/** @internal */
EventChannel.__pulumiType = 'azure-native:eventgrid/v20210601preview:EventChannel';
