// @pulumi/azure-native/machinelearningservices/v20200301/workspace.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An object that represents a machine learning workspace.
 */
class Workspace extends pulumi.CustomResource {
    /**
     * Create a Workspace resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["allowPublicAccessWhenBehindVnet"] = (_a = (args ? args.allowPublicAccessWhenBehindVnet : undefined)) !== null && _a !== void 0 ? _a : false;
            inputs["applicationInsights"] = args ? args.applicationInsights : undefined;
            inputs["containerRegistry"] = args ? args.containerRegistry : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["discoveryUrl"] = args ? args.discoveryUrl : undefined;
            inputs["encryption"] = args ? args.encryption : undefined;
            inputs["friendlyName"] = args ? args.friendlyName : undefined;
            inputs["hbiWorkspace"] = (_b = (args ? args.hbiWorkspace : undefined)) !== null && _b !== void 0 ? _b : false;
            inputs["identity"] = args ? args.identity : undefined;
            inputs["imageBuildCompute"] = args ? args.imageBuildCompute : undefined;
            inputs["keyVault"] = args ? args.keyVault : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sharedPrivateLinkResources"] = args ? args.sharedPrivateLinkResources : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["storageAccount"] = args ? args.storageAccount : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["creationTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["notebookInfo"] = undefined /*out*/;
            inputs["privateEndpointConnections"] = undefined /*out*/;
            inputs["privateLinkCount"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["serviceProvisionedResourceGroup"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["workspaceId"] = undefined /*out*/;
        }
        else {
            inputs["allowPublicAccessWhenBehindVnet"] = undefined /*out*/;
            inputs["applicationInsights"] = undefined /*out*/;
            inputs["containerRegistry"] = undefined /*out*/;
            inputs["creationTime"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["discoveryUrl"] = undefined /*out*/;
            inputs["encryption"] = undefined /*out*/;
            inputs["friendlyName"] = undefined /*out*/;
            inputs["hbiWorkspace"] = undefined /*out*/;
            inputs["identity"] = undefined /*out*/;
            inputs["imageBuildCompute"] = undefined /*out*/;
            inputs["keyVault"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["notebookInfo"] = undefined /*out*/;
            inputs["privateEndpointConnections"] = undefined /*out*/;
            inputs["privateLinkCount"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["serviceProvisionedResourceGroup"] = undefined /*out*/;
            inputs["sharedPrivateLinkResources"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["storageAccount"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["workspaceId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:machinelearningservices/v20200301:Workspace" }, { type: "azure-native:machinelearningservices:Workspace" }, { type: "azure-nextgen:machinelearningservices:Workspace" }, { type: "azure-native:machinelearningservices/v20180301preview:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20180301preview:Workspace" }, { type: "azure-native:machinelearningservices/v20181119:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20181119:Workspace" }, { type: "azure-native:machinelearningservices/v20190501:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20190501:Workspace" }, { type: "azure-native:machinelearningservices/v20190601:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20190601:Workspace" }, { type: "azure-native:machinelearningservices/v20191101:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20191101:Workspace" }, { type: "azure-native:machinelearningservices/v20200101:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20200101:Workspace" }, { type: "azure-native:machinelearningservices/v20200218preview:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20200218preview:Workspace" }, { type: "azure-native:machinelearningservices/v20200401:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20200401:Workspace" }, { type: "azure-native:machinelearningservices/v20200501preview:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20200501preview:Workspace" }, { type: "azure-native:machinelearningservices/v20200515preview:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20200515preview:Workspace" }, { type: "azure-native:machinelearningservices/v20200601:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20200601:Workspace" }, { type: "azure-native:machinelearningservices/v20200801:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20200801:Workspace" }, { type: "azure-native:machinelearningservices/v20200901preview:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20200901preview:Workspace" }, { type: "azure-native:machinelearningservices/v20210101:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20210101:Workspace" }, { type: "azure-native:machinelearningservices/v20210301preview:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20210301preview:Workspace" }, { type: "azure-native:machinelearningservices/v20210401:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20210401:Workspace" }, { type: "azure-native:machinelearningservices/v20210701:Workspace" }, { type: "azure-nextgen:machinelearningservices/v20210701:Workspace" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Workspace.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Workspace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Workspace(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Workspace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Workspace.__pulumiType;
    }
}
exports.Workspace = Workspace;
/** @internal */
Workspace.__pulumiType = 'azure-native:machinelearningservices/v20200301:Workspace';
