// @pulumi/azure-native/types/enums/streamanalytics/v20160301/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuName = exports.OutputStartMode = exports.OutputErrorPolicy = exports.JsonOutputSerializationFormat = exports.EventsOutOfOrderPolicy = exports.EventSerializationType = exports.Encoding = exports.CompatibilityLevel = void 0;
exports.CompatibilityLevel = {
    CompatibilityLevel_1_0: "1.0",
};
exports.Encoding = {
    UTF8: "UTF8",
};
exports.EventSerializationType = {
    Csv: "Csv",
    Avro: "Avro",
    Json: "Json",
};
exports.EventsOutOfOrderPolicy = {
    Adjust: "Adjust",
    Drop: "Drop",
};
exports.JsonOutputSerializationFormat = {
    LineSeparated: "LineSeparated",
    Array: "Array",
};
exports.OutputErrorPolicy = {
    Stop: "Stop",
    Drop: "Drop",
};
exports.OutputStartMode = {
    JobStartTime: "JobStartTime",
    CustomTime: "CustomTime",
    LastOutputEventTime: "LastOutputEventTime",
};
exports.SkuName = {
    Standard: "Standard",
};
