// @pulumi/azure-native/hanaonazure/hanaInstance.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HanaInstance = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * HANA instance info on Azure (ARM properties and HANA properties)
 * API Version: 2017-11-03-preview.
 */
class HanaInstance extends pulumi.CustomResource {
    /**
     * Create a HanaInstance resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["hanaInstanceName"] = args ? args.hanaInstanceName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["networkProfile"] = args ? args.networkProfile : undefined;
            inputs["osProfile"] = args ? args.osProfile : undefined;
            inputs["partnerNodeId"] = args ? args.partnerNodeId : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["storageProfile"] = args ? args.storageProfile : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["hanaInstanceId"] = undefined /*out*/;
            inputs["hardwareProfile"] = undefined /*out*/;
            inputs["hwRevision"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["powerState"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["proximityPlacementGroup"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["hanaInstanceId"] = undefined /*out*/;
            inputs["hardwareProfile"] = undefined /*out*/;
            inputs["hwRevision"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["networkProfile"] = undefined /*out*/;
            inputs["osProfile"] = undefined /*out*/;
            inputs["partnerNodeId"] = undefined /*out*/;
            inputs["powerState"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["proximityPlacementGroup"] = undefined /*out*/;
            inputs["storageProfile"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:hanaonazure:HanaInstance" }, { type: "azure-native:hanaonazure/v20171103preview:HanaInstance" }, { type: "azure-nextgen:hanaonazure/v20171103preview:HanaInstance" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(HanaInstance.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing HanaInstance resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new HanaInstance(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of HanaInstance.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HanaInstance.__pulumiType;
    }
}
exports.HanaInstance = HanaInstance;
/** @internal */
HanaInstance.__pulumiType = 'azure-native:hanaonazure:HanaInstance';
