// @pulumi/azure-native/costmanagement/cloudConnector.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The Connector model definition
 * API Version: 2019-03-01-preview.
 */
class CloudConnector extends pulumi.CustomResource {
    /**
     * Create a CloudConnector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["billingModel"] = args ? args.billingModel : undefined;
            inputs["connectorName"] = args ? args.connectorName : undefined;
            inputs["credentialsKey"] = args ? args.credentialsKey : undefined;
            inputs["credentialsSecret"] = args ? args.credentialsSecret : undefined;
            inputs["defaultManagementGroupId"] = args ? args.defaultManagementGroupId : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["reportId"] = args ? args.reportId : undefined;
            inputs["subscriptionId"] = args ? args.subscriptionId : undefined;
            inputs["collectionInfo"] = undefined /*out*/;
            inputs["createdOn"] = undefined /*out*/;
            inputs["daysTrialRemaining"] = undefined /*out*/;
            inputs["externalBillingAccountId"] = undefined /*out*/;
            inputs["modifiedOn"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["providerBillingAccountDisplayName"] = undefined /*out*/;
            inputs["providerBillingAccountId"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["billingModel"] = undefined /*out*/;
            inputs["collectionInfo"] = undefined /*out*/;
            inputs["createdOn"] = undefined /*out*/;
            inputs["credentialsKey"] = undefined /*out*/;
            inputs["daysTrialRemaining"] = undefined /*out*/;
            inputs["defaultManagementGroupId"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["externalBillingAccountId"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["modifiedOn"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["providerBillingAccountDisplayName"] = undefined /*out*/;
            inputs["providerBillingAccountId"] = undefined /*out*/;
            inputs["reportId"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["subscriptionId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:costmanagement:CloudConnector" }, { type: "azure-native:costmanagement/v20180801preview:CloudConnector" }, { type: "azure-nextgen:costmanagement/v20180801preview:CloudConnector" }, { type: "azure-native:costmanagement/v20190301preview:CloudConnector" }, { type: "azure-nextgen:costmanagement/v20190301preview:CloudConnector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CloudConnector.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing CloudConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CloudConnector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CloudConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CloudConnector.__pulumiType;
    }
}
exports.CloudConnector = CloudConnector;
/** @internal */
CloudConnector.__pulumiType = 'azure-native:costmanagement:CloudConnector';
