// @pulumi/azure-native/web/v20200601/webAppVnetConnection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppVnetConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Virtual Network information contract.
 */
class WebAppVnetConnection extends pulumi.CustomResource {
    /**
     * Create a WebAppVnetConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["certBlob"] = args ? args.certBlob : undefined;
            inputs["dnsServers"] = args ? args.dnsServers : undefined;
            inputs["isSwift"] = args ? args.isSwift : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["vnetName"] = args ? args.vnetName : undefined;
            inputs["vnetResourceId"] = args ? args.vnetResourceId : undefined;
            inputs["certThumbprint"] = undefined /*out*/;
            inputs["resyncRequired"] = undefined /*out*/;
            inputs["routes"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["certBlob"] = undefined /*out*/;
            inputs["certThumbprint"] = undefined /*out*/;
            inputs["dnsServers"] = undefined /*out*/;
            inputs["isSwift"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["resyncRequired"] = undefined /*out*/;
            inputs["routes"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["vnetResourceId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20200601:WebAppVnetConnection" }, { type: "azure-native:web:WebAppVnetConnection" }, { type: "azure-nextgen:web:WebAppVnetConnection" }, { type: "azure-native:web/v20150801:WebAppVnetConnection" }, { type: "azure-nextgen:web/v20150801:WebAppVnetConnection" }, { type: "azure-native:web/v20160801:WebAppVnetConnection" }, { type: "azure-nextgen:web/v20160801:WebAppVnetConnection" }, { type: "azure-native:web/v20180201:WebAppVnetConnection" }, { type: "azure-nextgen:web/v20180201:WebAppVnetConnection" }, { type: "azure-native:web/v20181101:WebAppVnetConnection" }, { type: "azure-nextgen:web/v20181101:WebAppVnetConnection" }, { type: "azure-native:web/v20190801:WebAppVnetConnection" }, { type: "azure-nextgen:web/v20190801:WebAppVnetConnection" }, { type: "azure-native:web/v20200901:WebAppVnetConnection" }, { type: "azure-nextgen:web/v20200901:WebAppVnetConnection" }, { type: "azure-native:web/v20201001:WebAppVnetConnection" }, { type: "azure-nextgen:web/v20201001:WebAppVnetConnection" }, { type: "azure-native:web/v20201201:WebAppVnetConnection" }, { type: "azure-nextgen:web/v20201201:WebAppVnetConnection" }, { type: "azure-native:web/v20210101:WebAppVnetConnection" }, { type: "azure-nextgen:web/v20210101:WebAppVnetConnection" }, { type: "azure-native:web/v20210115:WebAppVnetConnection" }, { type: "azure-nextgen:web/v20210115:WebAppVnetConnection" }, { type: "azure-native:web/v20210201:WebAppVnetConnection" }, { type: "azure-nextgen:web/v20210201:WebAppVnetConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppVnetConnection.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WebAppVnetConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppVnetConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppVnetConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppVnetConnection.__pulumiType;
    }
}
exports.WebAppVnetConnection = WebAppVnetConnection;
/** @internal */
WebAppVnetConnection.__pulumiType = 'azure-native:web/v20200601:WebAppVnetConnection';
