// @pulumi/azure-native/types/enums/storagesync/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointServiceConnectionStatus = exports.LocalCacheMode = exports.InitialDownloadPolicy = exports.IncomingTrafficPolicy = exports.FeatureStatus = exports.v20200901 = exports.v20200301 = void 0;
// Export sub-modules:
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
exports.FeatureStatus = {
    On: "on",
    Off: "off",
};
exports.IncomingTrafficPolicy = {
    AllowAllTraffic: "AllowAllTraffic",
    AllowVirtualNetworksOnly: "AllowVirtualNetworksOnly",
};
exports.InitialDownloadPolicy = {
    NamespaceOnly: "NamespaceOnly",
    NamespaceThenModifiedFiles: "NamespaceThenModifiedFiles",
    AvoidTieredFiles: "AvoidTieredFiles",
};
exports.LocalCacheMode = {
    DownloadNewAndModifiedFiles: "DownloadNewAndModifiedFiles",
    UpdateLocallyCachedFiles: "UpdateLocallyCachedFiles",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
