// @pulumi/azure-native/synapse/bigDataPool.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BigDataPool = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A Big Data pool
 * API Version: 2021-03-01.
 */
class BigDataPool extends pulumi.CustomResource {
    /**
     * Create a BigDataPool resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["autoPause"] = args ? args.autoPause : undefined;
            inputs["autoScale"] = args ? args.autoScale : undefined;
            inputs["bigDataPoolName"] = args ? args.bigDataPoolName : undefined;
            inputs["cacheSize"] = args ? args.cacheSize : undefined;
            inputs["creationDate"] = args ? args.creationDate : undefined;
            inputs["customLibraries"] = args ? args.customLibraries : undefined;
            inputs["defaultSparkLogFolder"] = args ? args.defaultSparkLogFolder : undefined;
            inputs["dynamicExecutorAllocation"] = args ? args.dynamicExecutorAllocation : undefined;
            inputs["force"] = args ? args.force : undefined;
            inputs["isComputeIsolationEnabled"] = args ? args.isComputeIsolationEnabled : undefined;
            inputs["libraryRequirements"] = args ? args.libraryRequirements : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["nodeCount"] = args ? args.nodeCount : undefined;
            inputs["nodeSize"] = args ? args.nodeSize : undefined;
            inputs["nodeSizeFamily"] = args ? args.nodeSizeFamily : undefined;
            inputs["provisioningState"] = args ? args.provisioningState : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sessionLevelPackagesEnabled"] = args ? args.sessionLevelPackagesEnabled : undefined;
            inputs["sparkConfigProperties"] = args ? args.sparkConfigProperties : undefined;
            inputs["sparkEventsFolder"] = args ? args.sparkEventsFolder : undefined;
            inputs["sparkVersion"] = args ? args.sparkVersion : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["lastSucceededTimestamp"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["autoPause"] = undefined /*out*/;
            inputs["autoScale"] = undefined /*out*/;
            inputs["cacheSize"] = undefined /*out*/;
            inputs["creationDate"] = undefined /*out*/;
            inputs["customLibraries"] = undefined /*out*/;
            inputs["defaultSparkLogFolder"] = undefined /*out*/;
            inputs["dynamicExecutorAllocation"] = undefined /*out*/;
            inputs["isComputeIsolationEnabled"] = undefined /*out*/;
            inputs["lastSucceededTimestamp"] = undefined /*out*/;
            inputs["libraryRequirements"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["nodeCount"] = undefined /*out*/;
            inputs["nodeSize"] = undefined /*out*/;
            inputs["nodeSizeFamily"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["sessionLevelPackagesEnabled"] = undefined /*out*/;
            inputs["sparkConfigProperties"] = undefined /*out*/;
            inputs["sparkEventsFolder"] = undefined /*out*/;
            inputs["sparkVersion"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:synapse:BigDataPool" }, { type: "azure-native:synapse/v20190601preview:BigDataPool" }, { type: "azure-nextgen:synapse/v20190601preview:BigDataPool" }, { type: "azure-native:synapse/v20201201:BigDataPool" }, { type: "azure-nextgen:synapse/v20201201:BigDataPool" }, { type: "azure-native:synapse/v20210301:BigDataPool" }, { type: "azure-nextgen:synapse/v20210301:BigDataPool" }, { type: "azure-native:synapse/v20210401preview:BigDataPool" }, { type: "azure-nextgen:synapse/v20210401preview:BigDataPool" }, { type: "azure-native:synapse/v20210501:BigDataPool" }, { type: "azure-nextgen:synapse/v20210501:BigDataPool" }, { type: "azure-native:synapse/v20210601preview:BigDataPool" }, { type: "azure-nextgen:synapse/v20210601preview:BigDataPool" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BigDataPool.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing BigDataPool resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BigDataPool(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BigDataPool.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BigDataPool.__pulumiType;
    }
}
exports.BigDataPool = BigDataPool;
/** @internal */
BigDataPool.__pulumiType = 'azure-native:synapse:BigDataPool';
