// @pulumi/azure-native/web/v20160801/webAppHybridConnectionSlot.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppHybridConnectionSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Hybrid Connection contract. This is used to configure a Hybrid Connection.
 */
class WebAppHybridConnectionSlot extends pulumi.CustomResource {
    /**
     * Create a WebAppHybridConnectionSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.namespaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'namespaceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            inputs["hostname"] = args ? args.hostname : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["namespaceName"] = args ? args.namespaceName : undefined;
            inputs["port"] = args ? args.port : undefined;
            inputs["relayArmUri"] = args ? args.relayArmUri : undefined;
            inputs["relayName"] = args ? args.relayName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sendKeyName"] = args ? args.sendKeyName : undefined;
            inputs["sendKeyValue"] = args ? args.sendKeyValue : undefined;
            inputs["serviceBusNamespace"] = args ? args.serviceBusNamespace : undefined;
            inputs["serviceBusSuffix"] = args ? args.serviceBusSuffix : undefined;
            inputs["slot"] = args ? args.slot : undefined;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["hostname"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["port"] = undefined /*out*/;
            inputs["relayArmUri"] = undefined /*out*/;
            inputs["relayName"] = undefined /*out*/;
            inputs["sendKeyName"] = undefined /*out*/;
            inputs["sendKeyValue"] = undefined /*out*/;
            inputs["serviceBusNamespace"] = undefined /*out*/;
            inputs["serviceBusSuffix"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20160801:WebAppHybridConnectionSlot" }, { type: "azure-native:web:WebAppHybridConnectionSlot" }, { type: "azure-nextgen:web:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20180201:WebAppHybridConnectionSlot" }, { type: "azure-nextgen:web/v20180201:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20181101:WebAppHybridConnectionSlot" }, { type: "azure-nextgen:web/v20181101:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20190801:WebAppHybridConnectionSlot" }, { type: "azure-nextgen:web/v20190801:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20200601:WebAppHybridConnectionSlot" }, { type: "azure-nextgen:web/v20200601:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20200901:WebAppHybridConnectionSlot" }, { type: "azure-nextgen:web/v20200901:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20201001:WebAppHybridConnectionSlot" }, { type: "azure-nextgen:web/v20201001:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20201201:WebAppHybridConnectionSlot" }, { type: "azure-nextgen:web/v20201201:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20210101:WebAppHybridConnectionSlot" }, { type: "azure-nextgen:web/v20210101:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20210115:WebAppHybridConnectionSlot" }, { type: "azure-nextgen:web/v20210115:WebAppHybridConnectionSlot" }, { type: "azure-native:web/v20210201:WebAppHybridConnectionSlot" }, { type: "azure-nextgen:web/v20210201:WebAppHybridConnectionSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppHybridConnectionSlot.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WebAppHybridConnectionSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppHybridConnectionSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppHybridConnectionSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppHybridConnectionSlot.__pulumiType;
    }
}
exports.WebAppHybridConnectionSlot = WebAppHybridConnectionSlot;
/** @internal */
WebAppHybridConnectionSlot.__pulumiType = 'azure-native:web/v20160801:WebAppHybridConnectionSlot';
