// @pulumi/azure-native/devtestlab/v20150521preview/labResource.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LabResource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A lab.
 */
class LabResource extends pulumi.CustomResource {
    /**
     * Create a LabResource resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["artifactsStorageAccount"] = args ? args.artifactsStorageAccount : undefined;
            inputs["createdDate"] = args ? args.createdDate : undefined;
            inputs["defaultStorageAccount"] = args ? args.defaultStorageAccount : undefined;
            inputs["defaultVirtualNetworkId"] = args ? args.defaultVirtualNetworkId : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["labStorageType"] = args ? args.labStorageType : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["provisioningState"] = args ? args.provisioningState : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["storageAccounts"] = args ? args.storageAccounts : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["type"] = args ? args.type : undefined;
            inputs["vaultName"] = args ? args.vaultName : undefined;
        }
        else {
            inputs["artifactsStorageAccount"] = undefined /*out*/;
            inputs["createdDate"] = undefined /*out*/;
            inputs["defaultStorageAccount"] = undefined /*out*/;
            inputs["defaultVirtualNetworkId"] = undefined /*out*/;
            inputs["labStorageType"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["storageAccounts"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["vaultName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:devtestlab/v20150521preview:LabResource" }, { type: "azure-native:devtestlab:LabResource" }, { type: "azure-nextgen:devtestlab:LabResource" }, { type: "azure-native:devtestlab/v20160515:LabResource" }, { type: "azure-nextgen:devtestlab/v20160515:LabResource" }, { type: "azure-native:devtestlab/v20180915:LabResource" }, { type: "azure-nextgen:devtestlab/v20180915:LabResource" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(LabResource.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing LabResource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LabResource(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LabResource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LabResource.__pulumiType;
    }
}
exports.LabResource = LabResource;
/** @internal */
LabResource.__pulumiType = 'azure-native:devtestlab/v20150521preview:LabResource';
