// @pulumi/azure-native/devtestlab/v20150521preview/scheduleResource.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleResource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A schedule.
 */
class ScheduleResource extends pulumi.CustomResource {
    /**
     * Create a ScheduleResource resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.labName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'labName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["dailyRecurrence"] = args ? args.dailyRecurrence : undefined;
            inputs["hourlyRecurrence"] = args ? args.hourlyRecurrence : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["labName"] = args ? args.labName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["provisioningState"] = args ? args.provisioningState : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["status"] = args ? args.status : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["taskType"] = args ? args.taskType : undefined;
            inputs["timeZoneId"] = args ? args.timeZoneId : undefined;
            inputs["type"] = args ? args.type : undefined;
            inputs["weeklyRecurrence"] = args ? args.weeklyRecurrence : undefined;
        }
        else {
            inputs["dailyRecurrence"] = undefined /*out*/;
            inputs["hourlyRecurrence"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["taskType"] = undefined /*out*/;
            inputs["timeZoneId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["weeklyRecurrence"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:devtestlab/v20150521preview:ScheduleResource" }, { type: "azure-native:devtestlab:ScheduleResource" }, { type: "azure-nextgen:devtestlab:ScheduleResource" }, { type: "azure-native:devtestlab/v20160515:ScheduleResource" }, { type: "azure-nextgen:devtestlab/v20160515:ScheduleResource" }, { type: "azure-native:devtestlab/v20180915:ScheduleResource" }, { type: "azure-nextgen:devtestlab/v20180915:ScheduleResource" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ScheduleResource.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ScheduleResource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ScheduleResource(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ScheduleResource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ScheduleResource.__pulumiType;
    }
}
exports.ScheduleResource = ScheduleResource;
/** @internal */
ScheduleResource.__pulumiType = 'azure-native:devtestlab/v20150521preview:ScheduleResource';
