// @pulumi/azure-native/intune/v20150114preview/androidMAMPolicyByName.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AndroidMAMPolicyByName = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Android Policy entity for Intune MAM.
 */
class AndroidMAMPolicyByName extends pulumi.CustomResource {
    /**
     * Create a AndroidMAMPolicyByName resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.friendlyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'friendlyName'");
            }
            if ((!args || args.hostName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hostName'");
            }
            inputs["accessRecheckOfflineTimeout"] = args ? args.accessRecheckOfflineTimeout : undefined;
            inputs["accessRecheckOnlineTimeout"] = args ? args.accessRecheckOnlineTimeout : undefined;
            inputs["appSharingFromLevel"] = (_a = (args ? args.appSharingFromLevel : undefined)) !== null && _a !== void 0 ? _a : "none";
            inputs["appSharingToLevel"] = (_b = (args ? args.appSharingToLevel : undefined)) !== null && _b !== void 0 ? _b : "none";
            inputs["authentication"] = (_c = (args ? args.authentication : undefined)) !== null && _c !== void 0 ? _c : "required";
            inputs["clipboardSharingLevel"] = (_d = (args ? args.clipboardSharingLevel : undefined)) !== null && _d !== void 0 ? _d : "blocked";
            inputs["dataBackup"] = (_e = (args ? args.dataBackup : undefined)) !== null && _e !== void 0 ? _e : "allow";
            inputs["description"] = args ? args.description : undefined;
            inputs["deviceCompliance"] = (_f = (args ? args.deviceCompliance : undefined)) !== null && _f !== void 0 ? _f : "enable";
            inputs["fileEncryption"] = (_g = (args ? args.fileEncryption : undefined)) !== null && _g !== void 0 ? _g : "required";
            inputs["fileSharingSaveAs"] = (_h = (args ? args.fileSharingSaveAs : undefined)) !== null && _h !== void 0 ? _h : "allow";
            inputs["friendlyName"] = args ? args.friendlyName : undefined;
            inputs["hostName"] = args ? args.hostName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["managedBrowser"] = (_j = (args ? args.managedBrowser : undefined)) !== null && _j !== void 0 ? _j : "required";
            inputs["offlineWipeTimeout"] = args ? args.offlineWipeTimeout : undefined;
            inputs["pin"] = (_k = (args ? args.pin : undefined)) !== null && _k !== void 0 ? _k : "required";
            inputs["pinNumRetry"] = args ? args.pinNumRetry : undefined;
            inputs["policyName"] = args ? args.policyName : undefined;
            inputs["screenCapture"] = (_l = (args ? args.screenCapture : undefined)) !== null && _l !== void 0 ? _l : "allow";
            inputs["tags"] = args ? args.tags : undefined;
            inputs["groupStatus"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["numOfApps"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["accessRecheckOfflineTimeout"] = undefined /*out*/;
            inputs["accessRecheckOnlineTimeout"] = undefined /*out*/;
            inputs["appSharingFromLevel"] = undefined /*out*/;
            inputs["appSharingToLevel"] = undefined /*out*/;
            inputs["authentication"] = undefined /*out*/;
            inputs["clipboardSharingLevel"] = undefined /*out*/;
            inputs["dataBackup"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["deviceCompliance"] = undefined /*out*/;
            inputs["fileEncryption"] = undefined /*out*/;
            inputs["fileSharingSaveAs"] = undefined /*out*/;
            inputs["friendlyName"] = undefined /*out*/;
            inputs["groupStatus"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["managedBrowser"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["numOfApps"] = undefined /*out*/;
            inputs["offlineWipeTimeout"] = undefined /*out*/;
            inputs["pin"] = undefined /*out*/;
            inputs["pinNumRetry"] = undefined /*out*/;
            inputs["screenCapture"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:intune/v20150114preview:AndroidMAMPolicyByName" }, { type: "azure-native:intune:AndroidMAMPolicyByName" }, { type: "azure-nextgen:intune:AndroidMAMPolicyByName" }, { type: "azure-native:intune/v20150114privatepreview:AndroidMAMPolicyByName" }, { type: "azure-nextgen:intune/v20150114privatepreview:AndroidMAMPolicyByName" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AndroidMAMPolicyByName.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing AndroidMAMPolicyByName resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AndroidMAMPolicyByName(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AndroidMAMPolicyByName.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AndroidMAMPolicyByName.__pulumiType;
    }
}
exports.AndroidMAMPolicyByName = AndroidMAMPolicyByName;
/** @internal */
AndroidMAMPolicyByName.__pulumiType = 'azure-native:intune/v20150114preview:AndroidMAMPolicyByName';
