// @pulumi/azure-native/security/v20200101/assessmentMetadataInSubscription.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssessmentMetadataInSubscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Security assessment metadata
 */
class AssessmentMetadataInSubscription extends pulumi.CustomResource {
    /**
     * Create a AssessmentMetadataInSubscription resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.assessmentType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'assessmentType'");
            }
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.severity === undefined) && !opts.urn) {
                throw new Error("Missing required property 'severity'");
            }
            inputs["assessmentMetadataName"] = args ? args.assessmentMetadataName : undefined;
            inputs["assessmentType"] = args ? args.assessmentType : undefined;
            inputs["categories"] = args ? args.categories : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["implementationEffort"] = args ? args.implementationEffort : undefined;
            inputs["partnerData"] = args ? args.partnerData : undefined;
            inputs["preview"] = args ? args.preview : undefined;
            inputs["remediationDescription"] = args ? args.remediationDescription : undefined;
            inputs["severity"] = args ? args.severity : undefined;
            inputs["threats"] = args ? args.threats : undefined;
            inputs["userImpact"] = args ? args.userImpact : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["policyDefinitionId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["assessmentType"] = undefined /*out*/;
            inputs["categories"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["implementationEffort"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["partnerData"] = undefined /*out*/;
            inputs["policyDefinitionId"] = undefined /*out*/;
            inputs["preview"] = undefined /*out*/;
            inputs["remediationDescription"] = undefined /*out*/;
            inputs["severity"] = undefined /*out*/;
            inputs["threats"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["userImpact"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:security/v20200101:AssessmentMetadataInSubscription" }, { type: "azure-native:security:AssessmentMetadataInSubscription" }, { type: "azure-nextgen:security:AssessmentMetadataInSubscription" }, { type: "azure-native:security/v20190101preview:AssessmentMetadataInSubscription" }, { type: "azure-nextgen:security/v20190101preview:AssessmentMetadataInSubscription" }, { type: "azure-native:security/v20210601:AssessmentMetadataInSubscription" }, { type: "azure-nextgen:security/v20210601:AssessmentMetadataInSubscription" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AssessmentMetadataInSubscription.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing AssessmentMetadataInSubscription resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AssessmentMetadataInSubscription(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AssessmentMetadataInSubscription.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AssessmentMetadataInSubscription.__pulumiType;
    }
}
exports.AssessmentMetadataInSubscription = AssessmentMetadataInSubscription;
/** @internal */
AssessmentMetadataInSubscription.__pulumiType = 'azure-native:security/v20200101:AssessmentMetadataInSubscription';
