// @pulumi/azure-native/security/v20200101/jitNetworkAccessPolicy.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.JitNetworkAccessPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
class JitNetworkAccessPolicy extends pulumi.CustomResource {
    /**
     * Create a JitNetworkAccessPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.ascLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ascLocation'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.virtualMachines === undefined) && !opts.urn) {
                throw new Error("Missing required property 'virtualMachines'");
            }
            inputs["ascLocation"] = args ? args.ascLocation : undefined;
            inputs["jitNetworkAccessPolicyName"] = args ? args.jitNetworkAccessPolicyName : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["requests"] = args ? args.requests : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["virtualMachines"] = args ? args.virtualMachines : undefined;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["requests"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["virtualMachines"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:security/v20200101:JitNetworkAccessPolicy" }, { type: "azure-native:security:JitNetworkAccessPolicy" }, { type: "azure-nextgen:security:JitNetworkAccessPolicy" }, { type: "azure-native:security/v20150601preview:JitNetworkAccessPolicy" }, { type: "azure-nextgen:security/v20150601preview:JitNetworkAccessPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(JitNetworkAccessPolicy.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing JitNetworkAccessPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new JitNetworkAccessPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of JitNetworkAccessPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === JitNetworkAccessPolicy.__pulumiType;
    }
}
exports.JitNetworkAccessPolicy = JitNetworkAccessPolicy;
/** @internal */
JitNetworkAccessPolicy.__pulumiType = 'azure-native:security/v20200101:JitNetworkAccessPolicy';
