// @pulumi/azure-native/storage/storageAccountStaticWebsite.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageAccountStaticWebsite = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Enables the static website feature of a storage account.
 */
class StorageAccountStaticWebsite extends pulumi.CustomResource {
    /**
     * Create a StorageAccountStaticWebsite resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["error404Document"] = args ? args.error404Document : undefined;
            inputs["indexDocument"] = args ? args.indexDocument : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["containerName"] = undefined /*out*/;
        }
        else {
            inputs["containerName"] = undefined /*out*/;
            inputs["error404Document"] = undefined /*out*/;
            inputs["indexDocument"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(StorageAccountStaticWebsite.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing StorageAccountStaticWebsite resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StorageAccountStaticWebsite(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StorageAccountStaticWebsite.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StorageAccountStaticWebsite.__pulumiType;
    }
}
exports.StorageAccountStaticWebsite = StorageAccountStaticWebsite;
/** @internal */
StorageAccountStaticWebsite.__pulumiType = 'azure-native:storage:StorageAccountStaticWebsite';
