// @pulumi/azure-native/devtestlab/v20180915/customImage.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomImage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A custom image.
 */
class CustomImage extends pulumi.CustomResource {
    /**
     * Create a CustomImage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.labName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'labName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["author"] = args ? args.author : undefined;
            inputs["customImagePlan"] = args ? args.customImagePlan : undefined;
            inputs["dataDiskStorageInfo"] = args ? args.dataDiskStorageInfo : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["isPlanAuthorized"] = args ? args.isPlanAuthorized : undefined;
            inputs["labName"] = args ? args.labName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["managedImageId"] = args ? args.managedImageId : undefined;
            inputs["managedSnapshotId"] = args ? args.managedSnapshotId : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["vhd"] = args ? args.vhd : undefined;
            inputs["vm"] = args ? args.vm : undefined;
            inputs["creationDate"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["uniqueIdentifier"] = undefined /*out*/;
        }
        else {
            inputs["author"] = undefined /*out*/;
            inputs["creationDate"] = undefined /*out*/;
            inputs["customImagePlan"] = undefined /*out*/;
            inputs["dataDiskStorageInfo"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["isPlanAuthorized"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["managedImageId"] = undefined /*out*/;
            inputs["managedSnapshotId"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["uniqueIdentifier"] = undefined /*out*/;
            inputs["vhd"] = undefined /*out*/;
            inputs["vm"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:devtestlab/v20180915:CustomImage" }, { type: "azure-native:devtestlab:CustomImage" }, { type: "azure-nextgen:devtestlab:CustomImage" }, { type: "azure-native:devtestlab/v20150521preview:CustomImage" }, { type: "azure-nextgen:devtestlab/v20150521preview:CustomImage" }, { type: "azure-native:devtestlab/v20160515:CustomImage" }, { type: "azure-nextgen:devtestlab/v20160515:CustomImage" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CustomImage.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing CustomImage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CustomImage(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CustomImage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomImage.__pulumiType;
    }
}
exports.CustomImage = CustomImage;
/** @internal */
CustomImage.__pulumiType = 'azure-native:devtestlab/v20180915:CustomImage';
