// @pulumi/azure-native/devtestlab/v20180915/disk.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Disk = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A Disk.
 */
class Disk extends pulumi.CustomResource {
    /**
     * Create a Disk resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.labName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'labName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.userName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'userName'");
            }
            inputs["diskBlobName"] = args ? args.diskBlobName : undefined;
            inputs["diskSizeGiB"] = args ? args.diskSizeGiB : undefined;
            inputs["diskType"] = args ? args.diskType : undefined;
            inputs["diskUri"] = args ? args.diskUri : undefined;
            inputs["hostCaching"] = args ? args.hostCaching : undefined;
            inputs["labName"] = args ? args.labName : undefined;
            inputs["leasedByLabVmId"] = args ? args.leasedByLabVmId : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["managedDiskId"] = args ? args.managedDiskId : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["storageAccountId"] = args ? args.storageAccountId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["userName"] = args ? args.userName : undefined;
            inputs["createdDate"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["uniqueIdentifier"] = undefined /*out*/;
        }
        else {
            inputs["createdDate"] = undefined /*out*/;
            inputs["diskBlobName"] = undefined /*out*/;
            inputs["diskSizeGiB"] = undefined /*out*/;
            inputs["diskType"] = undefined /*out*/;
            inputs["diskUri"] = undefined /*out*/;
            inputs["hostCaching"] = undefined /*out*/;
            inputs["leasedByLabVmId"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["managedDiskId"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["storageAccountId"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["uniqueIdentifier"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:devtestlab/v20180915:Disk" }, { type: "azure-native:devtestlab:Disk" }, { type: "azure-nextgen:devtestlab:Disk" }, { type: "azure-native:devtestlab/v20160515:Disk" }, { type: "azure-nextgen:devtestlab/v20160515:Disk" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Disk.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Disk resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Disk(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Disk.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Disk.__pulumiType;
    }
}
exports.Disk = Disk;
/** @internal */
Disk.__pulumiType = 'azure-native:devtestlab/v20180915:Disk';
