// @pulumi/azure-native/policyinsights/v20190701/remediationAtManagementGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemediationAtManagementGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The remediation definition.
 */
class RemediationAtManagementGroup extends pulumi.CustomResource {
    /**
     * Create a RemediationAtManagementGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.managementGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managementGroupId'");
            }
            if ((!args || args.managementGroupsNamespace === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managementGroupsNamespace'");
            }
            inputs["filters"] = args ? args.filters : undefined;
            inputs["managementGroupId"] = args ? args.managementGroupId : undefined;
            inputs["managementGroupsNamespace"] = args ? args.managementGroupsNamespace : undefined;
            inputs["policyAssignmentId"] = args ? args.policyAssignmentId : undefined;
            inputs["policyDefinitionReferenceId"] = args ? args.policyDefinitionReferenceId : undefined;
            inputs["remediationName"] = args ? args.remediationName : undefined;
            inputs["resourceDiscoveryMode"] = args ? args.resourceDiscoveryMode : undefined;
            inputs["createdOn"] = undefined /*out*/;
            inputs["deploymentStatus"] = undefined /*out*/;
            inputs["lastUpdatedOn"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["createdOn"] = undefined /*out*/;
            inputs["deploymentStatus"] = undefined /*out*/;
            inputs["filters"] = undefined /*out*/;
            inputs["lastUpdatedOn"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["policyAssignmentId"] = undefined /*out*/;
            inputs["policyDefinitionReferenceId"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["resourceDiscoveryMode"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:policyinsights/v20190701:RemediationAtManagementGroup" }, { type: "azure-native:policyinsights:RemediationAtManagementGroup" }, { type: "azure-nextgen:policyinsights:RemediationAtManagementGroup" }, { type: "azure-native:policyinsights/v20180701preview:RemediationAtManagementGroup" }, { type: "azure-nextgen:policyinsights/v20180701preview:RemediationAtManagementGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RemediationAtManagementGroup.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing RemediationAtManagementGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RemediationAtManagementGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RemediationAtManagementGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RemediationAtManagementGroup.__pulumiType;
    }
}
exports.RemediationAtManagementGroup = RemediationAtManagementGroup;
/** @internal */
RemediationAtManagementGroup.__pulumiType = 'azure-native:policyinsights/v20190701:RemediationAtManagementGroup';
