// @pulumi/azure-native/blockchain/v20180601preview/blockchainMember.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlockchainMember = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Payload of the blockchain member which is exposed in the request/response of the resource provider.
 */
class BlockchainMember extends pulumi.CustomResource {
    /**
     * Create a BlockchainMember resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["blockchainMemberName"] = args ? args.blockchainMemberName : undefined;
            inputs["consortium"] = args ? args.consortium : undefined;
            inputs["consortiumManagementAccountPassword"] = args ? args.consortiumManagementAccountPassword : undefined;
            inputs["consortiumMemberDisplayName"] = args ? args.consortiumMemberDisplayName : undefined;
            inputs["consortiumRole"] = args ? args.consortiumRole : undefined;
            inputs["firewallRules"] = args ? args.firewallRules : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["password"] = args ? args.password : undefined;
            inputs["protocol"] = args ? args.protocol : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["validatorNodesSku"] = args ? args.validatorNodesSku : undefined;
            inputs["consortiumManagementAccountAddress"] = undefined /*out*/;
            inputs["dns"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["publicKey"] = undefined /*out*/;
            inputs["rootContractAddress"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["userName"] = undefined /*out*/;
        }
        else {
            inputs["consortium"] = undefined /*out*/;
            inputs["consortiumManagementAccountAddress"] = undefined /*out*/;
            inputs["consortiumManagementAccountPassword"] = undefined /*out*/;
            inputs["consortiumMemberDisplayName"] = undefined /*out*/;
            inputs["consortiumRole"] = undefined /*out*/;
            inputs["dns"] = undefined /*out*/;
            inputs["firewallRules"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["password"] = undefined /*out*/;
            inputs["protocol"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["publicKey"] = undefined /*out*/;
            inputs["rootContractAddress"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["userName"] = undefined /*out*/;
            inputs["validatorNodesSku"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:blockchain/v20180601preview:BlockchainMember" }, { type: "azure-native:blockchain:BlockchainMember" }, { type: "azure-nextgen:blockchain:BlockchainMember" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BlockchainMember.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing BlockchainMember resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BlockchainMember(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BlockchainMember.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BlockchainMember.__pulumiType;
    }
}
exports.BlockchainMember = BlockchainMember;
/** @internal */
BlockchainMember.__pulumiType = 'azure-native:blockchain/v20180601preview:BlockchainMember';
