// @pulumi/azure-native/healthcareapis/iotConnectorFhirDestination.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IotConnectorFhirDestination = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * IoT Connector FHIR destination definition.
 * API Version: 2021-06-01-preview.
 */
class IotConnectorFhirDestination extends pulumi.CustomResource {
    /**
     * Create a IotConnectorFhirDestination resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fhirMapping === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fhirMapping'");
            }
            if ((!args || args.fhirServiceResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fhirServiceResourceId'");
            }
            if ((!args || args.iotConnectorName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iotConnectorName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceIdentityResolutionType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceIdentityResolutionType'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["etag"] = args ? args.etag : undefined;
            inputs["fhirDestinationName"] = args ? args.fhirDestinationName : undefined;
            inputs["fhirMapping"] = args ? args.fhirMapping : undefined;
            inputs["fhirServiceResourceId"] = args ? args.fhirServiceResourceId : undefined;
            inputs["iotConnectorName"] = args ? args.iotConnectorName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["resourceIdentityResolutionType"] = args ? args.resourceIdentityResolutionType : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["etag"] = undefined /*out*/;
            inputs["fhirMapping"] = undefined /*out*/;
            inputs["fhirServiceResourceId"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["resourceIdentityResolutionType"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:healthcareapis:IotConnectorFhirDestination" }, { type: "azure-native:healthcareapis/v20210601preview:IotConnectorFhirDestination" }, { type: "azure-nextgen:healthcareapis/v20210601preview:IotConnectorFhirDestination" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IotConnectorFhirDestination.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing IotConnectorFhirDestination resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IotConnectorFhirDestination(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IotConnectorFhirDestination.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IotConnectorFhirDestination.__pulumiType;
    }
}
exports.IotConnectorFhirDestination = IotConnectorFhirDestination;
/** @internal */
IotConnectorFhirDestination.__pulumiType = 'azure-native:healthcareapis:IotConnectorFhirDestination';
