// @pulumi/azure-native/migrate/v20180202/assessment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Assessment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An assessment created for a group in the Migration project.
 */
class Assessment extends pulumi.CustomResource {
    /**
     * Create a Assessment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.azureHybridUseBenefit === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azureHybridUseBenefit'");
            }
            if ((!args || args.azureLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azureLocation'");
            }
            if ((!args || args.azureOfferCode === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azureOfferCode'");
            }
            if ((!args || args.azurePricingTier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azurePricingTier'");
            }
            if ((!args || args.azureStorageRedundancy === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azureStorageRedundancy'");
            }
            if ((!args || args.currency === undefined) && !opts.urn) {
                throw new Error("Missing required property 'currency'");
            }
            if ((!args || args.discountPercentage === undefined) && !opts.urn) {
                throw new Error("Missing required property 'discountPercentage'");
            }
            if ((!args || args.groupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'groupName'");
            }
            if ((!args || args.percentile === undefined) && !opts.urn) {
                throw new Error("Missing required property 'percentile'");
            }
            if ((!args || args.projectName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'projectName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.scalingFactor === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scalingFactor'");
            }
            if ((!args || args.sizingCriterion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sizingCriterion'");
            }
            if ((!args || args.stage === undefined) && !opts.urn) {
                throw new Error("Missing required property 'stage'");
            }
            if ((!args || args.timeRange === undefined) && !opts.urn) {
                throw new Error("Missing required property 'timeRange'");
            }
            inputs["assessmentName"] = args ? args.assessmentName : undefined;
            inputs["azureHybridUseBenefit"] = args ? args.azureHybridUseBenefit : undefined;
            inputs["azureLocation"] = args ? args.azureLocation : undefined;
            inputs["azureOfferCode"] = args ? args.azureOfferCode : undefined;
            inputs["azurePricingTier"] = args ? args.azurePricingTier : undefined;
            inputs["azureStorageRedundancy"] = args ? args.azureStorageRedundancy : undefined;
            inputs["currency"] = args ? args.currency : undefined;
            inputs["discountPercentage"] = args ? args.discountPercentage : undefined;
            inputs["eTag"] = args ? args.eTag : undefined;
            inputs["groupName"] = args ? args.groupName : undefined;
            inputs["percentile"] = args ? args.percentile : undefined;
            inputs["projectName"] = args ? args.projectName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["scalingFactor"] = args ? args.scalingFactor : undefined;
            inputs["sizingCriterion"] = args ? args.sizingCriterion : undefined;
            inputs["stage"] = args ? args.stage : undefined;
            inputs["timeRange"] = args ? args.timeRange : undefined;
            inputs["confidenceRatingInPercentage"] = undefined /*out*/;
            inputs["createdTimestamp"] = undefined /*out*/;
            inputs["monthlyBandwidthCost"] = undefined /*out*/;
            inputs["monthlyComputeCost"] = undefined /*out*/;
            inputs["monthlyStorageCost"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["numberOfMachines"] = undefined /*out*/;
            inputs["pricesTimestamp"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["updatedTimestamp"] = undefined /*out*/;
        }
        else {
            inputs["azureHybridUseBenefit"] = undefined /*out*/;
            inputs["azureLocation"] = undefined /*out*/;
            inputs["azureOfferCode"] = undefined /*out*/;
            inputs["azurePricingTier"] = undefined /*out*/;
            inputs["azureStorageRedundancy"] = undefined /*out*/;
            inputs["confidenceRatingInPercentage"] = undefined /*out*/;
            inputs["createdTimestamp"] = undefined /*out*/;
            inputs["currency"] = undefined /*out*/;
            inputs["discountPercentage"] = undefined /*out*/;
            inputs["eTag"] = undefined /*out*/;
            inputs["monthlyBandwidthCost"] = undefined /*out*/;
            inputs["monthlyComputeCost"] = undefined /*out*/;
            inputs["monthlyStorageCost"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["numberOfMachines"] = undefined /*out*/;
            inputs["percentile"] = undefined /*out*/;
            inputs["pricesTimestamp"] = undefined /*out*/;
            inputs["scalingFactor"] = undefined /*out*/;
            inputs["sizingCriterion"] = undefined /*out*/;
            inputs["stage"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["timeRange"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["updatedTimestamp"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:migrate/v20180202:Assessment" }, { type: "azure-native:migrate/v20171111preview:Assessment" }, { type: "azure-nextgen:migrate/v20171111preview:Assessment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Assessment.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Assessment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Assessment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Assessment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Assessment.__pulumiType;
    }
}
exports.Assessment = Assessment;
/** @internal */
Assessment.__pulumiType = 'azure-native:migrate/v20180202:Assessment';
