// @pulumi/azure-native/analysisservices/v20170801beta/serverDetails.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerDetails = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents an instance of an Analysis Services resource.
 */
class ServerDetails extends pulumi.CustomResource {
    /**
     * Create a ServerDetails resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            inputs["asAdministrators"] = args ? args.asAdministrators : undefined;
            inputs["backupBlobContainerUri"] = args ? args.backupBlobContainerUri : undefined;
            inputs["gatewayDetails"] = args ? args.gatewayDetails : undefined;
            inputs["ipV4FirewallSettings"] = args ? args.ipV4FirewallSettings : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["managedMode"] = (_a = (args ? args.managedMode : undefined)) !== null && _a !== void 0 ? _a : 1;
            inputs["querypoolConnectionMode"] = (_b = (args ? args.querypoolConnectionMode : undefined)) !== null && _b !== void 0 ? _b : "All";
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serverMonitorMode"] = (_c = (args ? args.serverMonitorMode : undefined)) !== null && _c !== void 0 ? _c : 1;
            inputs["serverName"] = args ? args.serverName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["serverFullName"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["asAdministrators"] = undefined /*out*/;
            inputs["backupBlobContainerUri"] = undefined /*out*/;
            inputs["gatewayDetails"] = undefined /*out*/;
            inputs["ipV4FirewallSettings"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["managedMode"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["querypoolConnectionMode"] = undefined /*out*/;
            inputs["serverFullName"] = undefined /*out*/;
            inputs["serverMonitorMode"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["state"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:analysisservices/v20170801beta:ServerDetails" }, { type: "azure-native:analysisservices:ServerDetails" }, { type: "azure-nextgen:analysisservices:ServerDetails" }, { type: "azure-native:analysisservices/v20160516:ServerDetails" }, { type: "azure-nextgen:analysisservices/v20160516:ServerDetails" }, { type: "azure-native:analysisservices/v20170714:ServerDetails" }, { type: "azure-nextgen:analysisservices/v20170714:ServerDetails" }, { type: "azure-native:analysisservices/v20170801:ServerDetails" }, { type: "azure-nextgen:analysisservices/v20170801:ServerDetails" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ServerDetails.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ServerDetails resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ServerDetails(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ServerDetails.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServerDetails.__pulumiType;
    }
}
exports.ServerDetails = ServerDetails;
/** @internal */
ServerDetails.__pulumiType = 'azure-native:analysisservices/v20170801beta:ServerDetails';
