// @pulumi/azure-native/types/enums/blueprint/v20171111preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplateParameterType = exports.ManagedServiceIdentityType = exports.BlueprintTargetScope = exports.AssignmentLockMode = exports.ArtifactKind = void 0;
exports.ArtifactKind = {
    Template: "template",
    RoleAssignment: "roleAssignment",
    PolicyAssignment: "policyAssignment",
};
exports.AssignmentLockMode = {
    None: "None",
    AllResources: "AllResources",
};
exports.BlueprintTargetScope = {
    Subscription: "subscription",
    ManagementGroup: "managementGroup",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.TemplateParameterType = {
    String: "string",
    Array: "array",
    Bool: "bool",
    Int: "int",
    Object: "object",
    SecureObject: "secureObject",
    SecureString: "secureString",
};
