// @pulumi/azure-native/types/enums/machinelearning/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParameterType = exports.OutputPortType = exports.InputPortType = exports.DiagnosticsLevel = exports.ColumnType = exports.ColumnFormat = exports.AssetType = exports.v20170101 = exports.v20160501preview = void 0;
// Export sub-modules:
const v20160501preview = require("./v20160501preview");
exports.v20160501preview = v20160501preview;
const v20170101 = require("./v20170101");
exports.v20170101 = v20170101;
exports.AssetType = {
    Module: "Module",
    Resource: "Resource",
};
exports.ColumnFormat = {
    Byte: "Byte",
    Char: "Char",
    Complex64: "Complex64",
    Complex128: "Complex128",
    Date_time: "Date-time",
    Date_timeOffset: "Date-timeOffset",
    Double: "Double",
    Duration: "Duration",
    Float: "Float",
    Int8: "Int8",
    Int16: "Int16",
    Int32: "Int32",
    Int64: "Int64",
    Uint8: "Uint8",
    Uint16: "Uint16",
    Uint32: "Uint32",
    Uint64: "Uint64",
};
exports.ColumnType = {
    Boolean: "Boolean",
    Integer: "Integer",
    Number: "Number",
    String: "String",
};
exports.DiagnosticsLevel = {
    None: "None",
    Error: "Error",
    All: "All",
};
exports.InputPortType = {
    Dataset: "Dataset",
};
exports.OutputPortType = {
    Dataset: "Dataset",
};
exports.ParameterType = {
    String: "String",
    Int: "Int",
    Float: "Float",
    Enumerated: "Enumerated",
    Script: "Script",
    Mode: "Mode",
    Credential: "Credential",
    Boolean: "Boolean",
    Double: "Double",
    ColumnPicker: "ColumnPicker",
    ParameterRange: "ParameterRange",
    DataGatewayName: "DataGatewayName",
};
