// @pulumi/azure-native/types/enums/healthcareapis/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.ManagedServiceIdentityType = exports.Kind = exports.IotIdentityResolutionType = exports.FhirServiceKind = exports.v20210601preview = exports.v20210111 = exports.v20200330 = exports.v20200315 = exports.v20190916 = exports.v20180820preview = void 0;
// Export sub-modules:
const v20180820preview = require("./v20180820preview");
exports.v20180820preview = v20180820preview;
const v20190916 = require("./v20190916");
exports.v20190916 = v20190916;
const v20200315 = require("./v20200315");
exports.v20200315 = v20200315;
const v20200330 = require("./v20200330");
exports.v20200330 = v20200330;
const v20210111 = require("./v20210111");
exports.v20210111 = v20210111;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
exports.FhirServiceKind = {
    Fhir_Stu3: "fhir-Stu3",
    Fhir_R4: "fhir-R4",
};
exports.IotIdentityResolutionType = {
    Create: "Create",
    Lookup: "Lookup",
};
exports.Kind = {
    Fhir: "fhir",
    Fhir_Stu3: "fhir-Stu3",
    Fhir_R4: "fhir-R4",
};
exports.ManagedServiceIdentityType = {
    SystemAssigned: "SystemAssigned",
    None: "None",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
