// @pulumi/azure-native/network/v20181001/virtualNetworkTap.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualNetworkTap = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Virtual Network Tap resource
 */
class VirtualNetworkTap extends pulumi.CustomResource {
    /**
     * Create a VirtualNetworkTap resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["destinationLoadBalancerFrontEndIPConfiguration"] = args ? args.destinationLoadBalancerFrontEndIPConfiguration : undefined;
            inputs["destinationNetworkInterfaceIPConfiguration"] = args ? args.destinationNetworkInterfaceIPConfiguration : undefined;
            inputs["destinationPort"] = args ? args.destinationPort : undefined;
            inputs["etag"] = args ? args.etag : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["tapName"] = args ? args.tapName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["networkInterfaceTapConfigurations"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["resourceGuid"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["destinationLoadBalancerFrontEndIPConfiguration"] = undefined /*out*/;
            inputs["destinationNetworkInterfaceIPConfiguration"] = undefined /*out*/;
            inputs["destinationPort"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["networkInterfaceTapConfigurations"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["resourceGuid"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20181001:VirtualNetworkTap" }, { type: "azure-native:network:VirtualNetworkTap" }, { type: "azure-nextgen:network:VirtualNetworkTap" }, { type: "azure-native:network/v20180801:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20180801:VirtualNetworkTap" }, { type: "azure-native:network/v20181101:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20181101:VirtualNetworkTap" }, { type: "azure-native:network/v20181201:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20181201:VirtualNetworkTap" }, { type: "azure-native:network/v20190201:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20190201:VirtualNetworkTap" }, { type: "azure-native:network/v20190401:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20190401:VirtualNetworkTap" }, { type: "azure-native:network/v20190601:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20190601:VirtualNetworkTap" }, { type: "azure-native:network/v20190701:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20190701:VirtualNetworkTap" }, { type: "azure-native:network/v20190801:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20190801:VirtualNetworkTap" }, { type: "azure-native:network/v20190901:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20190901:VirtualNetworkTap" }, { type: "azure-native:network/v20191101:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20191101:VirtualNetworkTap" }, { type: "azure-native:network/v20191201:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20191201:VirtualNetworkTap" }, { type: "azure-native:network/v20200301:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20200301:VirtualNetworkTap" }, { type: "azure-native:network/v20200401:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20200401:VirtualNetworkTap" }, { type: "azure-native:network/v20200501:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20200501:VirtualNetworkTap" }, { type: "azure-native:network/v20200601:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20200601:VirtualNetworkTap" }, { type: "azure-native:network/v20200701:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20200701:VirtualNetworkTap" }, { type: "azure-native:network/v20200801:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20200801:VirtualNetworkTap" }, { type: "azure-native:network/v20201101:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20201101:VirtualNetworkTap" }, { type: "azure-native:network/v20210201:VirtualNetworkTap" }, { type: "azure-nextgen:network/v20210201:VirtualNetworkTap" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VirtualNetworkTap.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing VirtualNetworkTap resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VirtualNetworkTap(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualNetworkTap.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualNetworkTap.__pulumiType;
    }
}
exports.VirtualNetworkTap = VirtualNetworkTap;
/** @internal */
VirtualNetworkTap.__pulumiType = 'azure-native:network/v20181001:VirtualNetworkTap';
