// @pulumi/azure-native/datamigration/v20171115preview/project.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Project = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A project resource
 */
class Project extends pulumi.CustomResource {
    /**
     * Create a Project resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.groupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'groupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            if ((!args || args.sourcePlatform === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourcePlatform'");
            }
            if ((!args || args.targetPlatform === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetPlatform'");
            }
            inputs["databasesInfo"] = args ? args.databasesInfo : undefined;
            inputs["groupName"] = args ? args.groupName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["projectName"] = args ? args.projectName : undefined;
            inputs["serviceName"] = args ? args.serviceName : undefined;
            inputs["sourceConnectionInfo"] = args ? args.sourceConnectionInfo : undefined;
            inputs["sourcePlatform"] = args ? args.sourcePlatform : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["targetConnectionInfo"] = args ? args.targetConnectionInfo : undefined;
            inputs["targetPlatform"] = args ? args.targetPlatform : undefined;
            inputs["creationTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["creationTime"] = undefined /*out*/;
            inputs["databasesInfo"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["sourceConnectionInfo"] = undefined /*out*/;
            inputs["sourcePlatform"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["targetConnectionInfo"] = undefined /*out*/;
            inputs["targetPlatform"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:datamigration/v20171115preview:Project" }, { type: "azure-native:datamigration:Project" }, { type: "azure-nextgen:datamigration:Project" }, { type: "azure-native:datamigration/v20180315preview:Project" }, { type: "azure-nextgen:datamigration/v20180315preview:Project" }, { type: "azure-native:datamigration/v20180331preview:Project" }, { type: "azure-nextgen:datamigration/v20180331preview:Project" }, { type: "azure-native:datamigration/v20180419:Project" }, { type: "azure-nextgen:datamigration/v20180419:Project" }, { type: "azure-native:datamigration/v20180715preview:Project" }, { type: "azure-nextgen:datamigration/v20180715preview:Project" }, { type: "azure-native:datamigration/v20210630:Project" }, { type: "azure-nextgen:datamigration/v20210630:Project" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Project.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Project resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Project(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Project.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Project.__pulumiType;
    }
}
exports.Project = Project;
/** @internal */
Project.__pulumiType = 'azure-native:datamigration/v20171115preview:Project';
