// @pulumi/azure-native/recoveryservices/v20210210/replicationMigrationItem.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationMigrationItem = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Migration item.
 */
class ReplicationMigrationItem extends pulumi.CustomResource {
    /**
     * Create a ReplicationMigrationItem resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fabricName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fabricName'");
            }
            if ((!args || args.properties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'properties'");
            }
            if ((!args || args.protectionContainerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'protectionContainerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            inputs["fabricName"] = args ? args.fabricName : undefined;
            inputs["migrationItemName"] = args ? args.migrationItemName : undefined;
            inputs["properties"] = args ? args.properties : undefined;
            inputs["protectionContainerName"] = args ? args.protectionContainerName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["resourceName"] = args ? args.resourceName : undefined;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:recoveryservices/v20210210:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices:ReplicationMigrationItem" }, { type: "azure-nextgen:recoveryservices:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20180110:ReplicationMigrationItem" }, { type: "azure-nextgen:recoveryservices/v20180110:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20180710:ReplicationMigrationItem" }, { type: "azure-nextgen:recoveryservices/v20180710:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20210301:ReplicationMigrationItem" }, { type: "azure-nextgen:recoveryservices/v20210301:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20210401:ReplicationMigrationItem" }, { type: "azure-nextgen:recoveryservices/v20210401:ReplicationMigrationItem" }, { type: "azure-native:recoveryservices/v20210601:ReplicationMigrationItem" }, { type: "azure-nextgen:recoveryservices/v20210601:ReplicationMigrationItem" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ReplicationMigrationItem.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ReplicationMigrationItem resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ReplicationMigrationItem(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReplicationMigrationItem.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicationMigrationItem.__pulumiType;
    }
}
exports.ReplicationMigrationItem = ReplicationMigrationItem;
/** @internal */
ReplicationMigrationItem.__pulumiType = 'azure-native:recoveryservices/v20210210:ReplicationMigrationItem';
