// @pulumi/azure-native/providerhub/v20210501preview/skusNestedResourceTypeSecond.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkusNestedResourceTypeSecond = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
class SkusNestedResourceTypeSecond extends pulumi.CustomResource {
    /**
     * Create a SkusNestedResourceTypeSecond resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.nestedResourceTypeFirst === undefined) && !opts.urn) {
                throw new Error("Missing required property 'nestedResourceTypeFirst'");
            }
            if ((!args || args.nestedResourceTypeSecond === undefined) && !opts.urn) {
                throw new Error("Missing required property 'nestedResourceTypeSecond'");
            }
            if ((!args || args.providerNamespace === undefined) && !opts.urn) {
                throw new Error("Missing required property 'providerNamespace'");
            }
            if ((!args || args.resourceType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceType'");
            }
            if ((!args || args.skuSettings === undefined) && !opts.urn) {
                throw new Error("Missing required property 'skuSettings'");
            }
            inputs["nestedResourceTypeFirst"] = args ? args.nestedResourceTypeFirst : undefined;
            inputs["nestedResourceTypeSecond"] = args ? args.nestedResourceTypeSecond : undefined;
            inputs["providerNamespace"] = args ? args.providerNamespace : undefined;
            inputs["provisioningState"] = args ? args.provisioningState : undefined;
            inputs["resourceType"] = args ? args.resourceType : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["skuSettings"] = args ? args.skuSettings : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:providerhub/v20210501preview:SkusNestedResourceTypeSecond" }, { type: "azure-native:providerhub:SkusNestedResourceTypeSecond" }, { type: "azure-nextgen:providerhub:SkusNestedResourceTypeSecond" }, { type: "azure-native:providerhub/v20201120:SkusNestedResourceTypeSecond" }, { type: "azure-nextgen:providerhub/v20201120:SkusNestedResourceTypeSecond" }, { type: "azure-native:providerhub/v20210601preview:SkusNestedResourceTypeSecond" }, { type: "azure-nextgen:providerhub/v20210601preview:SkusNestedResourceTypeSecond" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SkusNestedResourceTypeSecond.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SkusNestedResourceTypeSecond resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SkusNestedResourceTypeSecond(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SkusNestedResourceTypeSecond.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SkusNestedResourceTypeSecond.__pulumiType;
    }
}
exports.SkusNestedResourceTypeSecond = SkusNestedResourceTypeSecond;
/** @internal */
SkusNestedResourceTypeSecond.__pulumiType = 'azure-native:providerhub/v20210501preview:SkusNestedResourceTypeSecond';
