// @pulumi/azure-native/network/v20191201/expressRouteCircuitConnection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpressRouteCircuitConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Express Route Circuit Connection in an ExpressRouteCircuitPeering resource.
 */
class ExpressRouteCircuitConnection extends pulumi.CustomResource {
    /**
     * Create a ExpressRouteCircuitConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.circuitName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'circuitName'");
            }
            if ((!args || args.peeringName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peeringName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["addressPrefix"] = args ? args.addressPrefix : undefined;
            inputs["authorizationKey"] = args ? args.authorizationKey : undefined;
            inputs["circuitName"] = args ? args.circuitName : undefined;
            inputs["connectionName"] = args ? args.connectionName : undefined;
            inputs["expressRouteCircuitPeering"] = args ? args.expressRouteCircuitPeering : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["ipv6CircuitConnectionConfig"] = args ? args.ipv6CircuitConnectionConfig : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["peerExpressRouteCircuitPeering"] = args ? args.peerExpressRouteCircuitPeering : undefined;
            inputs["peeringName"] = args ? args.peeringName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["circuitConnectionStatus"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["addressPrefix"] = undefined /*out*/;
            inputs["authorizationKey"] = undefined /*out*/;
            inputs["circuitConnectionStatus"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["expressRouteCircuitPeering"] = undefined /*out*/;
            inputs["ipv6CircuitConnectionConfig"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["peerExpressRouteCircuitPeering"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20191201:ExpressRouteCircuitConnection" }, { type: "azure-native:network:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20180201:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20180201:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20180401:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20180401:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20180601:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20180601:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20180701:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20180701:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20180801:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20180801:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20181001:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20181001:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20181101:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20181101:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20181201:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20181201:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20190201:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20190201:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20190401:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20190401:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20190601:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20190601:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20190701:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20190701:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20190801:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20190801:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20190901:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20190901:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20191101:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20191101:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20200301:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20200301:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20200401:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20200401:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20200501:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20200501:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20200601:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20200601:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20200701:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20200701:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20200801:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20200801:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20201101:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20201101:ExpressRouteCircuitConnection" }, { type: "azure-native:network/v20210201:ExpressRouteCircuitConnection" }, { type: "azure-nextgen:network/v20210201:ExpressRouteCircuitConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ExpressRouteCircuitConnection.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ExpressRouteCircuitConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ExpressRouteCircuitConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ExpressRouteCircuitConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExpressRouteCircuitConnection.__pulumiType;
    }
}
exports.ExpressRouteCircuitConnection = ExpressRouteCircuitConnection;
/** @internal */
ExpressRouteCircuitConnection.__pulumiType = 'azure-native:network/v20191201:ExpressRouteCircuitConnection';
