// @pulumi/azure-native/web/v20201201/webAppPremierAddOn.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppPremierAddOn = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Premier add-on.
 */
class WebAppPremierAddOn extends pulumi.CustomResource {
    /**
     * Create a WebAppPremierAddOn resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["kind"] = args ? args.kind : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["marketplaceOffer"] = args ? args.marketplaceOffer : undefined;
            inputs["marketplacePublisher"] = args ? args.marketplacePublisher : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["premierAddOnName"] = args ? args.premierAddOnName : undefined;
            inputs["product"] = args ? args.product : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["vendor"] = args ? args.vendor : undefined;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["marketplaceOffer"] = undefined /*out*/;
            inputs["marketplacePublisher"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["product"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["vendor"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20201201:WebAppPremierAddOn" }, { type: "azure-native:web:WebAppPremierAddOn" }, { type: "azure-nextgen:web:WebAppPremierAddOn" }, { type: "azure-native:web/v20150801:WebAppPremierAddOn" }, { type: "azure-nextgen:web/v20150801:WebAppPremierAddOn" }, { type: "azure-native:web/v20160801:WebAppPremierAddOn" }, { type: "azure-nextgen:web/v20160801:WebAppPremierAddOn" }, { type: "azure-native:web/v20180201:WebAppPremierAddOn" }, { type: "azure-nextgen:web/v20180201:WebAppPremierAddOn" }, { type: "azure-native:web/v20181101:WebAppPremierAddOn" }, { type: "azure-nextgen:web/v20181101:WebAppPremierAddOn" }, { type: "azure-native:web/v20190801:WebAppPremierAddOn" }, { type: "azure-nextgen:web/v20190801:WebAppPremierAddOn" }, { type: "azure-native:web/v20200601:WebAppPremierAddOn" }, { type: "azure-nextgen:web/v20200601:WebAppPremierAddOn" }, { type: "azure-native:web/v20200901:WebAppPremierAddOn" }, { type: "azure-nextgen:web/v20200901:WebAppPremierAddOn" }, { type: "azure-native:web/v20201001:WebAppPremierAddOn" }, { type: "azure-nextgen:web/v20201001:WebAppPremierAddOn" }, { type: "azure-native:web/v20210101:WebAppPremierAddOn" }, { type: "azure-nextgen:web/v20210101:WebAppPremierAddOn" }, { type: "azure-native:web/v20210115:WebAppPremierAddOn" }, { type: "azure-nextgen:web/v20210115:WebAppPremierAddOn" }, { type: "azure-native:web/v20210201:WebAppPremierAddOn" }, { type: "azure-nextgen:web/v20210201:WebAppPremierAddOn" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppPremierAddOn.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WebAppPremierAddOn resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppPremierAddOn(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppPremierAddOn.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppPremierAddOn.__pulumiType;
    }
}
exports.WebAppPremierAddOn = WebAppPremierAddOn;
/** @internal */
WebAppPremierAddOn.__pulumiType = 'azure-native:web/v20201201:WebAppPremierAddOn';
