// @pulumi/azure-native/types/enums/desktopvirtualization/v20210513preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMachineDiskType = exports.StopHostsWhen = exports.SkuTier = exports.SessionHostLoadBalancingAlgorithm = exports.SessionHostComponentUpdateType = exports.SSOSecretType = exports.ResourceIdentityType = exports.RemoteApplicationType = exports.RegistrationTokenOperation = exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.PreferredAppGroupType = exports.PersonalDesktopAssignmentType = exports.Operation = exports.LoadBalancerType = exports.ImageType = exports.HostPoolType = exports.DomainJoinType = exports.DayOfWeek = exports.CommandLineSetting = exports.ApplicationGroupType = void 0;
exports.ApplicationGroupType = {
    RemoteApp: "RemoteApp",
    Desktop: "Desktop",
};
exports.CommandLineSetting = {
    DoNotAllow: "DoNotAllow",
    Allow: "Allow",
    Require: "Require",
};
exports.DayOfWeek = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.DomainJoinType = {
    /**
     * Using microsoft active directory.
     */
    ActiveDirectory: "ActiveDirectory",
    /**
     * Using microsoft azure active directory.
     */
    AzureActiveDirectory: "AzureActiveDirectory",
};
exports.HostPoolType = {
    /**
     * Users will be assigned a SessionHost either by administrators (PersonalDesktopAssignmentType = Direct) or upon connecting to the pool (PersonalDesktopAssignmentType = Automatic). They will always be redirected to their assigned SessionHost.
     */
    Personal: "Personal",
    /**
     * Users get a new (random) SessionHost every time it connects to the HostPool.
     */
    Pooled: "Pooled",
    /**
     * Users assign their own machines, load balancing logic remains the same as Personal. PersonalDesktopAssignmentType must be Direct.
     */
    BYODesktop: "BYODesktop",
};
exports.ImageType = {
    /**
     * Using default gallery images offered by Azure Marketplace.
     */
    Gallery: "Gallery",
    /**
     * Using a VHD stored in a storage blob.
     */
    StorageBlob: "StorageBlob",
    /**
     * Using custom image or custom shared image.
     */
    CustomImage: "CustomImage",
};
exports.LoadBalancerType = {
    BreadthFirst: "BreadthFirst",
    DepthFirst: "DepthFirst",
    Persistent: "Persistent",
};
exports.Operation = {
    /**
     * Start the migration.
     */
    Start: "Start",
    /**
     * Revoke the migration.
     */
    Revoke: "Revoke",
    /**
     * Complete the migration.
     */
    Complete: "Complete",
    /**
     * Hide the hostpool.
     */
    Hide: "Hide",
    /**
     * Unhide the hostpool.
     */
    Unhide: "Unhide",
};
exports.PersonalDesktopAssignmentType = {
    Automatic: "Automatic",
    Direct: "Direct",
};
exports.PreferredAppGroupType = {
    None: "None",
    Desktop: "Desktop",
    RailApplications: "RailApplications",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RegistrationTokenOperation = {
    Delete: "Delete",
    None: "None",
    Update: "Update",
};
exports.RemoteApplicationType = {
    InBuilt: "InBuilt",
    MsixApplication: "MsixApplication",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.SSOSecretType = {
    SharedKey: "SharedKey",
    Certificate: "Certificate",
    SharedKeyInKeyVault: "SharedKeyInKeyVault",
    CertificateInKeyVault: "CertificateInKeyVault",
};
exports.SessionHostComponentUpdateType = {
    /**
     * Agent and other agent side components are delivery schedule is controlled by WVD Infra.
     */
    Default: "Default",
    /**
     * TenantAdmin have opted in for Scheduled Component Update feature.
     */
    TenantAdminControlled: "TenantAdminControlled",
};
exports.SessionHostLoadBalancingAlgorithm = {
    BreadthFirst: "BreadthFirst",
    DepthFirst: "DepthFirst",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.StopHostsWhen = {
    ZeroSessions: "ZeroSessions",
    ZeroActiveSessions: "ZeroActiveSessions",
};
exports.VirtualMachineDiskType = {
    /**
     * Standard HDD locally redundant storage. Best for backup, non-critical, and infrequent access.
     */
    Standard_LRS: "Standard_LRS",
    /**
     * Premium SSD locally redundant storage. Best for production and performance sensitive workloads.
     */
    Premium_LRS: "Premium_LRS",
    /**
     * Standard SSD locally redundant storage. Best for web servers, lightly used enterprise applications and dev/test.
     */
    StandardSSD_LRS: "StandardSSD_LRS",
    /**
     * Ultra SSD locally redundant storage. Best for IO-intensive workloads such as SAP HANA, top tier databases (for example, SQL, Oracle), and other transaction-heavy workloads.
     */
    UltraSSD_LRS: "UltraSSD_LRS",
};
