// @pulumi/azure-native/types/enums/synapse/v20190601preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransparentDataEncryptionStatus = exports.SensitivityLabelRank = exports.ResourceIdentityType = exports.NodeSizeFamily = exports.NodeSize = exports.IntegrationRuntimeType = exports.IntegrationRuntimeSsisCatalogPricingTier = exports.IntegrationRuntimeLicenseType = exports.IntegrationRuntimeEntityReferenceType = exports.IntegrationRuntimeEdition = exports.DataFlowComputeType = void 0;
exports.DataFlowComputeType = {
    General: "General",
    MemoryOptimized: "MemoryOptimized",
    ComputeOptimized: "ComputeOptimized",
};
exports.IntegrationRuntimeEdition = {
    Standard: "Standard",
    Enterprise: "Enterprise",
};
exports.IntegrationRuntimeEntityReferenceType = {
    IntegrationRuntimeReference: "IntegrationRuntimeReference",
    LinkedServiceReference: "LinkedServiceReference",
};
exports.IntegrationRuntimeLicenseType = {
    BasePrice: "BasePrice",
    LicenseIncluded: "LicenseIncluded",
};
exports.IntegrationRuntimeSsisCatalogPricingTier = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
    PremiumRS: "PremiumRS",
};
exports.IntegrationRuntimeType = {
    Managed: "Managed",
    SelfHosted: "SelfHosted",
};
exports.NodeSize = {
    None: "None",
    Small: "Small",
    Medium: "Medium",
    Large: "Large",
    XLarge: "XLarge",
    XXLarge: "XXLarge",
    XXXLarge: "XXXLarge",
};
exports.NodeSizeFamily = {
    None: "None",
    MemoryOptimized: "MemoryOptimized",
};
exports.ResourceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
exports.SensitivityLabelRank = {
    None: "None",
    Low: "Low",
    Medium: "Medium",
    High: "High",
    Critical: "Critical",
};
exports.TransparentDataEncryptionStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
