// @pulumi/azure-native/types/enums/servicebus/v20210601preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.SkuName = exports.PublicNetworkAccessFlag = exports.PrivateLinkConnectionStatus = exports.NetworkRuleIPAction = exports.ManagedServiceIdentityType = exports.KeySource = exports.FilterType = exports.EntityStatus = exports.EndPointProvisioningState = exports.DefaultAction = exports.AccessRights = void 0;
exports.AccessRights = {
    Manage: "Manage",
    Send: "Send",
    Listen: "Listen",
};
exports.DefaultAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.EndPointProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
exports.EntityStatus = {
    Active: "Active",
    Disabled: "Disabled",
    Restoring: "Restoring",
    SendDisabled: "SendDisabled",
    ReceiveDisabled: "ReceiveDisabled",
    Creating: "Creating",
    Deleting: "Deleting",
    Renaming: "Renaming",
    Unknown: "Unknown",
};
exports.FilterType = {
    SqlFilter: "SqlFilter",
    CorrelationFilter: "CorrelationFilter",
};
exports.KeySource = {
    Microsoft_KeyVault: "Microsoft.KeyVault",
};
exports.ManagedServiceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.NetworkRuleIPAction = {
    Allow: "Allow",
};
exports.PrivateLinkConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccessFlag = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SkuName = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.SkuTier = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
