// @pulumi/azure-native/synapse/v20210401preview/key.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Key = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A workspace key
 */
class Key extends pulumi.CustomResource {
    /**
     * Create a Key resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["isActiveCMK"] = args ? args.isActiveCMK : undefined;
            inputs["keyName"] = args ? args.keyName : undefined;
            inputs["keyVaultUrl"] = args ? args.keyVaultUrl : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["isActiveCMK"] = undefined /*out*/;
            inputs["keyVaultUrl"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:synapse/v20210401preview:Key" }, { type: "azure-native:synapse:Key" }, { type: "azure-nextgen:synapse:Key" }, { type: "azure-native:synapse/v20190601preview:Key" }, { type: "azure-nextgen:synapse/v20190601preview:Key" }, { type: "azure-native:synapse/v20201201:Key" }, { type: "azure-nextgen:synapse/v20201201:Key" }, { type: "azure-native:synapse/v20210301:Key" }, { type: "azure-nextgen:synapse/v20210301:Key" }, { type: "azure-native:synapse/v20210501:Key" }, { type: "azure-nextgen:synapse/v20210501:Key" }, { type: "azure-native:synapse/v20210601preview:Key" }, { type: "azure-nextgen:synapse/v20210601preview:Key" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Key.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Key resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Key(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Key.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Key.__pulumiType;
    }
}
exports.Key = Key;
/** @internal */
Key.__pulumiType = 'azure-native:synapse/v20210401preview:Key';
