// @pulumi/azure-native/synapse/v20210401preview/privateLinkHub.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateLinkHub = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A privateLinkHub
 */
class PrivateLinkHub extends pulumi.CustomResource {
    /**
     * Create a PrivateLinkHub resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["location"] = args ? args.location : undefined;
            inputs["privateLinkHubName"] = args ? args.privateLinkHubName : undefined;
            inputs["provisioningState"] = args ? args.provisioningState : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["privateEndpointConnections"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["privateEndpointConnections"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:synapse/v20210401preview:PrivateLinkHub" }, { type: "azure-native:synapse:PrivateLinkHub" }, { type: "azure-nextgen:synapse:PrivateLinkHub" }, { type: "azure-native:synapse/v20190601preview:PrivateLinkHub" }, { type: "azure-nextgen:synapse/v20190601preview:PrivateLinkHub" }, { type: "azure-native:synapse/v20201201:PrivateLinkHub" }, { type: "azure-nextgen:synapse/v20201201:PrivateLinkHub" }, { type: "azure-native:synapse/v20210301:PrivateLinkHub" }, { type: "azure-nextgen:synapse/v20210301:PrivateLinkHub" }, { type: "azure-native:synapse/v20210501:PrivateLinkHub" }, { type: "azure-nextgen:synapse/v20210501:PrivateLinkHub" }, { type: "azure-native:synapse/v20210601preview:PrivateLinkHub" }, { type: "azure-nextgen:synapse/v20210601preview:PrivateLinkHub" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateLinkHub.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing PrivateLinkHub resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateLinkHub(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivateLinkHub.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateLinkHub.__pulumiType;
    }
}
exports.PrivateLinkHub = PrivateLinkHub;
/** @internal */
PrivateLinkHub.__pulumiType = 'azure-native:synapse/v20210401preview:PrivateLinkHub';
