// @pulumi/azure-native/synapse/v20210401preview/workspaceAadAdmin.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceAadAdmin = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Workspace active directory administrator
 */
class WorkspaceAadAdmin extends pulumi.CustomResource {
    /**
     * Create a WorkspaceAadAdmin resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["administratorType"] = args ? args.administratorType : undefined;
            inputs["login"] = args ? args.login : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sid"] = args ? args.sid : undefined;
            inputs["tenantId"] = args ? args.tenantId : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["administratorType"] = undefined /*out*/;
            inputs["login"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["sid"] = undefined /*out*/;
            inputs["tenantId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:synapse/v20210401preview:WorkspaceAadAdmin" }, { type: "azure-native:synapse:WorkspaceAadAdmin" }, { type: "azure-nextgen:synapse:WorkspaceAadAdmin" }, { type: "azure-native:synapse/v20190601preview:WorkspaceAadAdmin" }, { type: "azure-nextgen:synapse/v20190601preview:WorkspaceAadAdmin" }, { type: "azure-native:synapse/v20201201:WorkspaceAadAdmin" }, { type: "azure-nextgen:synapse/v20201201:WorkspaceAadAdmin" }, { type: "azure-native:synapse/v20210301:WorkspaceAadAdmin" }, { type: "azure-nextgen:synapse/v20210301:WorkspaceAadAdmin" }, { type: "azure-native:synapse/v20210501:WorkspaceAadAdmin" }, { type: "azure-nextgen:synapse/v20210501:WorkspaceAadAdmin" }, { type: "azure-native:synapse/v20210601preview:WorkspaceAadAdmin" }, { type: "azure-nextgen:synapse/v20210601preview:WorkspaceAadAdmin" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WorkspaceAadAdmin.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WorkspaceAadAdmin resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WorkspaceAadAdmin(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WorkspaceAadAdmin.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkspaceAadAdmin.__pulumiType;
    }
}
exports.WorkspaceAadAdmin = WorkspaceAadAdmin;
/** @internal */
WorkspaceAadAdmin.__pulumiType = 'azure-native:synapse/v20210401preview:WorkspaceAadAdmin';
