// @pulumi/azure-native/blockchain/transactionNode.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionNode = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Payload of the transaction node which is the request/response of the resource provider.
 * API Version: 2018-06-01-preview.
 */
class TransactionNode extends pulumi.CustomResource {
    /**
     * Create a TransactionNode resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.blockchainMemberName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'blockchainMemberName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["blockchainMemberName"] = args ? args.blockchainMemberName : undefined;
            inputs["firewallRules"] = args ? args.firewallRules : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["password"] = args ? args.password : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["transactionNodeName"] = args ? args.transactionNodeName : undefined;
            inputs["dns"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["publicKey"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["userName"] = undefined /*out*/;
        }
        else {
            inputs["dns"] = undefined /*out*/;
            inputs["firewallRules"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["password"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["publicKey"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["userName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:blockchain:TransactionNode" }, { type: "azure-native:blockchain/v20180601preview:TransactionNode" }, { type: "azure-nextgen:blockchain/v20180601preview:TransactionNode" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(TransactionNode.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing TransactionNode resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TransactionNode(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TransactionNode.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TransactionNode.__pulumiType;
    }
}
exports.TransactionNode = TransactionNode;
/** @internal */
TransactionNode.__pulumiType = 'azure-native:blockchain:TransactionNode';
