// @pulumi/azure-native/types/enums/datamigration/v20180419/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlSourcePlatform = exports.ServerLevelPermissionsGroup = exports.ProjectTargetPlatform = exports.ProjectSourcePlatform = exports.BackupMode = exports.AuthenticationType = void 0;
exports.AuthenticationType = {
    None: "None",
    WindowsAuthentication: "WindowsAuthentication",
    SqlAuthentication: "SqlAuthentication",
    ActiveDirectoryIntegrated: "ActiveDirectoryIntegrated",
    ActiveDirectoryPassword: "ActiveDirectoryPassword",
};
exports.BackupMode = {
    CreateBackup: "CreateBackup",
    ExistingBackup: "ExistingBackup",
};
exports.ProjectSourcePlatform = {
    SQL: "SQL",
    Unknown: "Unknown",
};
exports.ProjectTargetPlatform = {
    SQLDB: "SQLDB",
    Unknown: "Unknown",
};
exports.ServerLevelPermissionsGroup = {
    Default: "Default",
    MigrationFromSqlServerToAzureDB: "MigrationFromSqlServerToAzureDB",
    MigrationFromSqlServerToAzureMI: "MigrationFromSqlServerToAzureMI",
    MigrationFromMySQLToAzureDBForMySQL: "MigrationFromMySQLToAzureDBForMySQL",
};
exports.SqlSourcePlatform = {
    SqlOnPrem: "SqlOnPrem",
};
