// @pulumi/azure-native/timeseriesinsights/v20180815preview/longTermEnvironment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LongTermEnvironment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An environment is a set of time-series data available for query, and is the top level Azure Time Series Insights resource. LongTerm environments do not have set data retention limits.
 */
class LongTermEnvironment extends pulumi.CustomResource {
    /**
     * Create a LongTermEnvironment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            if ((!args || args.storageConfiguration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'storageConfiguration'");
            }
            if ((!args || args.timeSeriesIdProperties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'timeSeriesIdProperties'");
            }
            inputs["environmentName"] = args ? args.environmentName : undefined;
            inputs["kind"] = "LongTerm";
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["storageConfiguration"] = args ? args.storageConfiguration : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["timeSeriesIdProperties"] = args ? args.timeSeriesIdProperties : undefined;
            inputs["warmStoreConfiguration"] = args ? args.warmStoreConfiguration : undefined;
            inputs["creationTime"] = undefined /*out*/;
            inputs["dataAccessFqdn"] = undefined /*out*/;
            inputs["dataAccessId"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["creationTime"] = undefined /*out*/;
            inputs["dataAccessFqdn"] = undefined /*out*/;
            inputs["dataAccessId"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["storageConfiguration"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["timeSeriesIdProperties"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["warmStoreConfiguration"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:timeseriesinsights/v20180815preview:LongTermEnvironment" }, { type: "azure-native:timeseriesinsights:LongTermEnvironment" }, { type: "azure-nextgen:timeseriesinsights:LongTermEnvironment" }, { type: "azure-native:timeseriesinsights/v20170228preview:LongTermEnvironment" }, { type: "azure-nextgen:timeseriesinsights/v20170228preview:LongTermEnvironment" }, { type: "azure-native:timeseriesinsights/v20171115:LongTermEnvironment" }, { type: "azure-nextgen:timeseriesinsights/v20171115:LongTermEnvironment" }, { type: "azure-native:timeseriesinsights/v20200515:LongTermEnvironment" }, { type: "azure-nextgen:timeseriesinsights/v20200515:LongTermEnvironment" }, { type: "azure-native:timeseriesinsights/v20210630preview:LongTermEnvironment" }, { type: "azure-nextgen:timeseriesinsights/v20210630preview:LongTermEnvironment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(LongTermEnvironment.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing LongTermEnvironment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new LongTermEnvironment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of LongTermEnvironment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LongTermEnvironment.__pulumiType;
    }
}
exports.LongTermEnvironment = LongTermEnvironment;
/** @internal */
LongTermEnvironment.__pulumiType = 'azure-native:timeseriesinsights/v20180815preview:LongTermEnvironment';
