// @pulumi/azure-native/securityinsights/v20190101preview/bookmarkRelation.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BookmarkRelation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents a relation between two resources
 */
class BookmarkRelation extends pulumi.CustomResource {
    /**
     * Create a BookmarkRelation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.bookmarkId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'bookmarkId'");
            }
            if ((!args || args.operationalInsightsResourceProvider === undefined) && !opts.urn) {
                throw new Error("Missing required property 'operationalInsightsResourceProvider'");
            }
            if ((!args || args.relatedResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'relatedResourceId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["bookmarkId"] = args ? args.bookmarkId : undefined;
            inputs["etag"] = args ? args.etag : undefined;
            inputs["operationalInsightsResourceProvider"] = args ? args.operationalInsightsResourceProvider : undefined;
            inputs["relatedResourceId"] = args ? args.relatedResourceId : undefined;
            inputs["relationName"] = args ? args.relationName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["relatedResourceKind"] = undefined /*out*/;
            inputs["relatedResourceName"] = undefined /*out*/;
            inputs["relatedResourceType"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["relatedResourceId"] = undefined /*out*/;
            inputs["relatedResourceKind"] = undefined /*out*/;
            inputs["relatedResourceName"] = undefined /*out*/;
            inputs["relatedResourceType"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:securityinsights/v20190101preview:BookmarkRelation" }, { type: "azure-native:securityinsights:BookmarkRelation" }, { type: "azure-nextgen:securityinsights:BookmarkRelation" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BookmarkRelation.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing BookmarkRelation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BookmarkRelation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BookmarkRelation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BookmarkRelation.__pulumiType;
    }
}
exports.BookmarkRelation = BookmarkRelation;
/** @internal */
BookmarkRelation.__pulumiType = 'azure-native:securityinsights/v20190101preview:BookmarkRelation';
