// @pulumi/azure-native/securityinsights/v20190101preview/watchlist.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Watchlist = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents a Watchlist in Azure Security Insights.
 */
class Watchlist extends pulumi.CustomResource {
    /**
     * Create a Watchlist resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.operationalInsightsResourceProvider === undefined) && !opts.urn) {
                throw new Error("Missing required property 'operationalInsightsResourceProvider'");
            }
            if ((!args || args.provider === undefined) && !opts.urn) {
                throw new Error("Missing required property 'provider'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.source === undefined) && !opts.urn) {
                throw new Error("Missing required property 'source'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["contentType"] = args ? args.contentType : undefined;
            inputs["created"] = args ? args.created : undefined;
            inputs["createdBy"] = args ? args.createdBy : undefined;
            inputs["defaultDuration"] = args ? args.defaultDuration : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["etag"] = args ? args.etag : undefined;
            inputs["isDeleted"] = args ? args.isDeleted : undefined;
            inputs["labels"] = args ? args.labels : undefined;
            inputs["numberOfLinesToSkip"] = args ? args.numberOfLinesToSkip : undefined;
            inputs["operationalInsightsResourceProvider"] = args ? args.operationalInsightsResourceProvider : undefined;
            inputs["provider"] = args ? args.provider : undefined;
            inputs["rawContent"] = args ? args.rawContent : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["source"] = args ? args.source : undefined;
            inputs["tenantId"] = args ? args.tenantId : undefined;
            inputs["updated"] = args ? args.updated : undefined;
            inputs["updatedBy"] = args ? args.updatedBy : undefined;
            inputs["uploadStatus"] = args ? args.uploadStatus : undefined;
            inputs["watchlistAlias"] = args ? args.watchlistAlias : undefined;
            inputs["watchlistId"] = args ? args.watchlistId : undefined;
            inputs["watchlistItemsCount"] = args ? args.watchlistItemsCount : undefined;
            inputs["watchlistType"] = args ? args.watchlistType : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["contentType"] = undefined /*out*/;
            inputs["created"] = undefined /*out*/;
            inputs["createdBy"] = undefined /*out*/;
            inputs["defaultDuration"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["isDeleted"] = undefined /*out*/;
            inputs["labels"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["numberOfLinesToSkip"] = undefined /*out*/;
            inputs["provider"] = undefined /*out*/;
            inputs["rawContent"] = undefined /*out*/;
            inputs["source"] = undefined /*out*/;
            inputs["tenantId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["updated"] = undefined /*out*/;
            inputs["updatedBy"] = undefined /*out*/;
            inputs["uploadStatus"] = undefined /*out*/;
            inputs["watchlistAlias"] = undefined /*out*/;
            inputs["watchlistId"] = undefined /*out*/;
            inputs["watchlistItemsCount"] = undefined /*out*/;
            inputs["watchlistType"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:securityinsights/v20190101preview:Watchlist" }, { type: "azure-native:securityinsights:Watchlist" }, { type: "azure-nextgen:securityinsights:Watchlist" }, { type: "azure-native:securityinsights/v20210301preview:Watchlist" }, { type: "azure-nextgen:securityinsights/v20210301preview:Watchlist" }, { type: "azure-native:securityinsights/v20210401:Watchlist" }, { type: "azure-nextgen:securityinsights/v20210401:Watchlist" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Watchlist.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Watchlist resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Watchlist(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Watchlist.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Watchlist.__pulumiType;
    }
}
exports.Watchlist = Watchlist;
/** @internal */
Watchlist.__pulumiType = 'azure-native:securityinsights/v20190101preview:Watchlist';
