// @pulumi/azure-native/provider.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Provider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * The provider type for the native Azure package.
 */
class Provider extends pulumi.ProviderResource {
    /**
     * Create a Provider resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l;
        let inputs = {};
        opts = opts || {};
        {
            inputs["auxiliaryTenantIds"] = pulumi.output(args ? args.auxiliaryTenantIds : undefined).apply(JSON.stringify);
            inputs["clientCertificatePassword"] = (_a = (args ? args.clientCertificatePassword : undefined)) !== null && _a !== void 0 ? _a : utilities.getEnv("ARM_CLIENT_CERTIFICATE_PASSWORD");
            inputs["clientCertificatePath"] = (_b = (args ? args.clientCertificatePath : undefined)) !== null && _b !== void 0 ? _b : utilities.getEnv("ARM_CLIENT_CERTIFICATE_PATH");
            inputs["clientId"] = (_c = (args ? args.clientId : undefined)) !== null && _c !== void 0 ? _c : utilities.getEnv("ARM_CLIENT_ID");
            inputs["clientSecret"] = (_d = (args ? args.clientSecret : undefined)) !== null && _d !== void 0 ? _d : utilities.getEnv("ARM_CLIENT_SECRET");
            inputs["disablePulumiPartnerId"] = pulumi.output((_e = (args ? args.disablePulumiPartnerId : undefined)) !== null && _e !== void 0 ? _e : utilities.getEnvBoolean("ARM_DISABLE_PULUMI_PARTNER_ID")).apply(JSON.stringify);
            inputs["environment"] = (_f = (args ? args.environment : undefined)) !== null && _f !== void 0 ? _f : (utilities.getEnv("ARM_ENVIRONMENT") || "public");
            inputs["msiEndpoint"] = (_g = (args ? args.msiEndpoint : undefined)) !== null && _g !== void 0 ? _g : utilities.getEnv("ARM_MSI_ENDPOINT");
            inputs["partnerId"] = (_h = (args ? args.partnerId : undefined)) !== null && _h !== void 0 ? _h : utilities.getEnv("ARM_PARTNER_ID");
            inputs["subscriptionId"] = (_j = (args ? args.subscriptionId : undefined)) !== null && _j !== void 0 ? _j : utilities.getEnv("ARM_SUBSCRIPTION_ID");
            inputs["tenantId"] = (_k = (args ? args.tenantId : undefined)) !== null && _k !== void 0 ? _k : utilities.getEnv("ARM_TENANT_ID");
            inputs["useMsi"] = pulumi.output((_l = (args ? args.useMsi : undefined)) !== null && _l !== void 0 ? _l : (utilities.getEnvBoolean("ARM_USE_MSI") || false)).apply(JSON.stringify);
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "pulumi:providers:azure-nextgen" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Provider.__pulumiType, name, inputs, opts);
    }
    /**
     * Returns true if the given object is an instance of Provider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Provider.__pulumiType;
    }
}
exports.Provider = Provider;
/** @internal */
Provider.__pulumiType = 'azure-native';
