// @pulumi/azure-native/network/v20200501/frontDoor.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FrontDoor = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Front Door represents a collection of backend endpoints to route traffic to along with rules that specify how traffic is sent there.
 */
class FrontDoor extends pulumi.CustomResource {
    /**
     * Create a FrontDoor resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["backendPools"] = args ? args.backendPools : undefined;
            inputs["backendPoolsSettings"] = args ? args.backendPoolsSettings : undefined;
            inputs["enabledState"] = args ? args.enabledState : undefined;
            inputs["friendlyName"] = args ? args.friendlyName : undefined;
            inputs["frontDoorName"] = args ? args.frontDoorName : undefined;
            inputs["frontendEndpoints"] = args ? args.frontendEndpoints : undefined;
            inputs["healthProbeSettings"] = args ? args.healthProbeSettings : undefined;
            inputs["loadBalancingSettings"] = args ? args.loadBalancingSettings : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["routingRules"] = args ? args.routingRules : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["cname"] = undefined /*out*/;
            inputs["frontdoorId"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["resourceState"] = undefined /*out*/;
            inputs["rulesEngines"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["backendPools"] = undefined /*out*/;
            inputs["backendPoolsSettings"] = undefined /*out*/;
            inputs["cname"] = undefined /*out*/;
            inputs["enabledState"] = undefined /*out*/;
            inputs["friendlyName"] = undefined /*out*/;
            inputs["frontdoorId"] = undefined /*out*/;
            inputs["frontendEndpoints"] = undefined /*out*/;
            inputs["healthProbeSettings"] = undefined /*out*/;
            inputs["loadBalancingSettings"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["resourceState"] = undefined /*out*/;
            inputs["routingRules"] = undefined /*out*/;
            inputs["rulesEngines"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20200501:FrontDoor" }, { type: "azure-native:network:FrontDoor" }, { type: "azure-nextgen:network:FrontDoor" }, { type: "azure-native:network/v20190401:FrontDoor" }, { type: "azure-nextgen:network/v20190401:FrontDoor" }, { type: "azure-native:network/v20190501:FrontDoor" }, { type: "azure-nextgen:network/v20190501:FrontDoor" }, { type: "azure-native:network/v20200101:FrontDoor" }, { type: "azure-nextgen:network/v20200101:FrontDoor" }, { type: "azure-native:network/v20200401:FrontDoor" }, { type: "azure-nextgen:network/v20200401:FrontDoor" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FrontDoor.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing FrontDoor resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FrontDoor(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FrontDoor.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FrontDoor.__pulumiType;
    }
}
exports.FrontDoor = FrontDoor;
/** @internal */
FrontDoor.__pulumiType = 'azure-native:network/v20200501:FrontDoor';
