// @pulumi/azure-native/sql/v20150501preview/syncMember.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyncMember = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An Azure SQL Database sync member.
 */
class SyncMember extends pulumi.CustomResource {
    /**
     * Create a SyncMember resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serverName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serverName'");
            }
            if ((!args || args.syncGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'syncGroupName'");
            }
            inputs["databaseName"] = args ? args.databaseName : undefined;
            inputs["databaseType"] = args ? args.databaseType : undefined;
            inputs["password"] = args ? args.password : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serverName"] = args ? args.serverName : undefined;
            inputs["sqlServerDatabaseId"] = args ? args.sqlServerDatabaseId : undefined;
            inputs["syncAgentId"] = args ? args.syncAgentId : undefined;
            inputs["syncDirection"] = args ? args.syncDirection : undefined;
            inputs["syncGroupName"] = args ? args.syncGroupName : undefined;
            inputs["syncMemberName"] = args ? args.syncMemberName : undefined;
            inputs["userName"] = args ? args.userName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["syncState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["databaseName"] = undefined /*out*/;
            inputs["databaseType"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["serverName"] = undefined /*out*/;
            inputs["sqlServerDatabaseId"] = undefined /*out*/;
            inputs["syncAgentId"] = undefined /*out*/;
            inputs["syncDirection"] = undefined /*out*/;
            inputs["syncState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["userName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:sql/v20150501preview:SyncMember" }, { type: "azure-native:sql:SyncMember" }, { type: "azure-nextgen:sql:SyncMember" }, { type: "azure-native:sql/v20190601preview:SyncMember" }, { type: "azure-nextgen:sql/v20190601preview:SyncMember" }, { type: "azure-native:sql/v20200202preview:SyncMember" }, { type: "azure-nextgen:sql/v20200202preview:SyncMember" }, { type: "azure-native:sql/v20200801preview:SyncMember" }, { type: "azure-nextgen:sql/v20200801preview:SyncMember" }, { type: "azure-native:sql/v20201101preview:SyncMember" }, { type: "azure-nextgen:sql/v20201101preview:SyncMember" }, { type: "azure-native:sql/v20210201preview:SyncMember" }, { type: "azure-nextgen:sql/v20210201preview:SyncMember" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SyncMember.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SyncMember resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SyncMember(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SyncMember.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SyncMember.__pulumiType;
    }
}
exports.SyncMember = SyncMember;
/** @internal */
SyncMember.__pulumiType = 'azure-native:sql/v20150501preview:SyncMember';
