// @pulumi/azure-native/peering/v20210101/registeredAsn.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegisteredAsn = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The customer's ASN that is registered by the peering service provider.
 */
class RegisteredAsn extends pulumi.CustomResource {
    /**
     * Create a RegisteredAsn resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.peeringName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peeringName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["asn"] = args ? args.asn : undefined;
            inputs["peeringName"] = args ? args.peeringName : undefined;
            inputs["registeredAsnName"] = args ? args.registeredAsnName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["peeringServicePrefixKey"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["asn"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["peeringServicePrefixKey"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:peering/v20210101:RegisteredAsn" }, { type: "azure-native:peering:RegisteredAsn" }, { type: "azure-nextgen:peering:RegisteredAsn" }, { type: "azure-native:peering/v20200101preview:RegisteredAsn" }, { type: "azure-nextgen:peering/v20200101preview:RegisteredAsn" }, { type: "azure-native:peering/v20200401:RegisteredAsn" }, { type: "azure-nextgen:peering/v20200401:RegisteredAsn" }, { type: "azure-native:peering/v20201001:RegisteredAsn" }, { type: "azure-nextgen:peering/v20201001:RegisteredAsn" }, { type: "azure-native:peering/v20210601:RegisteredAsn" }, { type: "azure-nextgen:peering/v20210601:RegisteredAsn" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RegisteredAsn.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing RegisteredAsn resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RegisteredAsn(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RegisteredAsn.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegisteredAsn.__pulumiType;
    }
}
exports.RegisteredAsn = RegisteredAsn;
/** @internal */
RegisteredAsn.__pulumiType = 'azure-native:peering/v20210101:RegisteredAsn';
