// @pulumi/azure-native/softwareplan/v20191201/hybridUseBenefit.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HybridUseBenefit = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Response on GET of a hybrid use benefit
 */
class HybridUseBenefit extends pulumi.CustomResource {
    /**
     * Create a HybridUseBenefit resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.scope === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scope'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            inputs["planId"] = args ? args.planId : undefined;
            inputs["scope"] = args ? args.scope : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["createdDate"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["lastUpdatedDate"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["createdDate"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["lastUpdatedDate"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:softwareplan/v20191201:HybridUseBenefit" }, { type: "azure-native:softwareplan:HybridUseBenefit" }, { type: "azure-nextgen:softwareplan:HybridUseBenefit" }, { type: "azure-native:softwareplan/v20190601preview:HybridUseBenefit" }, { type: "azure-nextgen:softwareplan/v20190601preview:HybridUseBenefit" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(HybridUseBenefit.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing HybridUseBenefit resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new HybridUseBenefit(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of HybridUseBenefit.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HybridUseBenefit.__pulumiType;
    }
}
exports.HybridUseBenefit = HybridUseBenefit;
/** @internal */
HybridUseBenefit.__pulumiType = 'azure-native:softwareplan/v20191201:HybridUseBenefit';
