// @pulumi/azure-native/recoveryservices/replicationRecoveryServicesProvider.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationRecoveryServicesProvider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provider details.
 * API Version: 2018-07-10.
 */
class ReplicationRecoveryServicesProvider extends pulumi.CustomResource {
    /**
     * Create a ReplicationRecoveryServicesProvider resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fabricName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fabricName'");
            }
            if ((!args || args.properties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'properties'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            inputs["fabricName"] = args ? args.fabricName : undefined;
            inputs["properties"] = args ? args.properties : undefined;
            inputs["providerName"] = args ? args.providerName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["resourceName"] = args ? args.resourceName : undefined;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:recoveryservices:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20180110:ReplicationRecoveryServicesProvider" }, { type: "azure-nextgen:recoveryservices/v20180110:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20180710:ReplicationRecoveryServicesProvider" }, { type: "azure-nextgen:recoveryservices/v20180710:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20210210:ReplicationRecoveryServicesProvider" }, { type: "azure-nextgen:recoveryservices/v20210210:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20210301:ReplicationRecoveryServicesProvider" }, { type: "azure-nextgen:recoveryservices/v20210301:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20210401:ReplicationRecoveryServicesProvider" }, { type: "azure-nextgen:recoveryservices/v20210401:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20210601:ReplicationRecoveryServicesProvider" }, { type: "azure-nextgen:recoveryservices/v20210601:ReplicationRecoveryServicesProvider" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ReplicationRecoveryServicesProvider.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ReplicationRecoveryServicesProvider resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ReplicationRecoveryServicesProvider(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReplicationRecoveryServicesProvider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicationRecoveryServicesProvider.__pulumiType;
    }
}
exports.ReplicationRecoveryServicesProvider = ReplicationRecoveryServicesProvider;
/** @internal */
ReplicationRecoveryServicesProvider.__pulumiType = 'azure-native:recoveryservices:ReplicationRecoveryServicesProvider';
