// @pulumi/azure-native/types/enums/appplatform/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSourceType = exports.RuntimeVersion = exports.ManagedIdentityType = exports.v20210601preview = exports.v20201101preview = exports.v20200701 = exports.v20190501preview = void 0;
// Export sub-modules:
const v20190501preview = require("./v20190501preview");
exports.v20190501preview = v20190501preview;
const v20200701 = require("./v20200701");
exports.v20200701 = v20200701;
const v20201101preview = require("./v20201101preview");
exports.v20201101preview = v20201101preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
exports.ManagedIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.RuntimeVersion = {
    Java_8: "Java_8",
    Java_11: "Java_11",
    NetCore_31: "NetCore_31",
};
exports.UserSourceType = {
    Jar: "Jar",
    NetCoreZip: "NetCoreZip",
    Source: "Source",
};
