// @pulumi/azure-native/config/vars.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const __config = new pulumi.Config("azure-native");
Object.defineProperty(exports, "auxiliaryTenantIds", {
    get() {
        return __config.getObject("auxiliaryTenantIds");
    },
    enumerable: true,
});
Object.defineProperty(exports, "clientCertificatePassword", {
    get() {
        return __config.get("clientCertificatePassword");
    },
    enumerable: true,
});
Object.defineProperty(exports, "clientCertificatePath", {
    get() {
        return __config.get("clientCertificatePath");
    },
    enumerable: true,
});
Object.defineProperty(exports, "clientId", {
    get() {
        return __config.get("clientId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "clientSecret", {
    get() {
        return __config.get("clientSecret");
    },
    enumerable: true,
});
Object.defineProperty(exports, "disablePulumiPartnerId", {
    get() {
        return __config.getObject("disablePulumiPartnerId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "environment", {
    get() {
        return __config.get("environment");
    },
    enumerable: true,
});
Object.defineProperty(exports, "msiEndpoint", {
    get() {
        return __config.get("msiEndpoint");
    },
    enumerable: true,
});
Object.defineProperty(exports, "partnerId", {
    get() {
        return __config.get("partnerId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "subscriptionId", {
    get() {
        return __config.get("subscriptionId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "tenantId", {
    get() {
        return __config.get("tenantId");
    },
    enumerable: true,
});
Object.defineProperty(exports, "useMsi", {
    get() {
        return __config.getObject("useMsi");
    },
    enumerable: true,
});
