// @pulumi/azure-native/customerinsights/v20170101/kpi.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Kpi = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The KPI resource format.
 */
class Kpi extends pulumi.CustomResource {
    /**
     * Create a Kpi resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.calculationWindow === undefined) && !opts.urn) {
                throw new Error("Missing required property 'calculationWindow'");
            }
            if ((!args || args.entityType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'entityType'");
            }
            if ((!args || args.entityTypeName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'entityTypeName'");
            }
            if ((!args || args.expression === undefined) && !opts.urn) {
                throw new Error("Missing required property 'expression'");
            }
            if ((!args || args.function === undefined) && !opts.urn) {
                throw new Error("Missing required property 'function'");
            }
            if ((!args || args.hubName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hubName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["aliases"] = args ? args.aliases : undefined;
            inputs["calculationWindow"] = args ? args.calculationWindow : undefined;
            inputs["calculationWindowFieldName"] = args ? args.calculationWindowFieldName : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["entityType"] = args ? args.entityType : undefined;
            inputs["entityTypeName"] = args ? args.entityTypeName : undefined;
            inputs["expression"] = args ? args.expression : undefined;
            inputs["extracts"] = args ? args.extracts : undefined;
            inputs["filter"] = args ? args.filter : undefined;
            inputs["function"] = args ? args.function : undefined;
            inputs["groupBy"] = args ? args.groupBy : undefined;
            inputs["hubName"] = args ? args.hubName : undefined;
            inputs["kpiName"] = args ? args.kpiName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["thresHolds"] = args ? args.thresHolds : undefined;
            inputs["unit"] = args ? args.unit : undefined;
            inputs["groupByMetadata"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["participantProfilesMetadata"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["tenantId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["aliases"] = undefined /*out*/;
            inputs["calculationWindow"] = undefined /*out*/;
            inputs["calculationWindowFieldName"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["entityType"] = undefined /*out*/;
            inputs["entityTypeName"] = undefined /*out*/;
            inputs["expression"] = undefined /*out*/;
            inputs["extracts"] = undefined /*out*/;
            inputs["filter"] = undefined /*out*/;
            inputs["function"] = undefined /*out*/;
            inputs["groupBy"] = undefined /*out*/;
            inputs["groupByMetadata"] = undefined /*out*/;
            inputs["kpiName"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["participantProfilesMetadata"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["tenantId"] = undefined /*out*/;
            inputs["thresHolds"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["unit"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:customerinsights/v20170101:Kpi" }, { type: "azure-native:customerinsights:Kpi" }, { type: "azure-nextgen:customerinsights:Kpi" }, { type: "azure-native:customerinsights/v20170426:Kpi" }, { type: "azure-nextgen:customerinsights/v20170426:Kpi" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Kpi.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Kpi resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Kpi(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Kpi.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Kpi.__pulumiType;
    }
}
exports.Kpi = Kpi;
/** @internal */
Kpi.__pulumiType = 'azure-native:customerinsights/v20170101:Kpi';
