// @pulumi/azure-native/customerinsights/v20170101/relationshipLink.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RelationshipLink = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The relationship link resource format.
 */
class RelationshipLink extends pulumi.CustomResource {
    /**
     * Create a RelationshipLink resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.hubName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hubName'");
            }
            if ((!args || args.interactionType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'interactionType'");
            }
            if ((!args || args.profilePropertyReferences === undefined) && !opts.urn) {
                throw new Error("Missing required property 'profilePropertyReferences'");
            }
            if ((!args || args.relatedProfilePropertyReferences === undefined) && !opts.urn) {
                throw new Error("Missing required property 'relatedProfilePropertyReferences'");
            }
            if ((!args || args.relationshipName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'relationshipName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["description"] = args ? args.description : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["hubName"] = args ? args.hubName : undefined;
            inputs["interactionType"] = args ? args.interactionType : undefined;
            inputs["mappings"] = args ? args.mappings : undefined;
            inputs["profilePropertyReferences"] = args ? args.profilePropertyReferences : undefined;
            inputs["relatedProfilePropertyReferences"] = args ? args.relatedProfilePropertyReferences : undefined;
            inputs["relationshipLinkName"] = args ? args.relationshipLinkName : undefined;
            inputs["relationshipName"] = args ? args.relationshipName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["linkName"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["relationshipGuidId"] = undefined /*out*/;
            inputs["tenantId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["interactionType"] = undefined /*out*/;
            inputs["linkName"] = undefined /*out*/;
            inputs["mappings"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["profilePropertyReferences"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["relatedProfilePropertyReferences"] = undefined /*out*/;
            inputs["relationshipGuidId"] = undefined /*out*/;
            inputs["relationshipName"] = undefined /*out*/;
            inputs["tenantId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:customerinsights/v20170101:RelationshipLink" }, { type: "azure-native:customerinsights:RelationshipLink" }, { type: "azure-nextgen:customerinsights:RelationshipLink" }, { type: "azure-native:customerinsights/v20170426:RelationshipLink" }, { type: "azure-nextgen:customerinsights/v20170426:RelationshipLink" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RelationshipLink.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing RelationshipLink resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RelationshipLink(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RelationshipLink.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RelationshipLink.__pulumiType;
    }
}
exports.RelationshipLink = RelationshipLink;
/** @internal */
RelationshipLink.__pulumiType = 'azure-native:customerinsights/v20170101:RelationshipLink';
