// @pulumi/azure-native/kubernetesconfiguration/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210501preview = exports.v20210301 = exports.v20201001preview = exports.v20200701preview = exports.v20191101preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./extension"), exports);
__exportStar(require("./getExtension"), exports);
__exportStar(require("./getSourceControlConfiguration"), exports);
__exportStar(require("./sourceControlConfiguration"), exports);
// Export enums:
__exportStar(require("../types/enums/kubernetesconfiguration"), exports);
// Export sub-modules:
const v20191101preview = require("./v20191101preview");
exports.v20191101preview = v20191101preview;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const v20201001preview = require("./v20201001preview");
exports.v20201001preview = v20201001preview;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210501preview = require("./v20210501preview");
exports.v20210501preview = v20210501preview;
// Import resources to register:
const extension_1 = require("./extension");
const sourceControlConfiguration_1 = require("./sourceControlConfiguration");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:kubernetesconfiguration:Extension":
                return new extension_1.Extension(name, undefined, { urn });
            case "azure-native:kubernetesconfiguration:SourceControlConfiguration":
                return new sourceControlConfiguration_1.SourceControlConfiguration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "kubernetesconfiguration", _module);
