// @pulumi/azure-native/storagepool/v20210401preview/iscsiTarget.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IscsiTarget = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Response for iSCSI Target requests.
 */
class IscsiTarget extends pulumi.CustomResource {
    /**
     * Create a IscsiTarget resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.aclMode === undefined) && !opts.urn) {
                throw new Error("Missing required property 'aclMode'");
            }
            if ((!args || args.diskPoolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'diskPoolName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["aclMode"] = args ? args.aclMode : undefined;
            inputs["diskPoolName"] = args ? args.diskPoolName : undefined;
            inputs["iscsiTargetName"] = args ? args.iscsiTargetName : undefined;
            inputs["luns"] = args ? args.luns : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["staticAcls"] = args ? args.staticAcls : undefined;
            inputs["targetIqn"] = args ? args.targetIqn : undefined;
            inputs["endpoints"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["port"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["aclMode"] = undefined /*out*/;
            inputs["endpoints"] = undefined /*out*/;
            inputs["luns"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["port"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["staticAcls"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["targetIqn"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:storagepool/v20210401preview:IscsiTarget" }, { type: "azure-native:storagepool:IscsiTarget" }, { type: "azure-nextgen:storagepool:IscsiTarget" }, { type: "azure-native:storagepool/v20200315preview:IscsiTarget" }, { type: "azure-nextgen:storagepool/v20200315preview:IscsiTarget" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IscsiTarget.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing IscsiTarget resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IscsiTarget(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IscsiTarget.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IscsiTarget.__pulumiType;
    }
}
exports.IscsiTarget = IscsiTarget;
/** @internal */
IscsiTarget.__pulumiType = 'azure-native:storagepool/v20210401preview:IscsiTarget';
