// @pulumi/azure-native/synapse/v20210301/workspace.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workspace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A workspace
 */
class Workspace extends pulumi.CustomResource {
    /**
     * Create a Workspace resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["connectivityEndpoints"] = args ? args.connectivityEndpoints : undefined;
            inputs["defaultDataLakeStorage"] = args ? args.defaultDataLakeStorage : undefined;
            inputs["encryption"] = args ? args.encryption : undefined;
            inputs["identity"] = args ? args.identity : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["managedResourceGroupName"] = args ? args.managedResourceGroupName : undefined;
            inputs["managedVirtualNetwork"] = args ? args.managedVirtualNetwork : undefined;
            inputs["managedVirtualNetworkSettings"] = args ? args.managedVirtualNetworkSettings : undefined;
            inputs["privateEndpointConnections"] = args ? args.privateEndpointConnections : undefined;
            inputs["publicNetworkAccess"] = args ? args.publicNetworkAccess : undefined;
            inputs["purviewConfiguration"] = args ? args.purviewConfiguration : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sqlAdministratorLogin"] = args ? args.sqlAdministratorLogin : undefined;
            inputs["sqlAdministratorLoginPassword"] = args ? args.sqlAdministratorLoginPassword : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["virtualNetworkProfile"] = args ? args.virtualNetworkProfile : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["workspaceRepositoryConfiguration"] = args ? args.workspaceRepositoryConfiguration : undefined;
            inputs["adlaResourceId"] = undefined /*out*/;
            inputs["extraProperties"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["workspaceUID"] = undefined /*out*/;
        }
        else {
            inputs["adlaResourceId"] = undefined /*out*/;
            inputs["connectivityEndpoints"] = undefined /*out*/;
            inputs["defaultDataLakeStorage"] = undefined /*out*/;
            inputs["encryption"] = undefined /*out*/;
            inputs["extraProperties"] = undefined /*out*/;
            inputs["identity"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["managedResourceGroupName"] = undefined /*out*/;
            inputs["managedVirtualNetwork"] = undefined /*out*/;
            inputs["managedVirtualNetworkSettings"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["privateEndpointConnections"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["publicNetworkAccess"] = undefined /*out*/;
            inputs["purviewConfiguration"] = undefined /*out*/;
            inputs["sqlAdministratorLogin"] = undefined /*out*/;
            inputs["sqlAdministratorLoginPassword"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["virtualNetworkProfile"] = undefined /*out*/;
            inputs["workspaceRepositoryConfiguration"] = undefined /*out*/;
            inputs["workspaceUID"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:synapse/v20210301:Workspace" }, { type: "azure-native:synapse:Workspace" }, { type: "azure-nextgen:synapse:Workspace" }, { type: "azure-native:synapse/v20190601preview:Workspace" }, { type: "azure-nextgen:synapse/v20190601preview:Workspace" }, { type: "azure-native:synapse/v20201201:Workspace" }, { type: "azure-nextgen:synapse/v20201201:Workspace" }, { type: "azure-native:synapse/v20210401preview:Workspace" }, { type: "azure-nextgen:synapse/v20210401preview:Workspace" }, { type: "azure-native:synapse/v20210501:Workspace" }, { type: "azure-nextgen:synapse/v20210501:Workspace" }, { type: "azure-native:synapse/v20210601preview:Workspace" }, { type: "azure-nextgen:synapse/v20210601preview:Workspace" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Workspace.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Workspace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Workspace(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Workspace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Workspace.__pulumiType;
    }
}
exports.Workspace = Workspace;
/** @internal */
Workspace.__pulumiType = 'azure-native:synapse/v20210301:Workspace';
