// @pulumi/azure-native/types/enums/documentdb/v20200601preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TriggerType = exports.TriggerOperation = exports.SpatialType = exports.ServerVersion = exports.RoleDefinitionType = exports.RestoreMode = exports.ResourceIdentityType = exports.PartitionKind = exports.IndexingMode = exports.IndexKind = exports.DefaultConsistencyLevel = exports.DatabaseAccountOfferType = exports.DatabaseAccountKind = exports.DataType = exports.CreateMode = exports.ConnectorOffer = exports.ConflictResolutionMode = exports.CompositePathSortOrder = exports.BackupType = exports.BackupStorageRedundancy = void 0;
exports.BackupStorageRedundancy = {
    Geo: "Geo",
    Local: "Local",
    Zone: "Zone",
};
exports.BackupType = {
    Periodic: "Periodic",
    Continuous: "Continuous",
};
exports.CompositePathSortOrder = {
    Ascending: "Ascending",
    Descending: "Descending",
};
exports.ConflictResolutionMode = {
    LastWriterWins: "LastWriterWins",
    Custom: "Custom",
};
exports.ConnectorOffer = {
    Small: "Small",
};
exports.CreateMode = {
    Default: "Default",
    Restore: "Restore",
};
exports.DataType = {
    String: "String",
    Number: "Number",
    Point: "Point",
    Polygon: "Polygon",
    LineString: "LineString",
    MultiPolygon: "MultiPolygon",
};
exports.DatabaseAccountKind = {
    GlobalDocumentDB: "GlobalDocumentDB",
    MongoDB: "MongoDB",
    Parse: "Parse",
};
exports.DatabaseAccountOfferType = {
    Standard: "Standard",
};
exports.DefaultConsistencyLevel = {
    Eventual: "Eventual",
    Session: "Session",
    BoundedStaleness: "BoundedStaleness",
    Strong: "Strong",
    ConsistentPrefix: "ConsistentPrefix",
};
exports.IndexKind = {
    Hash: "Hash",
    Range: "Range",
    Spatial: "Spatial",
};
exports.IndexingMode = {
    Consistent: "Consistent",
    Lazy: "Lazy",
    None: "None",
};
exports.PartitionKind = {
    Hash: "Hash",
    Range: "Range",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
    None: "None",
};
exports.RestoreMode = {
    PointInTime: "PointInTime",
};
exports.RoleDefinitionType = {
    BuiltInRole: "BuiltInRole",
    CustomRole: "CustomRole",
};
exports.ServerVersion = {
    ServerVersion_3_2: "3.2",
    ServerVersion_3_6: "3.6",
};
exports.SpatialType = {
    Point: "Point",
    LineString: "LineString",
    Polygon: "Polygon",
    MultiPolygon: "MultiPolygon",
};
exports.TriggerOperation = {
    All: "All",
    Create: "Create",
    Update: "Update",
    Delete: "Delete",
    Replace: "Replace",
};
exports.TriggerType = {
    Pre: "Pre",
    Post: "Post",
};
