// @pulumi/azure-native/databoxedge/v20201201/kubernetesRole.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.KubernetesRole = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Kubernetes role.
 */
class KubernetesRole extends pulumi.CustomResource {
    /**
     * Create a KubernetesRole resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.deviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deviceName'");
            }
            if ((!args || args.hostPlatform === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hostPlatform'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.kubernetesClusterInfo === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kubernetesClusterInfo'");
            }
            if ((!args || args.kubernetesRoleResources === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kubernetesRoleResources'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.roleStatus === undefined) && !opts.urn) {
                throw new Error("Missing required property 'roleStatus'");
            }
            inputs["deviceName"] = args ? args.deviceName : undefined;
            inputs["hostPlatform"] = args ? args.hostPlatform : undefined;
            inputs["kind"] = "Kubernetes";
            inputs["kubernetesClusterInfo"] = args ? args.kubernetesClusterInfo : undefined;
            inputs["kubernetesRoleResources"] = args ? args.kubernetesRoleResources : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["roleStatus"] = args ? args.roleStatus : undefined;
            inputs["hostPlatformType"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["hostPlatform"] = undefined /*out*/;
            inputs["hostPlatformType"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["kubernetesClusterInfo"] = undefined /*out*/;
            inputs["kubernetesRoleResources"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["roleStatus"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:databoxedge/v20201201:KubernetesRole" }, { type: "azure-native:databoxedge:KubernetesRole" }, { type: "azure-nextgen:databoxedge:KubernetesRole" }, { type: "azure-native:databoxedge/v20190301:KubernetesRole" }, { type: "azure-nextgen:databoxedge/v20190301:KubernetesRole" }, { type: "azure-native:databoxedge/v20190701:KubernetesRole" }, { type: "azure-nextgen:databoxedge/v20190701:KubernetesRole" }, { type: "azure-native:databoxedge/v20190801:KubernetesRole" }, { type: "azure-nextgen:databoxedge/v20190801:KubernetesRole" }, { type: "azure-native:databoxedge/v20200501preview:KubernetesRole" }, { type: "azure-nextgen:databoxedge/v20200501preview:KubernetesRole" }, { type: "azure-native:databoxedge/v20200901:KubernetesRole" }, { type: "azure-nextgen:databoxedge/v20200901:KubernetesRole" }, { type: "azure-native:databoxedge/v20200901preview:KubernetesRole" }, { type: "azure-nextgen:databoxedge/v20200901preview:KubernetesRole" }, { type: "azure-native:databoxedge/v20210201:KubernetesRole" }, { type: "azure-nextgen:databoxedge/v20210201:KubernetesRole" }, { type: "azure-native:databoxedge/v20210201preview:KubernetesRole" }, { type: "azure-nextgen:databoxedge/v20210201preview:KubernetesRole" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(KubernetesRole.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing KubernetesRole resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new KubernetesRole(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of KubernetesRole.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KubernetesRole.__pulumiType;
    }
}
exports.KubernetesRole = KubernetesRole;
/** @internal */
KubernetesRole.__pulumiType = 'azure-native:databoxedge/v20201201:KubernetesRole';
