// @pulumi/azure-native/types/enums/storsimple/v20161001/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SslStatus = exports.ShareStatus = exports.MonitoringStatus = exports.ManagerType = exports.ManagerSkuType = exports.EncryptionStatus = exports.EncryptionAlgorithm = exports.DiskStatus = exports.DataPolicy = exports.CloudType = void 0;
exports.CloudType = {
    Azure: "Azure",
    S3: "S3",
    S3_RRS: "S3_RRS",
    OpenStack: "OpenStack",
    HP: "HP",
};
exports.DataPolicy = {
    Invalid: "Invalid",
    Local: "Local",
    Tiered: "Tiered",
    Cloud: "Cloud",
};
exports.DiskStatus = {
    Online: "Online",
    Offline: "Offline",
};
exports.EncryptionAlgorithm = {
    None: "None",
    AES256: "AES256",
    RSAES_PKCS1_v_1_5: "RSAES_PKCS1_v_1_5",
};
exports.EncryptionStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ManagerSkuType = {
    Standard: "Standard",
};
exports.ManagerType = {
    GardaV1: "GardaV1",
    HelsinkiV1: "HelsinkiV1",
};
exports.MonitoringStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ShareStatus = {
    Online: "Online",
    Offline: "Offline",
};
exports.SslStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
