// @pulumi/azure-native/appconfiguration/v20210301preview/keyValue.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.KeyValue = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The key-value resource along with all resource properties.
 */
class KeyValue extends pulumi.CustomResource {
    /**
     * Create a KeyValue resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.configStoreName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'configStoreName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["configStoreName"] = args ? args.configStoreName : undefined;
            inputs["contentType"] = args ? args.contentType : undefined;
            inputs["keyValueName"] = args ? args.keyValueName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["value"] = args ? args.value : undefined;
            inputs["eTag"] = undefined /*out*/;
            inputs["key"] = undefined /*out*/;
            inputs["label"] = undefined /*out*/;
            inputs["lastModified"] = undefined /*out*/;
            inputs["locked"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["contentType"] = undefined /*out*/;
            inputs["eTag"] = undefined /*out*/;
            inputs["key"] = undefined /*out*/;
            inputs["label"] = undefined /*out*/;
            inputs["lastModified"] = undefined /*out*/;
            inputs["locked"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["value"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:appconfiguration/v20210301preview:KeyValue" }, { type: "azure-native:appconfiguration:KeyValue" }, { type: "azure-nextgen:appconfiguration:KeyValue" }, { type: "azure-native:appconfiguration/v20200701preview:KeyValue" }, { type: "azure-nextgen:appconfiguration/v20200701preview:KeyValue" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(KeyValue.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing KeyValue resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new KeyValue(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of KeyValue.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KeyValue.__pulumiType;
    }
}
exports.KeyValue = KeyValue;
/** @internal */
KeyValue.__pulumiType = 'azure-native:appconfiguration/v20210301preview:KeyValue';
