// @pulumi/azure-native/storage/v20200801preview/objectReplicationPolicy.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectReplicationPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The replication policy between two storage accounts. Multiple rules can be defined in one policy.
 */
class ObjectReplicationPolicy extends pulumi.CustomResource {
    /**
     * Create a ObjectReplicationPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.destinationAccount === undefined) && !opts.urn) {
                throw new Error("Missing required property 'destinationAccount'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sourceAccount === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceAccount'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["destinationAccount"] = args ? args.destinationAccount : undefined;
            inputs["objectReplicationPolicyId"] = args ? args.objectReplicationPolicyId : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["rules"] = args ? args.rules : undefined;
            inputs["sourceAccount"] = args ? args.sourceAccount : undefined;
            inputs["enabledTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["policyId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["destinationAccount"] = undefined /*out*/;
            inputs["enabledTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["policyId"] = undefined /*out*/;
            inputs["rules"] = undefined /*out*/;
            inputs["sourceAccount"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:storage/v20200801preview:ObjectReplicationPolicy" }, { type: "azure-native:storage:ObjectReplicationPolicy" }, { type: "azure-nextgen:storage:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20190601:ObjectReplicationPolicy" }, { type: "azure-nextgen:storage/v20190601:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20210101:ObjectReplicationPolicy" }, { type: "azure-nextgen:storage/v20210101:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20210201:ObjectReplicationPolicy" }, { type: "azure-nextgen:storage/v20210201:ObjectReplicationPolicy" }, { type: "azure-native:storage/v20210401:ObjectReplicationPolicy" }, { type: "azure-nextgen:storage/v20210401:ObjectReplicationPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ObjectReplicationPolicy.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ObjectReplicationPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ObjectReplicationPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ObjectReplicationPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ObjectReplicationPolicy.__pulumiType;
    }
}
exports.ObjectReplicationPolicy = ObjectReplicationPolicy;
/** @internal */
ObjectReplicationPolicy.__pulumiType = 'azure-native:storage/v20200801preview:ObjectReplicationPolicy';
