// @pulumi/azure-native/types/enums/kubernetesconfiguration/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceIdentityType = exports.OperatorType = exports.OperatorScopeType = exports.LevelType = exports.v20210501preview = exports.v20210301 = exports.v20201001preview = exports.v20200701preview = exports.v20191101preview = void 0;
// Export sub-modules:
const v20191101preview = require("./v20191101preview");
exports.v20191101preview = v20191101preview;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const v20201001preview = require("./v20201001preview");
exports.v20201001preview = v20201001preview;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210501preview = require("./v20210501preview");
exports.v20210501preview = v20210501preview;
exports.LevelType = {
    Error: "Error",
    Warning: "Warning",
    Information: "Information",
};
exports.OperatorScopeType = {
    Cluster: "cluster",
    Namespace: "namespace",
};
exports.OperatorType = {
    Flux: "Flux",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    None: "None",
};
