// @pulumi/azure-native/hybridcompute/v20200802/machine.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Machine = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Describes a hybrid machine.
 */
class Machine extends pulumi.CustomResource {
    /**
     * Create a Machine resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["clientPublicKey"] = args ? args.clientPublicKey : undefined;
            inputs["identity"] = args ? args.identity : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["locationData"] = args ? args.locationData : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["vmId"] = args ? args.vmId : undefined;
            inputs["adFqdn"] = undefined /*out*/;
            inputs["agentVersion"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["dnsFqdn"] = undefined /*out*/;
            inputs["domainName"] = undefined /*out*/;
            inputs["errorDetails"] = undefined /*out*/;
            inputs["extensions"] = undefined /*out*/;
            inputs["lastStatusChange"] = undefined /*out*/;
            inputs["machineFqdn"] = undefined /*out*/;
            inputs["osName"] = undefined /*out*/;
            inputs["osProfile"] = undefined /*out*/;
            inputs["osSku"] = undefined /*out*/;
            inputs["osVersion"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["vmUuid"] = undefined /*out*/;
        }
        else {
            inputs["adFqdn"] = undefined /*out*/;
            inputs["agentVersion"] = undefined /*out*/;
            inputs["clientPublicKey"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["dnsFqdn"] = undefined /*out*/;
            inputs["domainName"] = undefined /*out*/;
            inputs["errorDetails"] = undefined /*out*/;
            inputs["extensions"] = undefined /*out*/;
            inputs["identity"] = undefined /*out*/;
            inputs["lastStatusChange"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["locationData"] = undefined /*out*/;
            inputs["machineFqdn"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["osName"] = undefined /*out*/;
            inputs["osProfile"] = undefined /*out*/;
            inputs["osSku"] = undefined /*out*/;
            inputs["osVersion"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["vmId"] = undefined /*out*/;
            inputs["vmUuid"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:hybridcompute/v20200802:Machine" }, { type: "azure-native:hybridcompute:Machine" }, { type: "azure-nextgen:hybridcompute:Machine" }, { type: "azure-native:hybridcompute/v20190318preview:Machine" }, { type: "azure-nextgen:hybridcompute/v20190318preview:Machine" }, { type: "azure-native:hybridcompute/v20190802preview:Machine" }, { type: "azure-nextgen:hybridcompute/v20190802preview:Machine" }, { type: "azure-native:hybridcompute/v20191212:Machine" }, { type: "azure-nextgen:hybridcompute/v20191212:Machine" }, { type: "azure-native:hybridcompute/v20200730preview:Machine" }, { type: "azure-nextgen:hybridcompute/v20200730preview:Machine" }, { type: "azure-native:hybridcompute/v20200815preview:Machine" }, { type: "azure-nextgen:hybridcompute/v20200815preview:Machine" }, { type: "azure-native:hybridcompute/v20210128preview:Machine" }, { type: "azure-nextgen:hybridcompute/v20210128preview:Machine" }, { type: "azure-native:hybridcompute/v20210325preview:Machine" }, { type: "azure-nextgen:hybridcompute/v20210325preview:Machine" }, { type: "azure-native:hybridcompute/v20210422preview:Machine" }, { type: "azure-nextgen:hybridcompute/v20210422preview:Machine" }, { type: "azure-native:hybridcompute/v20210517preview:Machine" }, { type: "azure-nextgen:hybridcompute/v20210517preview:Machine" }, { type: "azure-native:hybridcompute/v20210520:Machine" }, { type: "azure-nextgen:hybridcompute/v20210520:Machine" }, { type: "azure-native:hybridcompute/v20210610preview:Machine" }, { type: "azure-nextgen:hybridcompute/v20210610preview:Machine" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Machine.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Machine resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Machine(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Machine.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Machine.__pulumiType;
    }
}
exports.Machine = Machine;
/** @internal */
Machine.__pulumiType = 'azure-native:hybridcompute/v20200802:Machine';
