// @pulumi/azure-native/types/enums/documentdb/v20150401/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartitionKind = exports.IndexingMode = exports.IndexKind = exports.DefaultConsistencyLevel = exports.DatabaseAccountOfferType = exports.DatabaseAccountKind = exports.DataType = exports.ConnectorOffer = exports.ConflictResolutionMode = void 0;
exports.ConflictResolutionMode = {
    LastWriterWins: "LastWriterWins",
    Custom: "Custom",
};
exports.ConnectorOffer = {
    Small: "Small",
};
exports.DataType = {
    String: "String",
    Number: "Number",
    Point: "Point",
    Polygon: "Polygon",
    LineString: "LineString",
    MultiPolygon: "MultiPolygon",
};
exports.DatabaseAccountKind = {
    GlobalDocumentDB: "GlobalDocumentDB",
    MongoDB: "MongoDB",
    Parse: "Parse",
};
exports.DatabaseAccountOfferType = {
    Standard: "Standard",
};
exports.DefaultConsistencyLevel = {
    Eventual: "Eventual",
    Session: "Session",
    BoundedStaleness: "BoundedStaleness",
    Strong: "Strong",
    ConsistentPrefix: "ConsistentPrefix",
};
exports.IndexKind = {
    Hash: "Hash",
    Range: "Range",
    Spatial: "Spatial",
};
exports.IndexingMode = {
    Consistent: "Consistent",
    Lazy: "Lazy",
    None: "None",
};
exports.PartitionKind = {
    Hash: "Hash",
    Range: "Range",
};
