// @pulumi/azure-native/types/enums/eventgrid/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceProvisioningState = exports.PublicNetworkAccess = exports.PersistedConnectionStatus = exports.PartnerRegistrationVisibilityState = exports.IpActionType = exports.InputSchemaMappingType = exports.InputSchema = exports.IdentityType = exports.EventSubscriptionIdentityType = exports.EventDeliverySchema = exports.EndpointType = exports.DeadLetterEndPointType = exports.AdvancedFilterOperatorType = exports.v20210601preview = exports.v20201015preview = exports.v20200601 = exports.v20200401preview = exports.v20200101preview = exports.v20190601 = exports.v20190201preview = exports.v20190101 = exports.v20180915preview = exports.v20180501preview = exports.v20180101 = exports.v20170915preview = exports.v20170615preview = void 0;
// Export sub-modules:
const v20170615preview = require("./v20170615preview");
exports.v20170615preview = v20170615preview;
const v20170915preview = require("./v20170915preview");
exports.v20170915preview = v20170915preview;
const v20180101 = require("./v20180101");
exports.v20180101 = v20180101;
const v20180501preview = require("./v20180501preview");
exports.v20180501preview = v20180501preview;
const v20180915preview = require("./v20180915preview");
exports.v20180915preview = v20180915preview;
const v20190101 = require("./v20190101");
exports.v20190101 = v20190101;
const v20190201preview = require("./v20190201preview");
exports.v20190201preview = v20190201preview;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20200101preview = require("./v20200101preview");
exports.v20200101preview = v20200101preview;
const v20200401preview = require("./v20200401preview");
exports.v20200401preview = v20200401preview;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20201015preview = require("./v20201015preview");
exports.v20201015preview = v20201015preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
exports.AdvancedFilterOperatorType = {
    NumberIn: "NumberIn",
    NumberNotIn: "NumberNotIn",
    NumberLessThan: "NumberLessThan",
    NumberGreaterThan: "NumberGreaterThan",
    NumberLessThanOrEquals: "NumberLessThanOrEquals",
    NumberGreaterThanOrEquals: "NumberGreaterThanOrEquals",
    BoolEquals: "BoolEquals",
    StringIn: "StringIn",
    StringNotIn: "StringNotIn",
    StringBeginsWith: "StringBeginsWith",
    StringEndsWith: "StringEndsWith",
    StringContains: "StringContains",
};
exports.DeadLetterEndPointType = {
    StorageBlob: "StorageBlob",
};
exports.EndpointType = {
    WebHook: "WebHook",
    EventHub: "EventHub",
    StorageQueue: "StorageQueue",
    HybridConnection: "HybridConnection",
    ServiceBusQueue: "ServiceBusQueue",
    ServiceBusTopic: "ServiceBusTopic",
    AzureFunction: "AzureFunction",
};
exports.EventDeliverySchema = {
    EventGridSchema: "EventGridSchema",
    CustomInputSchema: "CustomInputSchema",
    CloudEventSchemaV1_0: "CloudEventSchemaV1_0",
};
exports.EventSubscriptionIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.InputSchema = {
    EventGridSchema: "EventGridSchema",
    CustomEventSchema: "CustomEventSchema",
    CloudEventSchemaV1_0: "CloudEventSchemaV1_0",
};
exports.InputSchemaMappingType = {
    Json: "Json",
};
exports.IpActionType = {
    Allow: "Allow",
};
exports.PartnerRegistrationVisibilityState = {
    Hidden: "Hidden",
    PublicPreview: "PublicPreview",
    GenerallyAvailable: "GenerallyAvailable",
};
exports.PersistedConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ResourceProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
