// @pulumi/azure-native/devtestlab/virtualMachine.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMachine = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A virtual machine.
 * API Version: 2018-09-15.
 */
class VirtualMachine extends pulumi.CustomResource {
    /**
     * Create a VirtualMachine resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b, _c, _d;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.labName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'labName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["allowClaim"] = (_a = (args ? args.allowClaim : undefined)) !== null && _a !== void 0 ? _a : false;
            inputs["artifacts"] = args ? args.artifacts : undefined;
            inputs["createdDate"] = args ? args.createdDate : undefined;
            inputs["customImageId"] = args ? args.customImageId : undefined;
            inputs["dataDiskParameters"] = args ? args.dataDiskParameters : undefined;
            inputs["disallowPublicIpAddress"] = (_b = (args ? args.disallowPublicIpAddress : undefined)) !== null && _b !== void 0 ? _b : false;
            inputs["environmentId"] = args ? args.environmentId : undefined;
            inputs["expirationDate"] = args ? args.expirationDate : undefined;
            inputs["galleryImageReference"] = args ? args.galleryImageReference : undefined;
            inputs["isAuthenticationWithSshKey"] = args ? args.isAuthenticationWithSshKey : undefined;
            inputs["labName"] = args ? args.labName : undefined;
            inputs["labSubnetName"] = args ? args.labSubnetName : undefined;
            inputs["labVirtualNetworkId"] = args ? args.labVirtualNetworkId : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["networkInterface"] = args ? args.networkInterface : undefined;
            inputs["notes"] = args ? args.notes : undefined;
            inputs["ownerObjectId"] = (_c = (args ? args.ownerObjectId : undefined)) !== null && _c !== void 0 ? _c : "dynamicValue";
            inputs["ownerUserPrincipalName"] = args ? args.ownerUserPrincipalName : undefined;
            inputs["password"] = args ? args.password : undefined;
            inputs["planId"] = args ? args.planId : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["scheduleParameters"] = args ? args.scheduleParameters : undefined;
            inputs["size"] = args ? args.size : undefined;
            inputs["sshKey"] = args ? args.sshKey : undefined;
            inputs["storageType"] = (_d = (args ? args.storageType : undefined)) !== null && _d !== void 0 ? _d : "labStorageType";
            inputs["tags"] = args ? args.tags : undefined;
            inputs["userName"] = args ? args.userName : undefined;
            inputs["applicableSchedule"] = undefined /*out*/;
            inputs["artifactDeploymentStatus"] = undefined /*out*/;
            inputs["computeId"] = undefined /*out*/;
            inputs["computeVm"] = undefined /*out*/;
            inputs["createdByUser"] = undefined /*out*/;
            inputs["createdByUserId"] = undefined /*out*/;
            inputs["fqdn"] = undefined /*out*/;
            inputs["lastKnownPowerState"] = undefined /*out*/;
            inputs["osType"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["uniqueIdentifier"] = undefined /*out*/;
            inputs["virtualMachineCreationSource"] = undefined /*out*/;
        }
        else {
            inputs["allowClaim"] = undefined /*out*/;
            inputs["applicableSchedule"] = undefined /*out*/;
            inputs["artifactDeploymentStatus"] = undefined /*out*/;
            inputs["artifacts"] = undefined /*out*/;
            inputs["computeId"] = undefined /*out*/;
            inputs["computeVm"] = undefined /*out*/;
            inputs["createdByUser"] = undefined /*out*/;
            inputs["createdByUserId"] = undefined /*out*/;
            inputs["createdDate"] = undefined /*out*/;
            inputs["customImageId"] = undefined /*out*/;
            inputs["dataDiskParameters"] = undefined /*out*/;
            inputs["disallowPublicIpAddress"] = undefined /*out*/;
            inputs["environmentId"] = undefined /*out*/;
            inputs["expirationDate"] = undefined /*out*/;
            inputs["fqdn"] = undefined /*out*/;
            inputs["galleryImageReference"] = undefined /*out*/;
            inputs["isAuthenticationWithSshKey"] = undefined /*out*/;
            inputs["labSubnetName"] = undefined /*out*/;
            inputs["labVirtualNetworkId"] = undefined /*out*/;
            inputs["lastKnownPowerState"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["networkInterface"] = undefined /*out*/;
            inputs["notes"] = undefined /*out*/;
            inputs["osType"] = undefined /*out*/;
            inputs["ownerObjectId"] = undefined /*out*/;
            inputs["ownerUserPrincipalName"] = undefined /*out*/;
            inputs["password"] = undefined /*out*/;
            inputs["planId"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["scheduleParameters"] = undefined /*out*/;
            inputs["size"] = undefined /*out*/;
            inputs["sshKey"] = undefined /*out*/;
            inputs["storageType"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["uniqueIdentifier"] = undefined /*out*/;
            inputs["userName"] = undefined /*out*/;
            inputs["virtualMachineCreationSource"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:devtestlab:VirtualMachine" }, { type: "azure-native:devtestlab/v20150521preview:VirtualMachine" }, { type: "azure-nextgen:devtestlab/v20150521preview:VirtualMachine" }, { type: "azure-native:devtestlab/v20160515:VirtualMachine" }, { type: "azure-nextgen:devtestlab/v20160515:VirtualMachine" }, { type: "azure-native:devtestlab/v20180915:VirtualMachine" }, { type: "azure-nextgen:devtestlab/v20180915:VirtualMachine" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VirtualMachine.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing VirtualMachine resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VirtualMachine(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualMachine.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualMachine.__pulumiType;
    }
}
exports.VirtualMachine = VirtualMachine;
/** @internal */
VirtualMachine.__pulumiType = 'azure-native:devtestlab:VirtualMachine';
