// @pulumi/azure-native/domainregistration/v20190801/domainOwnershipIdentifier.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainOwnershipIdentifier = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Domain ownership Identifier.
 */
class DomainOwnershipIdentifier extends pulumi.CustomResource {
    /**
     * Create a DomainOwnershipIdentifier resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.domainName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'domainName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["domainName"] = args ? args.domainName : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["ownershipId"] = args ? args.ownershipId : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["ownershipId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:domainregistration/v20190801:DomainOwnershipIdentifier" }, { type: "azure-native:domainregistration:DomainOwnershipIdentifier" }, { type: "azure-nextgen:domainregistration:DomainOwnershipIdentifier" }, { type: "azure-native:domainregistration/v20150401:DomainOwnershipIdentifier" }, { type: "azure-nextgen:domainregistration/v20150401:DomainOwnershipIdentifier" }, { type: "azure-native:domainregistration/v20180201:DomainOwnershipIdentifier" }, { type: "azure-nextgen:domainregistration/v20180201:DomainOwnershipIdentifier" }, { type: "azure-native:domainregistration/v20200601:DomainOwnershipIdentifier" }, { type: "azure-nextgen:domainregistration/v20200601:DomainOwnershipIdentifier" }, { type: "azure-native:domainregistration/v20200901:DomainOwnershipIdentifier" }, { type: "azure-nextgen:domainregistration/v20200901:DomainOwnershipIdentifier" }, { type: "azure-native:domainregistration/v20201001:DomainOwnershipIdentifier" }, { type: "azure-nextgen:domainregistration/v20201001:DomainOwnershipIdentifier" }, { type: "azure-native:domainregistration/v20201201:DomainOwnershipIdentifier" }, { type: "azure-nextgen:domainregistration/v20201201:DomainOwnershipIdentifier" }, { type: "azure-native:domainregistration/v20210101:DomainOwnershipIdentifier" }, { type: "azure-nextgen:domainregistration/v20210101:DomainOwnershipIdentifier" }, { type: "azure-native:domainregistration/v20210115:DomainOwnershipIdentifier" }, { type: "azure-nextgen:domainregistration/v20210115:DomainOwnershipIdentifier" }, { type: "azure-native:domainregistration/v20210201:DomainOwnershipIdentifier" }, { type: "azure-nextgen:domainregistration/v20210201:DomainOwnershipIdentifier" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DomainOwnershipIdentifier.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DomainOwnershipIdentifier resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DomainOwnershipIdentifier(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DomainOwnershipIdentifier.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DomainOwnershipIdentifier.__pulumiType;
    }
}
exports.DomainOwnershipIdentifier = DomainOwnershipIdentifier;
/** @internal */
DomainOwnershipIdentifier.__pulumiType = 'azure-native:domainregistration/v20190801:DomainOwnershipIdentifier';
