// @pulumi/azure-native/eventgrid/v20201015preview/partnerRegistration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartnerRegistration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Information about a partner registration.
 */
class PartnerRegistration extends pulumi.CustomResource {
    /**
     * Create a PartnerRegistration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["authorizedAzureSubscriptionIds"] = args ? args.authorizedAzureSubscriptionIds : undefined;
            inputs["customerServiceUri"] = args ? args.customerServiceUri : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["logoUri"] = args ? args.logoUri : undefined;
            inputs["longDescription"] = args ? args.longDescription : undefined;
            inputs["partnerCustomerServiceExtension"] = args ? args.partnerCustomerServiceExtension : undefined;
            inputs["partnerCustomerServiceNumber"] = args ? args.partnerCustomerServiceNumber : undefined;
            inputs["partnerName"] = args ? args.partnerName : undefined;
            inputs["partnerRegistrationName"] = args ? args.partnerRegistrationName : undefined;
            inputs["partnerResourceTypeDescription"] = args ? args.partnerResourceTypeDescription : undefined;
            inputs["partnerResourceTypeDisplayName"] = args ? args.partnerResourceTypeDisplayName : undefined;
            inputs["partnerResourceTypeName"] = args ? args.partnerResourceTypeName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["setupUri"] = args ? args.setupUri : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["visibilityState"] = args ? args.visibilityState : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["authorizedAzureSubscriptionIds"] = undefined /*out*/;
            inputs["customerServiceUri"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["logoUri"] = undefined /*out*/;
            inputs["longDescription"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["partnerCustomerServiceExtension"] = undefined /*out*/;
            inputs["partnerCustomerServiceNumber"] = undefined /*out*/;
            inputs["partnerName"] = undefined /*out*/;
            inputs["partnerResourceTypeDescription"] = undefined /*out*/;
            inputs["partnerResourceTypeDisplayName"] = undefined /*out*/;
            inputs["partnerResourceTypeName"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["setupUri"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["visibilityState"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:eventgrid/v20201015preview:PartnerRegistration" }, { type: "azure-native:eventgrid:PartnerRegistration" }, { type: "azure-nextgen:eventgrid:PartnerRegistration" }, { type: "azure-native:eventgrid/v20200401preview:PartnerRegistration" }, { type: "azure-nextgen:eventgrid/v20200401preview:PartnerRegistration" }, { type: "azure-native:eventgrid/v20210601preview:PartnerRegistration" }, { type: "azure-nextgen:eventgrid/v20210601preview:PartnerRegistration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PartnerRegistration.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing PartnerRegistration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PartnerRegistration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PartnerRegistration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PartnerRegistration.__pulumiType;
    }
}
exports.PartnerRegistration = PartnerRegistration;
/** @internal */
PartnerRegistration.__pulumiType = 'azure-native:eventgrid/v20201015preview:PartnerRegistration';
