// @pulumi/azure-native/hybridnetwork/v20210501/vendor.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Vendor = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Vendor resource.
 */
class Vendor extends pulumi.CustomResource {
    /**
     * Create a Vendor resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["vendorName"] = args ? args.vendorName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["skus"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["skus"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:hybridnetwork/v20210501:Vendor" }, { type: "azure-native:hybridnetwork:Vendor" }, { type: "azure-nextgen:hybridnetwork:Vendor" }, { type: "azure-native:hybridnetwork/v20200101preview:Vendor" }, { type: "azure-nextgen:hybridnetwork/v20200101preview:Vendor" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Vendor.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Vendor resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Vendor(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Vendor.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Vendor.__pulumiType;
    }
}
exports.Vendor = Vendor;
/** @internal */
Vendor.__pulumiType = 'azure-native:hybridnetwork/v20210501:Vendor';
