// @pulumi/azure-native/types/enums/dbformysql/v20171201preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageAutogrow = exports.SslEnforcementEnum = exports.SkuTier = exports.ServerVersion = exports.MinimalTlsVersionEnum = exports.GeoRedundantBackup = exports.CreateMode = exports.AdministratorType = void 0;
exports.AdministratorType = {
    ActiveDirectory: "ActiveDirectory",
};
exports.CreateMode = {
    Default: "Default",
    PointInTimeRestore: "PointInTimeRestore",
    GeoRestore: "GeoRestore",
    Replica: "Replica",
};
exports.GeoRedundantBackup = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.MinimalTlsVersionEnum = {
    TLS1_0: "TLS1_0",
    TLS1_1: "TLS1_1",
    TLS1_2: "TLS1_2",
    TLSEnforcementDisabled: "TLSEnforcementDisabled",
};
exports.ServerVersion = {
    ServerVersion_5_6: "5.6",
    ServerVersion_5_7: "5.7",
    ServerVersion_8_0: "8.0",
};
exports.SkuTier = {
    Basic: "Basic",
    GeneralPurpose: "GeneralPurpose",
    MemoryOptimized: "MemoryOptimized",
};
exports.SslEnforcementEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.StorageAutogrow = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
