// @pulumi/azure-native/synapse/v20200401preview/sqlPoolsV3.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlPoolsV3 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A sql pool resource.
 */
class SqlPoolsV3 extends pulumi.CustomResource {
    /**
     * Create a SqlPoolsV3 resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["autoPauseTimer"] = args ? args.autoPauseTimer : undefined;
            inputs["autoResume"] = args ? args.autoResume : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["maxServiceObjectiveName"] = args ? args.maxServiceObjectiveName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["sqlPoolName"] = args ? args.sqlPoolName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["currentServiceObjectiveName"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["requestedServiceObjectiveName"] = undefined /*out*/;
            inputs["sqlPoolGuid"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["autoPauseTimer"] = undefined /*out*/;
            inputs["autoResume"] = undefined /*out*/;
            inputs["currentServiceObjectiveName"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["maxServiceObjectiveName"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["requestedServiceObjectiveName"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["sqlPoolGuid"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:synapse/v20200401preview:SqlPoolsV3" }, { type: "azure-native:synapse:SqlPoolsV3" }, { type: "azure-nextgen:synapse:SqlPoolsV3" }, { type: "azure-native:synapse/v20190601preview:SqlPoolsV3" }, { type: "azure-nextgen:synapse/v20190601preview:SqlPoolsV3" }, { type: "azure-native:synapse/v20201201:SqlPoolsV3" }, { type: "azure-nextgen:synapse/v20201201:SqlPoolsV3" }, { type: "azure-native:synapse/v20210301:SqlPoolsV3" }, { type: "azure-nextgen:synapse/v20210301:SqlPoolsV3" }, { type: "azure-native:synapse/v20210401preview:SqlPoolsV3" }, { type: "azure-nextgen:synapse/v20210401preview:SqlPoolsV3" }, { type: "azure-native:synapse/v20210501:SqlPoolsV3" }, { type: "azure-nextgen:synapse/v20210501:SqlPoolsV3" }, { type: "azure-native:synapse/v20210601preview:SqlPoolsV3" }, { type: "azure-nextgen:synapse/v20210601preview:SqlPoolsV3" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlPoolsV3.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SqlPoolsV3 resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SqlPoolsV3(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlPoolsV3.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlPoolsV3.__pulumiType;
    }
}
exports.SqlPoolsV3 = SqlPoolsV3;
/** @internal */
SqlPoolsV3.__pulumiType = 'azure-native:synapse/v20200401preview:SqlPoolsV3';
