// @pulumi/azure-native/automation/v20200113preview/schedule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Definition of the schedule.
 */
class Schedule extends pulumi.CustomResource {
    /**
     * Create a Schedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.automationAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'automationAccountName'");
            }
            if ((!args || args.frequency === undefined) && !opts.urn) {
                throw new Error("Missing required property 'frequency'");
            }
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.startTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'startTime'");
            }
            inputs["advancedSchedule"] = args ? args.advancedSchedule : undefined;
            inputs["automationAccountName"] = args ? args.automationAccountName : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["expiryTime"] = args ? args.expiryTime : undefined;
            inputs["frequency"] = args ? args.frequency : undefined;
            inputs["interval"] = args ? args.interval : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["scheduleName"] = args ? args.scheduleName : undefined;
            inputs["startTime"] = args ? args.startTime : undefined;
            inputs["timeZone"] = args ? args.timeZone : undefined;
            inputs["creationTime"] = undefined /*out*/;
            inputs["expiryTimeOffsetMinutes"] = undefined /*out*/;
            inputs["isEnabled"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["nextRun"] = undefined /*out*/;
            inputs["nextRunOffsetMinutes"] = undefined /*out*/;
            inputs["startTimeOffsetMinutes"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["advancedSchedule"] = undefined /*out*/;
            inputs["creationTime"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["expiryTime"] = undefined /*out*/;
            inputs["expiryTimeOffsetMinutes"] = undefined /*out*/;
            inputs["frequency"] = undefined /*out*/;
            inputs["interval"] = undefined /*out*/;
            inputs["isEnabled"] = undefined /*out*/;
            inputs["lastModifiedTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["nextRun"] = undefined /*out*/;
            inputs["nextRunOffsetMinutes"] = undefined /*out*/;
            inputs["startTime"] = undefined /*out*/;
            inputs["startTimeOffsetMinutes"] = undefined /*out*/;
            inputs["timeZone"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:automation/v20200113preview:Schedule" }, { type: "azure-native:automation:Schedule" }, { type: "azure-nextgen:automation:Schedule" }, { type: "azure-native:automation/v20151031:Schedule" }, { type: "azure-nextgen:automation/v20151031:Schedule" }, { type: "azure-native:automation/v20190601:Schedule" }, { type: "azure-nextgen:automation/v20190601:Schedule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Schedule.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Schedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Schedule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Schedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Schedule.__pulumiType;
    }
}
exports.Schedule = Schedule;
/** @internal */
Schedule.__pulumiType = 'azure-native:automation/v20200113preview:Schedule';
