// @pulumi/azure-native/types/enums/hdinsight/v20150301preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tier = exports.ResourceProviderConnection = exports.ResourceIdentityType = exports.PrivateLink = exports.OSType = exports.JsonWebKeyEncryptionAlgorithm = exports.DirectoryType = exports.DaysOfWeek = void 0;
exports.DaysOfWeek = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.DirectoryType = {
    ActiveDirectory: "ActiveDirectory",
};
exports.JsonWebKeyEncryptionAlgorithm = {
    RSA_OAEP: "RSA-OAEP",
    RSA_OAEP_256: "RSA-OAEP-256",
    RSA1_5: "RSA1_5",
};
exports.OSType = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.PrivateLink = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.ResourceProviderConnection = {
    Inbound: "Inbound",
    Outbound: "Outbound",
};
exports.Tier = {
    Standard: "Standard",
    Premium: "Premium",
};
