// @pulumi/azure-native/documentdb/v20160319/databaseAccountGremlinGraph.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseAccountGremlinGraph = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An Azure Cosmos DB Gremlin graph.
 */
class DatabaseAccountGremlinGraph extends pulumi.CustomResource {
    /**
     * Create a DatabaseAccountGremlinGraph resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.options === undefined) && !opts.urn) {
                throw new Error("Missing required property 'options'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["databaseName"] = args ? args.databaseName : undefined;
            inputs["graphName"] = args ? args.graphName : undefined;
            inputs["options"] = args ? args.options : undefined;
            inputs["resource"] = args ? args.resource : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["conflictResolutionPolicy"] = undefined /*out*/;
            inputs["defaultTtl"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["indexingPolicy"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["partitionKey"] = undefined /*out*/;
            inputs["rid"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["ts"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["uniqueKeyPolicy"] = undefined /*out*/;
        }
        else {
            inputs["conflictResolutionPolicy"] = undefined /*out*/;
            inputs["defaultTtl"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["indexingPolicy"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["partitionKey"] = undefined /*out*/;
            inputs["rid"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["ts"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["uniqueKeyPolicy"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:documentdb/v20160319:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20150401:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20150401:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20150408:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20150408:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20151106:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20151106:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20160331:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20160331:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20190801:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20190801:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20191212:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20191212:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20200301:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20200301:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20200401:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20200401:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20200601preview:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20200601preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20200901:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20200901:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20210115:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20210115:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20210301preview:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20210301preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20210315:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20210315:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20210401preview:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20210401preview:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20210415:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20210415:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20210515:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20210515:DatabaseAccountGremlinGraph" }, { type: "azure-native:documentdb/v20210615:DatabaseAccountGremlinGraph" }, { type: "azure-nextgen:documentdb/v20210615:DatabaseAccountGremlinGraph" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DatabaseAccountGremlinGraph.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DatabaseAccountGremlinGraph resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DatabaseAccountGremlinGraph(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DatabaseAccountGremlinGraph.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DatabaseAccountGremlinGraph.__pulumiType;
    }
}
exports.DatabaseAccountGremlinGraph = DatabaseAccountGremlinGraph;
/** @internal */
DatabaseAccountGremlinGraph.__pulumiType = 'azure-native:documentdb/v20160319:DatabaseAccountGremlinGraph';
