// @pulumi/azure-native/peering/v20201001/registeredPrefix.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegisteredPrefix = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The customer's prefix that is registered by the peering service provider.
 */
class RegisteredPrefix extends pulumi.CustomResource {
    /**
     * Create a RegisteredPrefix resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.peeringName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peeringName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["peeringName"] = args ? args.peeringName : undefined;
            inputs["prefix"] = args ? args.prefix : undefined;
            inputs["registeredPrefixName"] = args ? args.registeredPrefixName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["errorMessage"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["peeringServicePrefixKey"] = undefined /*out*/;
            inputs["prefixValidationState"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["errorMessage"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["peeringServicePrefixKey"] = undefined /*out*/;
            inputs["prefix"] = undefined /*out*/;
            inputs["prefixValidationState"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:peering/v20201001:RegisteredPrefix" }, { type: "azure-native:peering:RegisteredPrefix" }, { type: "azure-nextgen:peering:RegisteredPrefix" }, { type: "azure-native:peering/v20200101preview:RegisteredPrefix" }, { type: "azure-nextgen:peering/v20200101preview:RegisteredPrefix" }, { type: "azure-native:peering/v20200401:RegisteredPrefix" }, { type: "azure-nextgen:peering/v20200401:RegisteredPrefix" }, { type: "azure-native:peering/v20210101:RegisteredPrefix" }, { type: "azure-nextgen:peering/v20210101:RegisteredPrefix" }, { type: "azure-native:peering/v20210601:RegisteredPrefix" }, { type: "azure-nextgen:peering/v20210601:RegisteredPrefix" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RegisteredPrefix.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing RegisteredPrefix resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RegisteredPrefix(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RegisteredPrefix.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RegisteredPrefix.__pulumiType;
    }
}
exports.RegisteredPrefix = RegisteredPrefix;
/** @internal */
RegisteredPrefix.__pulumiType = 'azure-native:peering/v20201001:RegisteredPrefix';
