// @pulumi/azure-native/azurestack/customerSubscription.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerSubscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Customer subscription.
 * API Version: 2017-06-01.
 */
class CustomerSubscription extends pulumi.CustomResource {
    /**
     * Create a CustomerSubscription resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.registrationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'registrationName'");
            }
            if ((!args || args.resourceGroup === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroup'");
            }
            inputs["customerSubscriptionName"] = args ? args.customerSubscriptionName : undefined;
            inputs["etag"] = args ? args.etag : undefined;
            inputs["registrationName"] = args ? args.registrationName : undefined;
            inputs["resourceGroup"] = args ? args.resourceGroup : undefined;
            inputs["tenantId"] = args ? args.tenantId : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["tenantId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:azurestack:CustomerSubscription" }, { type: "azure-native:azurestack/v20170601:CustomerSubscription" }, { type: "azure-nextgen:azurestack/v20170601:CustomerSubscription" }, { type: "azure-native:azurestack/v20200601preview:CustomerSubscription" }, { type: "azure-nextgen:azurestack/v20200601preview:CustomerSubscription" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CustomerSubscription.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing CustomerSubscription resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new CustomerSubscription(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CustomerSubscription.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomerSubscription.__pulumiType;
    }
}
exports.CustomerSubscription = CustomerSubscription;
/** @internal */
CustomerSubscription.__pulumiType = 'azure-native:azurestack:CustomerSubscription';
