// @pulumi/azure-native/marketplace/v20200101/privateStoreOffer.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateStoreOffer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The privateStore offer data structure.
 */
class PrivateStoreOffer extends pulumi.CustomResource {
    /**
     * Create a PrivateStoreOffer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.privateStoreId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'privateStoreId'");
            }
            inputs["eTag"] = args ? args.eTag : undefined;
            inputs["iconFileUris"] = args ? args.iconFileUris : undefined;
            inputs["offerId"] = args ? args.offerId : undefined;
            inputs["plans"] = args ? args.plans : undefined;
            inputs["privateStoreId"] = args ? args.privateStoreId : undefined;
            inputs["specificPlanIdsLimitation"] = args ? args.specificPlanIdsLimitation : undefined;
            inputs["updateSuppressedDueIdempotence"] = args ? args.updateSuppressedDueIdempotence : undefined;
            inputs["createdAt"] = undefined /*out*/;
            inputs["modifiedAt"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["offerDisplayName"] = undefined /*out*/;
            inputs["publisherDisplayName"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["uniqueOfferId"] = undefined /*out*/;
        }
        else {
            inputs["createdAt"] = undefined /*out*/;
            inputs["eTag"] = undefined /*out*/;
            inputs["iconFileUris"] = undefined /*out*/;
            inputs["modifiedAt"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["offerDisplayName"] = undefined /*out*/;
            inputs["plans"] = undefined /*out*/;
            inputs["privateStoreId"] = undefined /*out*/;
            inputs["publisherDisplayName"] = undefined /*out*/;
            inputs["specificPlanIdsLimitation"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["uniqueOfferId"] = undefined /*out*/;
            inputs["updateSuppressedDueIdempotence"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:marketplace/v20200101:PrivateStoreOffer" }, { type: "azure-native:marketplace:PrivateStoreOffer" }, { type: "azure-nextgen:marketplace:PrivateStoreOffer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateStoreOffer.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing PrivateStoreOffer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateStoreOffer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivateStoreOffer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateStoreOffer.__pulumiType;
    }
}
exports.PrivateStoreOffer = PrivateStoreOffer;
/** @internal */
PrivateStoreOffer.__pulumiType = 'azure-native:marketplace/v20200101:PrivateStoreOffer';
