// @pulumi/azure-native/types/enums/machinelearningservices/v20200901preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VmPriority = exports.VariantType = exports.SshPublicAccess = exports.ResourceIdentityType = exports.RemoteLoginPortPublicAccess = exports.PrivateEndpointServiceConnectionStatus = exports.OsType = exports.MediaType = exports.LinkedServiceLinkType = exports.ImageAnnotationType = exports.EncryptionStatus = exports.ComputeType = exports.ComputeInstanceAuthorizationType = exports.ComputeEnvironmentType = exports.ApplicationSharingPolicy = void 0;
exports.ApplicationSharingPolicy = {
    Personal: "Personal",
    Shared: "Shared",
};
exports.ComputeEnvironmentType = {
    ACI: "ACI",
    AKS: "AKS",
};
exports.ComputeInstanceAuthorizationType = {
    Personal: "personal",
};
exports.ComputeType = {
    AKS: "AKS",
    AmlCompute: "AmlCompute",
    ComputeInstance: "ComputeInstance",
    DataFactory: "DataFactory",
    VirtualMachine: "VirtualMachine",
    HDInsight: "HDInsight",
    Databricks: "Databricks",
    DataLakeAnalytics: "DataLakeAnalytics",
};
exports.EncryptionStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ImageAnnotationType = {
    Classification: "Classification",
    BoundingBox: "BoundingBox",
    InstanceSegmentation: "InstanceSegmentation",
};
exports.LinkedServiceLinkType = {
    Synapse: "Synapse",
};
exports.MediaType = {
    Image: "Image",
    Text: "Text",
};
exports.OsType = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
    Timeout: "Timeout",
};
exports.RemoteLoginPortPublicAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
    NotSpecified: "NotSpecified",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
    UserAssigned: "UserAssigned",
    None: "None",
};
exports.SshPublicAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.VariantType = {
    Control: "Control",
    Treatment: "Treatment",
};
exports.VmPriority = {
    Dedicated: "Dedicated",
    LowPriority: "LowPriority",
};
