// @pulumi/azure-native/azuredata/v20170301preview/sqlServerRegistration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlServerRegistration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A SQL server registration.
 */
class SqlServerRegistration extends pulumi.CustomResource {
    /**
     * Create a SqlServerRegistration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["location"] = args ? args.location : undefined;
            inputs["propertyBag"] = args ? args.propertyBag : undefined;
            inputs["resourceGroup"] = args ? args.resourceGroup : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sqlServerRegistrationName"] = args ? args.sqlServerRegistrationName : undefined;
            inputs["subscriptionId"] = args ? args.subscriptionId : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["propertyBag"] = undefined /*out*/;
            inputs["resourceGroup"] = undefined /*out*/;
            inputs["subscriptionId"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:azuredata/v20170301preview:SqlServerRegistration" }, { type: "azure-native:azuredata:SqlServerRegistration" }, { type: "azure-nextgen:azuredata:SqlServerRegistration" }, { type: "azure-native:azuredata/v20190724preview:SqlServerRegistration" }, { type: "azure-nextgen:azuredata/v20190724preview:SqlServerRegistration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlServerRegistration.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SqlServerRegistration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SqlServerRegistration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlServerRegistration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlServerRegistration.__pulumiType;
    }
}
exports.SqlServerRegistration = SqlServerRegistration;
/** @internal */
SqlServerRegistration.__pulumiType = 'azure-native:azuredata/v20170301preview:SqlServerRegistration';
