// @pulumi/azure-native/fluidrelay/v20210312preview/fluidRelayServer.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FluidRelayServer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A FluidRelay Server.
 */
class FluidRelayServer extends pulumi.CustomResource {
    /**
     * Create a FluidRelayServer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroup === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroup'");
            }
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["provisioningState"] = args ? args.provisioningState : undefined;
            inputs["resourceGroup"] = args ? args.resourceGroup : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["fluidRelayEndpoints"] = undefined /*out*/;
            inputs["frsTenantId"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["fluidRelayEndpoints"] = undefined /*out*/;
            inputs["frsTenantId"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:fluidrelay/v20210312preview:FluidRelayServer" }, { type: "azure-native:fluidrelay:FluidRelayServer" }, { type: "azure-nextgen:fluidrelay:FluidRelayServer" }, { type: "azure-native:fluidrelay/v20210615preview:FluidRelayServer" }, { type: "azure-nextgen:fluidrelay/v20210615preview:FluidRelayServer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FluidRelayServer.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing FluidRelayServer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FluidRelayServer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FluidRelayServer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FluidRelayServer.__pulumiType;
    }
}
exports.FluidRelayServer = FluidRelayServer;
/** @internal */
FluidRelayServer.__pulumiType = 'azure-native:fluidrelay/v20210312preview:FluidRelayServer';
