// @pulumi/azure-native/storsimple/v20161001/fileShare.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileShare = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The File Share.
 */
class FileShare extends pulumi.CustomResource {
    /**
     * Create a FileShare resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.adminUser === undefined) && !opts.urn) {
                throw new Error("Missing required property 'adminUser'");
            }
            if ((!args || args.dataPolicy === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataPolicy'");
            }
            if ((!args || args.deviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deviceName'");
            }
            if ((!args || args.fileServerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fileServerName'");
            }
            if ((!args || args.managerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managerName'");
            }
            if ((!args || args.monitoringStatus === undefined) && !opts.urn) {
                throw new Error("Missing required property 'monitoringStatus'");
            }
            if ((!args || args.provisionedCapacityInBytes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'provisionedCapacityInBytes'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.shareStatus === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shareStatus'");
            }
            inputs["adminUser"] = args ? args.adminUser : undefined;
            inputs["dataPolicy"] = args ? args.dataPolicy : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["deviceName"] = args ? args.deviceName : undefined;
            inputs["fileServerName"] = args ? args.fileServerName : undefined;
            inputs["managerName"] = args ? args.managerName : undefined;
            inputs["monitoringStatus"] = args ? args.monitoringStatus : undefined;
            inputs["provisionedCapacityInBytes"] = args ? args.provisionedCapacityInBytes : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["shareName"] = args ? args.shareName : undefined;
            inputs["shareStatus"] = args ? args.shareStatus : undefined;
            inputs["localUsedCapacityInBytes"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["usedCapacityInBytes"] = undefined /*out*/;
        }
        else {
            inputs["adminUser"] = undefined /*out*/;
            inputs["dataPolicy"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["localUsedCapacityInBytes"] = undefined /*out*/;
            inputs["monitoringStatus"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisionedCapacityInBytes"] = undefined /*out*/;
            inputs["shareStatus"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["usedCapacityInBytes"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:storsimple/v20161001:FileShare" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FileShare.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing FileShare resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FileShare(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FileShare.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FileShare.__pulumiType;
    }
}
exports.FileShare = FileShare;
/** @internal */
FileShare.__pulumiType = 'azure-native:storsimple/v20161001:FileShare';
