// @pulumi/azure-native/kubernetes/v20210301/connectedCluster.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectedCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents a connected cluster.
 */
class ConnectedCluster extends pulumi.CustomResource {
    /**
     * Create a ConnectedCluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.agentPublicKeyCertificate === undefined) && !opts.urn) {
                throw new Error("Missing required property 'agentPublicKeyCertificate'");
            }
            if ((!args || args.identity === undefined) && !opts.urn) {
                throw new Error("Missing required property 'identity'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["agentPublicKeyCertificate"] = args ? args.agentPublicKeyCertificate : undefined;
            inputs["clusterName"] = args ? args.clusterName : undefined;
            inputs["distribution"] = args ? args.distribution : undefined;
            inputs["identity"] = args ? args.identity : undefined;
            inputs["infrastructure"] = args ? args.infrastructure : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["provisioningState"] = args ? args.provisioningState : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["agentVersion"] = undefined /*out*/;
            inputs["connectivityStatus"] = undefined /*out*/;
            inputs["kubernetesVersion"] = undefined /*out*/;
            inputs["lastConnectivityTime"] = undefined /*out*/;
            inputs["managedIdentityCertificateExpirationTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["offering"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["totalCoreCount"] = undefined /*out*/;
            inputs["totalNodeCount"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["agentPublicKeyCertificate"] = undefined /*out*/;
            inputs["agentVersion"] = undefined /*out*/;
            inputs["connectivityStatus"] = undefined /*out*/;
            inputs["distribution"] = undefined /*out*/;
            inputs["identity"] = undefined /*out*/;
            inputs["infrastructure"] = undefined /*out*/;
            inputs["kubernetesVersion"] = undefined /*out*/;
            inputs["lastConnectivityTime"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["managedIdentityCertificateExpirationTime"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["offering"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["totalCoreCount"] = undefined /*out*/;
            inputs["totalNodeCount"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:kubernetes/v20210301:ConnectedCluster" }, { type: "azure-native:kubernetes:ConnectedCluster" }, { type: "azure-nextgen:kubernetes:ConnectedCluster" }, { type: "azure-native:kubernetes/v20200101preview:ConnectedCluster" }, { type: "azure-nextgen:kubernetes/v20200101preview:ConnectedCluster" }, { type: "azure-native:kubernetes/v20210401preview:ConnectedCluster" }, { type: "azure-nextgen:kubernetes/v20210401preview:ConnectedCluster" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ConnectedCluster.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ConnectedCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ConnectedCluster(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ConnectedCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConnectedCluster.__pulumiType;
    }
}
exports.ConnectedCluster = ConnectedCluster;
/** @internal */
ConnectedCluster.__pulumiType = 'azure-native:kubernetes/v20210301:ConnectedCluster';
