// @pulumi/azure-native/notificationhubs/v20170401/notificationHub.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotificationHub = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Description of a NotificationHub Resource.
 */
class NotificationHub extends pulumi.CustomResource {
    /**
     * Create a NotificationHub resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.namespaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'namespaceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["admCredential"] = args ? args.admCredential : undefined;
            inputs["apnsCredential"] = args ? args.apnsCredential : undefined;
            inputs["authorizationRules"] = args ? args.authorizationRules : undefined;
            inputs["baiduCredential"] = args ? args.baiduCredential : undefined;
            inputs["gcmCredential"] = args ? args.gcmCredential : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["mpnsCredential"] = args ? args.mpnsCredential : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["namespaceName"] = args ? args.namespaceName : undefined;
            inputs["notificationHubName"] = args ? args.notificationHubName : undefined;
            inputs["registrationTtl"] = args ? args.registrationTtl : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["wnsCredential"] = args ? args.wnsCredential : undefined;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["admCredential"] = undefined /*out*/;
            inputs["apnsCredential"] = undefined /*out*/;
            inputs["authorizationRules"] = undefined /*out*/;
            inputs["baiduCredential"] = undefined /*out*/;
            inputs["gcmCredential"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["mpnsCredential"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["registrationTtl"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["wnsCredential"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:notificationhubs/v20170401:NotificationHub" }, { type: "azure-native:notificationhubs:NotificationHub" }, { type: "azure-nextgen:notificationhubs:NotificationHub" }, { type: "azure-native:notificationhubs/v20140901:NotificationHub" }, { type: "azure-nextgen:notificationhubs/v20140901:NotificationHub" }, { type: "azure-native:notificationhubs/v20160301:NotificationHub" }, { type: "azure-nextgen:notificationhubs/v20160301:NotificationHub" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NotificationHub.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing NotificationHub resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NotificationHub(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NotificationHub.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NotificationHub.__pulumiType;
    }
}
exports.NotificationHub = NotificationHub;
/** @internal */
NotificationHub.__pulumiType = 'azure-native:notificationhubs/v20170401:NotificationHub';
