// @pulumi/azure-native/network/v20181101/routeTable.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RouteTable = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Route table resource.
 */
class RouteTable extends pulumi.CustomResource {
    /**
     * Create a RouteTable resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["disableBgpRoutePropagation"] = args ? args.disableBgpRoutePropagation : undefined;
            inputs["etag"] = args ? args.etag : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["provisioningState"] = args ? args.provisioningState : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["routeTableName"] = args ? args.routeTableName : undefined;
            inputs["routes"] = args ? args.routes : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["subnets"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["disableBgpRoutePropagation"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["routes"] = undefined /*out*/;
            inputs["subnets"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20181101:RouteTable" }, { type: "azure-native:network:RouteTable" }, { type: "azure-nextgen:network:RouteTable" }, { type: "azure-native:network/v20150501preview:RouteTable" }, { type: "azure-nextgen:network/v20150501preview:RouteTable" }, { type: "azure-native:network/v20150615:RouteTable" }, { type: "azure-nextgen:network/v20150615:RouteTable" }, { type: "azure-native:network/v20160330:RouteTable" }, { type: "azure-nextgen:network/v20160330:RouteTable" }, { type: "azure-native:network/v20160601:RouteTable" }, { type: "azure-nextgen:network/v20160601:RouteTable" }, { type: "azure-native:network/v20160901:RouteTable" }, { type: "azure-nextgen:network/v20160901:RouteTable" }, { type: "azure-native:network/v20161201:RouteTable" }, { type: "azure-nextgen:network/v20161201:RouteTable" }, { type: "azure-native:network/v20170301:RouteTable" }, { type: "azure-nextgen:network/v20170301:RouteTable" }, { type: "azure-native:network/v20170601:RouteTable" }, { type: "azure-nextgen:network/v20170601:RouteTable" }, { type: "azure-native:network/v20170801:RouteTable" }, { type: "azure-nextgen:network/v20170801:RouteTable" }, { type: "azure-native:network/v20170901:RouteTable" }, { type: "azure-nextgen:network/v20170901:RouteTable" }, { type: "azure-native:network/v20171001:RouteTable" }, { type: "azure-nextgen:network/v20171001:RouteTable" }, { type: "azure-native:network/v20171101:RouteTable" }, { type: "azure-nextgen:network/v20171101:RouteTable" }, { type: "azure-native:network/v20180101:RouteTable" }, { type: "azure-nextgen:network/v20180101:RouteTable" }, { type: "azure-native:network/v20180201:RouteTable" }, { type: "azure-nextgen:network/v20180201:RouteTable" }, { type: "azure-native:network/v20180401:RouteTable" }, { type: "azure-nextgen:network/v20180401:RouteTable" }, { type: "azure-native:network/v20180601:RouteTable" }, { type: "azure-nextgen:network/v20180601:RouteTable" }, { type: "azure-native:network/v20180701:RouteTable" }, { type: "azure-nextgen:network/v20180701:RouteTable" }, { type: "azure-native:network/v20180801:RouteTable" }, { type: "azure-nextgen:network/v20180801:RouteTable" }, { type: "azure-native:network/v20181001:RouteTable" }, { type: "azure-nextgen:network/v20181001:RouteTable" }, { type: "azure-native:network/v20181201:RouteTable" }, { type: "azure-nextgen:network/v20181201:RouteTable" }, { type: "azure-native:network/v20190201:RouteTable" }, { type: "azure-nextgen:network/v20190201:RouteTable" }, { type: "azure-native:network/v20190401:RouteTable" }, { type: "azure-nextgen:network/v20190401:RouteTable" }, { type: "azure-native:network/v20190601:RouteTable" }, { type: "azure-nextgen:network/v20190601:RouteTable" }, { type: "azure-native:network/v20190701:RouteTable" }, { type: "azure-nextgen:network/v20190701:RouteTable" }, { type: "azure-native:network/v20190801:RouteTable" }, { type: "azure-nextgen:network/v20190801:RouteTable" }, { type: "azure-native:network/v20190901:RouteTable" }, { type: "azure-nextgen:network/v20190901:RouteTable" }, { type: "azure-native:network/v20191101:RouteTable" }, { type: "azure-nextgen:network/v20191101:RouteTable" }, { type: "azure-native:network/v20191201:RouteTable" }, { type: "azure-nextgen:network/v20191201:RouteTable" }, { type: "azure-native:network/v20200301:RouteTable" }, { type: "azure-nextgen:network/v20200301:RouteTable" }, { type: "azure-native:network/v20200401:RouteTable" }, { type: "azure-nextgen:network/v20200401:RouteTable" }, { type: "azure-native:network/v20200501:RouteTable" }, { type: "azure-nextgen:network/v20200501:RouteTable" }, { type: "azure-native:network/v20200601:RouteTable" }, { type: "azure-nextgen:network/v20200601:RouteTable" }, { type: "azure-native:network/v20200701:RouteTable" }, { type: "azure-nextgen:network/v20200701:RouteTable" }, { type: "azure-native:network/v20200801:RouteTable" }, { type: "azure-nextgen:network/v20200801:RouteTable" }, { type: "azure-native:network/v20201101:RouteTable" }, { type: "azure-nextgen:network/v20201101:RouteTable" }, { type: "azure-native:network/v20210201:RouteTable" }, { type: "azure-nextgen:network/v20210201:RouteTable" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RouteTable.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing RouteTable resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RouteTable(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RouteTable.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RouteTable.__pulumiType;
    }
}
exports.RouteTable = RouteTable;
/** @internal */
RouteTable.__pulumiType = 'azure-native:network/v20181101:RouteTable';
