// @pulumi/azure-native/servicebus/namespaceVirtualNetworkRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NamespaceVirtualNetworkRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Single item in a List or Get VirtualNetworkRules operation
 * API Version: 2018-01-01-preview.
 */
class NamespaceVirtualNetworkRule extends pulumi.CustomResource {
    /**
     * Create a NamespaceVirtualNetworkRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.namespaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'namespaceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["namespaceName"] = args ? args.namespaceName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["virtualNetworkRuleName"] = args ? args.virtualNetworkRuleName : undefined;
            inputs["virtualNetworkSubnetId"] = args ? args.virtualNetworkSubnetId : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["virtualNetworkSubnetId"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:servicebus:NamespaceVirtualNetworkRule" }, { type: "azure-native:servicebus/v20180101preview:NamespaceVirtualNetworkRule" }, { type: "azure-nextgen:servicebus/v20180101preview:NamespaceVirtualNetworkRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NamespaceVirtualNetworkRule.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing NamespaceVirtualNetworkRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NamespaceVirtualNetworkRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NamespaceVirtualNetworkRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NamespaceVirtualNetworkRule.__pulumiType;
    }
}
exports.NamespaceVirtualNetworkRule = NamespaceVirtualNetworkRule;
/** @internal */
NamespaceVirtualNetworkRule.__pulumiType = 'azure-native:servicebus:NamespaceVirtualNetworkRule';
