// @pulumi/azure-native/securityandcompliance/privateEndpointConnectionsAdtAPI.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointConnectionsAdtAPI = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The Private Endpoint Connection resource.
 * API Version: 2021-03-08.
 */
class PrivateEndpointConnectionsAdtAPI extends pulumi.CustomResource {
    /**
     * Create a PrivateEndpointConnectionsAdtAPI resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.privateLinkServiceConnectionState === undefined) && !opts.urn) {
                throw new Error("Missing required property 'privateLinkServiceConnectionState'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            inputs["privateEndpointConnectionName"] = args ? args.privateEndpointConnectionName : undefined;
            inputs["privateLinkServiceConnectionState"] = args ? args.privateLinkServiceConnectionState : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["resourceName"] = args ? args.resourceName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["privateEndpoint"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["name"] = undefined /*out*/;
            inputs["privateEndpoint"] = undefined /*out*/;
            inputs["privateLinkServiceConnectionState"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:securityandcompliance:PrivateEndpointConnectionsAdtAPI" }, { type: "azure-native:securityandcompliance/v20210111:PrivateEndpointConnectionsAdtAPI" }, { type: "azure-nextgen:securityandcompliance/v20210111:PrivateEndpointConnectionsAdtAPI" }, { type: "azure-native:securityandcompliance/v20210308:PrivateEndpointConnectionsAdtAPI" }, { type: "azure-nextgen:securityandcompliance/v20210308:PrivateEndpointConnectionsAdtAPI" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateEndpointConnectionsAdtAPI.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing PrivateEndpointConnectionsAdtAPI resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new PrivateEndpointConnectionsAdtAPI(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivateEndpointConnectionsAdtAPI.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateEndpointConnectionsAdtAPI.__pulumiType;
    }
}
exports.PrivateEndpointConnectionsAdtAPI = PrivateEndpointConnectionsAdtAPI;
/** @internal */
PrivateEndpointConnectionsAdtAPI.__pulumiType = 'azure-native:securityandcompliance:PrivateEndpointConnectionsAdtAPI';
