// @pulumi/azure-native/types/enums/dbformariadb/v20180601/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageAutogrow = exports.SslEnforcementEnum = exports.SkuTier = exports.ServerVersion = exports.PublicNetworkAccessEnum = exports.MinimalTlsVersionEnum = exports.GeoRedundantBackup = exports.CreateMode = void 0;
exports.CreateMode = {
    Default: "Default",
    PointInTimeRestore: "PointInTimeRestore",
    GeoRestore: "GeoRestore",
    Replica: "Replica",
};
exports.GeoRedundantBackup = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.MinimalTlsVersionEnum = {
    TLS1_0: "TLS1_0",
    TLS1_1: "TLS1_1",
    TLS1_2: "TLS1_2",
    TLSEnforcementDisabled: "TLSEnforcementDisabled",
};
exports.PublicNetworkAccessEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ServerVersion = {
    ServerVersion_10_2: "10.2",
    ServerVersion_10_3: "10.3",
};
exports.SkuTier = {
    Basic: "Basic",
    GeneralPurpose: "GeneralPurpose",
    MemoryOptimized: "MemoryOptimized",
};
exports.SslEnforcementEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.StorageAutogrow = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
