// @pulumi/azure-native/types/enums/certificateregistration/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificateProductType = exports.v20210201 = exports.v20210115 = exports.v20210101 = exports.v20201201 = exports.v20201001 = exports.v20200901 = exports.v20200601 = exports.v20190801 = exports.v20180201 = exports.v20150801 = void 0;
// Export sub-modules:
const v20150801 = require("./v20150801");
exports.v20150801 = v20150801;
const v20180201 = require("./v20180201");
exports.v20180201 = v20180201;
const v20190801 = require("./v20190801");
exports.v20190801 = v20190801;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20210115 = require("./v20210115");
exports.v20210115 = v20210115;
const v20210201 = require("./v20210201");
exports.v20210201 = v20210201;
exports.CertificateProductType = {
    StandardDomainValidatedSsl: "StandardDomainValidatedSsl",
    StandardDomainValidatedWildCardSsl: "StandardDomainValidatedWildCardSsl",
};
