// @pulumi/azure-native/types/enums/datamigration/v20210630/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SsisStoreType = exports.SsisMigrationOverwriteOption = exports.SqlSourcePlatform = exports.ServerLevelPermissionsGroup = exports.ProjectTargetPlatform = exports.ProjectSourcePlatform = exports.MySqlTargetPlatformType = exports.MongoDbShardKeyOrder = exports.MongoDbReplication = exports.BackupMode = exports.AuthenticationType = void 0;
exports.AuthenticationType = {
    None: "None",
    WindowsAuthentication: "WindowsAuthentication",
    SqlAuthentication: "SqlAuthentication",
    ActiveDirectoryIntegrated: "ActiveDirectoryIntegrated",
    ActiveDirectoryPassword: "ActiveDirectoryPassword",
};
exports.BackupMode = {
    CreateBackup: "CreateBackup",
    ExistingBackup: "ExistingBackup",
};
exports.MongoDbReplication = {
    Disabled: "Disabled",
    OneTime: "OneTime",
    Continuous: "Continuous",
};
exports.MongoDbShardKeyOrder = {
    Forward: "Forward",
    Reverse: "Reverse",
    Hashed: "Hashed",
};
exports.MySqlTargetPlatformType = {
    SqlServer: "SqlServer",
    AzureDbForMySQL: "AzureDbForMySQL",
};
exports.ProjectSourcePlatform = {
    SQL: "SQL",
    MySQL: "MySQL",
    PostgreSql: "PostgreSql",
    MongoDb: "MongoDb",
    Unknown: "Unknown",
};
exports.ProjectTargetPlatform = {
    SQLDB: "SQLDB",
    SQLMI: "SQLMI",
    AzureDbForMySql: "AzureDbForMySql",
    AzureDbForPostgreSql: "AzureDbForPostgreSql",
    MongoDb: "MongoDb",
    Unknown: "Unknown",
};
exports.ServerLevelPermissionsGroup = {
    Default: "Default",
    MigrationFromSqlServerToAzureDB: "MigrationFromSqlServerToAzureDB",
    MigrationFromSqlServerToAzureMI: "MigrationFromSqlServerToAzureMI",
    MigrationFromMySQLToAzureDBForMySQL: "MigrationFromMySQLToAzureDBForMySQL",
};
exports.SqlSourcePlatform = {
    SqlOnPrem: "SqlOnPrem",
};
exports.SsisMigrationOverwriteOption = {
    Ignore: "Ignore",
    Overwrite: "Overwrite",
};
exports.SsisStoreType = {
    SsisCatalog: "SsisCatalog",
};
