// @pulumi/azure-native/servicefabric/v20191101preview/service.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Service = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The service resource.
 */
class Service extends pulumi.CustomResource {
    /**
     * Create a Service resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.applicationName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'applicationName'");
            }
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceKind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceKind'");
            }
            inputs["applicationName"] = args ? args.applicationName : undefined;
            inputs["clusterName"] = args ? args.clusterName : undefined;
            inputs["correlationScheme"] = args ? args.correlationScheme : undefined;
            inputs["defaultMoveCost"] = args ? args.defaultMoveCost : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["partitionDescription"] = args ? args.partitionDescription : undefined;
            inputs["placementConstraints"] = args ? args.placementConstraints : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serviceDnsName"] = args ? args.serviceDnsName : undefined;
            inputs["serviceKind"] = args ? args.serviceKind : undefined;
            inputs["serviceLoadMetrics"] = args ? args.serviceLoadMetrics : undefined;
            inputs["serviceName"] = args ? args.serviceName : undefined;
            inputs["servicePackageActivationMode"] = args ? args.servicePackageActivationMode : undefined;
            inputs["servicePlacementPolicies"] = args ? args.servicePlacementPolicies : undefined;
            inputs["serviceTypeName"] = args ? args.serviceTypeName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["correlationScheme"] = undefined /*out*/;
            inputs["defaultMoveCost"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["partitionDescription"] = undefined /*out*/;
            inputs["placementConstraints"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["serviceDnsName"] = undefined /*out*/;
            inputs["serviceKind"] = undefined /*out*/;
            inputs["serviceLoadMetrics"] = undefined /*out*/;
            inputs["servicePackageActivationMode"] = undefined /*out*/;
            inputs["servicePlacementPolicies"] = undefined /*out*/;
            inputs["serviceTypeName"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:servicefabric/v20191101preview:Service" }, { type: "azure-native:servicefabric:Service" }, { type: "azure-nextgen:servicefabric:Service" }, { type: "azure-native:servicefabric/v20170701preview:Service" }, { type: "azure-nextgen:servicefabric/v20170701preview:Service" }, { type: "azure-native:servicefabric/v20190301:Service" }, { type: "azure-nextgen:servicefabric/v20190301:Service" }, { type: "azure-native:servicefabric/v20190301preview:Service" }, { type: "azure-nextgen:servicefabric/v20190301preview:Service" }, { type: "azure-native:servicefabric/v20190601preview:Service" }, { type: "azure-nextgen:servicefabric/v20190601preview:Service" }, { type: "azure-native:servicefabric/v20200301:Service" }, { type: "azure-nextgen:servicefabric/v20200301:Service" }, { type: "azure-native:servicefabric/v20201201preview:Service" }, { type: "azure-nextgen:servicefabric/v20201201preview:Service" }, { type: "azure-native:servicefabric/v20210601:Service" }, { type: "azure-nextgen:servicefabric/v20210601:Service" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Service.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Service resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Service(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Service.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Service.__pulumiType;
    }
}
exports.Service = Service;
/** @internal */
Service.__pulumiType = 'azure-native:servicefabric/v20191101preview:Service';
