// @pulumi/azure-native/apimanagement/v20191201/tagByOperation.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagByOperation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Tag Contract details.
 */
class TagByOperation extends pulumi.CustomResource {
    /**
     * Create a TagByOperation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.apiId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'apiId'");
            }
            if ((!args || args.operationId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'operationId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            inputs["apiId"] = args ? args.apiId : undefined;
            inputs["operationId"] = args ? args.operationId : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serviceName"] = args ? args.serviceName : undefined;
            inputs["tagId"] = args ? args.tagId : undefined;
            inputs["displayName"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["displayName"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:apimanagement/v20191201:TagByOperation" }, { type: "azure-native:apimanagement:TagByOperation" }, { type: "azure-nextgen:apimanagement:TagByOperation" }, { type: "azure-native:apimanagement/v20170301:TagByOperation" }, { type: "azure-nextgen:apimanagement/v20170301:TagByOperation" }, { type: "azure-native:apimanagement/v20180101:TagByOperation" }, { type: "azure-nextgen:apimanagement/v20180101:TagByOperation" }, { type: "azure-native:apimanagement/v20180601preview:TagByOperation" }, { type: "azure-nextgen:apimanagement/v20180601preview:TagByOperation" }, { type: "azure-native:apimanagement/v20190101:TagByOperation" }, { type: "azure-nextgen:apimanagement/v20190101:TagByOperation" }, { type: "azure-native:apimanagement/v20191201preview:TagByOperation" }, { type: "azure-nextgen:apimanagement/v20191201preview:TagByOperation" }, { type: "azure-native:apimanagement/v20200601preview:TagByOperation" }, { type: "azure-nextgen:apimanagement/v20200601preview:TagByOperation" }, { type: "azure-native:apimanagement/v20201201:TagByOperation" }, { type: "azure-nextgen:apimanagement/v20201201:TagByOperation" }, { type: "azure-native:apimanagement/v20210101preview:TagByOperation" }, { type: "azure-nextgen:apimanagement/v20210101preview:TagByOperation" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(TagByOperation.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing TagByOperation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TagByOperation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TagByOperation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TagByOperation.__pulumiType;
    }
}
exports.TagByOperation = TagByOperation;
/** @internal */
TagByOperation.__pulumiType = 'azure-native:apimanagement/v20191201:TagByOperation';
