// @pulumi/azure-native/types/enums/devtestlab/v20160515/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WindowsOsState = exports.VirtualMachineCreationSource = exports.UsagePermissionType = exports.TransportProtocol = exports.StorageType = exports.SourceControlType = exports.PremiumDataDisk = exports.PolicyStatus = exports.PolicyFactName = exports.PolicyEvaluatorType = exports.NotificationStatus = exports.NotificationChannelEventType = exports.LinuxOsState = exports.EnableStatus = exports.CustomImageOsType = void 0;
exports.CustomImageOsType = {
    Windows: "Windows",
    Linux: "Linux",
    None: "None",
};
exports.EnableStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.LinuxOsState = {
    NonDeprovisioned: "NonDeprovisioned",
    DeprovisionRequested: "DeprovisionRequested",
    DeprovisionApplied: "DeprovisionApplied",
};
exports.NotificationChannelEventType = {
    AutoShutdown: "AutoShutdown",
    Cost: "Cost",
};
exports.NotificationStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.PolicyEvaluatorType = {
    AllowedValuesPolicy: "AllowedValuesPolicy",
    MaxValuePolicy: "MaxValuePolicy",
};
exports.PolicyFactName = {
    UserOwnedLabVmCount: "UserOwnedLabVmCount",
    UserOwnedLabPremiumVmCount: "UserOwnedLabPremiumVmCount",
    LabVmCount: "LabVmCount",
    LabPremiumVmCount: "LabPremiumVmCount",
    LabVmSize: "LabVmSize",
    GalleryImage: "GalleryImage",
    UserOwnedLabVmCountInSubnet: "UserOwnedLabVmCountInSubnet",
    LabTargetCost: "LabTargetCost",
};
exports.PolicyStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PremiumDataDisk = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.SourceControlType = {
    VsoGit: "VsoGit",
    GitHub: "GitHub",
};
exports.StorageType = {
    Standard: "Standard",
    Premium: "Premium",
};
exports.TransportProtocol = {
    Tcp: "Tcp",
    Udp: "Udp",
};
exports.UsagePermissionType = {
    Default: "Default",
    Deny: "Deny",
    Allow: "Allow",
};
exports.VirtualMachineCreationSource = {
    FromCustomImage: "FromCustomImage",
    FromGalleryImage: "FromGalleryImage",
};
exports.WindowsOsState = {
    NonSysprepped: "NonSysprepped",
    SysprepRequested: "SysprepRequested",
    SysprepApplied: "SysprepApplied",
};
