// @pulumi/azure-native/insights/v20180301/actionGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An action group resource.
 */
class ActionGroup extends pulumi.CustomResource {
    /**
     * Create a ActionGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.enabled === undefined) && !opts.urn) {
                throw new Error("Missing required property 'enabled'");
            }
            if ((!args || args.groupShortName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'groupShortName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["actionGroupName"] = args ? args.actionGroupName : undefined;
            inputs["automationRunbookReceivers"] = args ? args.automationRunbookReceivers : undefined;
            inputs["azureAppPushReceivers"] = args ? args.azureAppPushReceivers : undefined;
            inputs["azureFunctionReceivers"] = args ? args.azureFunctionReceivers : undefined;
            inputs["emailReceivers"] = args ? args.emailReceivers : undefined;
            inputs["enabled"] = (_a = (args ? args.enabled : undefined)) !== null && _a !== void 0 ? _a : true;
            inputs["groupShortName"] = args ? args.groupShortName : undefined;
            inputs["itsmReceivers"] = args ? args.itsmReceivers : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["logicAppReceivers"] = args ? args.logicAppReceivers : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["smsReceivers"] = args ? args.smsReceivers : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["voiceReceivers"] = args ? args.voiceReceivers : undefined;
            inputs["webhookReceivers"] = args ? args.webhookReceivers : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["automationRunbookReceivers"] = undefined /*out*/;
            inputs["azureAppPushReceivers"] = undefined /*out*/;
            inputs["azureFunctionReceivers"] = undefined /*out*/;
            inputs["emailReceivers"] = undefined /*out*/;
            inputs["enabled"] = undefined /*out*/;
            inputs["groupShortName"] = undefined /*out*/;
            inputs["itsmReceivers"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["logicAppReceivers"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["smsReceivers"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["voiceReceivers"] = undefined /*out*/;
            inputs["webhookReceivers"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:insights/v20180301:ActionGroup" }, { type: "azure-native:insights:ActionGroup" }, { type: "azure-nextgen:insights:ActionGroup" }, { type: "azure-native:insights/v20170401:ActionGroup" }, { type: "azure-nextgen:insights/v20170401:ActionGroup" }, { type: "azure-native:insights/v20180901:ActionGroup" }, { type: "azure-nextgen:insights/v20180901:ActionGroup" }, { type: "azure-native:insights/v20190301:ActionGroup" }, { type: "azure-nextgen:insights/v20190301:ActionGroup" }, { type: "azure-native:insights/v20190601:ActionGroup" }, { type: "azure-nextgen:insights/v20190601:ActionGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ActionGroup.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ActionGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ActionGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ActionGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ActionGroup.__pulumiType;
    }
}
exports.ActionGroup = ActionGroup;
/** @internal */
ActionGroup.__pulumiType = 'azure-native:insights/v20180301:ActionGroup';
