// @pulumi/azure-native/datamigration/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210630 = exports.v20180715preview = exports.v20180419 = exports.v20180331preview = exports.v20180315preview = exports.v20171115preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./file"), exports);
__exportStar(require("./getFile"), exports);
__exportStar(require("./getProject"), exports);
__exportStar(require("./getService"), exports);
__exportStar(require("./getTask"), exports);
__exportStar(require("./project"), exports);
__exportStar(require("./service"), exports);
__exportStar(require("./task"), exports);
// Export enums:
__exportStar(require("../types/enums/datamigration"), exports);
// Export sub-modules:
const v20171115preview = require("./v20171115preview");
exports.v20171115preview = v20171115preview;
const v20180315preview = require("./v20180315preview");
exports.v20180315preview = v20180315preview;
const v20180331preview = require("./v20180331preview");
exports.v20180331preview = v20180331preview;
const v20180419 = require("./v20180419");
exports.v20180419 = v20180419;
const v20180715preview = require("./v20180715preview");
exports.v20180715preview = v20180715preview;
const v20210630 = require("./v20210630");
exports.v20210630 = v20210630;
// Import resources to register:
const file_1 = require("./file");
const project_1 = require("./project");
const service_1 = require("./service");
const task_1 = require("./task");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:datamigration:File":
                return new file_1.File(name, undefined, { urn });
            case "azure-native:datamigration:Project":
                return new project_1.Project(name, undefined, { urn });
            case "azure-native:datamigration:Service":
                return new service_1.Service(name, undefined, { urn });
            case "azure-native:datamigration:Task":
                return new task_1.Task(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "datamigration", _module);
