// @pulumi/azure-native/iotsecurity/v20210201preview/onPremiseSensor.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OnPremiseSensor = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * On-premise IoT sensor
 */
class OnPremiseSensor extends pulumi.CustomResource {
    /**
     * Create a OnPremiseSensor resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            inputs["onPremiseSensorName"] = args ? args.onPremiseSensorName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:iotsecurity/v20210201preview:OnPremiseSensor" }, { type: "azure-native:iotsecurity:OnPremiseSensor" }, { type: "azure-nextgen:iotsecurity:OnPremiseSensor" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(OnPremiseSensor.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing OnPremiseSensor resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new OnPremiseSensor(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of OnPremiseSensor.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OnPremiseSensor.__pulumiType;
    }
}
exports.OnPremiseSensor = OnPremiseSensor;
/** @internal */
OnPremiseSensor.__pulumiType = 'azure-native:iotsecurity/v20210201preview:OnPremiseSensor';
