// @pulumi/azure-native/keyvault/v20200401preview/managedHsm.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedHsm = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Resource information with extended details.
 */
class ManagedHsm extends pulumi.CustomResource {
    /**
     * Create a ManagedHsm resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["properties"] = args ? args.properties : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:keyvault/v20200401preview:ManagedHsm" }, { type: "azure-native:keyvault:ManagedHsm" }, { type: "azure-nextgen:keyvault:ManagedHsm" }, { type: "azure-native:keyvault/v20210401preview:ManagedHsm" }, { type: "azure-nextgen:keyvault/v20210401preview:ManagedHsm" }, { type: "azure-native:keyvault/v20210601preview:ManagedHsm" }, { type: "azure-nextgen:keyvault/v20210601preview:ManagedHsm" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagedHsm.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ManagedHsm resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ManagedHsm(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagedHsm.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagedHsm.__pulumiType;
    }
}
exports.ManagedHsm = ManagedHsm;
/** @internal */
ManagedHsm.__pulumiType = 'azure-native:keyvault/v20200401preview:ManagedHsm';
