// @pulumi/azure-native/devspaces/controller.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Controller = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * API Version: 2019-04-01.
 */
class Controller extends pulumi.CustomResource {
    /**
     * Create a Controller resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            if ((!args || args.targetContainerHostCredentialsBase64 === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetContainerHostCredentialsBase64'");
            }
            if ((!args || args.targetContainerHostResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetContainerHostResourceId'");
            }
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["targetContainerHostCredentialsBase64"] = args ? args.targetContainerHostCredentialsBase64 : undefined;
            inputs["targetContainerHostResourceId"] = args ? args.targetContainerHostResourceId : undefined;
            inputs["dataPlaneFqdn"] = undefined /*out*/;
            inputs["hostSuffix"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["targetContainerHostApiServerFqdn"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["dataPlaneFqdn"] = undefined /*out*/;
            inputs["hostSuffix"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["targetContainerHostApiServerFqdn"] = undefined /*out*/;
            inputs["targetContainerHostResourceId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:devspaces:Controller" }, { type: "azure-native:devspaces/v20190401:Controller" }, { type: "azure-nextgen:devspaces/v20190401:Controller" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Controller.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Controller resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Controller(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Controller.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Controller.__pulumiType;
    }
}
exports.Controller = Controller;
/** @internal */
Controller.__pulumiType = 'azure-native:devspaces:Controller';
