// @pulumi/azure-native/machinelearningcompute/v20170801preview/operationalizationCluster.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationalizationCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Instance of an Azure ML Operationalization Cluster resource.
 */
class OperationalizationCluster extends pulumi.CustomResource {
    /**
     * Create a OperationalizationCluster resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterType'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["appInsights"] = args ? args.appInsights : undefined;
            inputs["clusterName"] = args ? args.clusterName : undefined;
            inputs["clusterType"] = args ? args.clusterType : undefined;
            inputs["containerRegistry"] = args ? args.containerRegistry : undefined;
            inputs["containerService"] = args ? args.containerService : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["globalServiceConfiguration"] = args ? args.globalServiceConfiguration : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["storageAccount"] = args ? args.storageAccount : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["createdOn"] = undefined /*out*/;
            inputs["modifiedOn"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningErrors"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["appInsights"] = undefined /*out*/;
            inputs["clusterType"] = undefined /*out*/;
            inputs["containerRegistry"] = undefined /*out*/;
            inputs["containerService"] = undefined /*out*/;
            inputs["createdOn"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["globalServiceConfiguration"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["modifiedOn"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningErrors"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["storageAccount"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:machinelearningcompute/v20170801preview:OperationalizationCluster" }, { type: "azure-native:machinelearningcompute:OperationalizationCluster" }, { type: "azure-nextgen:machinelearningcompute:OperationalizationCluster" }, { type: "azure-native:machinelearningcompute/v20170601preview:OperationalizationCluster" }, { type: "azure-nextgen:machinelearningcompute/v20170601preview:OperationalizationCluster" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(OperationalizationCluster.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing OperationalizationCluster resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new OperationalizationCluster(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of OperationalizationCluster.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OperationalizationCluster.__pulumiType;
    }
}
exports.OperationalizationCluster = OperationalizationCluster;
/** @internal */
OperationalizationCluster.__pulumiType = 'azure-native:machinelearningcompute/v20170801preview:OperationalizationCluster';
