// @pulumi/azure-native/storsimple/v20170601/storageAccountCredential.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageAccountCredential = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The storage account credential.
 */
class StorageAccountCredential extends pulumi.CustomResource {
    /**
     * Create a StorageAccountCredential resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.endPoint === undefined) && !opts.urn) {
                throw new Error("Missing required property 'endPoint'");
            }
            if ((!args || args.managerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sslStatus === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sslStatus'");
            }
            inputs["accessKey"] = args ? args.accessKey : undefined;
            inputs["endPoint"] = args ? args.endPoint : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["managerName"] = args ? args.managerName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sslStatus"] = args ? args.sslStatus : undefined;
            inputs["storageAccountCredentialName"] = args ? args.storageAccountCredentialName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["volumesCount"] = undefined /*out*/;
        }
        else {
            inputs["accessKey"] = undefined /*out*/;
            inputs["endPoint"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["sslStatus"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["volumesCount"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:storsimple/v20170601:StorageAccountCredential" }, { type: "azure-native:storsimple:StorageAccountCredential" }, { type: "azure-nextgen:storsimple:StorageAccountCredential" }, { type: "azure-native:storsimple/v20161001:StorageAccountCredential" }, { type: "azure-nextgen:storsimple/v20161001:StorageAccountCredential" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(StorageAccountCredential.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing StorageAccountCredential resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StorageAccountCredential(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StorageAccountCredential.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StorageAccountCredential.__pulumiType;
    }
}
exports.StorageAccountCredential = StorageAccountCredential;
/** @internal */
StorageAccountCredential.__pulumiType = 'azure-native:storsimple/v20170601:StorageAccountCredential';
