// @pulumi/azure-native/types/enums/cognitiveservices/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.NetworkRuleAction = exports.KeySource = exports.IdentityType = exports.v20210430 = exports.v20170418 = exports.v20160201preview = void 0;
// Export sub-modules:
const v20160201preview = require("./v20160201preview");
exports.v20160201preview = v20160201preview;
const v20170418 = require("./v20170418");
exports.v20170418 = v20170418;
const v20210430 = require("./v20210430");
exports.v20210430 = v20210430;
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.KeySource = {
    Microsoft_CognitiveServices: "Microsoft.CognitiveServices",
    Microsoft_KeyVault: "Microsoft.KeyVault",
};
exports.NetworkRuleAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
