// @pulumi/azure-native/types/enums/containerregistry/v20180901/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Variant = exports.TriggerStatus = exports.TokenType = exports.TaskStatus = exports.SourceTriggerEvent = exports.SourceRegistryLoginMode = exports.SourceControlType = exports.SecretObjectType = exports.OS = exports.BaseImageTriggerType = exports.Architecture = void 0;
exports.Architecture = {
    Amd64: "amd64",
    X86: "x86",
    Arm: "arm",
};
exports.BaseImageTriggerType = {
    All: "All",
    Runtime: "Runtime",
};
exports.OS = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.SecretObjectType = {
    Opaque: "Opaque",
};
exports.SourceControlType = {
    Github: "Github",
    VisualStudioTeamService: "VisualStudioTeamService",
};
exports.SourceRegistryLoginMode = {
    None: "None",
    Default: "Default",
};
exports.SourceTriggerEvent = {
    Commit: "commit",
    Pullrequest: "pullrequest",
};
exports.TaskStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.TokenType = {
    PAT: "PAT",
    OAuth: "OAuth",
};
exports.TriggerStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.Variant = {
    V6: "v6",
    V7: "v7",
    V8: "v8",
};
