// @pulumi/azure-native/authorization/v20201001preview/roleManagementPolicyAssignment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoleManagementPolicyAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Role management policy
 */
class RoleManagementPolicyAssignment extends pulumi.CustomResource {
    /**
     * Create a RoleManagementPolicyAssignment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.scope === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scope'");
            }
            inputs["policyId"] = args ? args.policyId : undefined;
            inputs["roleDefinitionId"] = args ? args.roleDefinitionId : undefined;
            inputs["roleManagementPolicyAssignmentName"] = args ? args.roleManagementPolicyAssignmentName : undefined;
            inputs["scope"] = args ? args.scope : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["policyAssignmentProperties"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["name"] = undefined /*out*/;
            inputs["policyAssignmentProperties"] = undefined /*out*/;
            inputs["policyId"] = undefined /*out*/;
            inputs["roleDefinitionId"] = undefined /*out*/;
            inputs["scope"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:authorization/v20201001preview:RoleManagementPolicyAssignment" }, { type: "azure-native:authorization:RoleManagementPolicyAssignment" }, { type: "azure-nextgen:authorization:RoleManagementPolicyAssignment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RoleManagementPolicyAssignment.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing RoleManagementPolicyAssignment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new RoleManagementPolicyAssignment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RoleManagementPolicyAssignment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RoleManagementPolicyAssignment.__pulumiType;
    }
}
exports.RoleManagementPolicyAssignment = RoleManagementPolicyAssignment;
/** @internal */
RoleManagementPolicyAssignment.__pulumiType = 'azure-native:authorization/v20201001preview:RoleManagementPolicyAssignment';
