// @pulumi/azure-native/types/enums/servicebus/v20170401/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.SkuName = exports.NetworkRuleIPAction = exports.FilterType = exports.EntityStatus = exports.DefaultAction = exports.AccessRights = void 0;
exports.AccessRights = {
    Manage: "Manage",
    Send: "Send",
    Listen: "Listen",
};
exports.DefaultAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.EntityStatus = {
    Active: "Active",
    Disabled: "Disabled",
    Restoring: "Restoring",
    SendDisabled: "SendDisabled",
    ReceiveDisabled: "ReceiveDisabled",
    Creating: "Creating",
    Deleting: "Deleting",
    Renaming: "Renaming",
    Unknown: "Unknown",
};
exports.FilterType = {
    SqlFilter: "SqlFilter",
    CorrelationFilter: "CorrelationFilter",
};
exports.NetworkRuleIPAction = {
    Allow: "Allow",
};
exports.SkuName = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.SkuTier = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
