// @pulumi/azure-native/keyvault/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210601preview = exports.v20210401preview = exports.v20200401preview = exports.v20190901 = exports.v20180214preview = exports.v20180214 = exports.v20161001 = exports.v20150601 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./getKey"), exports);
__exportStar(require("./getMHSMPrivateEndpointConnection"), exports);
__exportStar(require("./getManagedHsm"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getSecret"), exports);
__exportStar(require("./getVault"), exports);
__exportStar(require("./key"), exports);
__exportStar(require("./managedHsm"), exports);
__exportStar(require("./mhsmprivateEndpointConnection"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./secret"), exports);
__exportStar(require("./vault"), exports);
// Export enums:
__exportStar(require("../types/enums/keyvault"), exports);
// Export sub-modules:
const v20150601 = require("./v20150601");
exports.v20150601 = v20150601;
const v20161001 = require("./v20161001");
exports.v20161001 = v20161001;
const v20180214 = require("./v20180214");
exports.v20180214 = v20180214;
const v20180214preview = require("./v20180214preview");
exports.v20180214preview = v20180214preview;
const v20190901 = require("./v20190901");
exports.v20190901 = v20190901;
const v20200401preview = require("./v20200401preview");
exports.v20200401preview = v20200401preview;
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
// Import resources to register:
const key_1 = require("./key");
const mhsmprivateEndpointConnection_1 = require("./mhsmprivateEndpointConnection");
const managedHsm_1 = require("./managedHsm");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const secret_1 = require("./secret");
const vault_1 = require("./vault");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:keyvault:Key":
                return new key_1.Key(name, undefined, { urn });
            case "azure-native:keyvault:MHSMPrivateEndpointConnection":
                return new mhsmprivateEndpointConnection_1.MHSMPrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:keyvault:ManagedHsm":
                return new managedHsm_1.ManagedHsm(name, undefined, { urn });
            case "azure-native:keyvault:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:keyvault:Secret":
                return new secret_1.Secret(name, undefined, { urn });
            case "azure-native:keyvault:Vault":
                return new vault_1.Vault(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "keyvault", _module);
