// @pulumi/azure-native/apimanagement/v20180101/policy.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Policy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Policy Contract details.
 */
class Policy extends pulumi.CustomResource {
    /**
     * Create a Policy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.policyContent === undefined) && !opts.urn) {
                throw new Error("Missing required property 'policyContent'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            inputs["contentFormat"] = (_a = (args ? args.contentFormat : undefined)) !== null && _a !== void 0 ? _a : "xml";
            inputs["policyContent"] = args ? args.policyContent : undefined;
            inputs["policyId"] = args ? args.policyId : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serviceName"] = args ? args.serviceName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["contentFormat"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["policyContent"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:apimanagement/v20180101:Policy" }, { type: "azure-native:apimanagement:Policy" }, { type: "azure-nextgen:apimanagement:Policy" }, { type: "azure-native:apimanagement/v20170301:Policy" }, { type: "azure-nextgen:apimanagement/v20170301:Policy" }, { type: "azure-native:apimanagement/v20180601preview:Policy" }, { type: "azure-nextgen:apimanagement/v20180601preview:Policy" }, { type: "azure-native:apimanagement/v20190101:Policy" }, { type: "azure-nextgen:apimanagement/v20190101:Policy" }, { type: "azure-native:apimanagement/v20191201:Policy" }, { type: "azure-nextgen:apimanagement/v20191201:Policy" }, { type: "azure-native:apimanagement/v20191201preview:Policy" }, { type: "azure-nextgen:apimanagement/v20191201preview:Policy" }, { type: "azure-native:apimanagement/v20200601preview:Policy" }, { type: "azure-nextgen:apimanagement/v20200601preview:Policy" }, { type: "azure-native:apimanagement/v20201201:Policy" }, { type: "azure-nextgen:apimanagement/v20201201:Policy" }, { type: "azure-native:apimanagement/v20210101preview:Policy" }, { type: "azure-nextgen:apimanagement/v20210101preview:Policy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Policy.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Policy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Policy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Policy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Policy.__pulumiType;
    }
}
exports.Policy = Policy;
/** @internal */
Policy.__pulumiType = 'azure-native:apimanagement/v20180101:Policy';
