// @pulumi/azure-native/apimanagement/v20180101/tag.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tag = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Tag Contract details.
 */
class Tag extends pulumi.CustomResource {
    /**
     * Create a Tag resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["serviceName"] = args ? args.serviceName : undefined;
            inputs["tagId"] = args ? args.tagId : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["displayName"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:apimanagement/v20180101:Tag" }, { type: "azure-native:apimanagement:Tag" }, { type: "azure-nextgen:apimanagement:Tag" }, { type: "azure-native:apimanagement/v20170301:Tag" }, { type: "azure-nextgen:apimanagement/v20170301:Tag" }, { type: "azure-native:apimanagement/v20180601preview:Tag" }, { type: "azure-nextgen:apimanagement/v20180601preview:Tag" }, { type: "azure-native:apimanagement/v20190101:Tag" }, { type: "azure-nextgen:apimanagement/v20190101:Tag" }, { type: "azure-native:apimanagement/v20191201:Tag" }, { type: "azure-nextgen:apimanagement/v20191201:Tag" }, { type: "azure-native:apimanagement/v20191201preview:Tag" }, { type: "azure-nextgen:apimanagement/v20191201preview:Tag" }, { type: "azure-native:apimanagement/v20200601preview:Tag" }, { type: "azure-nextgen:apimanagement/v20200601preview:Tag" }, { type: "azure-native:apimanagement/v20201201:Tag" }, { type: "azure-nextgen:apimanagement/v20201201:Tag" }, { type: "azure-native:apimanagement/v20210101preview:Tag" }, { type: "azure-nextgen:apimanagement/v20210101preview:Tag" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Tag.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Tag resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Tag(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Tag.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Tag.__pulumiType;
    }
}
exports.Tag = Tag;
/** @internal */
Tag.__pulumiType = 'azure-native:apimanagement/v20180101:Tag';
