// @pulumi/azure-native/documentdb/v20210401preview/sqlResourceSqlRoleAssignment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlResourceSqlRoleAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An Azure Cosmos DB Role Assignment
 */
class SqlResourceSqlRoleAssignment extends pulumi.CustomResource {
    /**
     * Create a SqlResourceSqlRoleAssignment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["principalId"] = args ? args.principalId : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["roleAssignmentId"] = args ? args.roleAssignmentId : undefined;
            inputs["roleDefinitionId"] = args ? args.roleDefinitionId : undefined;
            inputs["scope"] = args ? args.scope : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["name"] = undefined /*out*/;
            inputs["principalId"] = undefined /*out*/;
            inputs["roleDefinitionId"] = undefined /*out*/;
            inputs["scope"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:documentdb/v20210401preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb:SqlResourceSqlRoleAssignment" }, { type: "azure-nextgen:documentdb:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20200601preview:SqlResourceSqlRoleAssignment" }, { type: "azure-nextgen:documentdb/v20200601preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20210301preview:SqlResourceSqlRoleAssignment" }, { type: "azure-nextgen:documentdb/v20210301preview:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20210415:SqlResourceSqlRoleAssignment" }, { type: "azure-nextgen:documentdb/v20210415:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20210515:SqlResourceSqlRoleAssignment" }, { type: "azure-nextgen:documentdb/v20210515:SqlResourceSqlRoleAssignment" }, { type: "azure-native:documentdb/v20210615:SqlResourceSqlRoleAssignment" }, { type: "azure-nextgen:documentdb/v20210615:SqlResourceSqlRoleAssignment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlResourceSqlRoleAssignment.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SqlResourceSqlRoleAssignment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SqlResourceSqlRoleAssignment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlResourceSqlRoleAssignment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlResourceSqlRoleAssignment.__pulumiType;
    }
}
exports.SqlResourceSqlRoleAssignment = SqlResourceSqlRoleAssignment;
/** @internal */
SqlResourceSqlRoleAssignment.__pulumiType = 'azure-native:documentdb/v20210401preview:SqlResourceSqlRoleAssignment';
