// @pulumi/azure-native/avs/v20210601/workloadNetworkSegment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkloadNetworkSegment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * NSX Segment
 */
class WorkloadNetworkSegment extends pulumi.CustomResource {
    /**
     * Create a WorkloadNetworkSegment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.privateCloudName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'privateCloudName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["connectedGateway"] = args ? args.connectedGateway : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["privateCloudName"] = args ? args.privateCloudName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["revision"] = args ? args.revision : undefined;
            inputs["segmentId"] = args ? args.segmentId : undefined;
            inputs["subnet"] = args ? args.subnet : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["portVif"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["connectedGateway"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["portVif"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["revision"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["subnet"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:avs/v20210601:WorkloadNetworkSegment" }, { type: "azure-native:avs:WorkloadNetworkSegment" }, { type: "azure-nextgen:avs:WorkloadNetworkSegment" }, { type: "azure-native:avs/v20200717preview:WorkloadNetworkSegment" }, { type: "azure-nextgen:avs/v20200717preview:WorkloadNetworkSegment" }, { type: "azure-native:avs/v20210101preview:WorkloadNetworkSegment" }, { type: "azure-nextgen:avs/v20210101preview:WorkloadNetworkSegment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WorkloadNetworkSegment.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WorkloadNetworkSegment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WorkloadNetworkSegment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WorkloadNetworkSegment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkloadNetworkSegment.__pulumiType;
    }
}
exports.WorkloadNetworkSegment = WorkloadNetworkSegment;
/** @internal */
WorkloadNetworkSegment.__pulumiType = 'azure-native:avs/v20210601:WorkloadNetworkSegment';
