// @pulumi/azure-native/insights/v20180601preview/guestDiagnosticsSetting.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GuestDiagnosticsSetting = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Virtual machine guest diagnostics settings resource.
 */
class GuestDiagnosticsSetting extends pulumi.CustomResource {
    /**
     * Create a GuestDiagnosticsSetting resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["dataSources"] = args ? args.dataSources : undefined;
            inputs["diagnosticSettingsName"] = args ? args.diagnosticSettingsName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["osType"] = args ? args.osType : undefined;
            inputs["proxySetting"] = args ? args.proxySetting : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["dataSources"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["osType"] = undefined /*out*/;
            inputs["proxySetting"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:insights/v20180601preview:guestDiagnosticsSetting" }, { type: "azure-native:insights:guestDiagnosticsSetting" }, { type: "azure-nextgen:insights:guestDiagnosticsSetting" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GuestDiagnosticsSetting.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing GuestDiagnosticsSetting resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GuestDiagnosticsSetting(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GuestDiagnosticsSetting.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GuestDiagnosticsSetting.__pulumiType;
    }
}
exports.GuestDiagnosticsSetting = GuestDiagnosticsSetting;
/** @internal */
GuestDiagnosticsSetting.__pulumiType = 'azure-native:insights/v20180601preview:guestDiagnosticsSetting';
