// @pulumi/azure-native/security/v20170801preview/securityContact.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityContact = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Contact details for security issues
 */
class SecurityContact extends pulumi.CustomResource {
    /**
     * Create a SecurityContact resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.alertNotifications === undefined) && !opts.urn) {
                throw new Error("Missing required property 'alertNotifications'");
            }
            if ((!args || args.alertsToAdmins === undefined) && !opts.urn) {
                throw new Error("Missing required property 'alertsToAdmins'");
            }
            if ((!args || args.email === undefined) && !opts.urn) {
                throw new Error("Missing required property 'email'");
            }
            inputs["alertNotifications"] = args ? args.alertNotifications : undefined;
            inputs["alertsToAdmins"] = args ? args.alertsToAdmins : undefined;
            inputs["email"] = args ? args.email : undefined;
            inputs["phone"] = args ? args.phone : undefined;
            inputs["securityContactName"] = args ? args.securityContactName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["alertNotifications"] = undefined /*out*/;
            inputs["alertsToAdmins"] = undefined /*out*/;
            inputs["email"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["phone"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:security/v20170801preview:SecurityContact" }, { type: "azure-native:security:SecurityContact" }, { type: "azure-nextgen:security:SecurityContact" }, { type: "azure-native:security/v20200101preview:SecurityContact" }, { type: "azure-nextgen:security/v20200101preview:SecurityContact" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SecurityContact.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SecurityContact resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SecurityContact(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SecurityContact.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SecurityContact.__pulumiType;
    }
}
exports.SecurityContact = SecurityContact;
/** @internal */
SecurityContact.__pulumiType = 'azure-native:security/v20170801preview:SecurityContact';
