// @pulumi/azure-native/storagecache/v20190801preview/storageTarget.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageTarget = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A storage system being cached by a Cache.
 */
class StorageTarget extends pulumi.CustomResource {
    /**
     * Create a StorageTarget resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.cacheName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cacheName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["cacheName"] = args ? args.cacheName : undefined;
            inputs["clfs"] = args ? args.clfs : undefined;
            inputs["junctions"] = args ? args.junctions : undefined;
            inputs["nfs3"] = args ? args.nfs3 : undefined;
            inputs["provisioningState"] = args ? args.provisioningState : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["storageTargetName"] = args ? args.storageTargetName : undefined;
            inputs["targetType"] = args ? args.targetType : undefined;
            inputs["unknown"] = args ? args.unknown : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["clfs"] = undefined /*out*/;
            inputs["junctions"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["nfs3"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["targetType"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["unknown"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:storagecache/v20190801preview:StorageTarget" }, { type: "azure-native:storagecache:StorageTarget" }, { type: "azure-nextgen:storagecache:StorageTarget" }, { type: "azure-native:storagecache/v20191101:StorageTarget" }, { type: "azure-nextgen:storagecache/v20191101:StorageTarget" }, { type: "azure-native:storagecache/v20200301:StorageTarget" }, { type: "azure-nextgen:storagecache/v20200301:StorageTarget" }, { type: "azure-native:storagecache/v20201001:StorageTarget" }, { type: "azure-nextgen:storagecache/v20201001:StorageTarget" }, { type: "azure-native:storagecache/v20210301:StorageTarget" }, { type: "azure-nextgen:storagecache/v20210301:StorageTarget" }, { type: "azure-native:storagecache/v20210501:StorageTarget" }, { type: "azure-nextgen:storagecache/v20210501:StorageTarget" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(StorageTarget.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing StorageTarget resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StorageTarget(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StorageTarget.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StorageTarget.__pulumiType;
    }
}
exports.StorageTarget = StorageTarget;
/** @internal */
StorageTarget.__pulumiType = 'azure-native:storagecache/v20190801preview:StorageTarget';
