// @pulumi/azure-native/customerinsights/hub.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Hub = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Hub resource.
 * API Version: 2017-04-26.
 */
class Hub extends pulumi.CustomResource {
    /**
     * Create a Hub resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["hubBillingInfo"] = args ? args.hubBillingInfo : undefined;
            inputs["hubName"] = args ? args.hubName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["tenantFeatures"] = args ? args.tenantFeatures : undefined;
            inputs["apiEndpoint"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["webEndpoint"] = undefined /*out*/;
        }
        else {
            inputs["apiEndpoint"] = undefined /*out*/;
            inputs["hubBillingInfo"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["tenantFeatures"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["webEndpoint"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:customerinsights:Hub" }, { type: "azure-native:customerinsights/v20170101:Hub" }, { type: "azure-nextgen:customerinsights/v20170101:Hub" }, { type: "azure-native:customerinsights/v20170426:Hub" }, { type: "azure-nextgen:customerinsights/v20170426:Hub" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Hub.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Hub resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Hub(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Hub.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Hub.__pulumiType;
    }
}
exports.Hub = Hub;
/** @internal */
Hub.__pulumiType = 'azure-native:customerinsights:Hub';
