// @pulumi/azure-native/web/v20190801/webAppSourceControl.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppSourceControl = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Source control configuration for an app.
 */
class WebAppSourceControl extends pulumi.CustomResource {
    /**
     * Create a WebAppSourceControl resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["branch"] = args ? args.branch : undefined;
            inputs["deploymentRollbackEnabled"] = args ? args.deploymentRollbackEnabled : undefined;
            inputs["isManualIntegration"] = args ? args.isManualIntegration : undefined;
            inputs["isMercurial"] = args ? args.isMercurial : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["repoUrl"] = args ? args.repoUrl : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["branch"] = undefined /*out*/;
            inputs["deploymentRollbackEnabled"] = undefined /*out*/;
            inputs["isManualIntegration"] = undefined /*out*/;
            inputs["isMercurial"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["repoUrl"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20190801:WebAppSourceControl" }, { type: "azure-native:web:WebAppSourceControl" }, { type: "azure-nextgen:web:WebAppSourceControl" }, { type: "azure-native:web/v20150801:WebAppSourceControl" }, { type: "azure-nextgen:web/v20150801:WebAppSourceControl" }, { type: "azure-native:web/v20160801:WebAppSourceControl" }, { type: "azure-nextgen:web/v20160801:WebAppSourceControl" }, { type: "azure-native:web/v20180201:WebAppSourceControl" }, { type: "azure-nextgen:web/v20180201:WebAppSourceControl" }, { type: "azure-native:web/v20181101:WebAppSourceControl" }, { type: "azure-nextgen:web/v20181101:WebAppSourceControl" }, { type: "azure-native:web/v20200601:WebAppSourceControl" }, { type: "azure-nextgen:web/v20200601:WebAppSourceControl" }, { type: "azure-native:web/v20200901:WebAppSourceControl" }, { type: "azure-nextgen:web/v20200901:WebAppSourceControl" }, { type: "azure-native:web/v20201001:WebAppSourceControl" }, { type: "azure-nextgen:web/v20201001:WebAppSourceControl" }, { type: "azure-native:web/v20201201:WebAppSourceControl" }, { type: "azure-nextgen:web/v20201201:WebAppSourceControl" }, { type: "azure-native:web/v20210101:WebAppSourceControl" }, { type: "azure-nextgen:web/v20210101:WebAppSourceControl" }, { type: "azure-native:web/v20210115:WebAppSourceControl" }, { type: "azure-nextgen:web/v20210115:WebAppSourceControl" }, { type: "azure-native:web/v20210201:WebAppSourceControl" }, { type: "azure-nextgen:web/v20210201:WebAppSourceControl" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppSourceControl.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WebAppSourceControl resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppSourceControl(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppSourceControl.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppSourceControl.__pulumiType;
    }
}
exports.WebAppSourceControl = WebAppSourceControl;
/** @internal */
WebAppSourceControl.__pulumiType = 'azure-native:web/v20190801:WebAppSourceControl';
