// @pulumi/azure-native/devices/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210701preview = exports.v20210701 = exports.v20210331 = exports.v20210303preview = exports.v20210201preview = exports.v20200901preview = exports.v20200831preview = exports.v20200831 = exports.v20200801 = exports.v20200710preview = exports.v20200615 = exports.v20200401 = exports.v20200301 = exports.v20200101 = exports.v20191104 = exports.v20190701preview = exports.v20190322preview = exports.v20190322 = exports.v20181201preview = exports.v20180401 = exports.v20180122 = exports.v20171115 = exports.v20170821preview = exports.v20170701 = exports.v20170119 = exports.v20160203 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./certificate"), exports);
__exportStar(require("./dpsCertificate"), exports);
__exportStar(require("./getCertificate"), exports);
__exportStar(require("./getDpsCertificate"), exports);
__exportStar(require("./getIotDpsResource"), exports);
__exportStar(require("./getIotDpsResourcePrivateEndpointConnection"), exports);
__exportStar(require("./getIotHubResource"), exports);
__exportStar(require("./getIotHubResourceEventHubConsumerGroup"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./iotDpsResource"), exports);
__exportStar(require("./iotDpsResourcePrivateEndpointConnection"), exports);
__exportStar(require("./iotHubResource"), exports);
__exportStar(require("./iotHubResourceEventHubConsumerGroup"), exports);
__exportStar(require("./listIotDpsResourceKeys"), exports);
__exportStar(require("./listIotDpsResourceKeysForKeyName"), exports);
__exportStar(require("./listIotHubResourceKeys"), exports);
__exportStar(require("./listIotHubResourceKeysForKeyName"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
// Export enums:
__exportStar(require("../types/enums/devices"), exports);
// Export sub-modules:
const v20160203 = require("./v20160203");
exports.v20160203 = v20160203;
const v20170119 = require("./v20170119");
exports.v20170119 = v20170119;
const v20170701 = require("./v20170701");
exports.v20170701 = v20170701;
const v20170821preview = require("./v20170821preview");
exports.v20170821preview = v20170821preview;
const v20171115 = require("./v20171115");
exports.v20171115 = v20171115;
const v20180122 = require("./v20180122");
exports.v20180122 = v20180122;
const v20180401 = require("./v20180401");
exports.v20180401 = v20180401;
const v20181201preview = require("./v20181201preview");
exports.v20181201preview = v20181201preview;
const v20190322 = require("./v20190322");
exports.v20190322 = v20190322;
const v20190322preview = require("./v20190322preview");
exports.v20190322preview = v20190322preview;
const v20190701preview = require("./v20190701preview");
exports.v20190701preview = v20190701preview;
const v20191104 = require("./v20191104");
exports.v20191104 = v20191104;
const v20200101 = require("./v20200101");
exports.v20200101 = v20200101;
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20200401 = require("./v20200401");
exports.v20200401 = v20200401;
const v20200615 = require("./v20200615");
exports.v20200615 = v20200615;
const v20200710preview = require("./v20200710preview");
exports.v20200710preview = v20200710preview;
const v20200801 = require("./v20200801");
exports.v20200801 = v20200801;
const v20200831 = require("./v20200831");
exports.v20200831 = v20200831;
const v20200831preview = require("./v20200831preview");
exports.v20200831preview = v20200831preview;
const v20200901preview = require("./v20200901preview");
exports.v20200901preview = v20200901preview;
const v20210201preview = require("./v20210201preview");
exports.v20210201preview = v20210201preview;
const v20210303preview = require("./v20210303preview");
exports.v20210303preview = v20210303preview;
const v20210331 = require("./v20210331");
exports.v20210331 = v20210331;
const v20210701 = require("./v20210701");
exports.v20210701 = v20210701;
const v20210701preview = require("./v20210701preview");
exports.v20210701preview = v20210701preview;
// Import resources to register:
const certificate_1 = require("./certificate");
const dpsCertificate_1 = require("./dpsCertificate");
const iotDpsResource_1 = require("./iotDpsResource");
const iotDpsResourcePrivateEndpointConnection_1 = require("./iotDpsResourcePrivateEndpointConnection");
const iotHubResource_1 = require("./iotHubResource");
const iotHubResourceEventHubConsumerGroup_1 = require("./iotHubResourceEventHubConsumerGroup");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:devices:Certificate":
                return new certificate_1.Certificate(name, undefined, { urn });
            case "azure-native:devices:DpsCertificate":
                return new dpsCertificate_1.DpsCertificate(name, undefined, { urn });
            case "azure-native:devices:IotDpsResource":
                return new iotDpsResource_1.IotDpsResource(name, undefined, { urn });
            case "azure-native:devices:IotDpsResourcePrivateEndpointConnection":
                return new iotDpsResourcePrivateEndpointConnection_1.IotDpsResourcePrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:devices:IotHubResource":
                return new iotHubResource_1.IotHubResource(name, undefined, { urn });
            case "azure-native:devices:IotHubResourceEventHubConsumerGroup":
                return new iotHubResourceEventHubConsumerGroup_1.IotHubResourceEventHubConsumerGroup(name, undefined, { urn });
            case "azure-native:devices:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "devices", _module);
