// @pulumi/azure-native/network/v20200601/serviceEndpointPolicyDefinition.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceEndpointPolicyDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Service Endpoint policy definitions.
 */
class ServiceEndpointPolicyDefinition extends pulumi.CustomResource {
    /**
     * Create a ServiceEndpointPolicyDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceEndpointPolicyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceEndpointPolicyName'");
            }
            inputs["description"] = args ? args.description : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["service"] = args ? args.service : undefined;
            inputs["serviceEndpointPolicyDefinitionName"] = args ? args.serviceEndpointPolicyDefinitionName : undefined;
            inputs["serviceEndpointPolicyName"] = args ? args.serviceEndpointPolicyName : undefined;
            inputs["serviceResources"] = args ? args.serviceResources : undefined;
            inputs["etag"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
        }
        else {
            inputs["description"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["service"] = undefined /*out*/;
            inputs["serviceResources"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20200601:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20180701:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20180701:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20180801:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20180801:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20181001:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20181001:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20181101:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20181101:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20181201:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20181201:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20190201:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20190201:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20190401:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20190401:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20190601:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20190601:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20190701:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20190701:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20190801:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20190801:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20190901:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20190901:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20191101:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20191101:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20191201:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20191201:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20200301:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20200301:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20200401:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20200401:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20200501:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20200501:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20200701:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20200701:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20200801:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20200801:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20201101:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20201101:ServiceEndpointPolicyDefinition" }, { type: "azure-native:network/v20210201:ServiceEndpointPolicyDefinition" }, { type: "azure-nextgen:network/v20210201:ServiceEndpointPolicyDefinition" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ServiceEndpointPolicyDefinition.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ServiceEndpointPolicyDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ServiceEndpointPolicyDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ServiceEndpointPolicyDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServiceEndpointPolicyDefinition.__pulumiType;
    }
}
exports.ServiceEndpointPolicyDefinition = ServiceEndpointPolicyDefinition;
/** @internal */
ServiceEndpointPolicyDefinition.__pulumiType = 'azure-native:network/v20200601:ServiceEndpointPolicyDefinition';
