// @pulumi/azure-native/network/v20200601/virtualApplianceSite.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualApplianceSite = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Virtual Appliance Site resource.
 */
class VirtualApplianceSite extends pulumi.CustomResource {
    /**
     * Create a VirtualApplianceSite resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.networkVirtualApplianceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'networkVirtualApplianceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["addressPrefix"] = args ? args.addressPrefix : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["networkVirtualApplianceName"] = args ? args.networkVirtualApplianceName : undefined;
            inputs["o365Policy"] = args ? args.o365Policy : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["siteName"] = args ? args.siteName : undefined;
            inputs["etag"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["addressPrefix"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["o365Policy"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20200601:VirtualApplianceSite" }, { type: "azure-native:network:VirtualApplianceSite" }, { type: "azure-nextgen:network:VirtualApplianceSite" }, { type: "azure-native:network/v20200501:VirtualApplianceSite" }, { type: "azure-nextgen:network/v20200501:VirtualApplianceSite" }, { type: "azure-native:network/v20200701:VirtualApplianceSite" }, { type: "azure-nextgen:network/v20200701:VirtualApplianceSite" }, { type: "azure-native:network/v20200801:VirtualApplianceSite" }, { type: "azure-nextgen:network/v20200801:VirtualApplianceSite" }, { type: "azure-native:network/v20201101:VirtualApplianceSite" }, { type: "azure-nextgen:network/v20201101:VirtualApplianceSite" }, { type: "azure-native:network/v20210201:VirtualApplianceSite" }, { type: "azure-nextgen:network/v20210201:VirtualApplianceSite" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VirtualApplianceSite.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing VirtualApplianceSite resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VirtualApplianceSite(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualApplianceSite.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualApplianceSite.__pulumiType;
    }
}
exports.VirtualApplianceSite = VirtualApplianceSite;
/** @internal */
VirtualApplianceSite.__pulumiType = 'azure-native:network/v20200601:VirtualApplianceSite';
