// @pulumi/azure/costmanagement/resourceGroupExport.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Azure Cost Management Export for a Resource Group.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as azure from "@pulumi/azure";
 *
 * const exampleResourceGroup = new azure.core.ResourceGroup("exampleResourceGroup", {location: "West Europe"});
 * const exampleAccount = new azure.storage.Account("exampleAccount", {
 *     resourceGroupName: exampleResourceGroup.name,
 *     location: exampleResourceGroup.location,
 *     accountTier: "Standard",
 *     accountReplicationType: "LRS",
 * });
 * const exampleResourceGroupExport = new azure.costmanagement.ResourceGroupExport("exampleResourceGroupExport", {
 *     resourceGroupId: exampleResourceGroup.id,
 *     recurrenceType: "Monthly",
 *     recurrencePeriodStart: "2020-08-18T00:00:00Z",
 *     recurrencePeriodEnd: "2020-09-18T00:00:00Z",
 *     deliveryInfo: {
 *         storageAccountId: exampleAccount.id,
 *         containerName: "examplecontainer",
 *         rootFolderPath: "/root/updated",
 *     },
 *     query: {
 *         type: "Usage",
 *         timeFrame: "WeekToDate",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Cost Management Export for a Resource Group can be imported using the `resource id`, e.g.
 *
 * ```sh
 *  $ pulumi import azure:costmanagement/resourceGroupExport:ResourceGroupExport example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/example/providers/Microsoft.CostManagement/exports/example
 * ```
 */
class ResourceGroupExport extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["active"] = state ? state.active : undefined;
            inputs["deliveryInfo"] = state ? state.deliveryInfo : undefined;
            inputs["name"] = state ? state.name : undefined;
            inputs["query"] = state ? state.query : undefined;
            inputs["recurrencePeriodEnd"] = state ? state.recurrencePeriodEnd : undefined;
            inputs["recurrencePeriodStart"] = state ? state.recurrencePeriodStart : undefined;
            inputs["recurrenceType"] = state ? state.recurrenceType : undefined;
            inputs["resourceGroupId"] = state ? state.resourceGroupId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.deliveryInfo === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deliveryInfo'");
            }
            if ((!args || args.query === undefined) && !opts.urn) {
                throw new Error("Missing required property 'query'");
            }
            if ((!args || args.recurrencePeriodEnd === undefined) && !opts.urn) {
                throw new Error("Missing required property 'recurrencePeriodEnd'");
            }
            if ((!args || args.recurrencePeriodStart === undefined) && !opts.urn) {
                throw new Error("Missing required property 'recurrencePeriodStart'");
            }
            if ((!args || args.recurrenceType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'recurrenceType'");
            }
            if ((!args || args.resourceGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupId'");
            }
            inputs["active"] = args ? args.active : undefined;
            inputs["deliveryInfo"] = args ? args.deliveryInfo : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["query"] = args ? args.query : undefined;
            inputs["recurrencePeriodEnd"] = args ? args.recurrencePeriodEnd : undefined;
            inputs["recurrencePeriodStart"] = args ? args.recurrencePeriodStart : undefined;
            inputs["recurrenceType"] = args ? args.recurrenceType : undefined;
            inputs["resourceGroupId"] = args ? args.resourceGroupId : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(ResourceGroupExport.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ResourceGroupExport resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ResourceGroupExport(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ResourceGroupExport.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResourceGroupExport.__pulumiType;
    }
}
exports.ResourceGroupExport = ResourceGroupExport;
/** @internal */
ResourceGroupExport.__pulumiType = 'azure:costmanagement/resourceGroupExport:ResourceGroupExport';
