// @pulumi/azure-native/types/enums/servicebus/v20150801/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.SkuName = exports.NamespaceState = exports.EntityStatus = exports.EntityAvailabilityStatus = exports.AccessRights = void 0;
exports.AccessRights = {
    Manage: "Manage",
    Send: "Send",
    Listen: "Listen",
};
exports.EntityAvailabilityStatus = {
    Available: "Available",
    Limited: "Limited",
    Renaming: "Renaming",
    Restoring: "Restoring",
    Unknown: "Unknown",
};
exports.EntityStatus = {
    Active: "Active",
    Creating: "Creating",
    Deleting: "Deleting",
    Disabled: "Disabled",
    ReceiveDisabled: "ReceiveDisabled",
    Renaming: "Renaming",
    Restoring: "Restoring",
    SendDisabled: "SendDisabled",
    Unknown: "Unknown",
};
exports.NamespaceState = {
    Unknown: "Unknown",
    Creating: "Creating",
    Created: "Created",
    Activating: "Activating",
    Enabling: "Enabling",
    Active: "Active",
    Disabling: "Disabling",
    Disabled: "Disabled",
    SoftDeleting: "SoftDeleting",
    SoftDeleted: "SoftDeleted",
    Removing: "Removing",
    Removed: "Removed",
    Failed: "Failed",
};
exports.SkuName = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.SkuTier = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
