// @pulumi/azure-native/types/enums/machinelearningservices/v20200601/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VmPriority = exports.SshPublicAccess = exports.ResourceIdentityType = exports.RemoteLoginPortPublicAccess = exports.PrivateEndpointServiceConnectionStatus = exports.EncryptionStatus = exports.ComputeType = exports.ApplicationSharingPolicy = void 0;
exports.ApplicationSharingPolicy = {
    Personal: "Personal",
    Shared: "Shared",
};
exports.ComputeType = {
    AKS: "AKS",
    AmlCompute: "AmlCompute",
    ComputeInstance: "ComputeInstance",
    DataFactory: "DataFactory",
    VirtualMachine: "VirtualMachine",
    HDInsight: "HDInsight",
    Databricks: "Databricks",
    DataLakeAnalytics: "DataLakeAnalytics",
};
exports.EncryptionStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
    Timeout: "Timeout",
};
exports.RemoteLoginPortPublicAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
    NotSpecified: "NotSpecified",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
    None: "None",
};
exports.SshPublicAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.VmPriority = {
    Dedicated: "Dedicated",
    LowPriority: "LowPriority",
};
