// @pulumi/azure-native/compute/virtualMachineRunCommandByVirtualMachine.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMachineRunCommandByVirtualMachine = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Describes a Virtual Machine run command.
 * API Version: 2021-03-01.
 */
class VirtualMachineRunCommandByVirtualMachine extends pulumi.CustomResource {
    /**
     * Create a VirtualMachineRunCommandByVirtualMachine resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.vmName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vmName'");
            }
            inputs["asyncExecution"] = (_a = (args ? args.asyncExecution : undefined)) !== null && _a !== void 0 ? _a : false;
            inputs["errorBlobUri"] = args ? args.errorBlobUri : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["outputBlobUri"] = args ? args.outputBlobUri : undefined;
            inputs["parameters"] = args ? args.parameters : undefined;
            inputs["protectedParameters"] = args ? args.protectedParameters : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["runAsPassword"] = args ? args.runAsPassword : undefined;
            inputs["runAsUser"] = args ? args.runAsUser : undefined;
            inputs["runCommandName"] = args ? args.runCommandName : undefined;
            inputs["source"] = args ? args.source : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["timeoutInSeconds"] = args ? args.timeoutInSeconds : undefined;
            inputs["vmName"] = args ? args.vmName : undefined;
            inputs["instanceView"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["asyncExecution"] = undefined /*out*/;
            inputs["errorBlobUri"] = undefined /*out*/;
            inputs["instanceView"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["outputBlobUri"] = undefined /*out*/;
            inputs["parameters"] = undefined /*out*/;
            inputs["protectedParameters"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["runAsPassword"] = undefined /*out*/;
            inputs["runAsUser"] = undefined /*out*/;
            inputs["source"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["timeoutInSeconds"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:compute:VirtualMachineRunCommandByVirtualMachine" }, { type: "azure-native:compute/v20200601:VirtualMachineRunCommandByVirtualMachine" }, { type: "azure-nextgen:compute/v20200601:VirtualMachineRunCommandByVirtualMachine" }, { type: "azure-native:compute/v20201201:VirtualMachineRunCommandByVirtualMachine" }, { type: "azure-nextgen:compute/v20201201:VirtualMachineRunCommandByVirtualMachine" }, { type: "azure-native:compute/v20210301:VirtualMachineRunCommandByVirtualMachine" }, { type: "azure-nextgen:compute/v20210301:VirtualMachineRunCommandByVirtualMachine" }, { type: "azure-native:compute/v20210401:VirtualMachineRunCommandByVirtualMachine" }, { type: "azure-nextgen:compute/v20210401:VirtualMachineRunCommandByVirtualMachine" }, { type: "azure-native:compute/v20210701:VirtualMachineRunCommandByVirtualMachine" }, { type: "azure-nextgen:compute/v20210701:VirtualMachineRunCommandByVirtualMachine" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VirtualMachineRunCommandByVirtualMachine.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing VirtualMachineRunCommandByVirtualMachine resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VirtualMachineRunCommandByVirtualMachine(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualMachineRunCommandByVirtualMachine.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualMachineRunCommandByVirtualMachine.__pulumiType;
    }
}
exports.VirtualMachineRunCommandByVirtualMachine = VirtualMachineRunCommandByVirtualMachine;
/** @internal */
VirtualMachineRunCommandByVirtualMachine.__pulumiType = 'azure-native:compute:VirtualMachineRunCommandByVirtualMachine';
