// @pulumi/azure-native/media/v20210601/streamingLocator.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamingLocator = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A Streaming Locator resource
 */
class StreamingLocator extends pulumi.CustomResource {
    /**
     * Create a StreamingLocator resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.assetName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'assetName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.streamingPolicyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'streamingPolicyName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["alternativeMediaId"] = args ? args.alternativeMediaId : undefined;
            inputs["assetName"] = args ? args.assetName : undefined;
            inputs["contentKeys"] = args ? args.contentKeys : undefined;
            inputs["defaultContentKeyPolicyName"] = args ? args.defaultContentKeyPolicyName : undefined;
            inputs["endTime"] = args ? args.endTime : undefined;
            inputs["filters"] = args ? args.filters : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["startTime"] = args ? args.startTime : undefined;
            inputs["streamingLocatorId"] = args ? args.streamingLocatorId : undefined;
            inputs["streamingLocatorName"] = args ? args.streamingLocatorName : undefined;
            inputs["streamingPolicyName"] = args ? args.streamingPolicyName : undefined;
            inputs["created"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["alternativeMediaId"] = undefined /*out*/;
            inputs["assetName"] = undefined /*out*/;
            inputs["contentKeys"] = undefined /*out*/;
            inputs["created"] = undefined /*out*/;
            inputs["defaultContentKeyPolicyName"] = undefined /*out*/;
            inputs["endTime"] = undefined /*out*/;
            inputs["filters"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["startTime"] = undefined /*out*/;
            inputs["streamingLocatorId"] = undefined /*out*/;
            inputs["streamingPolicyName"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:media/v20210601:StreamingLocator" }, { type: "azure-native:media:StreamingLocator" }, { type: "azure-nextgen:media:StreamingLocator" }, { type: "azure-native:media/v20180330preview:StreamingLocator" }, { type: "azure-nextgen:media/v20180330preview:StreamingLocator" }, { type: "azure-native:media/v20180601preview:StreamingLocator" }, { type: "azure-nextgen:media/v20180601preview:StreamingLocator" }, { type: "azure-native:media/v20180701:StreamingLocator" }, { type: "azure-nextgen:media/v20180701:StreamingLocator" }, { type: "azure-native:media/v20200501:StreamingLocator" }, { type: "azure-nextgen:media/v20200501:StreamingLocator" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(StreamingLocator.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing StreamingLocator resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new StreamingLocator(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StreamingLocator.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StreamingLocator.__pulumiType;
    }
}
exports.StreamingLocator = StreamingLocator;
/** @internal */
StreamingLocator.__pulumiType = 'azure-native:media/v20210601:StreamingLocator';
