// @pulumi/azure-native/web/v20150801/siteHostNameBinding.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiteHostNameBinding = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A host name binding object
 */
class SiteHostNameBinding extends pulumi.CustomResource {
    /**
     * Create a SiteHostNameBinding resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["azureResourceName"] = args ? args.azureResourceName : undefined;
            inputs["azureResourceType"] = args ? args.azureResourceType : undefined;
            inputs["customHostNameDnsRecordType"] = args ? args.customHostNameDnsRecordType : undefined;
            inputs["domainId"] = args ? args.domainId : undefined;
            inputs["hostName"] = args ? args.hostName : undefined;
            inputs["hostNameType"] = args ? args.hostNameType : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["siteName"] = args ? args.siteName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["type"] = args ? args.type : undefined;
        }
        else {
            inputs["azureResourceName"] = undefined /*out*/;
            inputs["azureResourceType"] = undefined /*out*/;
            inputs["customHostNameDnsRecordType"] = undefined /*out*/;
            inputs["domainId"] = undefined /*out*/;
            inputs["hostNameType"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["siteName"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20150801:SiteHostNameBinding" }, { type: "azure-native:web:SiteHostNameBinding" }, { type: "azure-nextgen:web:SiteHostNameBinding" }, { type: "azure-native:web/v20160801:SiteHostNameBinding" }, { type: "azure-nextgen:web/v20160801:SiteHostNameBinding" }, { type: "azure-native:web/v20180201:SiteHostNameBinding" }, { type: "azure-nextgen:web/v20180201:SiteHostNameBinding" }, { type: "azure-native:web/v20181101:SiteHostNameBinding" }, { type: "azure-nextgen:web/v20181101:SiteHostNameBinding" }, { type: "azure-native:web/v20190801:SiteHostNameBinding" }, { type: "azure-nextgen:web/v20190801:SiteHostNameBinding" }, { type: "azure-native:web/v20200601:SiteHostNameBinding" }, { type: "azure-nextgen:web/v20200601:SiteHostNameBinding" }, { type: "azure-native:web/v20200901:SiteHostNameBinding" }, { type: "azure-nextgen:web/v20200901:SiteHostNameBinding" }, { type: "azure-native:web/v20201001:SiteHostNameBinding" }, { type: "azure-nextgen:web/v20201001:SiteHostNameBinding" }, { type: "azure-native:web/v20201201:SiteHostNameBinding" }, { type: "azure-nextgen:web/v20201201:SiteHostNameBinding" }, { type: "azure-native:web/v20210101:SiteHostNameBinding" }, { type: "azure-nextgen:web/v20210101:SiteHostNameBinding" }, { type: "azure-native:web/v20210115:SiteHostNameBinding" }, { type: "azure-nextgen:web/v20210115:SiteHostNameBinding" }, { type: "azure-native:web/v20210201:SiteHostNameBinding" }, { type: "azure-nextgen:web/v20210201:SiteHostNameBinding" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SiteHostNameBinding.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SiteHostNameBinding resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SiteHostNameBinding(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SiteHostNameBinding.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SiteHostNameBinding.__pulumiType;
    }
}
exports.SiteHostNameBinding = SiteHostNameBinding;
/** @internal */
SiteHostNameBinding.__pulumiType = 'azure-native:web/v20150801:SiteHostNameBinding';
