// @pulumi/azure-native/web/v20150801/siteRelayServiceConnection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiteRelayServiceConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Class that represents a BizTalk Hybrid Connection
 */
class SiteRelayServiceConnection extends pulumi.CustomResource {
    /**
     * Create a SiteRelayServiceConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["biztalkUri"] = args ? args.biztalkUri : undefined;
            inputs["entityConnectionString"] = args ? args.entityConnectionString : undefined;
            inputs["entityName"] = args ? args.entityName : undefined;
            inputs["hostname"] = args ? args.hostname : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["port"] = args ? args.port : undefined;
            inputs["resourceConnectionString"] = args ? args.resourceConnectionString : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["resourceType"] = args ? args.resourceType : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["type"] = args ? args.type : undefined;
        }
        else {
            inputs["biztalkUri"] = undefined /*out*/;
            inputs["entityConnectionString"] = undefined /*out*/;
            inputs["entityName"] = undefined /*out*/;
            inputs["hostname"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["port"] = undefined /*out*/;
            inputs["resourceConnectionString"] = undefined /*out*/;
            inputs["resourceType"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20150801:SiteRelayServiceConnection" }, { type: "azure-native:web:SiteRelayServiceConnection" }, { type: "azure-nextgen:web:SiteRelayServiceConnection" }, { type: "azure-native:web/v20160801:SiteRelayServiceConnection" }, { type: "azure-nextgen:web/v20160801:SiteRelayServiceConnection" }, { type: "azure-native:web/v20180201:SiteRelayServiceConnection" }, { type: "azure-nextgen:web/v20180201:SiteRelayServiceConnection" }, { type: "azure-native:web/v20181101:SiteRelayServiceConnection" }, { type: "azure-nextgen:web/v20181101:SiteRelayServiceConnection" }, { type: "azure-native:web/v20190801:SiteRelayServiceConnection" }, { type: "azure-nextgen:web/v20190801:SiteRelayServiceConnection" }, { type: "azure-native:web/v20200601:SiteRelayServiceConnection" }, { type: "azure-nextgen:web/v20200601:SiteRelayServiceConnection" }, { type: "azure-native:web/v20200901:SiteRelayServiceConnection" }, { type: "azure-nextgen:web/v20200901:SiteRelayServiceConnection" }, { type: "azure-native:web/v20201001:SiteRelayServiceConnection" }, { type: "azure-nextgen:web/v20201001:SiteRelayServiceConnection" }, { type: "azure-native:web/v20201201:SiteRelayServiceConnection" }, { type: "azure-nextgen:web/v20201201:SiteRelayServiceConnection" }, { type: "azure-native:web/v20210101:SiteRelayServiceConnection" }, { type: "azure-nextgen:web/v20210101:SiteRelayServiceConnection" }, { type: "azure-native:web/v20210115:SiteRelayServiceConnection" }, { type: "azure-nextgen:web/v20210115:SiteRelayServiceConnection" }, { type: "azure-native:web/v20210201:SiteRelayServiceConnection" }, { type: "azure-nextgen:web/v20210201:SiteRelayServiceConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SiteRelayServiceConnection.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SiteRelayServiceConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SiteRelayServiceConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SiteRelayServiceConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SiteRelayServiceConnection.__pulumiType;
    }
}
exports.SiteRelayServiceConnection = SiteRelayServiceConnection;
/** @internal */
SiteRelayServiceConnection.__pulumiType = 'azure-native:web/v20150801:SiteRelayServiceConnection';
