// @pulumi/azure-native/web/v20150801/siteBackupConfigurationSlot.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiteBackupConfigurationSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Description of a backup which will be performed
 */
class SiteBackupConfigurationSlot extends pulumi.CustomResource {
    /**
     * Create a SiteBackupConfigurationSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            inputs["backupSchedule"] = args ? args.backupSchedule : undefined;
            inputs["databases"] = args ? args.databases : undefined;
            inputs["enabled"] = args ? args.enabled : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["slot"] = args ? args.slot : undefined;
            inputs["storageAccountUrl"] = args ? args.storageAccountUrl : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["type"] = args ? args.type : undefined;
        }
        else {
            inputs["backupSchedule"] = undefined /*out*/;
            inputs["databases"] = undefined /*out*/;
            inputs["enabled"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["storageAccountUrl"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20150801:SiteBackupConfigurationSlot" }, { type: "azure-native:web:SiteBackupConfigurationSlot" }, { type: "azure-nextgen:web:SiteBackupConfigurationSlot" }, { type: "azure-native:web/v20160801:SiteBackupConfigurationSlot" }, { type: "azure-nextgen:web/v20160801:SiteBackupConfigurationSlot" }, { type: "azure-native:web/v20180201:SiteBackupConfigurationSlot" }, { type: "azure-nextgen:web/v20180201:SiteBackupConfigurationSlot" }, { type: "azure-native:web/v20181101:SiteBackupConfigurationSlot" }, { type: "azure-nextgen:web/v20181101:SiteBackupConfigurationSlot" }, { type: "azure-native:web/v20190801:SiteBackupConfigurationSlot" }, { type: "azure-nextgen:web/v20190801:SiteBackupConfigurationSlot" }, { type: "azure-native:web/v20200601:SiteBackupConfigurationSlot" }, { type: "azure-nextgen:web/v20200601:SiteBackupConfigurationSlot" }, { type: "azure-native:web/v20200901:SiteBackupConfigurationSlot" }, { type: "azure-nextgen:web/v20200901:SiteBackupConfigurationSlot" }, { type: "azure-native:web/v20201001:SiteBackupConfigurationSlot" }, { type: "azure-nextgen:web/v20201001:SiteBackupConfigurationSlot" }, { type: "azure-native:web/v20201201:SiteBackupConfigurationSlot" }, { type: "azure-nextgen:web/v20201201:SiteBackupConfigurationSlot" }, { type: "azure-native:web/v20210101:SiteBackupConfigurationSlot" }, { type: "azure-nextgen:web/v20210101:SiteBackupConfigurationSlot" }, { type: "azure-native:web/v20210115:SiteBackupConfigurationSlot" }, { type: "azure-nextgen:web/v20210115:SiteBackupConfigurationSlot" }, { type: "azure-native:web/v20210201:SiteBackupConfigurationSlot" }, { type: "azure-nextgen:web/v20210201:SiteBackupConfigurationSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SiteBackupConfigurationSlot.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SiteBackupConfigurationSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SiteBackupConfigurationSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SiteBackupConfigurationSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SiteBackupConfigurationSlot.__pulumiType;
    }
}
exports.SiteBackupConfigurationSlot = SiteBackupConfigurationSlot;
/** @internal */
SiteBackupConfigurationSlot.__pulumiType = 'azure-native:web/v20150801:SiteBackupConfigurationSlot';
