// @pulumi/azure-native/web/v20150801/siteAuthSettingsSlot.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiteAuthSettingsSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Configuration settings for the Azure App Service Authentication / Authorization feature.
 */
class SiteAuthSettingsSlot extends pulumi.CustomResource {
    /**
     * Create a SiteAuthSettingsSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            inputs["aadClientId"] = args ? args.aadClientId : undefined;
            inputs["additionalLoginParams"] = args ? args.additionalLoginParams : undefined;
            inputs["allowedAudiences"] = args ? args.allowedAudiences : undefined;
            inputs["allowedExternalRedirectUrls"] = args ? args.allowedExternalRedirectUrls : undefined;
            inputs["clientId"] = args ? args.clientId : undefined;
            inputs["clientSecret"] = args ? args.clientSecret : undefined;
            inputs["defaultProvider"] = args ? args.defaultProvider : undefined;
            inputs["enabled"] = args ? args.enabled : undefined;
            inputs["facebookAppId"] = args ? args.facebookAppId : undefined;
            inputs["facebookAppSecret"] = args ? args.facebookAppSecret : undefined;
            inputs["facebookOAuthScopes"] = args ? args.facebookOAuthScopes : undefined;
            inputs["googleClientId"] = args ? args.googleClientId : undefined;
            inputs["googleClientSecret"] = args ? args.googleClientSecret : undefined;
            inputs["googleOAuthScopes"] = args ? args.googleOAuthScopes : undefined;
            inputs["httpApiPrefixPath"] = args ? args.httpApiPrefixPath : undefined;
            inputs["issuer"] = args ? args.issuer : undefined;
            inputs["microsoftAccountClientId"] = args ? args.microsoftAccountClientId : undefined;
            inputs["microsoftAccountClientSecret"] = args ? args.microsoftAccountClientSecret : undefined;
            inputs["microsoftAccountOAuthScopes"] = args ? args.microsoftAccountOAuthScopes : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["openIdIssuer"] = args ? args.openIdIssuer : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["slot"] = args ? args.slot : undefined;
            inputs["tokenRefreshExtensionHours"] = args ? args.tokenRefreshExtensionHours : undefined;
            inputs["tokenStoreEnabled"] = args ? args.tokenStoreEnabled : undefined;
            inputs["twitterConsumerKey"] = args ? args.twitterConsumerKey : undefined;
            inputs["twitterConsumerSecret"] = args ? args.twitterConsumerSecret : undefined;
            inputs["unauthenticatedClientAction"] = args ? args.unauthenticatedClientAction : undefined;
        }
        else {
            inputs["aadClientId"] = undefined /*out*/;
            inputs["additionalLoginParams"] = undefined /*out*/;
            inputs["allowedAudiences"] = undefined /*out*/;
            inputs["allowedExternalRedirectUrls"] = undefined /*out*/;
            inputs["clientId"] = undefined /*out*/;
            inputs["clientSecret"] = undefined /*out*/;
            inputs["defaultProvider"] = undefined /*out*/;
            inputs["enabled"] = undefined /*out*/;
            inputs["facebookAppId"] = undefined /*out*/;
            inputs["facebookAppSecret"] = undefined /*out*/;
            inputs["facebookOAuthScopes"] = undefined /*out*/;
            inputs["googleClientId"] = undefined /*out*/;
            inputs["googleClientSecret"] = undefined /*out*/;
            inputs["googleOAuthScopes"] = undefined /*out*/;
            inputs["httpApiPrefixPath"] = undefined /*out*/;
            inputs["issuer"] = undefined /*out*/;
            inputs["microsoftAccountClientId"] = undefined /*out*/;
            inputs["microsoftAccountClientSecret"] = undefined /*out*/;
            inputs["microsoftAccountOAuthScopes"] = undefined /*out*/;
            inputs["openIdIssuer"] = undefined /*out*/;
            inputs["tokenRefreshExtensionHours"] = undefined /*out*/;
            inputs["tokenStoreEnabled"] = undefined /*out*/;
            inputs["twitterConsumerKey"] = undefined /*out*/;
            inputs["twitterConsumerSecret"] = undefined /*out*/;
            inputs["unauthenticatedClientAction"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20150801:SiteAuthSettingsSlot" }, { type: "azure-native:web:SiteAuthSettingsSlot" }, { type: "azure-nextgen:web:SiteAuthSettingsSlot" }, { type: "azure-native:web/v20160801:SiteAuthSettingsSlot" }, { type: "azure-nextgen:web/v20160801:SiteAuthSettingsSlot" }, { type: "azure-native:web/v20180201:SiteAuthSettingsSlot" }, { type: "azure-nextgen:web/v20180201:SiteAuthSettingsSlot" }, { type: "azure-native:web/v20181101:SiteAuthSettingsSlot" }, { type: "azure-nextgen:web/v20181101:SiteAuthSettingsSlot" }, { type: "azure-native:web/v20190801:SiteAuthSettingsSlot" }, { type: "azure-nextgen:web/v20190801:SiteAuthSettingsSlot" }, { type: "azure-native:web/v20200601:SiteAuthSettingsSlot" }, { type: "azure-nextgen:web/v20200601:SiteAuthSettingsSlot" }, { type: "azure-native:web/v20200901:SiteAuthSettingsSlot" }, { type: "azure-nextgen:web/v20200901:SiteAuthSettingsSlot" }, { type: "azure-native:web/v20201001:SiteAuthSettingsSlot" }, { type: "azure-nextgen:web/v20201001:SiteAuthSettingsSlot" }, { type: "azure-native:web/v20201201:SiteAuthSettingsSlot" }, { type: "azure-nextgen:web/v20201201:SiteAuthSettingsSlot" }, { type: "azure-native:web/v20210101:SiteAuthSettingsSlot" }, { type: "azure-nextgen:web/v20210101:SiteAuthSettingsSlot" }, { type: "azure-native:web/v20210115:SiteAuthSettingsSlot" }, { type: "azure-nextgen:web/v20210115:SiteAuthSettingsSlot" }, { type: "azure-native:web/v20210201:SiteAuthSettingsSlot" }, { type: "azure-nextgen:web/v20210201:SiteAuthSettingsSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SiteAuthSettingsSlot.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing SiteAuthSettingsSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SiteAuthSettingsSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SiteAuthSettingsSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SiteAuthSettingsSlot.__pulumiType;
    }
}
exports.SiteAuthSettingsSlot = SiteAuthSettingsSlot;
/** @internal */
SiteAuthSettingsSlot.__pulumiType = 'azure-native:web/v20150801:SiteAuthSettingsSlot';
