// @pulumi/azure-native/logic/v20190501/integrationServiceEnvironmentManagedApi.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntegrationServiceEnvironmentManagedApi = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The managed api definition.
 */
class IntegrationServiceEnvironmentManagedApi extends pulumi.CustomResource {
    /**
     * Create a IntegrationServiceEnvironmentManagedApi resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.integrationServiceEnvironmentName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'integrationServiceEnvironmentName'");
            }
            if ((!args || args.resourceGroup === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroup'");
            }
            inputs["apiName"] = args ? args.apiName : undefined;
            inputs["integrationServiceEnvironmentName"] = args ? args.integrationServiceEnvironmentName : undefined;
            inputs["resourceGroup"] = args ? args.resourceGroup : undefined;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:logic/v20190501:IntegrationServiceEnvironmentManagedApi" }, { type: "azure-native:logic:IntegrationServiceEnvironmentManagedApi" }, { type: "azure-nextgen:logic:IntegrationServiceEnvironmentManagedApi" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IntegrationServiceEnvironmentManagedApi.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing IntegrationServiceEnvironmentManagedApi resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IntegrationServiceEnvironmentManagedApi(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IntegrationServiceEnvironmentManagedApi.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IntegrationServiceEnvironmentManagedApi.__pulumiType;
    }
}
exports.IntegrationServiceEnvironmentManagedApi = IntegrationServiceEnvironmentManagedApi;
/** @internal */
IntegrationServiceEnvironmentManagedApi.__pulumiType = 'azure-native:logic/v20190501:IntegrationServiceEnvironmentManagedApi';
