// @pulumi/azure-native/datashare/v20191101/invitation.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Invitation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A Invitation data transfer object.
 */
class Invitation extends pulumi.CustomResource {
    /**
     * Create a Invitation resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.shareName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shareName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["invitationName"] = args ? args.invitationName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["shareName"] = args ? args.shareName : undefined;
            inputs["targetActiveDirectoryId"] = args ? args.targetActiveDirectoryId : undefined;
            inputs["targetEmail"] = args ? args.targetEmail : undefined;
            inputs["targetObjectId"] = args ? args.targetObjectId : undefined;
            inputs["invitationId"] = undefined /*out*/;
            inputs["invitationStatus"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["respondedAt"] = undefined /*out*/;
            inputs["sentAt"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["userEmail"] = undefined /*out*/;
            inputs["userName"] = undefined /*out*/;
        }
        else {
            inputs["invitationId"] = undefined /*out*/;
            inputs["invitationStatus"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["respondedAt"] = undefined /*out*/;
            inputs["sentAt"] = undefined /*out*/;
            inputs["targetActiveDirectoryId"] = undefined /*out*/;
            inputs["targetEmail"] = undefined /*out*/;
            inputs["targetObjectId"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["userEmail"] = undefined /*out*/;
            inputs["userName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:datashare/v20191101:Invitation" }, { type: "azure-native:datashare:Invitation" }, { type: "azure-nextgen:datashare:Invitation" }, { type: "azure-native:datashare/v20181101preview:Invitation" }, { type: "azure-nextgen:datashare/v20181101preview:Invitation" }, { type: "azure-native:datashare/v20200901:Invitation" }, { type: "azure-nextgen:datashare/v20200901:Invitation" }, { type: "azure-native:datashare/v20201001preview:Invitation" }, { type: "azure-nextgen:datashare/v20201001preview:Invitation" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Invitation.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Invitation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Invitation(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Invitation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Invitation.__pulumiType;
    }
}
exports.Invitation = Invitation;
/** @internal */
Invitation.__pulumiType = 'azure-native:datashare/v20191101:Invitation';
