// @pulumi/azure-native/batch/application.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Contains information about an application in a Batch account.
 * API Version: 2021-01-01.
 */
class Application extends pulumi.CustomResource {
    /**
     * Create a Application resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["accountName"] = args ? args.accountName : undefined;
            inputs["allowUpdates"] = args ? args.allowUpdates : undefined;
            inputs["applicationName"] = args ? args.applicationName : undefined;
            inputs["defaultVersion"] = args ? args.defaultVersion : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["allowUpdates"] = undefined /*out*/;
            inputs["defaultVersion"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:batch:Application" }, { type: "azure-native:batch/v20151201:Application" }, { type: "azure-nextgen:batch/v20151201:Application" }, { type: "azure-native:batch/v20170101:Application" }, { type: "azure-nextgen:batch/v20170101:Application" }, { type: "azure-native:batch/v20170501:Application" }, { type: "azure-nextgen:batch/v20170501:Application" }, { type: "azure-native:batch/v20170901:Application" }, { type: "azure-nextgen:batch/v20170901:Application" }, { type: "azure-native:batch/v20181201:Application" }, { type: "azure-nextgen:batch/v20181201:Application" }, { type: "azure-native:batch/v20190401:Application" }, { type: "azure-nextgen:batch/v20190401:Application" }, { type: "azure-native:batch/v20190801:Application" }, { type: "azure-nextgen:batch/v20190801:Application" }, { type: "azure-native:batch/v20200301:Application" }, { type: "azure-nextgen:batch/v20200301:Application" }, { type: "azure-native:batch/v20200501:Application" }, { type: "azure-nextgen:batch/v20200501:Application" }, { type: "azure-native:batch/v20200901:Application" }, { type: "azure-nextgen:batch/v20200901:Application" }, { type: "azure-native:batch/v20210101:Application" }, { type: "azure-nextgen:batch/v20210101:Application" }, { type: "azure-native:batch/v20210601:Application" }, { type: "azure-nextgen:batch/v20210601:Application" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Application.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing Application resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Application(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Application.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Application.__pulumiType;
    }
}
exports.Application = Application;
/** @internal */
Application.__pulumiType = 'azure-native:batch:Application';
