// @pulumi/azure-native/types/enums/botservice/v20180712/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuName = exports.Kind = exports.EnterpriseChannelState = exports.EnterpriseChannelNodeState = void 0;
exports.EnterpriseChannelNodeState = {
    Creating: "Creating",
    CreateFailed: "CreateFailed",
    Started: "Started",
    Starting: "Starting",
    StartFailed: "StartFailed",
    Stopped: "Stopped",
    Stopping: "Stopping",
    StopFailed: "StopFailed",
    Deleting: "Deleting",
    DeleteFailed: "DeleteFailed",
};
exports.EnterpriseChannelState = {
    Creating: "Creating",
    CreateFailed: "CreateFailed",
    Started: "Started",
    Starting: "Starting",
    StartFailed: "StartFailed",
    Stopped: "Stopped",
    Stopping: "Stopping",
    StopFailed: "StopFailed",
    Deleting: "Deleting",
    DeleteFailed: "DeleteFailed",
};
exports.Kind = {
    Sdk: "sdk",
    Designer: "designer",
    Bot: "bot",
    Function: "function",
};
exports.SkuName = {
    F0: "F0",
    S1: "S1",
};
