// @pulumi/azure-native/types/enums/media/v20180701/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrackPropertyType = exports.TrackPropertyCompareOperation = exports.StretchMode = exports.StreamOptionsFlag = exports.StorageAccountType = exports.Rotation = exports.Priority = exports.OnErrorType = exports.LiveEventInputProtocol = exports.LiveEventEncodingType = exports.InsightsType = exports.H264VideoProfile = exports.H264Complexity = exports.FilterTrackPropertyType = exports.FilterTrackPropertyCompareOperation = exports.EntropyMode = exports.EncoderNamedPreset = exports.DeinterlaceParity = exports.DeinterlaceMode = exports.ContentKeyPolicyRestrictionTokenType = exports.ContentKeyPolicyPlayReadyUnknownOutputPassingOption = exports.ContentKeyPolicyPlayReadyLicenseType = exports.ContentKeyPolicyPlayReadyContentType = exports.ContentKeyPolicyFairPlayRentalAndLeaseKeyType = exports.AssetContainerPermission = exports.AnalysisResolution = exports.AacAudioProfile = void 0;
exports.AacAudioProfile = {
    /**
     * Specifies that the output audio is to be encoded into AAC Low Complexity profile (AAC-LC).
     */
    AacLc: "AacLc",
    /**
     * Specifies that the output audio is to be encoded into HE-AAC v1 profile.
     */
    HeAacV1: "HeAacV1",
    /**
     * Specifies that the output audio is to be encoded into HE-AAC v2 profile.
     */
    HeAacV2: "HeAacV2",
};
exports.AnalysisResolution = {
    SourceResolution: "SourceResolution",
    StandardDefinition: "StandardDefinition",
};
exports.AssetContainerPermission = {
    /**
     * The SAS URL will allow read access to the container.
     */
    Read: "Read",
    /**
     * The SAS URL will allow read and write access to the container.
     */
    ReadWrite: "ReadWrite",
    /**
     * The SAS URL will allow read, write and delete access to the container.
     */
    ReadWriteDelete: "ReadWriteDelete",
};
exports.ContentKeyPolicyFairPlayRentalAndLeaseKeyType = {
    /**
     * Represents a ContentKeyPolicyFairPlayRentalAndLeaseKeyType that is unavailable in current API version.
     */
    Unknown: "Unknown",
    /**
     * Key duration is not specified.
     */
    Undefined: "Undefined",
    /**
     * Dual expiry for offline rental.
     */
    DualExpiry: "DualExpiry",
    /**
     * Content key can be persisted with an unlimited duration
     */
    PersistentUnlimited: "PersistentUnlimited",
    /**
     * Content key can be persisted and the valid duration is limited by the Rental Duration value
     */
    PersistentLimited: "PersistentLimited",
};
exports.ContentKeyPolicyPlayReadyContentType = {
    /**
     * Represents a ContentKeyPolicyPlayReadyContentType that is unavailable in current API version.
     */
    Unknown: "Unknown",
    /**
     * Unspecified content type.
     */
    Unspecified: "Unspecified",
    /**
     * Ultraviolet download content type.
     */
    UltraVioletDownload: "UltraVioletDownload",
    /**
     * Ultraviolet streaming content type.
     */
    UltraVioletStreaming: "UltraVioletStreaming",
};
exports.ContentKeyPolicyPlayReadyLicenseType = {
    /**
     * Represents a ContentKeyPolicyPlayReadyLicenseType that is unavailable in current API version.
     */
    Unknown: "Unknown",
    /**
     * Non persistent license.
     */
    NonPersistent: "NonPersistent",
    /**
     * Persistent license. Allows offline playback.
     */
    Persistent: "Persistent",
};
exports.ContentKeyPolicyPlayReadyUnknownOutputPassingOption = {
    /**
     * Represents a ContentKeyPolicyPlayReadyUnknownOutputPassingOption that is unavailable in current API version.
     */
    Unknown: "Unknown",
    /**
     * Passing the video portion of protected content to an Unknown Output is not allowed.
     */
    NotAllowed: "NotAllowed",
    /**
     * Passing the video portion of protected content to an Unknown Output is allowed.
     */
    Allowed: "Allowed",
    /**
     * Passing the video portion of protected content to an Unknown Output is allowed but with constrained resolution.
     */
    AllowedWithVideoConstriction: "AllowedWithVideoConstriction",
};
exports.ContentKeyPolicyRestrictionTokenType = {
    /**
     * Represents a ContentKeyPolicyRestrictionTokenType that is unavailable in current API version.
     */
    Unknown: "Unknown",
    /**
     * Simple Web Token.
     */
    Swt: "Swt",
    /**
     * JSON Web Token.
     */
    Jwt: "Jwt",
};
exports.DeinterlaceMode = {
    /**
     * Disables de-interlacing of the source video.
     */
    Off: "Off",
    /**
     * Apply automatic pixel adaptive de-interlacing on each frame in the input video.
     */
    AutoPixelAdaptive: "AutoPixelAdaptive",
};
exports.DeinterlaceParity = {
    /**
     * Automatically detect the order of fields
     */
    Auto: "Auto",
    /**
     * Apply top field first processing of input video.
     */
    TopFieldFirst: "TopFieldFirst",
    /**
     * Apply bottom field first processing of input video.
     */
    BottomFieldFirst: "BottomFieldFirst",
};
exports.EncoderNamedPreset = {
    /**
     * Produces an MP4 file where the video is encoded with H.264 codec at 2200 kbps and a picture height of 480 pixels, and the stereo audio is encoded with AAC-LC codec at 64 kbps.
     */
    H264SingleBitrateSD: "H264SingleBitrateSD",
    /**
     * Produces an MP4 file where the video is encoded with H.264 codec at 4500 kbps and a picture height of 720 pixels, and the stereo audio is encoded with AAC-LC codec at 64 kbps.
     */
    H264SingleBitrate720p: "H264SingleBitrate720p",
    /**
     * Produces an MP4 file where the video is encoded with H.264 codec at 6750 kbps and a picture height of 1080 pixels, and the stereo audio is encoded with AAC-LC codec at 64 kbps.
     */
    H264SingleBitrate1080p: "H264SingleBitrate1080p",
    /**
     * Produces a set of GOP aligned MP4 files with H.264 video and stereo AAC audio. Auto-generates a bitrate ladder based on the input resolution and bitrate. The auto-generated preset will never exceed the input resolution and bitrate. For example, if the input is 720p at 3 Mbps, output will remain 720p at best, and will start at rates lower than 3 Mbps. The output will have video and audio in separate MP4 files, which is optimal for adaptive streaming.
     */
    AdaptiveStreaming: "AdaptiveStreaming",
    /**
     * Produces a single MP4 file containing only stereo audio encoded at 192 kbps.
     */
    AACGoodQualityAudio: "AACGoodQualityAudio",
    /**
     * Exposes an experimental preset for content-aware encoding. Given any input content, the service attempts to automatically determine the optimal number of layers, appropriate bitrate and resolution settings for delivery by adaptive streaming. The underlying algorithms will continue to evolve over time. The output will contain MP4 files with video and audio interleaved.
     */
    ContentAwareEncodingExperimental: "ContentAwareEncodingExperimental",
    /**
     * Produces a set of GOP-aligned MP4s by using content-aware encoding. Given any input content, the service performs an initial lightweight analysis of the input content, and uses the results to determine the optimal number of layers, appropriate bitrate and resolution settings for delivery by adaptive streaming. This preset is particularly effective for low and medium complexity videos, where the output files will be at lower bitrates but at a quality that still delivers a good experience to viewers. The output will contain MP4 files with video and audio interleaved.
     */
    ContentAwareEncoding: "ContentAwareEncoding",
    /**
     * Produces a set of 8 GOP-aligned MP4 files, ranging from 6000 kbps to 400 kbps, and stereo AAC audio. Resolution starts at 1080p and goes down to 360p.
     */
    H264MultipleBitrate1080p: "H264MultipleBitrate1080p",
    /**
     * Produces a set of 6 GOP-aligned MP4 files, ranging from 3400 kbps to 400 kbps, and stereo AAC audio. Resolution starts at 720p and goes down to 360p.
     */
    H264MultipleBitrate720p: "H264MultipleBitrate720p",
    /**
     * Produces a set of 5 GOP-aligned MP4 files, ranging from 1600kbps to 400 kbps, and stereo AAC audio. Resolution starts at 480p and goes down to 360p.
     */
    H264MultipleBitrateSD: "H264MultipleBitrateSD",
};
exports.EntropyMode = {
    /**
     * Context Adaptive Binary Arithmetic Coder (CABAC) entropy encoding.
     */
    Cabac: "Cabac",
    /**
     * Context Adaptive Variable Length Coder (CAVLC) entropy encoding.
     */
    Cavlc: "Cavlc",
};
exports.FilterTrackPropertyCompareOperation = {
    /**
     * The equal operation.
     */
    Equal: "Equal",
    /**
     * The not equal operation.
     */
    NotEqual: "NotEqual",
};
exports.FilterTrackPropertyType = {
    /**
     * The unknown track property type.
     */
    Unknown: "Unknown",
    /**
     * The type.
     */
    Type: "Type",
    /**
     * The name.
     */
    Name: "Name",
    /**
     * The language.
     */
    Language: "Language",
    /**
     * The fourCC.
     */
    FourCC: "FourCC",
    /**
     * The bitrate.
     */
    Bitrate: "Bitrate",
};
exports.H264Complexity = {
    /**
     * Tells the encoder to use settings that are optimized for faster encoding. Quality is sacrificed to decrease encoding time.
     */
    Speed: "Speed",
    /**
     * Tells the encoder to use settings that achieve a balance between speed and quality.
     */
    Balanced: "Balanced",
    /**
     * Tells the encoder to use settings that are optimized to produce higher quality output at the expense of slower overall encode time.
     */
    Quality: "Quality",
};
exports.H264VideoProfile = {
    /**
     * Tells the encoder to automatically determine the appropriate H.264 profile.
     */
    Auto: "Auto",
    /**
     * Baseline profile
     */
    Baseline: "Baseline",
    /**
     * Main profile
     */
    Main: "Main",
    /**
     * High profile.
     */
    High: "High",
    /**
     * High 4:2:2 profile.
     */
    High422: "High422",
    /**
     * High 4:4:4 predictive profile.
     */
    High444: "High444",
};
exports.InsightsType = {
    /**
     * Generate audio only insights. Ignore video even if present. Fails if no audio is present.
     */
    AudioInsightsOnly: "AudioInsightsOnly",
    /**
     * Generate video only insights. Ignore audio if present. Fails if no video is present.
     */
    VideoInsightsOnly: "VideoInsightsOnly",
    /**
     * Generate both audio and video insights. Fails if either audio or video Insights fail.
     */
    AllInsights: "AllInsights",
};
exports.LiveEventEncodingType = {
    None: "None",
    Basic: "Basic",
    Standard: "Standard",
    Premium1080p: "Premium1080p",
};
exports.LiveEventInputProtocol = {
    FragmentedMP4: "FragmentedMP4",
    RTMP: "RTMP",
};
exports.OnErrorType = {
    /**
     * Tells the service that if this TransformOutput fails, then any other incomplete TransformOutputs can be stopped.
     */
    StopProcessingJob: "StopProcessingJob",
    /**
     * Tells the service that if this TransformOutput fails, then allow any other TransformOutput to continue.
     */
    ContinueJob: "ContinueJob",
};
exports.Priority = {
    /**
     * Used for TransformOutputs that can be generated after Normal and High priority TransformOutputs.
     */
    Low: "Low",
    /**
     * Used for TransformOutputs that can be generated at Normal priority.
     */
    Normal: "Normal",
    /**
     * Used for TransformOutputs that should take precedence over others.
     */
    High: "High",
};
exports.Rotation = {
    /**
     * Automatically detect and rotate as needed.
     */
    Auto: "Auto",
    /**
     * Do not rotate the video.  If the output format supports it, any metadata about rotation is kept intact.
     */
    None: "None",
    /**
     * Do not rotate the video but remove any metadata about the rotation.
     */
    Rotate0: "Rotate0",
    /**
     * Rotate 90 degrees clockwise.
     */
    Rotate90: "Rotate90",
    /**
     * Rotate 180 degrees clockwise.
     */
    Rotate180: "Rotate180",
    /**
     * Rotate 270 degrees clockwise.
     */
    Rotate270: "Rotate270",
};
exports.StorageAccountType = {
    /**
     * The primary storage account for the Media Services account.
     */
    Primary: "Primary",
    /**
     * A secondary storage account for the Media Services account.
     */
    Secondary: "Secondary",
};
exports.StreamOptionsFlag = {
    Default: "Default",
    LowLatency: "LowLatency",
};
exports.StretchMode = {
    /**
     * Strictly respect the output resolution without considering the pixel aspect ratio or display aspect ratio of the input video.
     */
    None: "None",
    /**
     * Override the output resolution, and change it to match the display aspect ratio of the input, without padding. For example, if the input is 1920x1080 and the encoding preset asks for 1280x1280, then the value in the preset is overridden, and the output will be at 1280x720, which maintains the input aspect ratio of 16:9.
     */
    AutoSize: "AutoSize",
    /**
     * Pad the output (with either letterbox or pillar box) to honor the output resolution, while ensuring that the active video region in the output has the same aspect ratio as the input. For example, if the input is 1920x1080 and the encoding preset asks for 1280x1280, then the output will be at 1280x1280, which contains an inner rectangle of 1280x720 at aspect ratio of 16:9, and pillar box regions 280 pixels wide at the left and right.
     */
    AutoFit: "AutoFit",
};
exports.TrackPropertyCompareOperation = {
    /**
     * Unknown track property compare operation
     */
    Unknown: "Unknown",
    /**
     * Equal operation
     */
    Equal: "Equal",
};
exports.TrackPropertyType = {
    /**
     * Unknown track property
     */
    Unknown: "Unknown",
    /**
     * Track FourCC
     */
    FourCC: "FourCC",
};
