// @pulumi/azure-native/sql/v20201101preview/managedInstanceAzureADOnlyAuthentication.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedInstanceAzureADOnlyAuthentication = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Azure Active Directory only authentication.
 */
class ManagedInstanceAzureADOnlyAuthentication extends pulumi.CustomResource {
    /**
     * Create a ManagedInstanceAzureADOnlyAuthentication resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.azureADOnlyAuthentication === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azureADOnlyAuthentication'");
            }
            if ((!args || args.managedInstanceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managedInstanceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["authenticationName"] = args ? args.authenticationName : undefined;
            inputs["azureADOnlyAuthentication"] = args ? args.azureADOnlyAuthentication : undefined;
            inputs["managedInstanceName"] = args ? args.managedInstanceName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["azureADOnlyAuthentication"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:sql/v20201101preview:ManagedInstanceAzureADOnlyAuthentication" }, { type: "azure-native:sql:ManagedInstanceAzureADOnlyAuthentication" }, { type: "azure-nextgen:sql:ManagedInstanceAzureADOnlyAuthentication" }, { type: "azure-native:sql/v20200202preview:ManagedInstanceAzureADOnlyAuthentication" }, { type: "azure-nextgen:sql/v20200202preview:ManagedInstanceAzureADOnlyAuthentication" }, { type: "azure-native:sql/v20200801preview:ManagedInstanceAzureADOnlyAuthentication" }, { type: "azure-nextgen:sql/v20200801preview:ManagedInstanceAzureADOnlyAuthentication" }, { type: "azure-native:sql/v20210201preview:ManagedInstanceAzureADOnlyAuthentication" }, { type: "azure-nextgen:sql/v20210201preview:ManagedInstanceAzureADOnlyAuthentication" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagedInstanceAzureADOnlyAuthentication.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ManagedInstanceAzureADOnlyAuthentication resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ManagedInstanceAzureADOnlyAuthentication(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagedInstanceAzureADOnlyAuthentication.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagedInstanceAzureADOnlyAuthentication.__pulumiType;
    }
}
exports.ManagedInstanceAzureADOnlyAuthentication = ManagedInstanceAzureADOnlyAuthentication;
/** @internal */
ManagedInstanceAzureADOnlyAuthentication.__pulumiType = 'azure-native:sql/v20201101preview:ManagedInstanceAzureADOnlyAuthentication';
