// @pulumi/azure-native/types/enums/blueprint/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplateParameterType = exports.ManagedServiceIdentityType = exports.BlueprintTargetScope = exports.AssignmentLockMode = exports.ArtifactKind = exports.v20181101preview = exports.v20171111preview = void 0;
// Export sub-modules:
const v20171111preview = require("./v20171111preview");
exports.v20171111preview = v20171111preview;
const v20181101preview = require("./v20181101preview");
exports.v20181101preview = v20181101preview;
exports.ArtifactKind = {
    Template: "template",
    RoleAssignment: "roleAssignment",
    PolicyAssignment: "policyAssignment",
};
exports.AssignmentLockMode = {
    None: "None",
    AllResourcesReadOnly: "AllResourcesReadOnly",
    AllResourcesDoNotDelete: "AllResourcesDoNotDelete",
};
exports.BlueprintTargetScope = {
    /**
     * The blueprint targets a subscription during blueprint assignment.
     */
    Subscription: "subscription",
    /**
     * The blueprint targets a management group during blueprint assignment. This is reserved for future use.
     */
    ManagementGroup: "managementGroup",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.TemplateParameterType = {
    String: "string",
    Array: "array",
    Bool: "bool",
    Int: "int",
    Object: "object",
    SecureObject: "secureObject",
    SecureString: "secureString",
};
