// @pulumi/azure-native/types/enums/databox/v20180101/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuName = exports.NotificationStageName = exports.DataDestinationType = exports.ClassDiscriminator = exports.AddressType = void 0;
exports.AddressType = {
    /**
     * Address type not known.
     */
    None: "None",
    /**
     * Residential Address.
     */
    Residential: "Residential",
    /**
     * Commercial Address.
     */
    Commercial: "Commercial",
};
exports.ClassDiscriminator = {
    /**
     * DataBox orders.
     */
    DataBox: "DataBox",
    /**
     * DataBoxDisk orders.
     */
    DataBoxDisk: "DataBoxDisk",
    /**
     * DataBoxHeavy orders.
     */
    DataBoxHeavy: "DataBoxHeavy",
};
exports.DataDestinationType = {
    /**
     * Unknown type.
     */
    UnknownType: "UnknownType",
    /**
     * Storage Accounts .
     */
    StorageAccount: "StorageAccount",
    /**
     * Azure Managed disk storage.
     */
    ManagedDisk: "ManagedDisk",
};
exports.NotificationStageName = {
    /**
     * Notification at device prepared stage.
     */
    DevicePrepared: "DevicePrepared",
    /**
     * Notification at device dispatched stage.
     */
    Dispatched: "Dispatched",
    /**
     * Notification at device delivered stage.
     */
    Delivered: "Delivered",
    /**
     * Notification at device picked up from user stage.
     */
    PickedUp: "PickedUp",
    /**
     * Notification at device received at azure datacenter stage.
     */
    AtAzureDC: "AtAzureDC",
    /**
     * Notification at data copy started stage.
     */
    DataCopy: "DataCopy",
};
exports.SkuName = {
    /**
     * DataBox.
     */
    DataBox: "DataBox",
    /**
     * DataBoxDisk.
     */
    DataBoxDisk: "DataBoxDisk",
    /**
     * DataBoxHeavy.
     */
    DataBoxHeavy: "DataBoxHeavy",
};
