// @pulumi/azure-native/securityinsights/fusionAlertRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FusionAlertRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents Fusion alert rule.
 * API Version: 2020-01-01.
 */
class FusionAlertRule extends pulumi.CustomResource {
    /**
     * Create a FusionAlertRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.alertRuleTemplateName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'alertRuleTemplateName'");
            }
            if ((!args || args.enabled === undefined) && !opts.urn) {
                throw new Error("Missing required property 'enabled'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["alertRuleTemplateName"] = args ? args.alertRuleTemplateName : undefined;
            inputs["enabled"] = args ? args.enabled : undefined;
            inputs["etag"] = args ? args.etag : undefined;
            inputs["kind"] = "Fusion";
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["ruleId"] = args ? args.ruleId : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["lastModifiedUtc"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["severity"] = undefined /*out*/;
            inputs["tactics"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["alertRuleTemplateName"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["enabled"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["lastModifiedUtc"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["severity"] = undefined /*out*/;
            inputs["tactics"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:securityinsights:FusionAlertRule" }, { type: "azure-native:securityinsights/v20190101preview:FusionAlertRule" }, { type: "azure-nextgen:securityinsights/v20190101preview:FusionAlertRule" }, { type: "azure-native:securityinsights/v20200101:FusionAlertRule" }, { type: "azure-nextgen:securityinsights/v20200101:FusionAlertRule" }, { type: "azure-native:securityinsights/v20210301preview:FusionAlertRule" }, { type: "azure-nextgen:securityinsights/v20210301preview:FusionAlertRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(FusionAlertRule.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing FusionAlertRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new FusionAlertRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of FusionAlertRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FusionAlertRule.__pulumiType;
    }
}
exports.FusionAlertRule = FusionAlertRule;
/** @internal */
FusionAlertRule.__pulumiType = 'azure-native:securityinsights:FusionAlertRule';
