// @pulumi/azure-native/securityinsights/scheduledAlertRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduledAlertRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents scheduled alert rule.
 * API Version: 2020-01-01.
 */
class ScheduledAlertRule extends pulumi.CustomResource {
    /**
     * Create a ScheduledAlertRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.enabled === undefined) && !opts.urn) {
                throw new Error("Missing required property 'enabled'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.query === undefined) && !opts.urn) {
                throw new Error("Missing required property 'query'");
            }
            if ((!args || args.queryFrequency === undefined) && !opts.urn) {
                throw new Error("Missing required property 'queryFrequency'");
            }
            if ((!args || args.queryPeriod === undefined) && !opts.urn) {
                throw new Error("Missing required property 'queryPeriod'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.severity === undefined) && !opts.urn) {
                throw new Error("Missing required property 'severity'");
            }
            if ((!args || args.suppressionDuration === undefined) && !opts.urn) {
                throw new Error("Missing required property 'suppressionDuration'");
            }
            if ((!args || args.suppressionEnabled === undefined) && !opts.urn) {
                throw new Error("Missing required property 'suppressionEnabled'");
            }
            if ((!args || args.triggerOperator === undefined) && !opts.urn) {
                throw new Error("Missing required property 'triggerOperator'");
            }
            if ((!args || args.triggerThreshold === undefined) && !opts.urn) {
                throw new Error("Missing required property 'triggerThreshold'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            inputs["alertRuleTemplateName"] = args ? args.alertRuleTemplateName : undefined;
            inputs["description"] = args ? args.description : undefined;
            inputs["displayName"] = args ? args.displayName : undefined;
            inputs["enabled"] = args ? args.enabled : undefined;
            inputs["etag"] = args ? args.etag : undefined;
            inputs["kind"] = "Scheduled";
            inputs["query"] = args ? args.query : undefined;
            inputs["queryFrequency"] = args ? args.queryFrequency : undefined;
            inputs["queryPeriod"] = args ? args.queryPeriod : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["ruleId"] = args ? args.ruleId : undefined;
            inputs["severity"] = args ? args.severity : undefined;
            inputs["suppressionDuration"] = args ? args.suppressionDuration : undefined;
            inputs["suppressionEnabled"] = args ? args.suppressionEnabled : undefined;
            inputs["tactics"] = args ? args.tactics : undefined;
            inputs["triggerOperator"] = args ? args.triggerOperator : undefined;
            inputs["triggerThreshold"] = args ? args.triggerThreshold : undefined;
            inputs["workspaceName"] = args ? args.workspaceName : undefined;
            inputs["lastModifiedUtc"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["alertRuleTemplateName"] = undefined /*out*/;
            inputs["description"] = undefined /*out*/;
            inputs["displayName"] = undefined /*out*/;
            inputs["enabled"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["lastModifiedUtc"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["query"] = undefined /*out*/;
            inputs["queryFrequency"] = undefined /*out*/;
            inputs["queryPeriod"] = undefined /*out*/;
            inputs["severity"] = undefined /*out*/;
            inputs["suppressionDuration"] = undefined /*out*/;
            inputs["suppressionEnabled"] = undefined /*out*/;
            inputs["tactics"] = undefined /*out*/;
            inputs["triggerOperator"] = undefined /*out*/;
            inputs["triggerThreshold"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:securityinsights:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20190101preview:ScheduledAlertRule" }, { type: "azure-nextgen:securityinsights/v20190101preview:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20200101:ScheduledAlertRule" }, { type: "azure-nextgen:securityinsights/v20200101:ScheduledAlertRule" }, { type: "azure-native:securityinsights/v20210301preview:ScheduledAlertRule" }, { type: "azure-nextgen:securityinsights/v20210301preview:ScheduledAlertRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ScheduledAlertRule.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ScheduledAlertRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ScheduledAlertRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ScheduledAlertRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ScheduledAlertRule.__pulumiType;
    }
}
exports.ScheduledAlertRule = ScheduledAlertRule;
/** @internal */
ScheduledAlertRule.__pulumiType = 'azure-native:securityinsights:ScheduledAlertRule';
