// @pulumi/azure-native/security/v20190801/deviceSecurityGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceSecurityGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The device security group resource
 */
class DeviceSecurityGroup extends pulumi.CustomResource {
    /**
     * Create a DeviceSecurityGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceId'");
            }
            inputs["allowlistRules"] = args ? args.allowlistRules : undefined;
            inputs["denylistRules"] = args ? args.denylistRules : undefined;
            inputs["deviceSecurityGroupName"] = args ? args.deviceSecurityGroupName : undefined;
            inputs["resourceId"] = args ? args.resourceId : undefined;
            inputs["thresholdRules"] = args ? args.thresholdRules : undefined;
            inputs["timeWindowRules"] = args ? args.timeWindowRules : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["allowlistRules"] = undefined /*out*/;
            inputs["denylistRules"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["thresholdRules"] = undefined /*out*/;
            inputs["timeWindowRules"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:security/v20190801:DeviceSecurityGroup" }, { type: "azure-native:security:DeviceSecurityGroup" }, { type: "azure-nextgen:security:DeviceSecurityGroup" }, { type: "azure-native:security/v20170801preview:DeviceSecurityGroup" }, { type: "azure-nextgen:security/v20170801preview:DeviceSecurityGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DeviceSecurityGroup.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DeviceSecurityGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DeviceSecurityGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DeviceSecurityGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DeviceSecurityGroup.__pulumiType;
    }
}
exports.DeviceSecurityGroup = DeviceSecurityGroup;
/** @internal */
DeviceSecurityGroup.__pulumiType = 'azure-native:security/v20190801:DeviceSecurityGroup';
