// @pulumi/azure-native/types/enums/sql/v20170301preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SensitivityLabelRank = exports.SecurityAlertPolicyState = exports.SampleName = exports.ManagedInstanceAdministratorType = exports.ManagedDatabaseCreateMode = exports.JobTargetType = exports.JobTargetGroupMembershipType = exports.JobStepOutputType = exports.JobStepActionType = exports.JobStepActionSource = exports.JobScheduleType = exports.CreateMode = exports.CatalogCollationType = exports.BlobAuditingPolicyState = void 0;
exports.BlobAuditingPolicyState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.CatalogCollationType = {
    DATABASE_DEFAULT: "DATABASE_DEFAULT",
    SQL_Latin1_General_CP1_CI_AS: "SQL_Latin1_General_CP1_CI_AS",
};
exports.CreateMode = {
    Default: "Default",
    Copy: "Copy",
    Secondary: "Secondary",
    OnlineSecondary: "OnlineSecondary",
    PointInTimeRestore: "PointInTimeRestore",
    Restore: "Restore",
    Recovery: "Recovery",
    RestoreExternalBackup: "RestoreExternalBackup",
    RestoreExternalBackupSecondary: "RestoreExternalBackupSecondary",
    RestoreLongTermRetentionBackup: "RestoreLongTermRetentionBackup",
};
exports.JobScheduleType = {
    Once: "Once",
    Recurring: "Recurring",
};
exports.JobStepActionSource = {
    Inline: "Inline",
};
exports.JobStepActionType = {
    TSql: "TSql",
};
exports.JobStepOutputType = {
    SqlDatabase: "SqlDatabase",
};
exports.JobTargetGroupMembershipType = {
    Include: "Include",
    Exclude: "Exclude",
};
exports.JobTargetType = {
    TargetGroup: "TargetGroup",
    SqlDatabase: "SqlDatabase",
    SqlElasticPool: "SqlElasticPool",
    SqlShardMap: "SqlShardMap",
    SqlServer: "SqlServer",
};
exports.ManagedDatabaseCreateMode = {
    Default: "Default",
    RestoreExternalBackup: "RestoreExternalBackup",
    PointInTimeRestore: "PointInTimeRestore",
    Recovery: "Recovery",
    RestoreLongTermRetentionBackup: "RestoreLongTermRetentionBackup",
};
exports.ManagedInstanceAdministratorType = {
    ActiveDirectory: "ActiveDirectory",
};
exports.SampleName = {
    AdventureWorksLT: "AdventureWorksLT",
    WideWorldImportersStd: "WideWorldImportersStd",
    WideWorldImportersFull: "WideWorldImportersFull",
};
exports.SecurityAlertPolicyState = {
    New: "New",
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SensitivityLabelRank = {
    None: "None",
    Low: "Low",
    Medium: "Medium",
    High: "High",
    Critical: "Critical",
};
