// @pulumi/azure-native/types/enums/containerinstance/v20180901/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperatingSystemTypes = exports.LogAnalyticsLogType = exports.ContainerNetworkProtocol = exports.ContainerGroupRestartPolicy = exports.ContainerGroupNetworkProtocol = exports.ContainerGroupIpAddressType = void 0;
exports.ContainerGroupIpAddressType = {
    Public: "Public",
    Private: "Private",
};
exports.ContainerGroupNetworkProtocol = {
    TCP: "TCP",
    UDP: "UDP",
};
exports.ContainerGroupRestartPolicy = {
    Always: "Always",
    OnFailure: "OnFailure",
    Never: "Never",
};
exports.ContainerNetworkProtocol = {
    TCP: "TCP",
    UDP: "UDP",
};
exports.LogAnalyticsLogType = {
    ContainerInsights: "ContainerInsights",
    ContainerInstanceLogs: "ContainerInstanceLogs",
};
exports.OperatingSystemTypes = {
    Windows: "Windows",
    Linux: "Linux",
};
