// @pulumi/azure-native/types/enums/solutions/v20200821preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceIdentityType = exports.JitApproverType = exports.JitApprovalMode = exports.DeploymentMode = exports.ApplicationManagementMode = exports.ApplicationLockLevel = exports.ApplicationDefinitionArtifactName = exports.ApplicationArtifactType = void 0;
exports.ApplicationArtifactType = {
    NotSpecified: "NotSpecified",
    Template: "Template",
    Custom: "Custom",
};
exports.ApplicationDefinitionArtifactName = {
    NotSpecified: "NotSpecified",
    ApplicationResourceTemplate: "ApplicationResourceTemplate",
    CreateUiDefinition: "CreateUiDefinition",
    MainTemplateParameters: "MainTemplateParameters",
};
exports.ApplicationLockLevel = {
    CanNotDelete: "CanNotDelete",
    ReadOnly: "ReadOnly",
    None: "None",
};
exports.ApplicationManagementMode = {
    NotSpecified: "NotSpecified",
    Unmanaged: "Unmanaged",
    Managed: "Managed",
};
exports.DeploymentMode = {
    NotSpecified: "NotSpecified",
    Incremental: "Incremental",
    Complete: "Complete",
};
exports.JitApprovalMode = {
    NotSpecified: "NotSpecified",
    AutoApprove: "AutoApprove",
    ManualApprove: "ManualApprove",
};
exports.JitApproverType = {
    User: "user",
    Group: "group",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
