// @pulumi/azure-native/web/v20201001/webAppPrivateEndpointConnection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppPrivateEndpointConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Private Endpoint Connection ARM resource.
 */
class WebAppPrivateEndpointConnection extends pulumi.CustomResource {
    /**
     * Create a WebAppPrivateEndpointConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["kind"] = args ? args.kind : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["privateEndpointConnectionName"] = args ? args.privateEndpointConnectionName : undefined;
            inputs["privateLinkServiceConnectionState"] = args ? args.privateLinkServiceConnectionState : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["privateEndpoint"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["privateEndpoint"] = undefined /*out*/;
            inputs["privateLinkServiceConnectionState"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20201001:WebAppPrivateEndpointConnection" }, { type: "azure-native:web:WebAppPrivateEndpointConnection" }, { type: "azure-nextgen:web:WebAppPrivateEndpointConnection" }, { type: "azure-native:web/v20190801:WebAppPrivateEndpointConnection" }, { type: "azure-nextgen:web/v20190801:WebAppPrivateEndpointConnection" }, { type: "azure-native:web/v20200601:WebAppPrivateEndpointConnection" }, { type: "azure-nextgen:web/v20200601:WebAppPrivateEndpointConnection" }, { type: "azure-native:web/v20200901:WebAppPrivateEndpointConnection" }, { type: "azure-nextgen:web/v20200901:WebAppPrivateEndpointConnection" }, { type: "azure-native:web/v20201201:WebAppPrivateEndpointConnection" }, { type: "azure-nextgen:web/v20201201:WebAppPrivateEndpointConnection" }, { type: "azure-native:web/v20210101:WebAppPrivateEndpointConnection" }, { type: "azure-nextgen:web/v20210101:WebAppPrivateEndpointConnection" }, { type: "azure-native:web/v20210115:WebAppPrivateEndpointConnection" }, { type: "azure-nextgen:web/v20210115:WebAppPrivateEndpointConnection" }, { type: "azure-native:web/v20210201:WebAppPrivateEndpointConnection" }, { type: "azure-nextgen:web/v20210201:WebAppPrivateEndpointConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppPrivateEndpointConnection.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WebAppPrivateEndpointConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppPrivateEndpointConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppPrivateEndpointConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppPrivateEndpointConnection.__pulumiType;
    }
}
exports.WebAppPrivateEndpointConnection = WebAppPrivateEndpointConnection;
/** @internal */
WebAppPrivateEndpointConnection.__pulumiType = 'azure-native:web/v20201001:WebAppPrivateEndpointConnection';
