// @pulumi/azure-native/web/v20201001/webAppAuthSettingsV2Slot.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppAuthSettingsV2Slot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
class WebAppAuthSettingsV2Slot extends pulumi.CustomResource {
    /**
     * Create a WebAppAuthSettingsV2Slot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            inputs["globalValidation"] = args ? args.globalValidation : undefined;
            inputs["httpSettings"] = args ? args.httpSettings : undefined;
            inputs["identityProviders"] = args ? args.identityProviders : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["login"] = args ? args.login : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["platform"] = args ? args.platform : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["slot"] = args ? args.slot : undefined;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["globalValidation"] = undefined /*out*/;
            inputs["httpSettings"] = undefined /*out*/;
            inputs["identityProviders"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["login"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["platform"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20201001:WebAppAuthSettingsV2Slot" }, { type: "azure-native:web:WebAppAuthSettingsV2Slot" }, { type: "azure-nextgen:web:WebAppAuthSettingsV2Slot" }, { type: "azure-native:web/v20200601:WebAppAuthSettingsV2Slot" }, { type: "azure-nextgen:web/v20200601:WebAppAuthSettingsV2Slot" }, { type: "azure-native:web/v20200901:WebAppAuthSettingsV2Slot" }, { type: "azure-nextgen:web/v20200901:WebAppAuthSettingsV2Slot" }, { type: "azure-native:web/v20201201:WebAppAuthSettingsV2Slot" }, { type: "azure-nextgen:web/v20201201:WebAppAuthSettingsV2Slot" }, { type: "azure-native:web/v20210101:WebAppAuthSettingsV2Slot" }, { type: "azure-nextgen:web/v20210101:WebAppAuthSettingsV2Slot" }, { type: "azure-native:web/v20210115:WebAppAuthSettingsV2Slot" }, { type: "azure-nextgen:web/v20210115:WebAppAuthSettingsV2Slot" }, { type: "azure-native:web/v20210201:WebAppAuthSettingsV2Slot" }, { type: "azure-nextgen:web/v20210201:WebAppAuthSettingsV2Slot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppAuthSettingsV2Slot.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WebAppAuthSettingsV2Slot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppAuthSettingsV2Slot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppAuthSettingsV2Slot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppAuthSettingsV2Slot.__pulumiType;
    }
}
exports.WebAppAuthSettingsV2Slot = WebAppAuthSettingsV2Slot;
/** @internal */
WebAppAuthSettingsV2Slot.__pulumiType = 'azure-native:web/v20201001:WebAppAuthSettingsV2Slot';
