// @pulumi/azure-native/web/v20201001/webAppSlotConfigurationNames.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppSlotConfigurationNames = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Slot Config names azure resource.
 */
class WebAppSlotConfigurationNames extends pulumi.CustomResource {
    /**
     * Create a WebAppSlotConfigurationNames resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["appSettingNames"] = args ? args.appSettingNames : undefined;
            inputs["azureStorageConfigNames"] = args ? args.azureStorageConfigNames : undefined;
            inputs["connectionStringNames"] = args ? args.connectionStringNames : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["appSettingNames"] = undefined /*out*/;
            inputs["azureStorageConfigNames"] = undefined /*out*/;
            inputs["connectionStringNames"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20201001:WebAppSlotConfigurationNames" }, { type: "azure-native:web:WebAppSlotConfigurationNames" }, { type: "azure-nextgen:web:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20150801:WebAppSlotConfigurationNames" }, { type: "azure-nextgen:web/v20150801:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20160801:WebAppSlotConfigurationNames" }, { type: "azure-nextgen:web/v20160801:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20180201:WebAppSlotConfigurationNames" }, { type: "azure-nextgen:web/v20180201:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20181101:WebAppSlotConfigurationNames" }, { type: "azure-nextgen:web/v20181101:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20190801:WebAppSlotConfigurationNames" }, { type: "azure-nextgen:web/v20190801:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20200601:WebAppSlotConfigurationNames" }, { type: "azure-nextgen:web/v20200601:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20200901:WebAppSlotConfigurationNames" }, { type: "azure-nextgen:web/v20200901:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20201201:WebAppSlotConfigurationNames" }, { type: "azure-nextgen:web/v20201201:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20210101:WebAppSlotConfigurationNames" }, { type: "azure-nextgen:web/v20210101:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20210115:WebAppSlotConfigurationNames" }, { type: "azure-nextgen:web/v20210115:WebAppSlotConfigurationNames" }, { type: "azure-native:web/v20210201:WebAppSlotConfigurationNames" }, { type: "azure-nextgen:web/v20210201:WebAppSlotConfigurationNames" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppSlotConfigurationNames.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WebAppSlotConfigurationNames resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppSlotConfigurationNames(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppSlotConfigurationNames.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppSlotConfigurationNames.__pulumiType;
    }
}
exports.WebAppSlotConfigurationNames = WebAppSlotConfigurationNames;
/** @internal */
WebAppSlotConfigurationNames.__pulumiType = 'azure-native:web/v20201001:WebAppSlotConfigurationNames';
