// @pulumi/azure-native/web/v20201001/webAppMetadataSlot.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppMetadataSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * String dictionary resource.
 */
class WebAppMetadataSlot extends pulumi.CustomResource {
    /**
     * Create a WebAppMetadataSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            inputs["kind"] = args ? args.kind : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["properties"] = args ? args.properties : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["slot"] = args ? args.slot : undefined;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:web/v20201001:WebAppMetadataSlot" }, { type: "azure-native:web:WebAppMetadataSlot" }, { type: "azure-nextgen:web:WebAppMetadataSlot" }, { type: "azure-native:web/v20150801:WebAppMetadataSlot" }, { type: "azure-nextgen:web/v20150801:WebAppMetadataSlot" }, { type: "azure-native:web/v20160801:WebAppMetadataSlot" }, { type: "azure-nextgen:web/v20160801:WebAppMetadataSlot" }, { type: "azure-native:web/v20180201:WebAppMetadataSlot" }, { type: "azure-nextgen:web/v20180201:WebAppMetadataSlot" }, { type: "azure-native:web/v20181101:WebAppMetadataSlot" }, { type: "azure-nextgen:web/v20181101:WebAppMetadataSlot" }, { type: "azure-native:web/v20190801:WebAppMetadataSlot" }, { type: "azure-nextgen:web/v20190801:WebAppMetadataSlot" }, { type: "azure-native:web/v20200601:WebAppMetadataSlot" }, { type: "azure-nextgen:web/v20200601:WebAppMetadataSlot" }, { type: "azure-native:web/v20200901:WebAppMetadataSlot" }, { type: "azure-nextgen:web/v20200901:WebAppMetadataSlot" }, { type: "azure-native:web/v20201201:WebAppMetadataSlot" }, { type: "azure-nextgen:web/v20201201:WebAppMetadataSlot" }, { type: "azure-native:web/v20210101:WebAppMetadataSlot" }, { type: "azure-nextgen:web/v20210101:WebAppMetadataSlot" }, { type: "azure-native:web/v20210115:WebAppMetadataSlot" }, { type: "azure-nextgen:web/v20210115:WebAppMetadataSlot" }, { type: "azure-native:web/v20210201:WebAppMetadataSlot" }, { type: "azure-nextgen:web/v20210201:WebAppMetadataSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppMetadataSlot.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing WebAppMetadataSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppMetadataSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppMetadataSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppMetadataSlot.__pulumiType;
    }
}
exports.WebAppMetadataSlot = WebAppMetadataSlot;
/** @internal */
WebAppMetadataSlot.__pulumiType = 'azure-native:web/v20201001:WebAppMetadataSlot';
