// @pulumi/azure-native/network/v20200701/vpnSite.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpnSite = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * VpnSite Resource.
 */
class VpnSite extends pulumi.CustomResource {
    /**
     * Create a VpnSite resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["addressSpace"] = args ? args.addressSpace : undefined;
            inputs["bgpProperties"] = args ? args.bgpProperties : undefined;
            inputs["deviceProperties"] = args ? args.deviceProperties : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["ipAddress"] = args ? args.ipAddress : undefined;
            inputs["isSecuritySite"] = args ? args.isSecuritySite : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["o365Policy"] = args ? args.o365Policy : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["siteKey"] = args ? args.siteKey : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["virtualWan"] = args ? args.virtualWan : undefined;
            inputs["vpnSiteLinks"] = args ? args.vpnSiteLinks : undefined;
            inputs["vpnSiteName"] = args ? args.vpnSiteName : undefined;
            inputs["etag"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["addressSpace"] = undefined /*out*/;
            inputs["bgpProperties"] = undefined /*out*/;
            inputs["deviceProperties"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["ipAddress"] = undefined /*out*/;
            inputs["isSecuritySite"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["o365Policy"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["siteKey"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["virtualWan"] = undefined /*out*/;
            inputs["vpnSiteLinks"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:network/v20200701:VpnSite" }, { type: "azure-native:network:VpnSite" }, { type: "azure-nextgen:network:VpnSite" }, { type: "azure-native:network/v20180401:VpnSite" }, { type: "azure-nextgen:network/v20180401:VpnSite" }, { type: "azure-native:network/v20180601:VpnSite" }, { type: "azure-nextgen:network/v20180601:VpnSite" }, { type: "azure-native:network/v20180701:VpnSite" }, { type: "azure-nextgen:network/v20180701:VpnSite" }, { type: "azure-native:network/v20180801:VpnSite" }, { type: "azure-nextgen:network/v20180801:VpnSite" }, { type: "azure-native:network/v20181001:VpnSite" }, { type: "azure-nextgen:network/v20181001:VpnSite" }, { type: "azure-native:network/v20181101:VpnSite" }, { type: "azure-nextgen:network/v20181101:VpnSite" }, { type: "azure-native:network/v20181201:VpnSite" }, { type: "azure-nextgen:network/v20181201:VpnSite" }, { type: "azure-native:network/v20190201:VpnSite" }, { type: "azure-nextgen:network/v20190201:VpnSite" }, { type: "azure-native:network/v20190401:VpnSite" }, { type: "azure-nextgen:network/v20190401:VpnSite" }, { type: "azure-native:network/v20190601:VpnSite" }, { type: "azure-nextgen:network/v20190601:VpnSite" }, { type: "azure-native:network/v20190701:VpnSite" }, { type: "azure-nextgen:network/v20190701:VpnSite" }, { type: "azure-native:network/v20190801:VpnSite" }, { type: "azure-nextgen:network/v20190801:VpnSite" }, { type: "azure-native:network/v20190901:VpnSite" }, { type: "azure-nextgen:network/v20190901:VpnSite" }, { type: "azure-native:network/v20191101:VpnSite" }, { type: "azure-nextgen:network/v20191101:VpnSite" }, { type: "azure-native:network/v20191201:VpnSite" }, { type: "azure-nextgen:network/v20191201:VpnSite" }, { type: "azure-native:network/v20200301:VpnSite" }, { type: "azure-nextgen:network/v20200301:VpnSite" }, { type: "azure-native:network/v20200401:VpnSite" }, { type: "azure-nextgen:network/v20200401:VpnSite" }, { type: "azure-native:network/v20200501:VpnSite" }, { type: "azure-nextgen:network/v20200501:VpnSite" }, { type: "azure-native:network/v20200601:VpnSite" }, { type: "azure-nextgen:network/v20200601:VpnSite" }, { type: "azure-native:network/v20200801:VpnSite" }, { type: "azure-nextgen:network/v20200801:VpnSite" }, { type: "azure-native:network/v20201101:VpnSite" }, { type: "azure-nextgen:network/v20201101:VpnSite" }, { type: "azure-native:network/v20210201:VpnSite" }, { type: "azure-nextgen:network/v20210201:VpnSite" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VpnSite.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing VpnSite resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VpnSite(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VpnSite.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VpnSite.__pulumiType;
    }
}
exports.VpnSite = VpnSite;
/** @internal */
VpnSite.__pulumiType = 'azure-native:network/v20200701:VpnSite';
