// @pulumi/azure-native/types/enums/mixedreality/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.ResourceIdentityType = exports.v20210301preview = exports.v20210101 = exports.v20200501 = exports.v20200406preview = exports.v20191202preview = exports.v20190228preview = void 0;
// Export sub-modules:
const v20190228preview = require("./v20190228preview");
exports.v20190228preview = v20190228preview;
const v20191202preview = require("./v20191202preview");
exports.v20191202preview = v20191202preview;
const v20200406preview = require("./v20200406preview");
exports.v20200406preview = v20200406preview;
const v20200501 = require("./v20200501");
exports.v20200501 = v20200501;
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20210301preview = require("./v20210301preview");
exports.v20210301preview = v20210301preview;
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
