// @pulumi/azure-native/resources/v20200801/deploymentAtTenantScope.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentAtTenantScope = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Deployment information.
 */
class DeploymentAtTenantScope extends pulumi.CustomResource {
    /**
     * Create a DeploymentAtTenantScope resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.properties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'properties'");
            }
            inputs["deploymentName"] = args ? args.deploymentName : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["properties"] = args ? args.properties : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["properties"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:resources/v20200801:DeploymentAtTenantScope" }, { type: "azure-native:resources:DeploymentAtTenantScope" }, { type: "azure-nextgen:resources:DeploymentAtTenantScope" }, { type: "azure-native:resources/v20190701:DeploymentAtTenantScope" }, { type: "azure-nextgen:resources/v20190701:DeploymentAtTenantScope" }, { type: "azure-native:resources/v20190801:DeploymentAtTenantScope" }, { type: "azure-nextgen:resources/v20190801:DeploymentAtTenantScope" }, { type: "azure-native:resources/v20191001:DeploymentAtTenantScope" }, { type: "azure-nextgen:resources/v20191001:DeploymentAtTenantScope" }, { type: "azure-native:resources/v20200601:DeploymentAtTenantScope" }, { type: "azure-nextgen:resources/v20200601:DeploymentAtTenantScope" }, { type: "azure-native:resources/v20201001:DeploymentAtTenantScope" }, { type: "azure-nextgen:resources/v20201001:DeploymentAtTenantScope" }, { type: "azure-native:resources/v20210101:DeploymentAtTenantScope" }, { type: "azure-nextgen:resources/v20210101:DeploymentAtTenantScope" }, { type: "azure-native:resources/v20210401:DeploymentAtTenantScope" }, { type: "azure-nextgen:resources/v20210401:DeploymentAtTenantScope" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DeploymentAtTenantScope.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DeploymentAtTenantScope resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DeploymentAtTenantScope(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DeploymentAtTenantScope.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DeploymentAtTenantScope.__pulumiType;
    }
}
exports.DeploymentAtTenantScope = DeploymentAtTenantScope;
/** @internal */
DeploymentAtTenantScope.__pulumiType = 'azure-native:resources/v20200801:DeploymentAtTenantScope';
