// @pulumi/azure-native/types/enums/avs/v20210601/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SslEnum = exports.ScriptExecutionParameterType = exports.PortMirroringDirectionEnum = exports.MountOptionEnum = exports.InternetEnum = exports.DnsServiceLogLevelEnum = exports.DhcpTypeEnum = exports.AddonType = void 0;
exports.AddonType = {
    SRM: "SRM",
    VR: "VR",
    HCX: "HCX",
};
exports.DhcpTypeEnum = {
    SERVER: "SERVER",
    RELAY: "RELAY",
};
exports.DnsServiceLogLevelEnum = {
    DEBUG: "DEBUG",
    INFO: "INFO",
    WARNING: "WARNING",
    ERROR: "ERROR",
    FATAL: "FATAL",
};
exports.InternetEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.MountOptionEnum = {
    MOUNT: "MOUNT",
    ATTACH: "ATTACH",
};
exports.PortMirroringDirectionEnum = {
    INGRESS_EGRESS_BIDIRECTIONAL: "INGRESS, EGRESS, BIDIRECTIONAL",
};
exports.ScriptExecutionParameterType = {
    Value: "Value",
    SecureValue: "SecureValue",
    Credential: "Credential",
};
exports.SslEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
