// @pulumi/azure-native/resources/v20191001preview/azureCliScript.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AzureCliScript = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Object model for the Azure CLI script.
 */
class AzureCliScript extends pulumi.CustomResource {
    /**
     * Create a AzureCliScript resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a, _b;
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.azCliVersion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azCliVersion'");
            }
            if ((!args || args.identity === undefined) && !opts.urn) {
                throw new Error("Missing required property 'identity'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.retentionInterval === undefined) && !opts.urn) {
                throw new Error("Missing required property 'retentionInterval'");
            }
            inputs["arguments"] = args ? args.arguments : undefined;
            inputs["azCliVersion"] = args ? args.azCliVersion : undefined;
            inputs["cleanupPreference"] = (_a = (args ? args.cleanupPreference : undefined)) !== null && _a !== void 0 ? _a : "Always";
            inputs["containerSettings"] = args ? args.containerSettings : undefined;
            inputs["environmentVariables"] = args ? args.environmentVariables : undefined;
            inputs["forceUpdateTag"] = args ? args.forceUpdateTag : undefined;
            inputs["identity"] = args ? args.identity : undefined;
            inputs["kind"] = "AzureCLI";
            inputs["location"] = args ? args.location : undefined;
            inputs["primaryScriptUri"] = args ? args.primaryScriptUri : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["retentionInterval"] = args ? args.retentionInterval : undefined;
            inputs["scriptContent"] = args ? args.scriptContent : undefined;
            inputs["scriptName"] = args ? args.scriptName : undefined;
            inputs["storageAccountSettings"] = args ? args.storageAccountSettings : undefined;
            inputs["supportingScriptUris"] = args ? args.supportingScriptUris : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["timeout"] = (_b = (args ? args.timeout : undefined)) !== null && _b !== void 0 ? _b : "P1D";
            inputs["name"] = undefined /*out*/;
            inputs["outputs"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["arguments"] = undefined /*out*/;
            inputs["azCliVersion"] = undefined /*out*/;
            inputs["cleanupPreference"] = undefined /*out*/;
            inputs["containerSettings"] = undefined /*out*/;
            inputs["environmentVariables"] = undefined /*out*/;
            inputs["forceUpdateTag"] = undefined /*out*/;
            inputs["identity"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["outputs"] = undefined /*out*/;
            inputs["primaryScriptUri"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["retentionInterval"] = undefined /*out*/;
            inputs["scriptContent"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["storageAccountSettings"] = undefined /*out*/;
            inputs["supportingScriptUris"] = undefined /*out*/;
            inputs["systemData"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["timeout"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:resources/v20191001preview:AzureCliScript" }, { type: "azure-native:resources:AzureCliScript" }, { type: "azure-nextgen:resources:AzureCliScript" }, { type: "azure-native:resources/v20201001:AzureCliScript" }, { type: "azure-nextgen:resources/v20201001:AzureCliScript" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AzureCliScript.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing AzureCliScript resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new AzureCliScript(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AzureCliScript.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AzureCliScript.__pulumiType;
    }
}
exports.AzureCliScript = AzureCliScript;
/** @internal */
AzureCliScript.__pulumiType = 'azure-native:resources/v20191001preview:AzureCliScript';
