// @pulumi/azure-native/authorization/v20160901/managementLockAtSubscriptionLevel.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagementLockAtSubscriptionLevel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The lock information.
 */
class ManagementLockAtSubscriptionLevel extends pulumi.CustomResource {
    /**
     * Create a ManagementLockAtSubscriptionLevel resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.level === undefined) && !opts.urn) {
                throw new Error("Missing required property 'level'");
            }
            inputs["level"] = args ? args.level : undefined;
            inputs["lockName"] = args ? args.lockName : undefined;
            inputs["notes"] = args ? args.notes : undefined;
            inputs["owners"] = args ? args.owners : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["level"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["notes"] = undefined /*out*/;
            inputs["owners"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:authorization/v20160901:ManagementLockAtSubscriptionLevel" }, { type: "azure-native:authorization:ManagementLockAtSubscriptionLevel" }, { type: "azure-nextgen:authorization:ManagementLockAtSubscriptionLevel" }, { type: "azure-native:authorization/v20150101:ManagementLockAtSubscriptionLevel" }, { type: "azure-nextgen:authorization/v20150101:ManagementLockAtSubscriptionLevel" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagementLockAtSubscriptionLevel.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ManagementLockAtSubscriptionLevel resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ManagementLockAtSubscriptionLevel(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagementLockAtSubscriptionLevel.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagementLockAtSubscriptionLevel.__pulumiType;
    }
}
exports.ManagementLockAtSubscriptionLevel = ManagementLockAtSubscriptionLevel;
/** @internal */
ManagementLockAtSubscriptionLevel.__pulumiType = 'azure-native:authorization/v20160901:ManagementLockAtSubscriptionLevel';
