// @pulumi/azure-native/types/enums/containerregistry/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookStatus = exports.WebhookAction = exports.Variant = exports.UpdateTriggerPayloadType = exports.TrustPolicyType = exports.TriggerStatus = exports.TokenType = exports.TokenStatus = exports.TokenPasswordName = exports.TokenCertificateName = exports.TaskStatus = exports.StepType = exports.SourceTriggerEvent = exports.SourceRegistryLoginMode = exports.SourceControlType = exports.SkuName = exports.SecretObjectType = exports.ResourceIdentityType = exports.PolicyStatus = exports.PipelineSourceType = exports.PipelineRunTargetType = exports.PipelineRunSourceType = exports.PipelineOptions = exports.OS = exports.LogLevel = exports.DefaultAction = exports.ConnectionStatus = exports.ConnectedRegistryMode = exports.BaseImageTriggerType = exports.AuditLogStatus = exports.Architecture = exports.ActionsRequired = exports.Action = exports.v20210601preview = exports.v20201101preview = exports.v20191201preview = exports.v20190601preview = exports.v20190501preview = exports.v20190501 = exports.v20190401 = exports.v20180901 = exports.v20180201preview = exports.v20171001 = exports.v20170601preview = void 0;
// Export sub-modules:
const v20170601preview = require("./v20170601preview");
exports.v20170601preview = v20170601preview;
const v20171001 = require("./v20171001");
exports.v20171001 = v20171001;
const v20180201preview = require("./v20180201preview");
exports.v20180201preview = v20180201preview;
const v20180901 = require("./v20180901");
exports.v20180901 = v20180901;
const v20190401 = require("./v20190401");
exports.v20190401 = v20190401;
const v20190501 = require("./v20190501");
exports.v20190501 = v20190501;
const v20190501preview = require("./v20190501preview");
exports.v20190501preview = v20190501preview;
const v20190601preview = require("./v20190601preview");
exports.v20190601preview = v20190601preview;
const v20191201preview = require("./v20191201preview");
exports.v20191201preview = v20191201preview;
const v20201101preview = require("./v20201101preview");
exports.v20201101preview = v20201101preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
exports.Action = {
    Allow: "Allow",
};
exports.ActionsRequired = {
    None: "None",
    Recreate: "Recreate",
};
exports.Architecture = {
    Amd64: "amd64",
    X86: "x86",
    Architecture_386: "386",
    Arm: "arm",
    Arm64: "arm64",
};
exports.AuditLogStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.BaseImageTriggerType = {
    All: "All",
    Runtime: "Runtime",
};
exports.ConnectedRegistryMode = {
    Registry: "Registry",
    Mirror: "Mirror",
};
exports.ConnectionStatus = {
    Approved: "Approved",
    Pending: "Pending",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.DefaultAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.LogLevel = {
    Debug: "Debug",
    Information: "Information",
    Warning: "Warning",
    Error: "Error",
    None: "None",
};
exports.OS = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.PipelineOptions = {
    OverwriteTags: "OverwriteTags",
    OverwriteBlobs: "OverwriteBlobs",
    DeleteSourceBlobOnSuccess: "DeleteSourceBlobOnSuccess",
    ContinueOnErrors: "ContinueOnErrors",
};
exports.PipelineRunSourceType = {
    AzureStorageBlob: "AzureStorageBlob",
};
exports.PipelineRunTargetType = {
    AzureStorageBlob: "AzureStorageBlob",
};
exports.PipelineSourceType = {
    AzureStorageBlobContainer: "AzureStorageBlobContainer",
};
exports.PolicyStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.SecretObjectType = {
    Opaque: "Opaque",
    Vaultsecret: "Vaultsecret",
};
exports.SkuName = {
    Classic: "Classic",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.SourceControlType = {
    Github: "Github",
    VisualStudioTeamService: "VisualStudioTeamService",
};
exports.SourceRegistryLoginMode = {
    None: "None",
    Default: "Default",
};
exports.SourceTriggerEvent = {
    Commit: "commit",
    Pullrequest: "pullrequest",
};
exports.StepType = {
    Docker: "Docker",
    FileTask: "FileTask",
    EncodedTask: "EncodedTask",
};
exports.TaskStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.TokenCertificateName = {
    Certificate1: "certificate1",
    Certificate2: "certificate2",
};
exports.TokenPasswordName = {
    Password1: "password1",
    Password2: "password2",
};
exports.TokenStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.TokenType = {
    PAT: "PAT",
    OAuth: "OAuth",
};
exports.TriggerStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.TrustPolicyType = {
    Notary: "Notary",
};
exports.UpdateTriggerPayloadType = {
    Default: "Default",
    Token: "Token",
};
exports.Variant = {
    V6: "v6",
    V7: "v7",
    V8: "v8",
};
exports.WebhookAction = {
    Push: "push",
    Delete: "delete",
    Quarantine: "quarantine",
    Chart_push: "chart_push",
    Chart_delete: "chart_delete",
};
exports.WebhookStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
