// @pulumi/azure-native/storsimple/backupSchedule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackupSchedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The backup schedule.
 * API Version: 2017-06-01.
 */
class BackupSchedule extends pulumi.CustomResource {
    /**
     * Create a BackupSchedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.backupPolicyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'backupPolicyName'");
            }
            if ((!args || args.backupType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'backupType'");
            }
            if ((!args || args.deviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deviceName'");
            }
            if ((!args || args.managerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.retentionCount === undefined) && !opts.urn) {
                throw new Error("Missing required property 'retentionCount'");
            }
            if ((!args || args.scheduleRecurrence === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scheduleRecurrence'");
            }
            if ((!args || args.scheduleStatus === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scheduleStatus'");
            }
            if ((!args || args.startTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'startTime'");
            }
            inputs["backupPolicyName"] = args ? args.backupPolicyName : undefined;
            inputs["backupScheduleName"] = args ? args.backupScheduleName : undefined;
            inputs["backupType"] = args ? args.backupType : undefined;
            inputs["deviceName"] = args ? args.deviceName : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["managerName"] = args ? args.managerName : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["retentionCount"] = args ? args.retentionCount : undefined;
            inputs["scheduleRecurrence"] = args ? args.scheduleRecurrence : undefined;
            inputs["scheduleStatus"] = args ? args.scheduleStatus : undefined;
            inputs["startTime"] = args ? args.startTime : undefined;
            inputs["lastSuccessfulRun"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["backupType"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["lastSuccessfulRun"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["retentionCount"] = undefined /*out*/;
            inputs["scheduleRecurrence"] = undefined /*out*/;
            inputs["scheduleStatus"] = undefined /*out*/;
            inputs["startTime"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:storsimple:BackupSchedule" }, { type: "azure-native:storsimple/v20170601:BackupSchedule" }, { type: "azure-nextgen:storsimple/v20170601:BackupSchedule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BackupSchedule.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing BackupSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BackupSchedule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BackupSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BackupSchedule.__pulumiType;
    }
}
exports.BackupSchedule = BackupSchedule;
/** @internal */
BackupSchedule.__pulumiType = 'azure-native:storsimple:BackupSchedule';
