// @pulumi/azure-native/storsimple/managerExtendedInfo.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagerExtendedInfo = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * The extended info of the manager.
 * API Version: 2017-06-01.
 */
class ManagerExtendedInfo extends pulumi.CustomResource {
    /**
     * Create a ManagerExtendedInfo resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.algorithm === undefined) && !opts.urn) {
                throw new Error("Missing required property 'algorithm'");
            }
            if ((!args || args.integrityKey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'integrityKey'");
            }
            if ((!args || args.managerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["algorithm"] = args ? args.algorithm : undefined;
            inputs["encryptionKey"] = args ? args.encryptionKey : undefined;
            inputs["encryptionKeyThumbprint"] = args ? args.encryptionKeyThumbprint : undefined;
            inputs["etag"] = args ? args.etag : undefined;
            inputs["integrityKey"] = args ? args.integrityKey : undefined;
            inputs["kind"] = args ? args.kind : undefined;
            inputs["managerName"] = args ? args.managerName : undefined;
            inputs["portalCertificateThumbprint"] = args ? args.portalCertificateThumbprint : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["version"] = args ? args.version : undefined;
            inputs["name"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["algorithm"] = undefined /*out*/;
            inputs["encryptionKey"] = undefined /*out*/;
            inputs["encryptionKeyThumbprint"] = undefined /*out*/;
            inputs["etag"] = undefined /*out*/;
            inputs["integrityKey"] = undefined /*out*/;
            inputs["kind"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["portalCertificateThumbprint"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["version"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:storsimple:ManagerExtendedInfo" }, { type: "azure-native:storsimple/v20161001:ManagerExtendedInfo" }, { type: "azure-nextgen:storsimple/v20161001:ManagerExtendedInfo" }, { type: "azure-native:storsimple/v20170601:ManagerExtendedInfo" }, { type: "azure-nextgen:storsimple/v20170601:ManagerExtendedInfo" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagerExtendedInfo.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing ManagerExtendedInfo resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ManagerExtendedInfo(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagerExtendedInfo.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagerExtendedInfo.__pulumiType;
    }
}
exports.ManagerExtendedInfo = ManagerExtendedInfo;
/** @internal */
ManagerExtendedInfo.__pulumiType = 'azure-native:storsimple:ManagerExtendedInfo';
