// @pulumi/azure-native/vmwarecloudsimple/v20190401/dedicatedCloudService.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DedicatedCloudService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Dedicated cloud service model
 */
class DedicatedCloudService extends pulumi.CustomResource {
    /**
     * Create a DedicatedCloudService resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.gatewaySubnet === undefined) && !opts.urn) {
                throw new Error("Missing required property 'gatewaySubnet'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["dedicatedCloudServiceName"] = args ? args.dedicatedCloudServiceName : undefined;
            inputs["gatewaySubnet"] = args ? args.gatewaySubnet : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["isAccountOnboarded"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["nodes"] = undefined /*out*/;
            inputs["serviceURL"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        else {
            inputs["gatewaySubnet"] = undefined /*out*/;
            inputs["isAccountOnboarded"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["nodes"] = undefined /*out*/;
            inputs["serviceURL"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:vmwarecloudsimple/v20190401:DedicatedCloudService" }, { type: "azure-native:vmwarecloudsimple:DedicatedCloudService" }, { type: "azure-nextgen:vmwarecloudsimple:DedicatedCloudService" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DedicatedCloudService.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DedicatedCloudService resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DedicatedCloudService(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DedicatedCloudService.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DedicatedCloudService.__pulumiType;
    }
}
exports.DedicatedCloudService = DedicatedCloudService;
/** @internal */
DedicatedCloudService.__pulumiType = 'azure-native:vmwarecloudsimple/v20190401:DedicatedCloudService';
