// @pulumi/azure-native/vmwarecloudsimple/v20190401/dedicatedCloudNode.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DedicatedCloudNode = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Dedicated cloud node model
 */
class DedicatedCloudNode extends pulumi.CustomResource {
    /**
     * Create a DedicatedCloudNode resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let inputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.availabilityZoneId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'availabilityZoneId'");
            }
            if ((!args || args.id === undefined) && !opts.urn) {
                throw new Error("Missing required property 'id'");
            }
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.nodesCount === undefined) && !opts.urn) {
                throw new Error("Missing required property 'nodesCount'");
            }
            if ((!args || args.placementGroupId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'placementGroupId'");
            }
            if ((!args || args.purchaseId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'purchaseId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            inputs["availabilityZoneId"] = args ? args.availabilityZoneId : undefined;
            inputs["dedicatedCloudNodeName"] = args ? args.dedicatedCloudNodeName : undefined;
            inputs["id"] = args ? args.id : undefined;
            inputs["location"] = args ? args.location : undefined;
            inputs["name"] = args ? args.name : undefined;
            inputs["nodesCount"] = args ? args.nodesCount : undefined;
            inputs["placementGroupId"] = args ? args.placementGroupId : undefined;
            inputs["purchaseId"] = args ? args.purchaseId : undefined;
            inputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            inputs["sku"] = args ? args.sku : undefined;
            inputs["tags"] = args ? args.tags : undefined;
            inputs["availabilityZoneName"] = undefined /*out*/;
            inputs["cloudRackName"] = undefined /*out*/;
            inputs["created"] = undefined /*out*/;
            inputs["placementGroupName"] = undefined /*out*/;
            inputs["privateCloudId"] = undefined /*out*/;
            inputs["privateCloudName"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["vmwareClusterName"] = undefined /*out*/;
        }
        else {
            inputs["availabilityZoneId"] = undefined /*out*/;
            inputs["availabilityZoneName"] = undefined /*out*/;
            inputs["cloudRackName"] = undefined /*out*/;
            inputs["created"] = undefined /*out*/;
            inputs["location"] = undefined /*out*/;
            inputs["name"] = undefined /*out*/;
            inputs["nodesCount"] = undefined /*out*/;
            inputs["placementGroupId"] = undefined /*out*/;
            inputs["placementGroupName"] = undefined /*out*/;
            inputs["privateCloudId"] = undefined /*out*/;
            inputs["privateCloudName"] = undefined /*out*/;
            inputs["provisioningState"] = undefined /*out*/;
            inputs["purchaseId"] = undefined /*out*/;
            inputs["sku"] = undefined /*out*/;
            inputs["status"] = undefined /*out*/;
            inputs["tags"] = undefined /*out*/;
            inputs["type"] = undefined /*out*/;
            inputs["vmwareClusterName"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        const aliasOpts = { aliases: [{ type: "azure-nextgen:vmwarecloudsimple/v20190401:DedicatedCloudNode" }, { type: "azure-native:vmwarecloudsimple:DedicatedCloudNode" }, { type: "azure-nextgen:vmwarecloudsimple:DedicatedCloudNode" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DedicatedCloudNode.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing DedicatedCloudNode resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DedicatedCloudNode(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DedicatedCloudNode.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DedicatedCloudNode.__pulumiType;
    }
}
exports.DedicatedCloudNode = DedicatedCloudNode;
/** @internal */
DedicatedCloudNode.__pulumiType = 'azure-native:vmwarecloudsimple/v20190401:DedicatedCloudNode';
